{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "2ccb64fc-ce60-42c7-80a8-0a5f13cd8198",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Mon Feb 19 12:09:49 PST 2024\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import sys\n",
    "sys.path.append('../dev')\n",
    "\n",
    "import utils\n",
    "import enrich_ as enrich\n",
    "\n",
    "pd.options.display.max_colwidth = 300\n",
    "pd.set_option('display.max_rows', None)\n",
    "\n",
    "!date"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c7b2865c-c7f3-475f-b4d6-8fc6a6797e5a",
   "metadata": {
    "tags": []
   },
   "source": [
    "# p97 Inhibitors\n",
    "\n",
    "Wang et al., 2022\n",
    "\n",
    "https://www.sciencedirect.com/science/article/pii/S2451945621004827\n",
    "\n",
    "shRNA knockdown of p97 in a colon cancer cell line."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "f3531c18-6ec8-479b-95c4-305243d14ce8",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 480/480 [00:00<00:00, 15036.27it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 338 entities from 270 out of 766 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_f39dd caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_f39dd\">\n",
       "  <caption>Table S1: P97 combined up/downregulated proteins, unweighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_f39dd_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_f39dd_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_f39dd_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_f39dd_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_f39dd_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_f39dd_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_f39dd_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_f39dd_row0_col0\" class=\"data row0 col0\" >PI5P, PP2A and IER3 Regulate PI3K/AKT Signaling - Reactome</td>\n",
       "      <td id=\"T_f39dd_row0_col1\" class=\"data row0 col1\" >0.000002</td>\n",
       "      <td id=\"T_f39dd_row0_col2\" class=\"data row0 col2\" >8</td>\n",
       "      <td id=\"T_f39dd_row0_col3\" class=\"data row0 col3\" >13</td>\n",
       "      <td id=\"T_f39dd_row0_col4\" class=\"data row0 col4\" >['set:PP2A-subunit A', 'set:PIP5K1A/B', 'set:PIP5K1A-C', 'set:Activator:PI3K', 'set:PP2A-catalytic subunit C', 'set:SYNJ/MTM(1)', 'set:PI4K2A/2B', 'set:Activated SRC,LCK,EGFR,INSR']</td>\n",
       "      <td id=\"T_f39dd_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-6811558</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f39dd_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_f39dd_row1_col0\" class=\"data row1 col0\" >Unwinding of DNA - Reactome</td>\n",
       "      <td id=\"T_f39dd_row1_col1\" class=\"data row1 col1\" >0.000002</td>\n",
       "      <td id=\"T_f39dd_row1_col2\" class=\"data row1 col2\" >6</td>\n",
       "      <td id=\"T_f39dd_row1_col3\" class=\"data row1 col3\" >7</td>\n",
       "      <td id=\"T_f39dd_row1_col4\" class=\"data row1 col4\" >['MCM3', 'MCM6', 'MCM4', 'MCM2', 'MCM7', 'MCM5']</td>\n",
       "      <td id=\"T_f39dd_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-176974</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x13bb68e20>"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "p97_step = enrich.enrich_wrapper('P97.csv','uniprot',FDR = 0.05, method='set', fpath = '../test_data/processed',table_number = 1)\n",
    "p97_step"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "9a235543-bcaa-4e61-a495-fa8098506509",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 480/480 [00:00<00:00, 16725.78it/s]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_0a6c1 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_0a6c1\">\n",
       "  <caption>Table S2: P97 combined up/downregulated proteins, standard method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_0a6c1_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_0a6c1_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_0a6c1_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_0a6c1_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_0a6c1_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_0a6c1_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_0a6c1_row0_col0\" class=\"data row0 col0\" >Unwinding of DNA - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row0_col1\" class=\"data row0 col1\" >0.000000</td>\n",
       "      <td id=\"T_0a6c1_row0_col2\" class=\"data row0 col2\" >6</td>\n",
       "      <td id=\"T_0a6c1_row0_col3\" class=\"data row0 col3\" >7</td>\n",
       "      <td id=\"T_0a6c1_row0_col4\" class=\"data row0 col4\" >['MCM3', 'MCM6', 'MCM4', 'MCM2', 'MCM7', 'MCM5']</td>\n",
       "      <td id=\"T_0a6c1_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-176974</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_0a6c1_row1_col0\" class=\"data row1 col0\" >SCF(Skp2)-mediated degradation of p27/p21 - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row1_col1\" class=\"data row1 col1\" >0.000001</td>\n",
       "      <td id=\"T_0a6c1_row1_col2\" class=\"data row1 col2\" >13</td>\n",
       "      <td id=\"T_0a6c1_row1_col3\" class=\"data row1 col3\" >50</td>\n",
       "      <td id=\"T_0a6c1_row1_col4\" class=\"data row1 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'CDK2', 'CDKN1A', 'PSMF1', 'CDKN1B', 'CKS1B']</td>\n",
       "      <td id=\"T_0a6c1_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-187577</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "      <td id=\"T_0a6c1_row2_col0\" class=\"data row2 col0\" >Autodegradation of Cdh1 by Cdh1:APC/C - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row2_col1\" class=\"data row2 col1\" >0.000005</td>\n",
       "      <td id=\"T_0a6c1_row2_col2\" class=\"data row2 col2\" >13</td>\n",
       "      <td id=\"T_0a6c1_row2_col3\" class=\"data row2 col3\" >59</td>\n",
       "      <td id=\"T_0a6c1_row2_col4\" class=\"data row2 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1', 'UBE2S', 'UBE2C', 'UBE2E1', 'ANAPC2']</td>\n",
       "      <td id=\"T_0a6c1_row2_col5\" class=\"data row2 col5\" >http://model.geneontology.org/R-HSA-174084</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row3\" class=\"row_heading level0 row3\" >3</th>\n",
       "      <td id=\"T_0a6c1_row3_col0\" class=\"data row3 col0\" >CDK-mediated phosphorylation and removal of Cdc6 - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row3_col1\" class=\"data row3 col1\" >0.000005</td>\n",
       "      <td id=\"T_0a6c1_row3_col2\" class=\"data row3 col2\" >14</td>\n",
       "      <td id=\"T_0a6c1_row3_col3\" class=\"data row3 col3\" >68</td>\n",
       "      <td id=\"T_0a6c1_row3_col4\" class=\"data row3 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'CDK2', 'PSMF1', 'UBE2S', 'UBE2C', 'UBE2E1', 'ANAPC2']</td>\n",
       "      <td id=\"T_0a6c1_row3_col5\" class=\"data row3 col5\" >http://model.geneontology.org/R-HSA-69017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row4\" class=\"row_heading level0 row4\" >4</th>\n",
       "      <td id=\"T_0a6c1_row4_col0\" class=\"data row4 col0\" >Assembly of the pre-replicative complex - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row4_col1\" class=\"data row4 col1\" >0.000006</td>\n",
       "      <td id=\"T_0a6c1_row4_col2\" class=\"data row4 col2\" >13</td>\n",
       "      <td id=\"T_0a6c1_row4_col3\" class=\"data row4 col3\" >60</td>\n",
       "      <td id=\"T_0a6c1_row4_col4\" class=\"data row4 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1', 'UBE2S', 'UBE2C', 'UBE2E1', 'ANAPC2']</td>\n",
       "      <td id=\"T_0a6c1_row4_col5\" class=\"data row4 col5\" >http://model.geneontology.org/R-HSA-68867</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row5\" class=\"row_heading level0 row5\" >5</th>\n",
       "      <td id=\"T_0a6c1_row5_col0\" class=\"data row5 col0\" >APC/C:Cdh1 mediated degradation of Cdc20 and other APC/C:Cdh1 targeted proteins in late mitosis/early G1 - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row5_col1\" class=\"data row5 col1\" >0.000009</td>\n",
       "      <td id=\"T_0a6c1_row5_col2\" class=\"data row5 col2\" >13</td>\n",
       "      <td id=\"T_0a6c1_row5_col3\" class=\"data row5 col3\" >62</td>\n",
       "      <td id=\"T_0a6c1_row5_col4\" class=\"data row5 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1', 'UBE2S', 'UBE2C', 'UBE2E1', 'ANAPC2']</td>\n",
       "      <td id=\"T_0a6c1_row5_col5\" class=\"data row5 col5\" >http://model.geneontology.org/R-HSA-174178</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row6\" class=\"row_heading level0 row6\" >6</th>\n",
       "      <td id=\"T_0a6c1_row6_col0\" class=\"data row6 col0\" >APC/C:Cdc20 mediated degradation of Securin - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row6_col1\" class=\"data row6 col1\" >0.000009</td>\n",
       "      <td id=\"T_0a6c1_row6_col2\" class=\"data row6 col2\" >13</td>\n",
       "      <td id=\"T_0a6c1_row6_col3\" class=\"data row6 col3\" >62</td>\n",
       "      <td id=\"T_0a6c1_row6_col4\" class=\"data row6 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1', 'UBE2S', 'UBE2C', 'UBE2E1', 'ANAPC2']</td>\n",
       "      <td id=\"T_0a6c1_row6_col5\" class=\"data row6 col5\" >http://model.geneontology.org/R-HSA-174154</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row7\" class=\"row_heading level0 row7\" >7</th>\n",
       "      <td id=\"T_0a6c1_row7_col0\" class=\"data row7 col0\" >Cdc20:Phospho-APC/C mediated degradation of Cyclin A - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row7_col1\" class=\"data row7 col1\" >0.000009</td>\n",
       "      <td id=\"T_0a6c1_row7_col2\" class=\"data row7 col2\" >13</td>\n",
       "      <td id=\"T_0a6c1_row7_col3\" class=\"data row7 col3\" >62</td>\n",
       "      <td id=\"T_0a6c1_row7_col4\" class=\"data row7 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1', 'UBE2S', 'UBE2C', 'UBE2E1', 'ANAPC2']</td>\n",
       "      <td id=\"T_0a6c1_row7_col5\" class=\"data row7 col5\" >http://model.geneontology.org/R-HSA-174184</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row8\" class=\"row_heading level0 row8\" >8</th>\n",
       "      <td id=\"T_0a6c1_row8_col0\" class=\"data row8 col0\" >Separation of Sister Chromatids - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row8_col1\" class=\"data row8 col1\" >0.000010</td>\n",
       "      <td id=\"T_0a6c1_row8_col2\" class=\"data row8 col2\" >13</td>\n",
       "      <td id=\"T_0a6c1_row8_col3\" class=\"data row8 col3\" >63</td>\n",
       "      <td id=\"T_0a6c1_row8_col4\" class=\"data row8 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1', 'UBE2S', 'UBE2C', 'UBE2E1', 'ANAPC2']</td>\n",
       "      <td id=\"T_0a6c1_row8_col5\" class=\"data row8 col5\" >http://model.geneontology.org/R-HSA-2467813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row9\" class=\"row_heading level0 row9\" >9</th>\n",
       "      <td id=\"T_0a6c1_row9_col0\" class=\"data row9 col0\" >Antigen processing: Ubiquitination & Proteasome degradation - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row9_col1\" class=\"data row9 col1\" >0.000017</td>\n",
       "      <td id=\"T_0a6c1_row9_col2\" class=\"data row9 col2\" >33</td>\n",
       "      <td id=\"T_0a6c1_row9_col3\" class=\"data row9 col3\" >305</td>\n",
       "      <td id=\"T_0a6c1_row9_col4\" class=\"data row9 col4\" >['HUWE1', 'HECTD1', 'PSMD2', 'TRIP12', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'UBE3A', 'PSMD14', 'FBXO22', 'UBE2J1', 'PSMF1', 'UBE2S', 'UBE2D2', 'UBE2E2', 'UBE2L6', 'UBE2C', 'RNF7', 'UBE2H', 'UBE2A', 'UBE2E1', 'UBE2R2', 'KEAP1', 'ANAPC2', 'UBE2J2', 'CDC20', 'UBE2W', 'UBAC1', 'ASB6', 'KLHL25']</td>\n",
       "      <td id=\"T_0a6c1_row9_col5\" class=\"data row9 col5\" >http://model.geneontology.org/R-HSA-983168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row10\" class=\"row_heading level0 row10\" >10</th>\n",
       "      <td id=\"T_0a6c1_row10_col0\" class=\"data row10 col0\" >Cyclin A:Cdk2-associated events at S phase entry - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row10_col1\" class=\"data row10 col1\" >0.000018</td>\n",
       "      <td id=\"T_0a6c1_row10_col2\" class=\"data row10 col2\" >12</td>\n",
       "      <td id=\"T_0a6c1_row10_col3\" class=\"data row10 col3\" >57</td>\n",
       "      <td id=\"T_0a6c1_row10_col4\" class=\"data row10 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'CDK2', 'CDKN1A', 'PSMF1', 'CDKN1B']</td>\n",
       "      <td id=\"T_0a6c1_row10_col5\" class=\"data row10 col5\" >http://model.geneontology.org/R-HSA-69656</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row11\" class=\"row_heading level0 row11\" >11</th>\n",
       "      <td id=\"T_0a6c1_row11_col0\" class=\"data row11 col0\" >Conversion from APC/C:Cdc20 to APC/C:Cdh1 in late anaphase - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row11_col1\" class=\"data row11 col1\" >0.000022</td>\n",
       "      <td id=\"T_0a6c1_row11_col2\" class=\"data row11 col2\" >11</td>\n",
       "      <td id=\"T_0a6c1_row11_col3\" class=\"data row11 col3\" >49</td>\n",
       "      <td id=\"T_0a6c1_row11_col4\" class=\"data row11 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'CDK1', 'PSMD14', 'CDK2', 'PSMF1']</td>\n",
       "      <td id=\"T_0a6c1_row11_col5\" class=\"data row11 col5\" >http://model.geneontology.org/R-HSA-176407</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row12\" class=\"row_heading level0 row12\" >12</th>\n",
       "      <td id=\"T_0a6c1_row12_col0\" class=\"data row12 col0\" >The role of GTSE1 in G2/M progression after G2 checkpoint - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row12_col1\" class=\"data row12 col1\" >0.000027</td>\n",
       "      <td id=\"T_0a6c1_row12_col2\" class=\"data row12 col2\" >11</td>\n",
       "      <td id=\"T_0a6c1_row12_col3\" class=\"data row12 col3\" >50</td>\n",
       "      <td id=\"T_0a6c1_row12_col4\" class=\"data row12 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1', 'PLK1', 'TP53']</td>\n",
       "      <td id=\"T_0a6c1_row12_col5\" class=\"data row12 col5\" >http://model.geneontology.org/R-HSA-8852276</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row13\" class=\"row_heading level0 row13\" >13</th>\n",
       "      <td id=\"T_0a6c1_row13_col0\" class=\"data row13 col0\" >SCF-beta-TrCP mediated degradation of Emi1 - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row13_col1\" class=\"data row13 col1\" >0.000032</td>\n",
       "      <td id=\"T_0a6c1_row13_col2\" class=\"data row13 col2\" >11</td>\n",
       "      <td id=\"T_0a6c1_row13_col3\" class=\"data row13 col3\" >51</td>\n",
       "      <td id=\"T_0a6c1_row13_col4\" class=\"data row13 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'CDK1', 'PSMD14', 'PSMF1', 'PLK1']</td>\n",
       "      <td id=\"T_0a6c1_row13_col5\" class=\"data row13 col5\" >http://model.geneontology.org/R-HSA-174113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row14\" class=\"row_heading level0 row14\" >14</th>\n",
       "      <td id=\"T_0a6c1_row14_col0\" class=\"data row14 col0\" >ER-Phagosome pathway - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row14_col1\" class=\"data row14 col1\" >0.000039</td>\n",
       "      <td id=\"T_0a6c1_row14_col2\" class=\"data row14 col2\" >11</td>\n",
       "      <td id=\"T_0a6c1_row14_col3\" class=\"data row14 col3\" >52</td>\n",
       "      <td id=\"T_0a6c1_row14_col4\" class=\"data row14 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1', 'SEC61A1', 'SEC61G']</td>\n",
       "      <td id=\"T_0a6c1_row14_col5\" class=\"data row14 col5\" >http://model.geneontology.org/R-HSA-1236974</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row15\" class=\"row_heading level0 row15\" >15</th>\n",
       "      <td id=\"T_0a6c1_row15_col0\" class=\"data row15 col0\" >Regulation of APC/C activators between G1/S and early anaphase - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row15_col1\" class=\"data row15 col1\" >0.000057</td>\n",
       "      <td id=\"T_0a6c1_row15_col2\" class=\"data row15 col2\" >11</td>\n",
       "      <td id=\"T_0a6c1_row15_col3\" class=\"data row15 col3\" >54</td>\n",
       "      <td id=\"T_0a6c1_row15_col4\" class=\"data row15 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'CDK1', 'PSMD14', 'CDK2', 'PSMF1']</td>\n",
       "      <td id=\"T_0a6c1_row15_col5\" class=\"data row15 col5\" >http://model.geneontology.org/R-HSA-176408</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row16\" class=\"row_heading level0 row16\" >16</th>\n",
       "      <td id=\"T_0a6c1_row16_col0\" class=\"data row16 col0\" >G2/M Checkpoints - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row16_col1\" class=\"data row16 col1\" >0.000102</td>\n",
       "      <td id=\"T_0a6c1_row16_col2\" class=\"data row16 col2\" >10</td>\n",
       "      <td id=\"T_0a6c1_row16_col3\" class=\"data row16 col3\" >48</td>\n",
       "      <td id=\"T_0a6c1_row16_col4\" class=\"data row16 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1', 'TP53']</td>\n",
       "      <td id=\"T_0a6c1_row16_col5\" class=\"data row16 col5\" >http://model.geneontology.org/R-HSA-69481</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row17\" class=\"row_heading level0 row17\" >17</th>\n",
       "      <td id=\"T_0a6c1_row17_col0\" class=\"data row17 col0\" >Activation of NF-kappaB in B cells - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row17_col1\" class=\"data row17 col1\" >0.000114</td>\n",
       "      <td id=\"T_0a6c1_row17_col2\" class=\"data row17 col2\" >11</td>\n",
       "      <td id=\"T_0a6c1_row17_col3\" class=\"data row17 col3\" >58</td>\n",
       "      <td id=\"T_0a6c1_row17_col4\" class=\"data row17 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'CHUK', 'PSMD14', 'PSMF1', 'MAP3K7']</td>\n",
       "      <td id=\"T_0a6c1_row17_col5\" class=\"data row17 col5\" >http://model.geneontology.org/R-HSA-1169091</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row18\" class=\"row_heading level0 row18\" >18</th>\n",
       "      <td id=\"T_0a6c1_row18_col0\" class=\"data row18 col0\" >Cyclin E associated events during G1/S transition - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row18_col1\" class=\"data row18 col1\" >0.000133</td>\n",
       "      <td id=\"T_0a6c1_row18_col2\" class=\"data row18 col2\" >11</td>\n",
       "      <td id=\"T_0a6c1_row18_col3\" class=\"data row18 col3\" >59</td>\n",
       "      <td id=\"T_0a6c1_row18_col4\" class=\"data row18 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'CDKN1A', 'PSMF1', 'CDKN1B']</td>\n",
       "      <td id=\"T_0a6c1_row18_col5\" class=\"data row18 col5\" >http://model.geneontology.org/R-HSA-69202</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row19\" class=\"row_heading level0 row19\" >19</th>\n",
       "      <td id=\"T_0a6c1_row19_col0\" class=\"data row19 col0\" >Degradation of beta-catenin by the destruction complex - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row19_col1\" class=\"data row19 col1\" >0.000133</td>\n",
       "      <td id=\"T_0a6c1_row19_col2\" class=\"data row19 col2\" >11</td>\n",
       "      <td id=\"T_0a6c1_row19_col3\" class=\"data row19 col3\" >59</td>\n",
       "      <td id=\"T_0a6c1_row19_col4\" class=\"data row19 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PPP2R1B', 'PPP2CB', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_0a6c1_row19_col5\" class=\"data row19 col5\" >http://model.geneontology.org/R-HSA-195253</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row20\" class=\"row_heading level0 row20\" >20</th>\n",
       "      <td id=\"T_0a6c1_row20_col0\" class=\"data row20 col0\" >NIK-->noncanonical NF-kB signaling - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row20_col1\" class=\"data row20 col1\" >0.000174</td>\n",
       "      <td id=\"T_0a6c1_row20_col2\" class=\"data row20 col2\" >10</td>\n",
       "      <td id=\"T_0a6c1_row20_col3\" class=\"data row20 col3\" >51</td>\n",
       "      <td id=\"T_0a6c1_row20_col4\" class=\"data row20 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'CHUK', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_0a6c1_row20_col5\" class=\"data row20 col5\" >http://model.geneontology.org/R-HSA-5676590</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row21\" class=\"row_heading level0 row21\" >21</th>\n",
       "      <td id=\"T_0a6c1_row21_col0\" class=\"data row21 col0\" >Regulation of expression of SLITs and ROBOs - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row21_col1\" class=\"data row21 col1\" >0.000174</td>\n",
       "      <td id=\"T_0a6c1_row21_col2\" class=\"data row21 col2\" >10</td>\n",
       "      <td id=\"T_0a6c1_row21_col3\" class=\"data row21 col3\" >51</td>\n",
       "      <td id=\"T_0a6c1_row21_col4\" class=\"data row21 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1', 'USP33']</td>\n",
       "      <td id=\"T_0a6c1_row21_col5\" class=\"data row21 col5\" >http://model.geneontology.org/R-HSA-9010553</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row22\" class=\"row_heading level0 row22\" >22</th>\n",
       "      <td id=\"T_0a6c1_row22_col0\" class=\"data row22 col0\" >Dectin-1 mediated noncanonical NF-kB signaling - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row22_col1\" class=\"data row22 col1\" >0.000174</td>\n",
       "      <td id=\"T_0a6c1_row22_col2\" class=\"data row22 col2\" >10</td>\n",
       "      <td id=\"T_0a6c1_row22_col3\" class=\"data row22 col3\" >51</td>\n",
       "      <td id=\"T_0a6c1_row22_col4\" class=\"data row22 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'CHUK', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_0a6c1_row22_col5\" class=\"data row22 col5\" >http://model.geneontology.org/R-HSA-5607761</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row23\" class=\"row_heading level0 row23\" >23</th>\n",
       "      <td id=\"T_0a6c1_row23_col0\" class=\"data row23 col0\" >Hedgehog ligand biogenesis - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row23_col1\" class=\"data row23 col1\" >0.000206</td>\n",
       "      <td id=\"T_0a6c1_row23_col2\" class=\"data row23 col2\" >10</td>\n",
       "      <td id=\"T_0a6c1_row23_col3\" class=\"data row23 col3\" >52</td>\n",
       "      <td id=\"T_0a6c1_row23_col4\" class=\"data row23 col4\" >['VCP', 'PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_0a6c1_row23_col5\" class=\"data row23 col5\" >http://model.geneontology.org/R-HSA-5358346</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row24\" class=\"row_heading level0 row24\" >24</th>\n",
       "      <td id=\"T_0a6c1_row24_col0\" class=\"data row24 col0\" >Orc1 removal from chromatin - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row24_col1\" class=\"data row24 col1\" >0.000206</td>\n",
       "      <td id=\"T_0a6c1_row24_col2\" class=\"data row24 col2\" >10</td>\n",
       "      <td id=\"T_0a6c1_row24_col3\" class=\"data row24 col3\" >52</td>\n",
       "      <td id=\"T_0a6c1_row24_col4\" class=\"data row24 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'CDK2', 'PSMF1']</td>\n",
       "      <td id=\"T_0a6c1_row24_col5\" class=\"data row24 col5\" >http://model.geneontology.org/R-HSA-68949</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row25\" class=\"row_heading level0 row25\" >25</th>\n",
       "      <td id=\"T_0a6c1_row25_col0\" class=\"data row25 col0\" >Resolution of Sister Chromatid Cohesion - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row25_col1\" class=\"data row25 col1\" >0.000226</td>\n",
       "      <td id=\"T_0a6c1_row25_col2\" class=\"data row25 col2\" >15</td>\n",
       "      <td id=\"T_0a6c1_row25_col3\" class=\"data row25 col3\" >106</td>\n",
       "      <td id=\"T_0a6c1_row25_col4\" class=\"data row25 col4\" >['PPP2R1B', 'PPP2CB', 'CDK1', 'SEC13', 'TAOK1', 'KIF2C', 'MAD2L1', 'BUB1B', 'SPC24', 'DSN1', 'PLK1', 'NDE1', 'CDC20', 'INCENP', 'NSL1']</td>\n",
       "      <td id=\"T_0a6c1_row25_col5\" class=\"data row25 col5\" >http://model.geneontology.org/R-HSA-2500257</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row26\" class=\"row_heading level0 row26\" >26</th>\n",
       "      <td id=\"T_0a6c1_row26_col0\" class=\"data row26 col0\" >GSK3B and BTRC:CUL1-mediated-degradation of NFE2L2 - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row26_col1\" class=\"data row26 col1\" >0.000264</td>\n",
       "      <td id=\"T_0a6c1_row26_col2\" class=\"data row26 col2\" >9</td>\n",
       "      <td id=\"T_0a6c1_row26_col3\" class=\"data row26 col3\" >44</td>\n",
       "      <td id=\"T_0a6c1_row26_col4\" class=\"data row26 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_0a6c1_row26_col5\" class=\"data row26 col5\" >http://model.geneontology.org/R-HSA-9762114</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row27\" class=\"row_heading level0 row27\" >27</th>\n",
       "      <td id=\"T_0a6c1_row27_col0\" class=\"data row27 col0\" >AUF1 (hnRNP D0) binds and destabilizes mRNA - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row27_col1\" class=\"data row27 col1\" >0.000315</td>\n",
       "      <td id=\"T_0a6c1_row27_col2\" class=\"data row27 col2\" >9</td>\n",
       "      <td id=\"T_0a6c1_row27_col3\" class=\"data row27 col3\" >45</td>\n",
       "      <td id=\"T_0a6c1_row27_col4\" class=\"data row27 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_0a6c1_row27_col5\" class=\"data row27 col5\" >http://model.geneontology.org/R-HSA-450408</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row28\" class=\"row_heading level0 row28\" >28</th>\n",
       "      <td id=\"T_0a6c1_row28_col0\" class=\"data row28 col0\" >APC/C:Cdc20 mediated degradation of mitotic proteins - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row28_col1\" class=\"data row28 col1\" >0.000315</td>\n",
       "      <td id=\"T_0a6c1_row28_col2\" class=\"data row28 col2\" >9</td>\n",
       "      <td id=\"T_0a6c1_row28_col3\" class=\"data row28 col3\" >45</td>\n",
       "      <td id=\"T_0a6c1_row28_col4\" class=\"data row28 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_0a6c1_row28_col5\" class=\"data row28 col5\" >http://model.geneontology.org/R-HSA-176409</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row29\" class=\"row_heading level0 row29\" >29</th>\n",
       "      <td id=\"T_0a6c1_row29_col0\" class=\"data row29 col0\" >Cross-presentation of soluble exogenous antigens (endosomes) - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row29_col1\" class=\"data row29 col1\" >0.000315</td>\n",
       "      <td id=\"T_0a6c1_row29_col2\" class=\"data row29 col2\" >9</td>\n",
       "      <td id=\"T_0a6c1_row29_col3\" class=\"data row29 col3\" >45</td>\n",
       "      <td id=\"T_0a6c1_row29_col4\" class=\"data row29 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_0a6c1_row29_col5\" class=\"data row29 col5\" >http://model.geneontology.org/R-HSA-1236978</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row30\" class=\"row_heading level0 row30\" >30</th>\n",
       "      <td id=\"T_0a6c1_row30_col0\" class=\"data row30 col0\" >Regulation of RUNX3 expression and activity - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row30_col1\" class=\"data row30 col1\" >0.000375</td>\n",
       "      <td id=\"T_0a6c1_row30_col2\" class=\"data row30 col2\" >9</td>\n",
       "      <td id=\"T_0a6c1_row30_col3\" class=\"data row30 col3\" >46</td>\n",
       "      <td id=\"T_0a6c1_row30_col4\" class=\"data row30 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_0a6c1_row30_col5\" class=\"data row30 col5\" >http://model.geneontology.org/R-HSA-8941858</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row31\" class=\"row_heading level0 row31\" >31</th>\n",
       "      <td id=\"T_0a6c1_row31_col0\" class=\"data row31 col0\" >Ubiquitin-dependent degradation of Cyclin D - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row31_col1\" class=\"data row31 col1\" >0.000375</td>\n",
       "      <td id=\"T_0a6c1_row31_col2\" class=\"data row31 col2\" >9</td>\n",
       "      <td id=\"T_0a6c1_row31_col3\" class=\"data row31 col3\" >46</td>\n",
       "      <td id=\"T_0a6c1_row31_col4\" class=\"data row31 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_0a6c1_row31_col5\" class=\"data row31 col5\" >http://model.geneontology.org/R-HSA-75815</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row32\" class=\"row_heading level0 row32\" >32</th>\n",
       "      <td id=\"T_0a6c1_row32_col0\" class=\"data row32 col0\" >Regulation of activated PAK-2p34 by proteasome mediated degradation - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row32_col1\" class=\"data row32 col1\" >0.000375</td>\n",
       "      <td id=\"T_0a6c1_row32_col2\" class=\"data row32 col2\" >9</td>\n",
       "      <td id=\"T_0a6c1_row32_col3\" class=\"data row32 col3\" >46</td>\n",
       "      <td id=\"T_0a6c1_row32_col4\" class=\"data row32 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_0a6c1_row32_col5\" class=\"data row32 col5\" >http://model.geneontology.org/R-HSA-211733</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row33\" class=\"row_heading level0 row33\" >33</th>\n",
       "      <td id=\"T_0a6c1_row33_col0\" class=\"data row33 col0\" >FCERI mediated NF-kB activation - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row33_col1\" class=\"data row33 col1\" >0.000389</td>\n",
       "      <td id=\"T_0a6c1_row33_col2\" class=\"data row33 col2\" >10</td>\n",
       "      <td id=\"T_0a6c1_row33_col3\" class=\"data row33 col3\" >56</td>\n",
       "      <td id=\"T_0a6c1_row33_col4\" class=\"data row33 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1', 'MAP3K7']</td>\n",
       "      <td id=\"T_0a6c1_row33_col5\" class=\"data row33 col5\" >http://model.geneontology.org/R-HSA-2871837</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row34\" class=\"row_heading level0 row34\" >34</th>\n",
       "      <td id=\"T_0a6c1_row34_col0\" class=\"data row34 col0\" >CLEC7A (Dectin-1) signaling - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row34_col1\" class=\"data row34 col1\" >0.000389</td>\n",
       "      <td id=\"T_0a6c1_row34_col2\" class=\"data row34 col2\" >10</td>\n",
       "      <td id=\"T_0a6c1_row34_col3\" class=\"data row34 col3\" >56</td>\n",
       "      <td id=\"T_0a6c1_row34_col4\" class=\"data row34 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1', 'MAP3K7']</td>\n",
       "      <td id=\"T_0a6c1_row34_col5\" class=\"data row34 col5\" >http://model.geneontology.org/R-HSA-5607764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row35\" class=\"row_heading level0 row35\" >35</th>\n",
       "      <td id=\"T_0a6c1_row35_col0\" class=\"data row35 col0\" >Asymmetric localization of PCP proteins - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row35_col1\" class=\"data row35 col1\" >0.000444</td>\n",
       "      <td id=\"T_0a6c1_row35_col2\" class=\"data row35 col2\" >9</td>\n",
       "      <td id=\"T_0a6c1_row35_col3\" class=\"data row35 col3\" >47</td>\n",
       "      <td id=\"T_0a6c1_row35_col4\" class=\"data row35 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_0a6c1_row35_col5\" class=\"data row35 col5\" >http://model.geneontology.org/R-HSA-4608870</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row36\" class=\"row_heading level0 row36\" >36</th>\n",
       "      <td id=\"T_0a6c1_row36_col0\" class=\"data row36 col0\" >TNFR2 non-canonical NF-kB pathway - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row36_col1\" class=\"data row36 col1\" >0.000444</td>\n",
       "      <td id=\"T_0a6c1_row36_col2\" class=\"data row36 col2\" >9</td>\n",
       "      <td id=\"T_0a6c1_row36_col3\" class=\"data row36 col3\" >47</td>\n",
       "      <td id=\"T_0a6c1_row36_col4\" class=\"data row36 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_0a6c1_row36_col5\" class=\"data row36 col5\" >http://model.geneontology.org/R-HSA-5668541</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row37\" class=\"row_heading level0 row37\" >37</th>\n",
       "      <td id=\"T_0a6c1_row37_col0\" class=\"data row37 col0\" >Autodegradation of the E3 ubiquitin ligase COP1 - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row37_col1\" class=\"data row37 col1\" >0.000444</td>\n",
       "      <td id=\"T_0a6c1_row37_col2\" class=\"data row37 col2\" >9</td>\n",
       "      <td id=\"T_0a6c1_row37_col3\" class=\"data row37 col3\" >47</td>\n",
       "      <td id=\"T_0a6c1_row37_col4\" class=\"data row37 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_0a6c1_row37_col5\" class=\"data row37 col5\" >http://model.geneontology.org/R-HSA-349425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row38\" class=\"row_heading level0 row38\" >38</th>\n",
       "      <td id=\"T_0a6c1_row38_col0\" class=\"data row38 col0\" >Degradation of DVL - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row38_col1\" class=\"data row38 col1\" >0.000444</td>\n",
       "      <td id=\"T_0a6c1_row38_col2\" class=\"data row38 col2\" >9</td>\n",
       "      <td id=\"T_0a6c1_row38_col3\" class=\"data row38 col3\" >47</td>\n",
       "      <td id=\"T_0a6c1_row38_col4\" class=\"data row38 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_0a6c1_row38_col5\" class=\"data row38 col5\" >http://model.geneontology.org/R-HSA-4641258</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row39\" class=\"row_heading level0 row39\" >39</th>\n",
       "      <td id=\"T_0a6c1_row39_col0\" class=\"data row39 col0\" >KEAP1-NFE2L2 pathway - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row39_col1\" class=\"data row39 col1\" >0.000451</td>\n",
       "      <td id=\"T_0a6c1_row39_col2\" class=\"data row39 col2\" >10</td>\n",
       "      <td id=\"T_0a6c1_row39_col3\" class=\"data row39 col3\" >57</td>\n",
       "      <td id=\"T_0a6c1_row39_col4\" class=\"data row39 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'CSNK2A1', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_0a6c1_row39_col5\" class=\"data row39 col5\" >http://model.geneontology.org/R-HSA-9755511</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row40\" class=\"row_heading level0 row40\" >40</th>\n",
       "      <td id=\"T_0a6c1_row40_col0\" class=\"data row40 col0\" >RUNX1 regulates transcription of genes involved in differentiation of HSCs - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row40_col1\" class=\"data row40 col1\" >0.000613</td>\n",
       "      <td id=\"T_0a6c1_row40_col2\" class=\"data row40 col2\" >9</td>\n",
       "      <td id=\"T_0a6c1_row40_col3\" class=\"data row40 col3\" >49</td>\n",
       "      <td id=\"T_0a6c1_row40_col4\" class=\"data row40 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_0a6c1_row40_col5\" class=\"data row40 col5\" >http://model.geneontology.org/R-HSA-8939236</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row41\" class=\"row_heading level0 row41\" >41</th>\n",
       "      <td id=\"T_0a6c1_row41_col0\" class=\"data row41 col0\" >Degradation of AXIN - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row41_col1\" class=\"data row41 col1\" >0.000613</td>\n",
       "      <td id=\"T_0a6c1_row41_col2\" class=\"data row41 col2\" >9</td>\n",
       "      <td id=\"T_0a6c1_row41_col3\" class=\"data row41 col3\" >49</td>\n",
       "      <td id=\"T_0a6c1_row41_col4\" class=\"data row41 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_0a6c1_row41_col5\" class=\"data row41 col5\" >http://model.geneontology.org/R-HSA-4641257</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row42\" class=\"row_heading level0 row42\" >42</th>\n",
       "      <td id=\"T_0a6c1_row42_col0\" class=\"data row42 col0\" >Oxygen-dependent proline hydroxylation of Hypoxia-inducible Factor Alpha - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row42_col1\" class=\"data row42 col1\" >0.000613</td>\n",
       "      <td id=\"T_0a6c1_row42_col2\" class=\"data row42 col2\" >9</td>\n",
       "      <td id=\"T_0a6c1_row42_col3\" class=\"data row42 col3\" >49</td>\n",
       "      <td id=\"T_0a6c1_row42_col4\" class=\"data row42 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_0a6c1_row42_col5\" class=\"data row42 col5\" >http://model.geneontology.org/R-HSA-1234176</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row43\" class=\"row_heading level0 row43\" >43</th>\n",
       "      <td id=\"T_0a6c1_row43_col0\" class=\"data row43 col0\" >Ubiquitin Mediated Degradation of Phosphorylated Cdc25A - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row43_col1\" class=\"data row43 col1\" >0.000613</td>\n",
       "      <td id=\"T_0a6c1_row43_col2\" class=\"data row43 col2\" >9</td>\n",
       "      <td id=\"T_0a6c1_row43_col3\" class=\"data row43 col3\" >49</td>\n",
       "      <td id=\"T_0a6c1_row43_col4\" class=\"data row43 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_0a6c1_row43_col5\" class=\"data row43 col5\" >http://model.geneontology.org/R-HSA-69601</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row44\" class=\"row_heading level0 row44\" >44</th>\n",
       "      <td id=\"T_0a6c1_row44_col0\" class=\"data row44 col0\" >FBXL7 down-regulates AURKA during mitotic entry and in early mitosis - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row44_col1\" class=\"data row44 col1\" >0.000715</td>\n",
       "      <td id=\"T_0a6c1_row44_col2\" class=\"data row44 col2\" >9</td>\n",
       "      <td id=\"T_0a6c1_row44_col3\" class=\"data row44 col3\" >50</td>\n",
       "      <td id=\"T_0a6c1_row44_col4\" class=\"data row44 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_0a6c1_row44_col5\" class=\"data row44 col5\" >http://model.geneontology.org/R-HSA-8854050</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row45\" class=\"row_heading level0 row45\" >45</th>\n",
       "      <td id=\"T_0a6c1_row45_col0\" class=\"data row45 col0\" >Hedgehog 'on' state - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row45_col1\" class=\"data row45 col1\" >0.000715</td>\n",
       "      <td id=\"T_0a6c1_row45_col2\" class=\"data row45 col2\" >9</td>\n",
       "      <td id=\"T_0a6c1_row45_col3\" class=\"data row45 col3\" >50</td>\n",
       "      <td id=\"T_0a6c1_row45_col4\" class=\"data row45 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_0a6c1_row45_col5\" class=\"data row45 col5\" >http://model.geneontology.org/R-HSA-5632684</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row46\" class=\"row_heading level0 row46\" >46</th>\n",
       "      <td id=\"T_0a6c1_row46_col0\" class=\"data row46 col0\" >Negative regulation of NOTCH4 signaling - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row46_col1\" class=\"data row46 col1\" >0.000831</td>\n",
       "      <td id=\"T_0a6c1_row46_col2\" class=\"data row46 col2\" >9</td>\n",
       "      <td id=\"T_0a6c1_row46_col3\" class=\"data row46 col3\" >51</td>\n",
       "      <td id=\"T_0a6c1_row46_col4\" class=\"data row46 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_0a6c1_row46_col5\" class=\"data row46 col5\" >http://model.geneontology.org/R-HSA-9604323</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row47\" class=\"row_heading level0 row47\" >47</th>\n",
       "      <td id=\"T_0a6c1_row47_col0\" class=\"data row47 col0\" >Senescence-Associated Secretory Phenotype (SASP) - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row47_col1\" class=\"data row47 col1\" >0.000840</td>\n",
       "      <td id=\"T_0a6c1_row47_col2\" class=\"data row47 col2\" >7</td>\n",
       "      <td id=\"T_0a6c1_row47_col3\" class=\"data row47 col3\" >32</td>\n",
       "      <td id=\"T_0a6c1_row47_col4\" class=\"data row47 col4\" >['CDK2', 'CDKN1A', 'UBE2S', 'UBE2C', 'CDKN1B', 'UBE2E1', 'ANAPC2']</td>\n",
       "      <td id=\"T_0a6c1_row47_col5\" class=\"data row47 col5\" >http://model.geneontology.org/R-HSA-2559582</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row48\" class=\"row_heading level0 row48\" >48</th>\n",
       "      <td id=\"T_0a6c1_row48_col0\" class=\"data row48 col0\" >Regulation of RAS by GAPs - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row48_col1\" class=\"data row48 col1\" >0.001024</td>\n",
       "      <td id=\"T_0a6c1_row48_col2\" class=\"data row48 col2\" >10</td>\n",
       "      <td id=\"T_0a6c1_row48_col3\" class=\"data row48 col3\" >63</td>\n",
       "      <td id=\"T_0a6c1_row48_col4\" class=\"data row48 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'NRAS', 'PSMF1']</td>\n",
       "      <td id=\"T_0a6c1_row48_col5\" class=\"data row48 col5\" >http://model.geneontology.org/R-HSA-5658442</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row49\" class=\"row_heading level0 row49\" >49</th>\n",
       "      <td id=\"T_0a6c1_row49_col0\" class=\"data row49 col0\" >Degradation of GLI1 by the proteasome - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row49_col1\" class=\"data row49 col1\" >0.001109</td>\n",
       "      <td id=\"T_0a6c1_row49_col2\" class=\"data row49 col2\" >9</td>\n",
       "      <td id=\"T_0a6c1_row49_col3\" class=\"data row49 col3\" >53</td>\n",
       "      <td id=\"T_0a6c1_row49_col4\" class=\"data row49 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_0a6c1_row49_col5\" class=\"data row49 col5\" >http://model.geneontology.org/R-HSA-5610780</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row50\" class=\"row_heading level0 row50\" >50</th>\n",
       "      <td id=\"T_0a6c1_row50_col0\" class=\"data row50 col0\" >ABC-family proteins mediated transport - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row50_col1\" class=\"data row50 col1\" >0.001161</td>\n",
       "      <td id=\"T_0a6c1_row50_col2\" class=\"data row50 col2\" >10</td>\n",
       "      <td id=\"T_0a6c1_row50_col3\" class=\"data row50 col3\" >64</td>\n",
       "      <td id=\"T_0a6c1_row50_col4\" class=\"data row50 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'ABCC4', 'PSMF1']</td>\n",
       "      <td id=\"T_0a6c1_row50_col5\" class=\"data row50 col5\" >http://model.geneontology.org/R-HSA-382556</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row51\" class=\"row_heading level0 row51\" >51</th>\n",
       "      <td id=\"T_0a6c1_row51_col0\" class=\"data row51 col0\" >Regulation of PTEN stability and activity - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row51_col1\" class=\"data row51 col1\" >0.001161</td>\n",
       "      <td id=\"T_0a6c1_row51_col2\" class=\"data row51 col2\" >10</td>\n",
       "      <td id=\"T_0a6c1_row51_col3\" class=\"data row51 col3\" >64</td>\n",
       "      <td id=\"T_0a6c1_row51_col4\" class=\"data row51 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'CSNK2A1', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_0a6c1_row51_col5\" class=\"data row51 col5\" >http://model.geneontology.org/R-HSA-8948751</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row52\" class=\"row_heading level0 row52\" >52</th>\n",
       "      <td id=\"T_0a6c1_row52_col0\" class=\"data row52 col0\" >Neddylation - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row52_col1\" class=\"data row52 col1\" >0.001161</td>\n",
       "      <td id=\"T_0a6c1_row52_col2\" class=\"data row52 col2\" >10</td>\n",
       "      <td id=\"T_0a6c1_row52_col3\" class=\"data row52 col3\" >64</td>\n",
       "      <td id=\"T_0a6c1_row52_col4\" class=\"data row52 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1', 'RNF7']</td>\n",
       "      <td id=\"T_0a6c1_row52_col5\" class=\"data row52 col5\" >http://model.geneontology.org/R-HSA-8951664</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row53\" class=\"row_heading level0 row53\" >53</th>\n",
       "      <td id=\"T_0a6c1_row53_col0\" class=\"data row53 col0\" >Regulation of RUNX2 expression and activity - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row53_col1\" class=\"data row53 col1\" >0.001273</td>\n",
       "      <td id=\"T_0a6c1_row53_col2\" class=\"data row53 col2\" >9</td>\n",
       "      <td id=\"T_0a6c1_row53_col3\" class=\"data row53 col3\" >54</td>\n",
       "      <td id=\"T_0a6c1_row53_col4\" class=\"data row53 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_0a6c1_row53_col5\" class=\"data row53 col5\" >http://model.geneontology.org/R-HSA-8939902</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row54\" class=\"row_heading level0 row54\" >54</th>\n",
       "      <td id=\"T_0a6c1_row54_col0\" class=\"data row54 col0\" >Degradation of GLI2 by the proteasome - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row54_col1\" class=\"data row54 col1\" >0.001273</td>\n",
       "      <td id=\"T_0a6c1_row54_col2\" class=\"data row54 col2\" >9</td>\n",
       "      <td id=\"T_0a6c1_row54_col3\" class=\"data row54 col3\" >54</td>\n",
       "      <td id=\"T_0a6c1_row54_col4\" class=\"data row54 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_0a6c1_row54_col5\" class=\"data row54 col5\" >http://model.geneontology.org/R-HSA-5610783</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row55\" class=\"row_heading level0 row55\" >55</th>\n",
       "      <td id=\"T_0a6c1_row55_col0\" class=\"data row55 col0\" >Antigen Presentation: Folding, assembly and peptide loading of class I MHC - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row55_col1\" class=\"data row55 col1\" >0.001273</td>\n",
       "      <td id=\"T_0a6c1_row55_col2\" class=\"data row55 col2\" >9</td>\n",
       "      <td id=\"T_0a6c1_row55_col3\" class=\"data row55 col3\" >54</td>\n",
       "      <td id=\"T_0a6c1_row55_col4\" class=\"data row55 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_0a6c1_row55_col5\" class=\"data row55 col5\" >http://model.geneontology.org/R-HSA-983170</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row56\" class=\"row_heading level0 row56\" >56</th>\n",
       "      <td id=\"T_0a6c1_row56_col0\" class=\"data row56 col0\" >GLI3 is processed to GLI3R by the proteasome - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row56_col1\" class=\"data row56 col1\" >0.001273</td>\n",
       "      <td id=\"T_0a6c1_row56_col2\" class=\"data row56 col2\" >9</td>\n",
       "      <td id=\"T_0a6c1_row56_col3\" class=\"data row56 col3\" >54</td>\n",
       "      <td id=\"T_0a6c1_row56_col4\" class=\"data row56 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_0a6c1_row56_col5\" class=\"data row56 col5\" >http://model.geneontology.org/R-HSA-5610785</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row57\" class=\"row_heading level0 row57\" >57</th>\n",
       "      <td id=\"T_0a6c1_row57_col0\" class=\"data row57 col0\" >Interleukin-1 signaling - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row57_col1\" class=\"data row57 col1\" >0.001313</td>\n",
       "      <td id=\"T_0a6c1_row57_col2\" class=\"data row57 col2\" >10</td>\n",
       "      <td id=\"T_0a6c1_row57_col3\" class=\"data row57 col3\" >65</td>\n",
       "      <td id=\"T_0a6c1_row57_col4\" class=\"data row57 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1', 'MAP3K7']</td>\n",
       "      <td id=\"T_0a6c1_row57_col5\" class=\"data row57 col5\" >http://model.geneontology.org/R-HSA-9020702</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row58\" class=\"row_heading level0 row58\" >58</th>\n",
       "      <td id=\"T_0a6c1_row58_col0\" class=\"data row58 col0\" >Synthesis of PC - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row58_col1\" class=\"data row58 col1\" >0.001317</td>\n",
       "      <td id=\"T_0a6c1_row58_col2\" class=\"data row58 col2\" >8</td>\n",
       "      <td id=\"T_0a6c1_row58_col3\" class=\"data row58 col3\" >44</td>\n",
       "      <td id=\"T_0a6c1_row58_col4\" class=\"data row58 col4\" >['CSNK2A1', 'PNPLA2', 'PCYT1A', 'SLC44A1', 'GPAT4', 'GPAT3', 'ABHD3', 'LPCAT4']</td>\n",
       "      <td id=\"T_0a6c1_row58_col5\" class=\"data row58 col5\" >http://model.geneontology.org/R-HSA-1483191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row59\" class=\"row_heading level0 row59\" >59</th>\n",
       "      <td id=\"T_0a6c1_row59_col0\" class=\"data row59 col0\" >Hedgehog 'off' state - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row59_col1\" class=\"data row59 col1\" >0.001457</td>\n",
       "      <td id=\"T_0a6c1_row59_col2\" class=\"data row59 col2\" >9</td>\n",
       "      <td id=\"T_0a6c1_row59_col3\" class=\"data row59 col3\" >55</td>\n",
       "      <td id=\"T_0a6c1_row59_col4\" class=\"data row59 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_0a6c1_row59_col5\" class=\"data row59 col5\" >http://model.geneontology.org/R-HSA-5610787</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row60\" class=\"row_heading level0 row60\" >60</th>\n",
       "      <td id=\"T_0a6c1_row60_col0\" class=\"data row60 col0\" >Inhibition of replication initiation of damaged DNA by RB1/E2F1 - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row60_col1\" class=\"data row60 col1\" >0.001524</td>\n",
       "      <td id=\"T_0a6c1_row60_col2\" class=\"data row60 col2\" >5</td>\n",
       "      <td id=\"T_0a6c1_row60_col3\" class=\"data row60 col3\" >18</td>\n",
       "      <td id=\"T_0a6c1_row60_col4\" class=\"data row60 col4\" >['PPP2R1B', 'PPP2CB', 'CDK2', 'CDKN1A', 'CDKN1B']</td>\n",
       "      <td id=\"T_0a6c1_row60_col5\" class=\"data row60 col5\" >http://model.geneontology.org/R-HSA-113501</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row61\" class=\"row_heading level0 row61\" >61</th>\n",
       "      <td id=\"T_0a6c1_row61_col0\" class=\"data row61 col0\" >MAPK6/MAPK4 signaling - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row61_col1\" class=\"data row61 col1\" >0.001868</td>\n",
       "      <td id=\"T_0a6c1_row61_col2\" class=\"data row61 col2\" >10</td>\n",
       "      <td id=\"T_0a6c1_row61_col3\" class=\"data row61 col3\" >68</td>\n",
       "      <td id=\"T_0a6c1_row61_col4\" class=\"data row61 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'CDK1', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_0a6c1_row61_col5\" class=\"data row61 col5\" >http://model.geneontology.org/R-HSA-5687128</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row62\" class=\"row_heading level0 row62\" >62</th>\n",
       "      <td id=\"T_0a6c1_row62_col0\" class=\"data row62 col0\" >Activation of the mRNA upon binding of the cap-binding complex and eIFs, and subsequent binding to 43S - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row62_col1\" class=\"data row62 col1\" >0.002504</td>\n",
       "      <td id=\"T_0a6c1_row62_col2\" class=\"data row62 col2\" >2</td>\n",
       "      <td id=\"T_0a6c1_row62_col3\" class=\"data row62 col3\" >2</td>\n",
       "      <td id=\"T_0a6c1_row62_col4\" class=\"data row62 col4\" >['EIF4A1', 'EIF4A2']</td>\n",
       "      <td id=\"T_0a6c1_row62_col5\" class=\"data row62 col5\" >http://model.geneontology.org/R-HSA-72662</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0a6c1_level0_row63\" class=\"row_heading level0 row63\" >63</th>\n",
       "      <td id=\"T_0a6c1_row63_col0\" class=\"data row63 col0\" >Translation initiation complex formation - Reactome</td>\n",
       "      <td id=\"T_0a6c1_row63_col1\" class=\"data row63 col1\" >0.002504</td>\n",
       "      <td id=\"T_0a6c1_row63_col2\" class=\"data row63 col2\" >2</td>\n",
       "      <td id=\"T_0a6c1_row63_col3\" class=\"data row63 col3\" >2</td>\n",
       "      <td id=\"T_0a6c1_row63_col4\" class=\"data row63 col4\" >['EIF4A1', 'EIF4A2']</td>\n",
       "      <td id=\"T_0a6c1_row63_col5\" class=\"data row63 col5\" >http://model.geneontology.org/R-HSA-72649</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x14fc84940>"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "p97_standard = enrich.enrich_wrapper('P97.csv','uniprot', FDR = 0.05, method='standard', fpath = '../test_data/processed', table_number = 2)\n",
    "p97_standard"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "ffc59744-c865-41c9-b8f7-63530d7e639c",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|█████████████████████████████████████████| 480/480 [00:45<00:00, 10.48it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 338 entities from 270 out of 766 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_26fa1 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_26fa1\">\n",
       "  <caption>Table S3: P97 combined up/downregulated proteins, weighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_26fa1_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_26fa1_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_26fa1_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_26fa1_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_26fa1_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_26fa1_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_26fa1_row0_col0\" class=\"data row0 col0\" >CDK-mediated phosphorylation and removal of Cdc6 - Reactome</td>\n",
       "      <td id=\"T_26fa1_row0_col1\" class=\"data row0 col1\" >0.000001</td>\n",
       "      <td id=\"T_26fa1_row0_col2\" class=\"data row0 col2\" >14</td>\n",
       "      <td id=\"T_26fa1_row0_col3\" class=\"data row0 col3\" >65</td>\n",
       "      <td id=\"T_26fa1_row0_col4\" class=\"data row0 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'CDK2', 'PSMF1', 'UBE2S', 'UBE2C', 'UBE2E1', 'ANAPC2']</td>\n",
       "      <td id=\"T_26fa1_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-69017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_26fa1_row1_col0\" class=\"data row1 col0\" >Autodegradation of Cdh1 by Cdh1:APC/C - Reactome</td>\n",
       "      <td id=\"T_26fa1_row1_col1\" class=\"data row1 col1\" >0.000001</td>\n",
       "      <td id=\"T_26fa1_row1_col2\" class=\"data row1 col2\" >13</td>\n",
       "      <td id=\"T_26fa1_row1_col3\" class=\"data row1 col3\" >59</td>\n",
       "      <td id=\"T_26fa1_row1_col4\" class=\"data row1 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1', 'UBE2S', 'UBE2C', 'UBE2E1', 'ANAPC2']</td>\n",
       "      <td id=\"T_26fa1_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-174084</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "      <td id=\"T_26fa1_row2_col0\" class=\"data row2 col0\" >SCF(Skp2)-mediated degradation of p27/p21 - Reactome</td>\n",
       "      <td id=\"T_26fa1_row2_col1\" class=\"data row2 col1\" >0.000001</td>\n",
       "      <td id=\"T_26fa1_row2_col2\" class=\"data row2 col2\" >12</td>\n",
       "      <td id=\"T_26fa1_row2_col3\" class=\"data row2 col3\" >49</td>\n",
       "      <td id=\"T_26fa1_row2_col4\" class=\"data row2 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'CDK2', 'PSMF1', 'CKS1B', 'set:p21,p27']</td>\n",
       "      <td id=\"T_26fa1_row2_col5\" class=\"data row2 col5\" >http://model.geneontology.org/R-HSA-187577</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row3\" class=\"row_heading level0 row3\" >3</th>\n",
       "      <td id=\"T_26fa1_row3_col0\" class=\"data row3 col0\" >Assembly of the pre-replicative complex - Reactome</td>\n",
       "      <td id=\"T_26fa1_row3_col1\" class=\"data row3 col1\" >0.000001</td>\n",
       "      <td id=\"T_26fa1_row3_col2\" class=\"data row3 col2\" >13</td>\n",
       "      <td id=\"T_26fa1_row3_col3\" class=\"data row3 col3\" >60</td>\n",
       "      <td id=\"T_26fa1_row3_col4\" class=\"data row3 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1', 'UBE2S', 'UBE2C', 'UBE2E1', 'ANAPC2']</td>\n",
       "      <td id=\"T_26fa1_row3_col5\" class=\"data row3 col5\" >http://model.geneontology.org/R-HSA-68867</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row4\" class=\"row_heading level0 row4\" >4</th>\n",
       "      <td id=\"T_26fa1_row4_col0\" class=\"data row4 col0\" >Unwinding of DNA - Reactome</td>\n",
       "      <td id=\"T_26fa1_row4_col1\" class=\"data row4 col1\" >0.000001</td>\n",
       "      <td id=\"T_26fa1_row4_col2\" class=\"data row4 col2\" >6</td>\n",
       "      <td id=\"T_26fa1_row4_col3\" class=\"data row4 col3\" >7</td>\n",
       "      <td id=\"T_26fa1_row4_col4\" class=\"data row4 col4\" >['MCM3', 'MCM6', 'MCM4', 'MCM2', 'MCM7', 'MCM5']</td>\n",
       "      <td id=\"T_26fa1_row4_col5\" class=\"data row4 col5\" >http://model.geneontology.org/R-HSA-176974</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row5\" class=\"row_heading level0 row5\" >5</th>\n",
       "      <td id=\"T_26fa1_row5_col0\" class=\"data row5 col0\" >APC/C:Cdc20 mediated degradation of Securin - Reactome</td>\n",
       "      <td id=\"T_26fa1_row5_col1\" class=\"data row5 col1\" >0.000001</td>\n",
       "      <td id=\"T_26fa1_row5_col2\" class=\"data row5 col2\" >13</td>\n",
       "      <td id=\"T_26fa1_row5_col3\" class=\"data row5 col3\" >62</td>\n",
       "      <td id=\"T_26fa1_row5_col4\" class=\"data row5 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1', 'UBE2S', 'UBE2C', 'UBE2E1', 'ANAPC2']</td>\n",
       "      <td id=\"T_26fa1_row5_col5\" class=\"data row5 col5\" >http://model.geneontology.org/R-HSA-174154</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row6\" class=\"row_heading level0 row6\" >6</th>\n",
       "      <td id=\"T_26fa1_row6_col0\" class=\"data row6 col0\" >APC/C:Cdh1 mediated degradation of Cdc20 and other APC/C:Cdh1 targeted proteins in late mitosis/early G1 - Reactome</td>\n",
       "      <td id=\"T_26fa1_row6_col1\" class=\"data row6 col1\" >0.000001</td>\n",
       "      <td id=\"T_26fa1_row6_col2\" class=\"data row6 col2\" >13</td>\n",
       "      <td id=\"T_26fa1_row6_col3\" class=\"data row6 col3\" >62</td>\n",
       "      <td id=\"T_26fa1_row6_col4\" class=\"data row6 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1', 'UBE2S', 'UBE2C', 'UBE2E1', 'ANAPC2']</td>\n",
       "      <td id=\"T_26fa1_row6_col5\" class=\"data row6 col5\" >http://model.geneontology.org/R-HSA-174178</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row7\" class=\"row_heading level0 row7\" >7</th>\n",
       "      <td id=\"T_26fa1_row7_col0\" class=\"data row7 col0\" >Cdc20:Phospho-APC/C mediated degradation of Cyclin A - Reactome</td>\n",
       "      <td id=\"T_26fa1_row7_col1\" class=\"data row7 col1\" >0.000001</td>\n",
       "      <td id=\"T_26fa1_row7_col2\" class=\"data row7 col2\" >13</td>\n",
       "      <td id=\"T_26fa1_row7_col3\" class=\"data row7 col3\" >62</td>\n",
       "      <td id=\"T_26fa1_row7_col4\" class=\"data row7 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1', 'UBE2S', 'UBE2C', 'UBE2E1', 'ANAPC2']</td>\n",
       "      <td id=\"T_26fa1_row7_col5\" class=\"data row7 col5\" >http://model.geneontology.org/R-HSA-174184</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row8\" class=\"row_heading level0 row8\" >8</th>\n",
       "      <td id=\"T_26fa1_row8_col0\" class=\"data row8 col0\" >Separation of Sister Chromatids - Reactome</td>\n",
       "      <td id=\"T_26fa1_row8_col1\" class=\"data row8 col1\" >0.000002</td>\n",
       "      <td id=\"T_26fa1_row8_col2\" class=\"data row8 col2\" >13</td>\n",
       "      <td id=\"T_26fa1_row8_col3\" class=\"data row8 col3\" >63</td>\n",
       "      <td id=\"T_26fa1_row8_col4\" class=\"data row8 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1', 'UBE2S', 'UBE2C', 'UBE2E1', 'ANAPC2']</td>\n",
       "      <td id=\"T_26fa1_row8_col5\" class=\"data row8 col5\" >http://model.geneontology.org/R-HSA-2467813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row9\" class=\"row_heading level0 row9\" >9</th>\n",
       "      <td id=\"T_26fa1_row9_col0\" class=\"data row9 col0\" >Conversion from APC/C:Cdc20 to APC/C:Cdh1 in late anaphase - Reactome</td>\n",
       "      <td id=\"T_26fa1_row9_col1\" class=\"data row9 col1\" >0.000007</td>\n",
       "      <td id=\"T_26fa1_row9_col2\" class=\"data row9 col2\" >11</td>\n",
       "      <td id=\"T_26fa1_row9_col3\" class=\"data row9 col3\" >49</td>\n",
       "      <td id=\"T_26fa1_row9_col4\" class=\"data row9 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'CDK1', 'PSMD14', 'CDK2', 'PSMF1']</td>\n",
       "      <td id=\"T_26fa1_row9_col5\" class=\"data row9 col5\" >http://model.geneontology.org/R-HSA-176407</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row10\" class=\"row_heading level0 row10\" >10</th>\n",
       "      <td id=\"T_26fa1_row10_col0\" class=\"data row10 col0\" >The role of GTSE1 in G2/M progression after G2 checkpoint - Reactome</td>\n",
       "      <td id=\"T_26fa1_row10_col1\" class=\"data row10 col1\" >0.000007</td>\n",
       "      <td id=\"T_26fa1_row10_col2\" class=\"data row10 col2\" >11</td>\n",
       "      <td id=\"T_26fa1_row10_col3\" class=\"data row10 col3\" >50</td>\n",
       "      <td id=\"T_26fa1_row10_col4\" class=\"data row10 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1', 'PLK1', 'TP53']</td>\n",
       "      <td id=\"T_26fa1_row10_col5\" class=\"data row10 col5\" >http://model.geneontology.org/R-HSA-8852276</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row11\" class=\"row_heading level0 row11\" >11</th>\n",
       "      <td id=\"T_26fa1_row11_col0\" class=\"data row11 col0\" >SCF-beta-TrCP mediated degradation of Emi1 - Reactome</td>\n",
       "      <td id=\"T_26fa1_row11_col1\" class=\"data row11 col1\" >0.000010</td>\n",
       "      <td id=\"T_26fa1_row11_col2\" class=\"data row11 col2\" >11</td>\n",
       "      <td id=\"T_26fa1_row11_col3\" class=\"data row11 col3\" >51</td>\n",
       "      <td id=\"T_26fa1_row11_col4\" class=\"data row11 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'CDK1', 'PSMD14', 'PSMF1', 'PLK1']</td>\n",
       "      <td id=\"T_26fa1_row11_col5\" class=\"data row11 col5\" >http://model.geneontology.org/R-HSA-174113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row12\" class=\"row_heading level0 row12\" >12</th>\n",
       "      <td id=\"T_26fa1_row12_col0\" class=\"data row12 col0\" >ER-Phagosome pathway - Reactome</td>\n",
       "      <td id=\"T_26fa1_row12_col1\" class=\"data row12 col1\" >0.000012</td>\n",
       "      <td id=\"T_26fa1_row12_col2\" class=\"data row12 col2\" >11</td>\n",
       "      <td id=\"T_26fa1_row12_col3\" class=\"data row12 col3\" >51</td>\n",
       "      <td id=\"T_26fa1_row12_col4\" class=\"data row12 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1', 'SEC61G', 'set:SEC61 alpha']</td>\n",
       "      <td id=\"T_26fa1_row12_col5\" class=\"data row12 col5\" >http://model.geneontology.org/R-HSA-1236974</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row13\" class=\"row_heading level0 row13\" >13</th>\n",
       "      <td id=\"T_26fa1_row13_col0\" class=\"data row13 col0\" >Regulation of APC/C activators between G1/S and early anaphase - Reactome</td>\n",
       "      <td id=\"T_26fa1_row13_col1\" class=\"data row13 col1\" >0.000015</td>\n",
       "      <td id=\"T_26fa1_row13_col2\" class=\"data row13 col2\" >11</td>\n",
       "      <td id=\"T_26fa1_row13_col3\" class=\"data row13 col3\" >52</td>\n",
       "      <td id=\"T_26fa1_row13_col4\" class=\"data row13 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'CDK1', 'PSMD14', 'CDK2', 'PSMF1']</td>\n",
       "      <td id=\"T_26fa1_row13_col5\" class=\"data row13 col5\" >http://model.geneontology.org/R-HSA-176408</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row14\" class=\"row_heading level0 row14\" >14</th>\n",
       "      <td id=\"T_26fa1_row14_col0\" class=\"data row14 col0\" >Resolution of Sister Chromatid Cohesion - Reactome</td>\n",
       "      <td id=\"T_26fa1_row14_col1\" class=\"data row14 col1\" >0.000018</td>\n",
       "      <td id=\"T_26fa1_row14_col2\" class=\"data row14 col2\" >15</td>\n",
       "      <td id=\"T_26fa1_row14_col3\" class=\"data row14 col3\" >95</td>\n",
       "      <td id=\"T_26fa1_row14_col4\" class=\"data row14 col4\" >['CDK1', 'SEC13', 'TAOK1', 'KIF2C', 'MAD2L1', 'BUB1B', 'SPC24', 'DSN1', 'PLK1', 'NDE1', 'CDC20', 'INCENP', 'NSL1', 'set:PP2A-subunit A', 'set:PP2A-catalytic subunit C']</td>\n",
       "      <td id=\"T_26fa1_row14_col5\" class=\"data row14 col5\" >http://model.geneontology.org/R-HSA-2500257</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row15\" class=\"row_heading level0 row15\" >15</th>\n",
       "      <td id=\"T_26fa1_row15_col0\" class=\"data row15 col0\" >Cyclin A:Cdk2-associated events at S phase entry - Reactome</td>\n",
       "      <td id=\"T_26fa1_row15_col1\" class=\"data row15 col1\" >0.000023</td>\n",
       "      <td id=\"T_26fa1_row15_col2\" class=\"data row15 col2\" >11</td>\n",
       "      <td id=\"T_26fa1_row15_col3\" class=\"data row15 col3\" >52</td>\n",
       "      <td id=\"T_26fa1_row15_col4\" class=\"data row15 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'CDK2', 'PSMF1', 'set:p21,p27']</td>\n",
       "      <td id=\"T_26fa1_row15_col5\" class=\"data row15 col5\" >http://model.geneontology.org/R-HSA-69656</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row16\" class=\"row_heading level0 row16\" >16</th>\n",
       "      <td id=\"T_26fa1_row16_col0\" class=\"data row16 col0\" >Cyclin E associated events during G1/S transition - Reactome</td>\n",
       "      <td id=\"T_26fa1_row16_col1\" class=\"data row16 col1\" >0.000028</td>\n",
       "      <td id=\"T_26fa1_row16_col2\" class=\"data row16 col2\" >11</td>\n",
       "      <td id=\"T_26fa1_row16_col3\" class=\"data row16 col3\" >51</td>\n",
       "      <td id=\"T_26fa1_row16_col4\" class=\"data row16 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1', 'set:p-T172-CDK4,p-T177-CDK6:CCND:CDKN1A,CDKN1B,(CDKN1C); p-T172-CDK4,p-T177-CDK6:CCND; (p-T172-CDK4,p-T177-CDK6:CCND:p-Y88-CDKN1B)', 'set:p21,p27']</td>\n",
       "      <td id=\"T_26fa1_row16_col5\" class=\"data row16 col5\" >http://model.geneontology.org/R-HSA-69202</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row17\" class=\"row_heading level0 row17\" >17</th>\n",
       "      <td id=\"T_26fa1_row17_col0\" class=\"data row17 col0\" >Activation of NF-kappaB in B cells - Reactome</td>\n",
       "      <td id=\"T_26fa1_row17_col1\" class=\"data row17 col1\" >0.000028</td>\n",
       "      <td id=\"T_26fa1_row17_col2\" class=\"data row17 col2\" >11</td>\n",
       "      <td id=\"T_26fa1_row17_col3\" class=\"data row17 col3\" >54</td>\n",
       "      <td id=\"T_26fa1_row17_col4\" class=\"data row17 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'CHUK', 'PSMD14', 'PSMF1', 'MAP3K7']</td>\n",
       "      <td id=\"T_26fa1_row17_col5\" class=\"data row17 col5\" >http://model.geneontology.org/R-HSA-1169091</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row18\" class=\"row_heading level0 row18\" >18</th>\n",
       "      <td id=\"T_26fa1_row18_col0\" class=\"data row18 col0\" >G2/M Checkpoints - Reactome</td>\n",
       "      <td id=\"T_26fa1_row18_col1\" class=\"data row18 col1\" >0.000034</td>\n",
       "      <td id=\"T_26fa1_row18_col2\" class=\"data row18 col2\" >10</td>\n",
       "      <td id=\"T_26fa1_row18_col3\" class=\"data row18 col3\" >48</td>\n",
       "      <td id=\"T_26fa1_row18_col4\" class=\"data row18 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1', 'TP53']</td>\n",
       "      <td id=\"T_26fa1_row18_col5\" class=\"data row18 col5\" >http://model.geneontology.org/R-HSA-69481</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row19\" class=\"row_heading level0 row19\" >19</th>\n",
       "      <td id=\"T_26fa1_row19_col0\" class=\"data row19 col0\" >Degradation of beta-catenin by the destruction complex - Reactome</td>\n",
       "      <td id=\"T_26fa1_row19_col1\" class=\"data row19 col1\" >0.000038</td>\n",
       "      <td id=\"T_26fa1_row19_col2\" class=\"data row19 col2\" >11</td>\n",
       "      <td id=\"T_26fa1_row19_col3\" class=\"data row19 col3\" >54</td>\n",
       "      <td id=\"T_26fa1_row19_col4\" class=\"data row19 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1', 'set:PP2A-subunit A', 'set:PP2A-catalytic subunit C']</td>\n",
       "      <td id=\"T_26fa1_row19_col5\" class=\"data row19 col5\" >http://model.geneontology.org/R-HSA-195253</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row20\" class=\"row_heading level0 row20\" >20</th>\n",
       "      <td id=\"T_26fa1_row20_col0\" class=\"data row20 col0\" >NIK-->noncanonical NF-kB signaling - Reactome</td>\n",
       "      <td id=\"T_26fa1_row20_col1\" class=\"data row20 col1\" >0.000065</td>\n",
       "      <td id=\"T_26fa1_row20_col2\" class=\"data row20 col2\" >10</td>\n",
       "      <td id=\"T_26fa1_row20_col3\" class=\"data row20 col3\" >50</td>\n",
       "      <td id=\"T_26fa1_row20_col4\" class=\"data row20 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'CHUK', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_26fa1_row20_col5\" class=\"data row20 col5\" >http://model.geneontology.org/R-HSA-5676590</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row21\" class=\"row_heading level0 row21\" >21</th>\n",
       "      <td id=\"T_26fa1_row21_col0\" class=\"data row21 col0\" >Dectin-1 mediated noncanonical NF-kB signaling - Reactome</td>\n",
       "      <td id=\"T_26fa1_row21_col1\" class=\"data row21 col1\" >0.000065</td>\n",
       "      <td id=\"T_26fa1_row21_col2\" class=\"data row21 col2\" >10</td>\n",
       "      <td id=\"T_26fa1_row21_col3\" class=\"data row21 col3\" >50</td>\n",
       "      <td id=\"T_26fa1_row21_col4\" class=\"data row21 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'CHUK', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_26fa1_row21_col5\" class=\"data row21 col5\" >http://model.geneontology.org/R-HSA-5607761</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row22\" class=\"row_heading level0 row22\" >22</th>\n",
       "      <td id=\"T_26fa1_row22_col0\" class=\"data row22 col0\" >Regulation of expression of SLITs and ROBOs - Reactome</td>\n",
       "      <td id=\"T_26fa1_row22_col1\" class=\"data row22 col1\" >0.000067</td>\n",
       "      <td id=\"T_26fa1_row22_col2\" class=\"data row22 col2\" >10</td>\n",
       "      <td id=\"T_26fa1_row22_col3\" class=\"data row22 col3\" >51</td>\n",
       "      <td id=\"T_26fa1_row22_col4\" class=\"data row22 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1', 'USP33']</td>\n",
       "      <td id=\"T_26fa1_row22_col5\" class=\"data row22 col5\" >http://model.geneontology.org/R-HSA-9010553</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row23\" class=\"row_heading level0 row23\" >23</th>\n",
       "      <td id=\"T_26fa1_row23_col0\" class=\"data row23 col0\" >Hedgehog ligand biogenesis - Reactome</td>\n",
       "      <td id=\"T_26fa1_row23_col1\" class=\"data row23 col1\" >0.000073</td>\n",
       "      <td id=\"T_26fa1_row23_col2\" class=\"data row23 col2\" >10</td>\n",
       "      <td id=\"T_26fa1_row23_col3\" class=\"data row23 col3\" >50</td>\n",
       "      <td id=\"T_26fa1_row23_col4\" class=\"data row23 col4\" >['VCP', 'PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_26fa1_row23_col5\" class=\"data row23 col5\" >http://model.geneontology.org/R-HSA-5358346</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row24\" class=\"row_heading level0 row24\" >24</th>\n",
       "      <td id=\"T_26fa1_row24_col0\" class=\"data row24 col0\" >Orc1 removal from chromatin - Reactome</td>\n",
       "      <td id=\"T_26fa1_row24_col1\" class=\"data row24 col1\" >0.000076</td>\n",
       "      <td id=\"T_26fa1_row24_col2\" class=\"data row24 col2\" >10</td>\n",
       "      <td id=\"T_26fa1_row24_col3\" class=\"data row24 col3\" >51</td>\n",
       "      <td id=\"T_26fa1_row24_col4\" class=\"data row24 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'CDK2', 'PSMF1']</td>\n",
       "      <td id=\"T_26fa1_row24_col5\" class=\"data row24 col5\" >http://model.geneontology.org/R-HSA-68949</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row25\" class=\"row_heading level0 row25\" >25</th>\n",
       "      <td id=\"T_26fa1_row25_col0\" class=\"data row25 col0\" >Regulation of RAS by GAPs - Reactome</td>\n",
       "      <td id=\"T_26fa1_row25_col1\" class=\"data row25 col1\" >0.000130</td>\n",
       "      <td id=\"T_26fa1_row25_col2\" class=\"data row25 col2\" >10</td>\n",
       "      <td id=\"T_26fa1_row25_col3\" class=\"data row25 col3\" >48</td>\n",
       "      <td id=\"T_26fa1_row25_col4\" class=\"data row25 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1', 'set:mature p21 RAS']</td>\n",
       "      <td id=\"T_26fa1_row25_col5\" class=\"data row25 col5\" >http://model.geneontology.org/R-HSA-5658442</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row26\" class=\"row_heading level0 row26\" >26</th>\n",
       "      <td id=\"T_26fa1_row26_col0\" class=\"data row26 col0\" >FCERI mediated NF-kB activation - Reactome</td>\n",
       "      <td id=\"T_26fa1_row26_col1\" class=\"data row26 col1\" >0.000136</td>\n",
       "      <td id=\"T_26fa1_row26_col2\" class=\"data row26 col2\" >10</td>\n",
       "      <td id=\"T_26fa1_row26_col3\" class=\"data row26 col3\" >54</td>\n",
       "      <td id=\"T_26fa1_row26_col4\" class=\"data row26 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1', 'MAP3K7']</td>\n",
       "      <td id=\"T_26fa1_row26_col5\" class=\"data row26 col5\" >http://model.geneontology.org/R-HSA-2871837</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row27\" class=\"row_heading level0 row27\" >27</th>\n",
       "      <td id=\"T_26fa1_row27_col0\" class=\"data row27 col0\" >GSK3B and BTRC:CUL1-mediated-degradation of NFE2L2 - Reactome</td>\n",
       "      <td id=\"T_26fa1_row27_col1\" class=\"data row27 col1\" >0.000137</td>\n",
       "      <td id=\"T_26fa1_row27_col2\" class=\"data row27 col2\" >9</td>\n",
       "      <td id=\"T_26fa1_row27_col3\" class=\"data row27 col3\" >44</td>\n",
       "      <td id=\"T_26fa1_row27_col4\" class=\"data row27 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_26fa1_row27_col5\" class=\"data row27 col5\" >http://model.geneontology.org/R-HSA-9762114</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row28\" class=\"row_heading level0 row28\" >28</th>\n",
       "      <td id=\"T_26fa1_row28_col0\" class=\"data row28 col0\" >CLEC7A (Dectin-1) signaling - Reactome</td>\n",
       "      <td id=\"T_26fa1_row28_col1\" class=\"data row28 col1\" >0.000140</td>\n",
       "      <td id=\"T_26fa1_row28_col2\" class=\"data row28 col2\" >10</td>\n",
       "      <td id=\"T_26fa1_row28_col3\" class=\"data row28 col3\" >55</td>\n",
       "      <td id=\"T_26fa1_row28_col4\" class=\"data row28 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1', 'MAP3K7']</td>\n",
       "      <td id=\"T_26fa1_row28_col5\" class=\"data row28 col5\" >http://model.geneontology.org/R-HSA-5607764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row29\" class=\"row_heading level0 row29\" >29</th>\n",
       "      <td id=\"T_26fa1_row29_col0\" class=\"data row29 col0\" >AUF1 (hnRNP D0) binds and destabilizes mRNA - Reactome</td>\n",
       "      <td id=\"T_26fa1_row29_col1\" class=\"data row29 col1\" >0.000161</td>\n",
       "      <td id=\"T_26fa1_row29_col2\" class=\"data row29 col2\" >9</td>\n",
       "      <td id=\"T_26fa1_row29_col3\" class=\"data row29 col3\" >45</td>\n",
       "      <td id=\"T_26fa1_row29_col4\" class=\"data row29 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_26fa1_row29_col5\" class=\"data row29 col5\" >http://model.geneontology.org/R-HSA-450408</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row30\" class=\"row_heading level0 row30\" >30</th>\n",
       "      <td id=\"T_26fa1_row30_col0\" class=\"data row30 col0\" >Cross-presentation of soluble exogenous antigens (endosomes) - Reactome</td>\n",
       "      <td id=\"T_26fa1_row30_col1\" class=\"data row30 col1\" >0.000161</td>\n",
       "      <td id=\"T_26fa1_row30_col2\" class=\"data row30 col2\" >9</td>\n",
       "      <td id=\"T_26fa1_row30_col3\" class=\"data row30 col3\" >45</td>\n",
       "      <td id=\"T_26fa1_row30_col4\" class=\"data row30 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_26fa1_row30_col5\" class=\"data row30 col5\" >http://model.geneontology.org/R-HSA-1236978</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row31\" class=\"row_heading level0 row31\" >31</th>\n",
       "      <td id=\"T_26fa1_row31_col0\" class=\"data row31 col0\" >APC/C:Cdc20 mediated degradation of mitotic proteins - Reactome</td>\n",
       "      <td id=\"T_26fa1_row31_col1\" class=\"data row31 col1\" >0.000161</td>\n",
       "      <td id=\"T_26fa1_row31_col2\" class=\"data row31 col2\" >9</td>\n",
       "      <td id=\"T_26fa1_row31_col3\" class=\"data row31 col3\" >45</td>\n",
       "      <td id=\"T_26fa1_row31_col4\" class=\"data row31 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_26fa1_row31_col5\" class=\"data row31 col5\" >http://model.geneontology.org/R-HSA-176409</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row32\" class=\"row_heading level0 row32\" >32</th>\n",
       "      <td id=\"T_26fa1_row32_col0\" class=\"data row32 col0\" >KEAP1-NFE2L2 pathway - Reactome</td>\n",
       "      <td id=\"T_26fa1_row32_col1\" class=\"data row32 col1\" >0.000174</td>\n",
       "      <td id=\"T_26fa1_row32_col2\" class=\"data row32 col2\" >10</td>\n",
       "      <td id=\"T_26fa1_row32_col3\" class=\"data row32 col3\" >55</td>\n",
       "      <td id=\"T_26fa1_row32_col4\" class=\"data row32 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1', 'set:CSNK2(A1:A1/A1:A2/A2:A2)']</td>\n",
       "      <td id=\"T_26fa1_row32_col5\" class=\"data row32 col5\" >http://model.geneontology.org/R-HSA-9755511</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row33\" class=\"row_heading level0 row33\" >33</th>\n",
       "      <td id=\"T_26fa1_row33_col0\" class=\"data row33 col0\" >Regulation of RUNX3 expression and activity - Reactome</td>\n",
       "      <td id=\"T_26fa1_row33_col1\" class=\"data row33 col1\" >0.000184</td>\n",
       "      <td id=\"T_26fa1_row33_col2\" class=\"data row33 col2\" >9</td>\n",
       "      <td id=\"T_26fa1_row33_col3\" class=\"data row33 col3\" >45</td>\n",
       "      <td id=\"T_26fa1_row33_col4\" class=\"data row33 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_26fa1_row33_col5\" class=\"data row33 col5\" >http://model.geneontology.org/R-HSA-8941858</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row34\" class=\"row_heading level0 row34\" >34</th>\n",
       "      <td id=\"T_26fa1_row34_col0\" class=\"data row34 col0\" >Ubiquitin-dependent degradation of Cyclin D - Reactome</td>\n",
       "      <td id=\"T_26fa1_row34_col1\" class=\"data row34 col1\" >0.000190</td>\n",
       "      <td id=\"T_26fa1_row34_col2\" class=\"data row34 col2\" >9</td>\n",
       "      <td id=\"T_26fa1_row34_col3\" class=\"data row34 col3\" >46</td>\n",
       "      <td id=\"T_26fa1_row34_col4\" class=\"data row34 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_26fa1_row34_col5\" class=\"data row34 col5\" >http://model.geneontology.org/R-HSA-75815</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row35\" class=\"row_heading level0 row35\" >35</th>\n",
       "      <td id=\"T_26fa1_row35_col0\" class=\"data row35 col0\" >Regulation of activated PAK-2p34 by proteasome mediated degradation - Reactome</td>\n",
       "      <td id=\"T_26fa1_row35_col1\" class=\"data row35 col1\" >0.000190</td>\n",
       "      <td id=\"T_26fa1_row35_col2\" class=\"data row35 col2\" >9</td>\n",
       "      <td id=\"T_26fa1_row35_col3\" class=\"data row35 col3\" >46</td>\n",
       "      <td id=\"T_26fa1_row35_col4\" class=\"data row35 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_26fa1_row35_col5\" class=\"data row35 col5\" >http://model.geneontology.org/R-HSA-211733</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row36\" class=\"row_heading level0 row36\" >36</th>\n",
       "      <td id=\"T_26fa1_row36_col0\" class=\"data row36 col0\" >Asymmetric localization of PCP proteins - Reactome</td>\n",
       "      <td id=\"T_26fa1_row36_col1\" class=\"data row36 col1\" >0.000216</td>\n",
       "      <td id=\"T_26fa1_row36_col2\" class=\"data row36 col2\" >9</td>\n",
       "      <td id=\"T_26fa1_row36_col3\" class=\"data row36 col3\" >46</td>\n",
       "      <td id=\"T_26fa1_row36_col4\" class=\"data row36 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_26fa1_row36_col5\" class=\"data row36 col5\" >http://model.geneontology.org/R-HSA-4608870</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row37\" class=\"row_heading level0 row37\" >37</th>\n",
       "      <td id=\"T_26fa1_row37_col0\" class=\"data row37 col0\" >Degradation of DVL - Reactome</td>\n",
       "      <td id=\"T_26fa1_row37_col1\" class=\"data row37 col1\" >0.000222</td>\n",
       "      <td id=\"T_26fa1_row37_col2\" class=\"data row37 col2\" >9</td>\n",
       "      <td id=\"T_26fa1_row37_col3\" class=\"data row37 col3\" >47</td>\n",
       "      <td id=\"T_26fa1_row37_col4\" class=\"data row37 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_26fa1_row37_col5\" class=\"data row37 col5\" >http://model.geneontology.org/R-HSA-4641258</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row38\" class=\"row_heading level0 row38\" >38</th>\n",
       "      <td id=\"T_26fa1_row38_col0\" class=\"data row38 col0\" >Autodegradation of the E3 ubiquitin ligase COP1 - Reactome</td>\n",
       "      <td id=\"T_26fa1_row38_col1\" class=\"data row38 col1\" >0.000222</td>\n",
       "      <td id=\"T_26fa1_row38_col2\" class=\"data row38 col2\" >9</td>\n",
       "      <td id=\"T_26fa1_row38_col3\" class=\"data row38 col3\" >47</td>\n",
       "      <td id=\"T_26fa1_row38_col4\" class=\"data row38 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_26fa1_row38_col5\" class=\"data row38 col5\" >http://model.geneontology.org/R-HSA-349425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row39\" class=\"row_heading level0 row39\" >39</th>\n",
       "      <td id=\"T_26fa1_row39_col0\" class=\"data row39 col0\" >PI5P, PP2A and IER3 Regulate PI3K/AKT Signaling - Reactome</td>\n",
       "      <td id=\"T_26fa1_row39_col1\" class=\"data row39 col1\" >0.000251</td>\n",
       "      <td id=\"T_26fa1_row39_col2\" class=\"data row39 col2\" >8</td>\n",
       "      <td id=\"T_26fa1_row39_col3\" class=\"data row39 col3\" >13</td>\n",
       "      <td id=\"T_26fa1_row39_col4\" class=\"data row39 col4\" >['set:PP2A-subunit A', 'set:PIP5K1A/B', 'set:PIP5K1A-C', 'set:Activator:PI3K', 'set:PP2A-catalytic subunit C', 'set:SYNJ/MTM(1)', 'set:PI4K2A/2B', 'set:Activated SRC,LCK,EGFR,INSR']</td>\n",
       "      <td id=\"T_26fa1_row39_col5\" class=\"data row39 col5\" >http://model.geneontology.org/R-HSA-6811558</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row40\" class=\"row_heading level0 row40\" >40</th>\n",
       "      <td id=\"T_26fa1_row40_col0\" class=\"data row40 col0\" >ABC-family proteins mediated transport - Reactome</td>\n",
       "      <td id=\"T_26fa1_row40_col1\" class=\"data row40 col1\" >0.000283</td>\n",
       "      <td id=\"T_26fa1_row40_col2\" class=\"data row40 col2\" >10</td>\n",
       "      <td id=\"T_26fa1_row40_col3\" class=\"data row40 col3\" >54</td>\n",
       "      <td id=\"T_26fa1_row40_col4\" class=\"data row40 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1', 'set:ABCCs']</td>\n",
       "      <td id=\"T_26fa1_row40_col5\" class=\"data row40 col5\" >http://model.geneontology.org/R-HSA-382556</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row41\" class=\"row_heading level0 row41\" >41</th>\n",
       "      <td id=\"T_26fa1_row41_col0\" class=\"data row41 col0\" >TNFR2 non-canonical NF-kB pathway - Reactome</td>\n",
       "      <td id=\"T_26fa1_row41_col1\" class=\"data row41 col1\" >0.000284</td>\n",
       "      <td id=\"T_26fa1_row41_col2\" class=\"data row41 col2\" >9</td>\n",
       "      <td id=\"T_26fa1_row41_col3\" class=\"data row41 col3\" >47</td>\n",
       "      <td id=\"T_26fa1_row41_col4\" class=\"data row41 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_26fa1_row41_col5\" class=\"data row41 col5\" >http://model.geneontology.org/R-HSA-5668541</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row42\" class=\"row_heading level0 row42\" >42</th>\n",
       "      <td id=\"T_26fa1_row42_col0\" class=\"data row42 col0\" >Degradation of AXIN - Reactome</td>\n",
       "      <td id=\"T_26fa1_row42_col1\" class=\"data row42 col1\" >0.000292</td>\n",
       "      <td id=\"T_26fa1_row42_col2\" class=\"data row42 col2\" >9</td>\n",
       "      <td id=\"T_26fa1_row42_col3\" class=\"data row42 col3\" >48</td>\n",
       "      <td id=\"T_26fa1_row42_col4\" class=\"data row42 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_26fa1_row42_col5\" class=\"data row42 col5\" >http://model.geneontology.org/R-HSA-4641257</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row43\" class=\"row_heading level0 row43\" >43</th>\n",
       "      <td id=\"T_26fa1_row43_col0\" class=\"data row43 col0\" >Antigen processing: Ubiquitination & Proteasome degradation - Reactome</td>\n",
       "      <td id=\"T_26fa1_row43_col1\" class=\"data row43 col1\" >0.000297</td>\n",
       "      <td id=\"T_26fa1_row43_col2\" class=\"data row43 col2\" >11</td>\n",
       "      <td id=\"T_26fa1_row43_col3\" class=\"data row43 col3\" >50</td>\n",
       "      <td id=\"T_26fa1_row43_col4\" class=\"data row43 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1', 'set:E3 ligases in proteasomal degradation', 'set:E2 Ubiquitin-conjugating enzyme']</td>\n",
       "      <td id=\"T_26fa1_row43_col5\" class=\"data row43 col5\" >http://model.geneontology.org/R-HSA-983168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row44\" class=\"row_heading level0 row44\" >44</th>\n",
       "      <td id=\"T_26fa1_row44_col0\" class=\"data row44 col0\" >Ubiquitin Mediated Degradation of Phosphorylated Cdc25A - Reactome</td>\n",
       "      <td id=\"T_26fa1_row44_col1\" class=\"data row44 col1\" >0.000300</td>\n",
       "      <td id=\"T_26fa1_row44_col2\" class=\"data row44 col2\" >9</td>\n",
       "      <td id=\"T_26fa1_row44_col3\" class=\"data row44 col3\" >49</td>\n",
       "      <td id=\"T_26fa1_row44_col4\" class=\"data row44 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_26fa1_row44_col5\" class=\"data row44 col5\" >http://model.geneontology.org/R-HSA-69601</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row45\" class=\"row_heading level0 row45\" >45</th>\n",
       "      <td id=\"T_26fa1_row45_col0\" class=\"data row45 col0\" >RUNX1 regulates transcription of genes involved in differentiation of HSCs - Reactome</td>\n",
       "      <td id=\"T_26fa1_row45_col1\" class=\"data row45 col1\" >0.000300</td>\n",
       "      <td id=\"T_26fa1_row45_col2\" class=\"data row45 col2\" >9</td>\n",
       "      <td id=\"T_26fa1_row45_col3\" class=\"data row45 col3\" >49</td>\n",
       "      <td id=\"T_26fa1_row45_col4\" class=\"data row45 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_26fa1_row45_col5\" class=\"data row45 col5\" >http://model.geneontology.org/R-HSA-8939236</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row46\" class=\"row_heading level0 row46\" >46</th>\n",
       "      <td id=\"T_26fa1_row46_col0\" class=\"data row46 col0\" >Oxygen-dependent proline hydroxylation of Hypoxia-inducible Factor Alpha - Reactome</td>\n",
       "      <td id=\"T_26fa1_row46_col1\" class=\"data row46 col1\" >0.000329</td>\n",
       "      <td id=\"T_26fa1_row46_col2\" class=\"data row46 col2\" >9</td>\n",
       "      <td id=\"T_26fa1_row46_col3\" class=\"data row46 col3\" >48</td>\n",
       "      <td id=\"T_26fa1_row46_col4\" class=\"data row46 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_26fa1_row46_col5\" class=\"data row46 col5\" >http://model.geneontology.org/R-HSA-1234176</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row47\" class=\"row_heading level0 row47\" >47</th>\n",
       "      <td id=\"T_26fa1_row47_col0\" class=\"data row47 col0\" >Regulation of PTEN stability and activity - Reactome</td>\n",
       "      <td id=\"T_26fa1_row47_col1\" class=\"data row47 col1\" >0.000329</td>\n",
       "      <td id=\"T_26fa1_row47_col2\" class=\"data row47 col2\" >10</td>\n",
       "      <td id=\"T_26fa1_row47_col3\" class=\"data row47 col3\" >56</td>\n",
       "      <td id=\"T_26fa1_row47_col4\" class=\"data row47 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1', 'set:CSNK2(A1:A1/A1:A2/A2:A2)']</td>\n",
       "      <td id=\"T_26fa1_row47_col5\" class=\"data row47 col5\" >http://model.geneontology.org/R-HSA-8948751</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row48\" class=\"row_heading level0 row48\" >48</th>\n",
       "      <td id=\"T_26fa1_row48_col0\" class=\"data row48 col0\" >Hedgehog 'on' state - Reactome</td>\n",
       "      <td id=\"T_26fa1_row48_col1\" class=\"data row48 col1\" >0.000338</td>\n",
       "      <td id=\"T_26fa1_row48_col2\" class=\"data row48 col2\" >9</td>\n",
       "      <td id=\"T_26fa1_row48_col3\" class=\"data row48 col3\" >49</td>\n",
       "      <td id=\"T_26fa1_row48_col4\" class=\"data row48 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_26fa1_row48_col5\" class=\"data row48 col5\" >http://model.geneontology.org/R-HSA-5632684</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row49\" class=\"row_heading level0 row49\" >49</th>\n",
       "      <td id=\"T_26fa1_row49_col0\" class=\"data row49 col0\" >FBXL7 down-regulates AURKA during mitotic entry and in early mitosis - Reactome</td>\n",
       "      <td id=\"T_26fa1_row49_col1\" class=\"data row49 col1\" >0.000346</td>\n",
       "      <td id=\"T_26fa1_row49_col2\" class=\"data row49 col2\" >9</td>\n",
       "      <td id=\"T_26fa1_row49_col3\" class=\"data row49 col3\" >50</td>\n",
       "      <td id=\"T_26fa1_row49_col4\" class=\"data row49 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_26fa1_row49_col5\" class=\"data row49 col5\" >http://model.geneontology.org/R-HSA-8854050</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row50\" class=\"row_heading level0 row50\" >50</th>\n",
       "      <td id=\"T_26fa1_row50_col0\" class=\"data row50 col0\" >Hedgehog 'off' state - Reactome</td>\n",
       "      <td id=\"T_26fa1_row50_col1\" class=\"data row50 col1\" >0.000373</td>\n",
       "      <td id=\"T_26fa1_row50_col2\" class=\"data row50 col2\" >9</td>\n",
       "      <td id=\"T_26fa1_row50_col3\" class=\"data row50 col3\" >46</td>\n",
       "      <td id=\"T_26fa1_row50_col4\" class=\"data row50 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_26fa1_row50_col5\" class=\"data row50 col5\" >http://model.geneontology.org/R-HSA-5610787</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row51\" class=\"row_heading level0 row51\" >51</th>\n",
       "      <td id=\"T_26fa1_row51_col0\" class=\"data row51 col0\" >Negative regulation of NOTCH4 signaling - Reactome</td>\n",
       "      <td id=\"T_26fa1_row51_col1\" class=\"data row51 col1\" >0.000380</td>\n",
       "      <td id=\"T_26fa1_row51_col2\" class=\"data row51 col2\" >9</td>\n",
       "      <td id=\"T_26fa1_row51_col3\" class=\"data row51 col3\" >49</td>\n",
       "      <td id=\"T_26fa1_row51_col4\" class=\"data row51 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_26fa1_row51_col5\" class=\"data row51 col5\" >http://model.geneontology.org/R-HSA-9604323</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row52\" class=\"row_heading level0 row52\" >52</th>\n",
       "      <td id=\"T_26fa1_row52_col0\" class=\"data row52 col0\" >Neddylation - Reactome</td>\n",
       "      <td id=\"T_26fa1_row52_col1\" class=\"data row52 col1\" >0.000394</td>\n",
       "      <td id=\"T_26fa1_row52_col2\" class=\"data row52 col2\" >10</td>\n",
       "      <td id=\"T_26fa1_row52_col3\" class=\"data row52 col3\" >62</td>\n",
       "      <td id=\"T_26fa1_row52_col4\" class=\"data row52 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1', 'RNF7']</td>\n",
       "      <td id=\"T_26fa1_row52_col5\" class=\"data row52 col5\" >http://model.geneontology.org/R-HSA-8951664</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row53\" class=\"row_heading level0 row53\" >53</th>\n",
       "      <td id=\"T_26fa1_row53_col0\" class=\"data row53 col0\" >Degradation of GLI1 by the proteasome - Reactome</td>\n",
       "      <td id=\"T_26fa1_row53_col1\" class=\"data row53 col1\" >0.000491</td>\n",
       "      <td id=\"T_26fa1_row53_col2\" class=\"data row53 col2\" >9</td>\n",
       "      <td id=\"T_26fa1_row53_col3\" class=\"data row53 col3\" >51</td>\n",
       "      <td id=\"T_26fa1_row53_col4\" class=\"data row53 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_26fa1_row53_col5\" class=\"data row53 col5\" >http://model.geneontology.org/R-HSA-5610780</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row54\" class=\"row_heading level0 row54\" >54</th>\n",
       "      <td id=\"T_26fa1_row54_col0\" class=\"data row54 col0\" >Antigen Presentation: Folding, assembly and peptide loading of class I MHC - Reactome</td>\n",
       "      <td id=\"T_26fa1_row54_col1\" class=\"data row54 col1\" >0.000522</td>\n",
       "      <td id=\"T_26fa1_row54_col2\" class=\"data row54 col2\" >9</td>\n",
       "      <td id=\"T_26fa1_row54_col3\" class=\"data row54 col3\" >50</td>\n",
       "      <td id=\"T_26fa1_row54_col4\" class=\"data row54 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_26fa1_row54_col5\" class=\"data row54 col5\" >http://model.geneontology.org/R-HSA-983170</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row55\" class=\"row_heading level0 row55\" >55</th>\n",
       "      <td id=\"T_26fa1_row55_col0\" class=\"data row55 col0\" >GLI3 is processed to GLI3R by the proteasome - Reactome</td>\n",
       "      <td id=\"T_26fa1_row55_col1\" class=\"data row55 col1\" >0.000561</td>\n",
       "      <td id=\"T_26fa1_row55_col2\" class=\"data row55 col2\" >9</td>\n",
       "      <td id=\"T_26fa1_row55_col3\" class=\"data row55 col3\" >52</td>\n",
       "      <td id=\"T_26fa1_row55_col4\" class=\"data row55 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_26fa1_row55_col5\" class=\"data row55 col5\" >http://model.geneontology.org/R-HSA-5610785</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row56\" class=\"row_heading level0 row56\" >56</th>\n",
       "      <td id=\"T_26fa1_row56_col0\" class=\"data row56 col0\" >Degradation of GLI2 by the proteasome - Reactome</td>\n",
       "      <td id=\"T_26fa1_row56_col1\" class=\"data row56 col1\" >0.000561</td>\n",
       "      <td id=\"T_26fa1_row56_col2\" class=\"data row56 col2\" >9</td>\n",
       "      <td id=\"T_26fa1_row56_col3\" class=\"data row56 col3\" >52</td>\n",
       "      <td id=\"T_26fa1_row56_col4\" class=\"data row56 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_26fa1_row56_col5\" class=\"data row56 col5\" >http://model.geneontology.org/R-HSA-5610783</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row57\" class=\"row_heading level0 row57\" >57</th>\n",
       "      <td id=\"T_26fa1_row57_col0\" class=\"data row57 col0\" >Regulation of RUNX2 expression and activity - Reactome</td>\n",
       "      <td id=\"T_26fa1_row57_col1\" class=\"data row57 col1\" >0.000595</td>\n",
       "      <td id=\"T_26fa1_row57_col2\" class=\"data row57 col2\" >9</td>\n",
       "      <td id=\"T_26fa1_row57_col3\" class=\"data row57 col3\" >54</td>\n",
       "      <td id=\"T_26fa1_row57_col4\" class=\"data row57 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_26fa1_row57_col5\" class=\"data row57 col5\" >http://model.geneontology.org/R-HSA-8939902</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row58\" class=\"row_heading level0 row58\" >58</th>\n",
       "      <td id=\"T_26fa1_row58_col0\" class=\"data row58 col0\" >Interleukin-1 signaling - Reactome</td>\n",
       "      <td id=\"T_26fa1_row58_col1\" class=\"data row58 col1\" >0.000712</td>\n",
       "      <td id=\"T_26fa1_row58_col2\" class=\"data row58 col2\" >10</td>\n",
       "      <td id=\"T_26fa1_row58_col3\" class=\"data row58 col3\" >64</td>\n",
       "      <td id=\"T_26fa1_row58_col4\" class=\"data row58 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1', 'MAP3K7']</td>\n",
       "      <td id=\"T_26fa1_row58_col5\" class=\"data row58 col5\" >http://model.geneontology.org/R-HSA-9020702</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row59\" class=\"row_heading level0 row59\" >59</th>\n",
       "      <td id=\"T_26fa1_row59_col0\" class=\"data row59 col0\" >MAPK6/MAPK4 signaling - Reactome</td>\n",
       "      <td id=\"T_26fa1_row59_col1\" class=\"data row59 col1\" >0.000893</td>\n",
       "      <td id=\"T_26fa1_row59_col2\" class=\"data row59 col2\" >10</td>\n",
       "      <td id=\"T_26fa1_row59_col3\" class=\"data row59 col3\" >62</td>\n",
       "      <td id=\"T_26fa1_row59_col4\" class=\"data row59 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'CDK1', 'PSMD14', 'PSMF1']</td>\n",
       "      <td id=\"T_26fa1_row59_col5\" class=\"data row59 col5\" >http://model.geneontology.org/R-HSA-5687128</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row60\" class=\"row_heading level0 row60\" >60</th>\n",
       "      <td id=\"T_26fa1_row60_col0\" class=\"data row60 col0\" >Downstream TCR signaling - Reactome</td>\n",
       "      <td id=\"T_26fa1_row60_col1\" class=\"data row60 col1\" >0.001877</td>\n",
       "      <td id=\"T_26fa1_row60_col2\" class=\"data row60 col2\" >12</td>\n",
       "      <td id=\"T_26fa1_row60_col3\" class=\"data row60 col3\" >74</td>\n",
       "      <td id=\"T_26fa1_row60_col4\" class=\"data row60 col4\" >['PSMD2', 'PSMC4', 'PSMC5', 'PSMC1', 'PSMD11', 'PSMC2', 'PSMD13', 'PSMD14', 'PSMF1', 'MAP3K7', 'set:PIP5K1A-C', 'set:Activator:PI3K']</td>\n",
       "      <td id=\"T_26fa1_row60_col5\" class=\"data row60 col5\" >http://model.geneontology.org/R-HSA-202424</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row61\" class=\"row_heading level0 row61\" >61</th>\n",
       "      <td id=\"T_26fa1_row61_col0\" class=\"data row61 col0\" >Synthesis of PC - Reactome</td>\n",
       "      <td id=\"T_26fa1_row61_col1\" class=\"data row61 col1\" >0.001973</td>\n",
       "      <td id=\"T_26fa1_row61_col2\" class=\"data row61 col2\" >7</td>\n",
       "      <td id=\"T_26fa1_row61_col3\" class=\"data row61 col3\" >21</td>\n",
       "      <td id=\"T_26fa1_row61_col4\" class=\"data row61 col4\" >['ABHD3', 'set:LPEAT', 'set:CSNK2(A1:A1/A1:A2/A2:A2)', 'set:AGPAT', 'set:PCYT1 dimer', 'set:PNPLA2/3', 'set:CTL1-5']</td>\n",
       "      <td id=\"T_26fa1_row61_col5\" class=\"data row61 col5\" >http://model.geneontology.org/R-HSA-1483191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_26fa1_level0_row62\" class=\"row_heading level0 row62\" >62</th>\n",
       "      <td id=\"T_26fa1_row62_col0\" class=\"data row62 col0\" >COPII-mediated vesicle transport - Reactome</td>\n",
       "      <td id=\"T_26fa1_row62_col1\" class=\"data row62 col1\" >0.002506</td>\n",
       "      <td id=\"T_26fa1_row62_col2\" class=\"data row62 col2\" >7</td>\n",
       "      <td id=\"T_26fa1_row62_col3\" class=\"data row62 col3\" >27</td>\n",
       "      <td id=\"T_26fa1_row62_col4\" class=\"data row62 col4\" >['SEC31A', 'SEC23A', 'TFG', 'SEC13', 'CSNK1D', 'set:SEC24', 'set:COPII cargo:receptors']</td>\n",
       "      <td id=\"T_26fa1_row62_col5\" class=\"data row62 col5\" >http://model.geneontology.org/R-HSA-204005</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x14fa3c9d0>"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "p97_weighted = enrich.enrich_wrapper('P97.csv','uniprot', FDR = 0.05, method='ncHGT', fpath = '../test_data/processed', table_number = 3)\n",
    "p97_weighted "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a8118fcf-d8d2-4219-8b43-1c507ecc7b13",
   "metadata": {
    "tags": []
   },
   "source": [
    "# Dilated and Arrhythmogenic Cardiomyopathy\n",
    "\n",
    "Reichart et al., 2022 \n",
    "\n",
    "https://www.science.org/doi/10.1126/science.abo1984?url_ver=Z39.88-2003&rfr_id=ori:rid:crossref.org&rfr_dat=cr_pub%20%200pubmed#supplementary-materials\n",
    "\n",
    "snRNA seq on human left and right ventricles from healthy and diseased donors. 5 genotypes: LMNA, RBM20, PKP2, TTN, and PVneg (pathogenic variant negative)\n",
    "\n",
    "\n",
    "cardiomyocytes: tables S7,8 of Reichart et al\n",
    "\n",
    "fibroblasts: tables S16,17 of Reichart et al\n",
    "\n",
    "smooth muscle cells: tables S23-24 of Reichart et al"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "64cf127b-21e5-41d5-b6c6-803e148bf013",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Left Ventricle Cardiomyocytes"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "24408158-8c49-41d3-9554-db5d1b1ff7e0",
   "metadata": {
    "tags": []
   },
   "source": [
    "### LMNA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "31730308-d40b-41c8-82a4-4f8bbaa5063b",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 299/299 [00:00<00:00, 15944.07it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 191 entities from 141 out of 689 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_410c1 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_410c1\">\n",
       "  <caption>Table S4: LMNA cardiomyocytes combined up/downregulated genes, unweighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_410c1_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_410c1_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_410c1_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_410c1_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_410c1_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_410c1_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_410c1_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_410c1_row0_col0\" class=\"data row0 col0\" >Regulation of Complement cascade - Reactome</td>\n",
       "      <td id=\"T_410c1_row0_col1\" class=\"data row0 col1\" >0.000004</td>\n",
       "      <td id=\"T_410c1_row0_col2\" class=\"data row0 col2\" >6</td>\n",
       "      <td id=\"T_410c1_row0_col3\" class=\"data row0 col3\" >11</td>\n",
       "      <td id=\"T_410c1_row0_col4\" class=\"data row0 col4\" >['C1R', 'C3', 'set:C3 convertases', 'set:Complement factor D', 'set:C4 activators', 'set:CD46, CR1:C4b:C3b complexes']</td>\n",
       "      <td id=\"T_410c1_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-977606</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_410c1_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_410c1_row1_col0\" class=\"data row1 col0\" >CD28 dependent PI3K/Akt signaling - Reactome</td>\n",
       "      <td id=\"T_410c1_row1_col1\" class=\"data row1 col1\" >0.000060</td>\n",
       "      <td id=\"T_410c1_row1_col2\" class=\"data row1 col2\" >6</td>\n",
       "      <td id=\"T_410c1_row1_col3\" class=\"data row1 col3\" >16</td>\n",
       "      <td id=\"T_410c1_row1_col4\" class=\"data row1 col4\" >['PIK3R1', 'MAP3K8', 'set:G-protein gamma subunit', 'set:Activator:PI3K', 'set:PI3K-regulatory subunit', 'set:PI3K alpha, beta, gamma']</td>\n",
       "      <td id=\"T_410c1_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-389357</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x13bc3e290>"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "LMNA_set = enrich.enrich_wrapper('LMNA_CM_comb.csv','Gene Symbol',method='set',FDR = 0.05,fpath = '../test_data/processed/', table_number = 4)\n",
    "LMNA_set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "6462946f-a803-49f6-9aa0-8b504dff9dcb",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 299/299 [00:00<00:00, 15988.38it/s]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_a9088 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_a9088\">\n",
       "  <caption>Table S5: LMNA cardiomyocytes combined up/downregulated genes, standard method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_a9088_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_a9088_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_a9088_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_a9088_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_a9088_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_a9088_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x12fd7ab00>"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "LMNA_standard= enrich.enrich_wrapper('LMNA_CM_comb.csv','Gene Symbol',method='standard',FDR = 0.05,fpath = '../test_data/processed/', table_number = 5)\n",
    "LMNA_standard"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "5afcfeee-3ef8-472f-927f-c185cda0fc44",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|█████████████████████████████████████████| 299/299 [02:01<00:00,  2.46it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 191 entities from 141 out of 689 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_1b959 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_1b959\">\n",
       "  <caption>Table S6: LMNA cardiomyocytes combined up/downregulated genes, weighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_1b959_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_1b959_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_1b959_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_1b959_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_1b959_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_1b959_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x13bc04df0>"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "LMNA_weighted  = enrich.enrich_wrapper('LMNA_CM_comb.csv','Gene Symbol',method='ncHGT',FDR = 0.05,fpath = '../test_data/processed/', table_number = 6)\n",
    "LMNA_weighted"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7f3ed318-ba88-4b38-a5dc-d5fba90a630d",
   "metadata": {
    "tags": []
   },
   "source": [
    "### RBM20"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "bb832643-8529-4088-831b-cd248b98d8d7",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 207/207 [00:00<00:00, 17563.23it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 156 entities from 115 out of 533 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_a9081 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_a9081\">\n",
       "  <caption>Table S7: RBM20 cardiomyocytes combined up/downregulated genes, unweighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_a9081_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_a9081_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_a9081_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_a9081_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_a9081_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_a9081_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x13bcf7040>"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "RBM20_step = enrich.enrich_wrapper('RBM20_CM_comb.csv','Gene Symbol',method='set',FDR = 0.05,fpath = '../test_data/processed/', table_number = 7)\n",
    "RBM20_step"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "6564a873-32ac-406a-9238-74eabf2608eb",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 207/207 [00:00<00:00, 17383.19it/s]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_9f901 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_9f901\">\n",
       "  <caption>Table S8: RBM20 cardiomyocytes combined up/downregulated genes, standard method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_9f901_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_9f901_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_9f901_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_9f901_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_9f901_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_9f901_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x14e3e3eb0>"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "RBM20_standard= enrich.enrich_wrapper('RBM20_CM_comb.csv','Gene Symbol',method='standard',FDR = 0.05,fpath = '../test_data/processed/',table_number = 8)\n",
    "RBM20_standard"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "adc1de44-613f-4f2e-a908-9ecec617f932",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|█████████████████████████████████████████| 207/207 [00:47<00:00,  4.40it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 156 entities from 115 out of 533 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_58232 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_58232\">\n",
       "  <caption>Table S9: RBM20 cardiomyocytes combined up/downregulated genes, weighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_58232_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_58232_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_58232_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_58232_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_58232_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_58232_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x13bc3e380>"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "RBM20_weighted = enrich.enrich_wrapper('RBM20_CM_comb.csv','Gene Symbol',method='ncHGT',FDR = 0.05,fpath = '../test_data/processed/',table_number = 9)\n",
    "RBM20_weighted"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "aaf146d9-cb80-4a94-bc48-1317a587b4ee",
   "metadata": {
    "tags": []
   },
   "source": [
    "### PKP2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "66ee92e4-39fb-4c91-9b12-79a60bf74cc6",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 302/302 [00:00<00:00, 17554.98it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 151 entities from 94 out of 403 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_0624b caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_0624b\">\n",
       "  <caption>Table S10: PKP2 cardiomyocytes combined up/downregulated genes, unweighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_0624b_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_0624b_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_0624b_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_0624b_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_0624b_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_0624b_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_0624b_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_0624b_row0_col0\" class=\"data row0 col0\" >Synthesis of PIPs at the plasma membrane - Reactome</td>\n",
       "      <td id=\"T_0624b_row0_col1\" class=\"data row0 col1\" >0.000007</td>\n",
       "      <td id=\"T_0624b_row0_col2\" class=\"data row0 col2\" >7</td>\n",
       "      <td id=\"T_0624b_row0_col3\" class=\"data row0 col3\" >21</td>\n",
       "      <td id=\"T_0624b_row0_col4\" class=\"data row0 col4\" >['set:PIK3C(1)', 'set:PIK3(2)', 'set:Activator:PI3K', 'set:PI3K-regulatory subunit', 'set:SYNJ/INPP5(1)', 'set:SYNJ/MTM(1)', 'set:INPP5(2)']</td>\n",
       "      <td id=\"T_0624b_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-1660499</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x14e3e22c0>"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PKP2_step = enrich.enrich_wrapper('PKP2_CM_comb.csv','Gene Symbol',method='set',FDR = 0.05,fpath = '../test_data/processed/',table_number = 10)\n",
    "PKP2_step"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "3d73cb3e-dacd-4473-a9b1-07ea9ddb5853",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 302/302 [00:00<00:00, 17210.09it/s]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_cb3fd caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_cb3fd\">\n",
       "  <caption>Table S11: PKP2 cardiomyocytes combined up/downregulated genes, standard method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_cb3fd_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_cb3fd_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_cb3fd_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_cb3fd_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_cb3fd_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_cb3fd_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x13bb68a60>"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PKP2_standard = enrich.enrich_wrapper('PKP2_CM_comb.csv','Gene Symbol',method='standard',FDR = 0.05,fpath = '../test_data/processed/',table_number = 11)\n",
    "PKP2_standard"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "8adaf263-09dd-42ac-a923-b972f692b2fa",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|█████████████████████████████████████████| 302/302 [01:35<00:00,  3.15it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 151 entities from 94 out of 403 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_918a2 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_918a2\">\n",
       "  <caption>Table S12: PKP2 cardiomyocytes combined up/downregulated genes, weighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_918a2_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_918a2_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_918a2_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_918a2_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_918a2_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_918a2_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x14f5e1870>"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PKP2_weighted = enrich.enrich_wrapper('PKP2_CM_comb.csv','Gene Symbol',method='ncHGT',FDR = 0.05,fpath = '../test_data/processed/',table_number = 12)\n",
    "PKP2_weighted"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2e180a89-dc0f-4dd7-ab7d-1ba1f5d79219",
   "metadata": {
    "tags": []
   },
   "source": [
    "### TTN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "cfe2ec8c-3345-4eb1-8e97-68b1a122400d",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 362/362 [00:00<00:00, 17538.65it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 193 entities from 133 out of 602 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_371f2 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_371f2\">\n",
       "  <caption>Table S13: TTN cardiomyocytes combined up/downregulated genes, unweighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_371f2_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_371f2_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_371f2_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_371f2_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_371f2_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_371f2_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_371f2_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_371f2_row0_col0\" class=\"data row0 col0\" >Regulation of Complement cascade - Reactome</td>\n",
       "      <td id=\"T_371f2_row0_col1\" class=\"data row0 col1\" >0.000005</td>\n",
       "      <td id=\"T_371f2_row0_col2\" class=\"data row0 col2\" >6</td>\n",
       "      <td id=\"T_371f2_row0_col3\" class=\"data row0 col3\" >11</td>\n",
       "      <td id=\"T_371f2_row0_col4\" class=\"data row0 col4\" >['C1R', 'C3', 'set:C3 convertases', 'set:Complement factor D', 'set:C4 activators', 'set:CD46, CR1:C4b:C3b complexes']</td>\n",
       "      <td id=\"T_371f2_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-977606</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_371f2_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_371f2_row1_col0\" class=\"data row1 col0\" >CD28 dependent PI3K/Akt signaling - Reactome</td>\n",
       "      <td id=\"T_371f2_row1_col1\" class=\"data row1 col1\" >0.000064</td>\n",
       "      <td id=\"T_371f2_row1_col2\" class=\"data row1 col2\" >6</td>\n",
       "      <td id=\"T_371f2_row1_col3\" class=\"data row1 col3\" >16</td>\n",
       "      <td id=\"T_371f2_row1_col4\" class=\"data row1 col4\" >['PIK3R1', 'MAP3K8', 'set:G-protein gamma subunit', 'set:Activator:PI3K', 'set:PI3K-regulatory subunit', 'set:PI3K alpha, beta, gamma']</td>\n",
       "      <td id=\"T_371f2_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-389357</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x13bc3f250>"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TTN_step = enrich.enrich_wrapper('TTN_CM_comb.csv','Gene Symbol',method='set',FDR = 0.05,fpath = '../test_data/processed/',table_number = 13)\n",
    "TTN_step"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "ae1ec971-7730-4989-93f9-b71c1da1b41f",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 362/362 [00:00<00:00, 16767.95it/s]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_e7f92 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_e7f92\">\n",
       "  <caption>Table S14: TTN cardiomyocytes combined up/downregulated genes, standard method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_e7f92_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_e7f92_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_e7f92_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_e7f92_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_e7f92_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_e7f92_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x13bb68c70>"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TTN_standard = enrich.enrich_wrapper('TTN_CM_comb.csv','Gene Symbol',method='standard',FDR = 0.05,fpath = '../test_data/processed/',table_number = 14)\n",
    "TTN_standard"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "b239b526-ac75-4338-bd71-12708a46a661",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|█████████████████████████████████████████| 362/362 [01:58<00:00,  3.05it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 193 entities from 133 out of 602 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_75106 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_75106\">\n",
       "  <caption>Table S15: TTN cardiomyocytes combined up/downregulated genes, weighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_75106_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_75106_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_75106_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_75106_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_75106_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_75106_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x13bc3ee60>"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TTN_weighted = enrich.enrich_wrapper('TTN_CM_comb.csv','Gene Symbol',method='ncHGT',FDR = 0.05,fpath = '../test_data/processed/',table_number = 15)\n",
    "TTN_weighted"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3cb76d32-ee25-4e0d-8e0a-07e4845811e8",
   "metadata": {
    "tags": []
   },
   "source": [
    "### PVneg"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "cd3ef738-30f1-42aa-b684-0e36d86f0d97",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|████████████████████████████████████████| 93/93 [00:00<00:00, 16786.60it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 59 entities from 42 out of 178 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_47840 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_47840\">\n",
       "  <caption>Table S16: PVneg cardiomyocytes combined up/downregulated genes, unweighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_47840_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_47840_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_47840_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_47840_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_47840_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_47840_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x15a89faf0>"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PVneg_step = enrich.enrich_wrapper('PVneg_CM_comb.csv','Gene Symbol',method='set',FDR = 0.05,fpath = '../test_data/processed/',table_number = 16)\n",
    "PVneg_step"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "03fc82dc-0ced-4786-8a90-c3318ad88350",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|████████████████████████████████████████| 93/93 [00:00<00:00, 16594.50it/s]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_7501a caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_7501a\">\n",
       "  <caption>Table S17: PVneg cardiomyocytes combined up/downregulated genes, standard method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_7501a_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_7501a_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_7501a_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_7501a_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_7501a_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_7501a_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x15a89c9a0>"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PVneg_standard = enrich.enrich_wrapper('PVneg_CM_comb.csv','Gene Symbol',method='standard',FDR = 0.05,fpath = '../test_data/processed/',table_number = 17)\n",
    "PVneg_standard"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "b36594b6-2360-444f-8023-cd04c60a5147",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|███████████████████████████████████████████| 93/93 [00:46<00:00,  2.00it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 59 entities from 42 out of 178 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_dbcb4 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_dbcb4\">\n",
       "  <caption>Table S18: PVneg cardiomyocytes combined up/downregulated genes, weighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_dbcb4_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_dbcb4_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_dbcb4_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_dbcb4_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_dbcb4_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_dbcb4_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x12fd7ae00>"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PVneg_weighted = enrich.enrich_wrapper('PVneg_CM_comb.csv','Gene Symbol',method='ncHGT',FDR = 0.05,fpath = '../test_data/processed/',table_number = 18)\n",
    "PVneg_weighted"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bbe61947-445f-44f2-b955-ae16449051c2",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Left Ventricle Fibroblasts"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2758be91-9a1a-49a0-b620-138a09c60f1b",
   "metadata": {
    "tags": []
   },
   "source": [
    "### LMNA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "bfc0de35-087b-4202-9e0b-e63b9d53d157",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 401/401 [00:00<00:00, 17013.28it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 284 entities from 234 out of 872 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_279d1 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_279d1\">\n",
       "  <caption>Table S19: LMNA fibroblasts combined up/downregulated genes, unweighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_279d1_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_279d1_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_279d1_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_279d1_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_279d1_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_279d1_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_279d1_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_279d1_row0_col0\" class=\"data row0 col0\" >Peptide chain elongation - Reactome</td>\n",
       "      <td id=\"T_279d1_row0_col1\" class=\"data row0 col1\" >0.000000</td>\n",
       "      <td id=\"T_279d1_row0_col2\" class=\"data row0 col2\" >20</td>\n",
       "      <td id=\"T_279d1_row0_col3\" class=\"data row0 col3\" >79</td>\n",
       "      <td id=\"T_279d1_row0_col4\" class=\"data row0 col4\" >['RPS17', 'RPL13A', 'RPS18', 'RPLP0', 'RPLP2', 'RPL34', 'RPS20', 'RPL27A', 'RPL7', 'RPL35', 'RPL38', 'RPLP1', 'RPS15A', 'RPL23A', 'RPS5', 'RPS19', 'RPL27', 'RPS15', 'set:RPL36A,RPL36AL', 'set:RPL3,RPL3L']</td>\n",
       "      <td id=\"T_279d1_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-156902</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_279d1_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_279d1_row1_col0\" class=\"data row1 col0\" >Eukaryotic Translation Elongation - Reactome</td>\n",
       "      <td id=\"T_279d1_row1_col1\" class=\"data row1 col1\" >0.000001</td>\n",
       "      <td id=\"T_279d1_row1_col2\" class=\"data row1 col2\" >20</td>\n",
       "      <td id=\"T_279d1_row1_col3\" class=\"data row1 col3\" >83</td>\n",
       "      <td id=\"T_279d1_row1_col4\" class=\"data row1 col4\" >['RPS17', 'RPL13A', 'RPS18', 'RPLP0', 'RPLP2', 'RPL34', 'RPS20', 'RPL27A', 'RPL7', 'RPL35', 'RPL38', 'RPLP1', 'RPS15A', 'RPL23A', 'RPS5', 'RPS19', 'RPL27', 'RPS15', 'set:RPL36A,RPL36AL', 'set:RPL3,RPL3L']</td>\n",
       "      <td id=\"T_279d1_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-156842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_279d1_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "      <td id=\"T_279d1_row2_col0\" class=\"data row2 col0\" >Regulation of Complement cascade - Reactome</td>\n",
       "      <td id=\"T_279d1_row2_col1\" class=\"data row2 col1\" >0.000002</td>\n",
       "      <td id=\"T_279d1_row2_col2\" class=\"data row2 col2\" >7</td>\n",
       "      <td id=\"T_279d1_row2_col3\" class=\"data row2 col3\" >11</td>\n",
       "      <td id=\"T_279d1_row2_col4\" class=\"data row2 col4\" >['C3', 'C1R', 'set:C3 convertases', 'set:Complement factor D', 'set:C4 activators', 'set:Activated thrombin, (ELANE)', 'set:CD46, CR1:C4b:C3b complexes']</td>\n",
       "      <td id=\"T_279d1_row2_col5\" class=\"data row2 col5\" >http://model.geneontology.org/R-HSA-977606</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_279d1_level0_row3\" class=\"row_heading level0 row3\" >3</th>\n",
       "      <td id=\"T_279d1_row3_col0\" class=\"data row3 col0\" >G alpha (i) signalling events - Reactome</td>\n",
       "      <td id=\"T_279d1_row3_col1\" class=\"data row3 col1\" >0.000083</td>\n",
       "      <td id=\"T_279d1_row3_col2\" class=\"data row3 col2\" >5</td>\n",
       "      <td id=\"T_279d1_row3_col3\" class=\"data row3 col3\" >8</td>\n",
       "      <td id=\"T_279d1_row3_col4\" class=\"data row3 col4\" >['set:RGS1,3,4,5,6,7,8,9,10,11,12,13,14,16,17,18,19,20,21', 'set:GNB', 'set:Ligand:GPCR complexes that activate Gi', 'set:G alpha (i)', 'set:Light-sensing opsins']</td>\n",
       "      <td id=\"T_279d1_row3_col5\" class=\"data row3 col5\" >http://model.geneontology.org/R-HSA-418594</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x12fd78820>"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "LMNA_step = enrich.enrich_wrapper('LMNA_FB_comb.csv','Gene Symbol',method='set',FDR = 0.05,fpath = '../test_data/processed/',table_number = 19)\n",
    "LMNA_step"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "0a5e1d12-6a9d-4bdb-b38d-16fd15ebe188",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 401/401 [00:00<00:00, 16269.10it/s]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_322df caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_322df\">\n",
       "  <caption>Table S20: LMNA fibroblasts combined up/downregulated genes, standard method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_322df_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_322df_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_322df_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_322df_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_322df_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_322df_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_322df_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_322df_row0_col0\" class=\"data row0 col0\" >Peptide chain elongation - Reactome</td>\n",
       "      <td id=\"T_322df_row0_col1\" class=\"data row0 col1\" >0.000000</td>\n",
       "      <td id=\"T_322df_row0_col2\" class=\"data row0 col2\" >20</td>\n",
       "      <td id=\"T_322df_row0_col3\" class=\"data row0 col3\" >88</td>\n",
       "      <td id=\"T_322df_row0_col4\" class=\"data row0 col4\" >['RPS17', 'RPL13A', 'RPS18', 'RPLP0', 'RPLP2', 'RPL34', 'RPS20', 'RPL27A', 'RPL7', 'RPL3', 'RPL35', 'RPL38', 'RPLP1', 'RPS15A', 'RPL23A', 'RPS5', 'RPS19', 'RPL27', 'RPL36AL', 'RPS15']</td>\n",
       "      <td id=\"T_322df_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-156902</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_322df_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_322df_row1_col0\" class=\"data row1 col0\" >Eukaryotic Translation Elongation - Reactome</td>\n",
       "      <td id=\"T_322df_row1_col1\" class=\"data row1 col1\" >0.000000</td>\n",
       "      <td id=\"T_322df_row1_col2\" class=\"data row1 col2\" >20</td>\n",
       "      <td id=\"T_322df_row1_col3\" class=\"data row1 col3\" >94</td>\n",
       "      <td id=\"T_322df_row1_col4\" class=\"data row1 col4\" >['RPS17', 'RPL13A', 'RPS18', 'RPLP0', 'RPLP2', 'RPL34', 'RPS20', 'RPL27A', 'RPL7', 'RPL3', 'RPL35', 'RPL38', 'RPLP1', 'RPS15A', 'RPL23A', 'RPS5', 'RPS19', 'RPL27', 'RPL36AL', 'RPS15']</td>\n",
       "      <td id=\"T_322df_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-156842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_322df_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "      <td id=\"T_322df_row2_col0\" class=\"data row2 col0\" >Formation of ATP by chemiosmotic coupling - Reactome</td>\n",
       "      <td id=\"T_322df_row2_col1\" class=\"data row2 col1\" >0.000033</td>\n",
       "      <td id=\"T_322df_row2_col2\" class=\"data row2 col2\" >14</td>\n",
       "      <td id=\"T_322df_row2_col3\" class=\"data row2 col3\" >89</td>\n",
       "      <td id=\"T_322df_row2_col4\" class=\"data row2 col4\" >['MT-ND2', 'MT-ND1', 'MT-CO1', 'MT-ND4', 'SLC25A6', 'NDUFA1', 'UQCRB', 'MT-ND6', 'MT-CYB', 'MT-ATP6', 'MT-CO2', 'NDUFA3', 'SLC25A5', 'COX7C']</td>\n",
       "      <td id=\"T_322df_row2_col5\" class=\"data row2 col5\" >http://model.geneontology.org/R-HSA-163210</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x12fd786a0>"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "LMNA_standard = enrich.enrich_wrapper('LMNA_FB_comb.csv','Gene Symbol',method='standard',FDR = 0.05,fpath = '../test_data/processed/',table_number = 20)\n",
    "LMNA_standard"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "172c25a8-217e-4f7b-8e90-2564cccba993",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|█████████████████████████████████████████| 401/401 [02:38<00:00,  2.52it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 284 entities from 234 out of 872 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_96105 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_96105\">\n",
       "  <caption>Table S21: LMNA fibroblasts combined up/downregulated genes, weighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_96105_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_96105_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_96105_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_96105_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_96105_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_96105_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_96105_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_96105_row0_col0\" class=\"data row0 col0\" >Peptide chain elongation - Reactome</td>\n",
       "      <td id=\"T_96105_row0_col1\" class=\"data row0 col1\" >0.000000</td>\n",
       "      <td id=\"T_96105_row0_col2\" class=\"data row0 col2\" >20</td>\n",
       "      <td id=\"T_96105_row0_col3\" class=\"data row0 col3\" >79</td>\n",
       "      <td id=\"T_96105_row0_col4\" class=\"data row0 col4\" >['RPS17', 'RPL13A', 'RPS18', 'RPLP0', 'RPLP2', 'RPL34', 'RPS20', 'RPL27A', 'RPL7', 'RPL35', 'RPL38', 'RPLP1', 'RPS15A', 'RPL23A', 'RPS5', 'RPS19', 'RPL27', 'RPS15', 'set:RPL36A,RPL36AL', 'set:RPL3,RPL3L']</td>\n",
       "      <td id=\"T_96105_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-156902</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_96105_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_96105_row1_col0\" class=\"data row1 col0\" >Eukaryotic Translation Elongation - Reactome</td>\n",
       "      <td id=\"T_96105_row1_col1\" class=\"data row1 col1\" >0.000000</td>\n",
       "      <td id=\"T_96105_row1_col2\" class=\"data row1 col2\" >20</td>\n",
       "      <td id=\"T_96105_row1_col3\" class=\"data row1 col3\" >83</td>\n",
       "      <td id=\"T_96105_row1_col4\" class=\"data row1 col4\" >['RPS17', 'RPL13A', 'RPS18', 'RPLP0', 'RPLP2', 'RPL34', 'RPS20', 'RPL27A', 'RPL7', 'RPL35', 'RPL38', 'RPLP1', 'RPS15A', 'RPL23A', 'RPS5', 'RPS19', 'RPL27', 'RPS15', 'set:RPL36A,RPL36AL', 'set:RPL3,RPL3L']</td>\n",
       "      <td id=\"T_96105_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-156842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_96105_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "      <td id=\"T_96105_row2_col0\" class=\"data row2 col0\" >Formation of ATP by chemiosmotic coupling - Reactome</td>\n",
       "      <td id=\"T_96105_row2_col1\" class=\"data row2 col1\" >0.000016</td>\n",
       "      <td id=\"T_96105_row2_col2\" class=\"data row2 col2\" >13</td>\n",
       "      <td id=\"T_96105_row2_col3\" class=\"data row2 col3\" >88</td>\n",
       "      <td id=\"T_96105_row2_col4\" class=\"data row2 col4\" >['MT-ND2', 'MT-ND1', 'MT-CO1', 'MT-ND4', 'NDUFA1', 'UQCRB', 'MT-ND6', 'MT-CYB', 'MT-ATP6', 'MT-CO2', 'NDUFA3', 'COX7C', 'set:ANT homodimer']</td>\n",
       "      <td id=\"T_96105_row2_col5\" class=\"data row2 col5\" >http://model.geneontology.org/R-HSA-163210</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_96105_level0_row3\" class=\"row_heading level0 row3\" >3</th>\n",
       "      <td id=\"T_96105_row3_col0\" class=\"data row3 col0\" >TP53 Regulates Metabolic Genes - Reactome</td>\n",
       "      <td id=\"T_96105_row3_col1\" class=\"data row3 col1\" >0.000052</td>\n",
       "      <td id=\"T_96105_row3_col2\" class=\"data row3 col2\" >10</td>\n",
       "      <td id=\"T_96105_row3_col3\" class=\"data row3 col3\" >46</td>\n",
       "      <td id=\"T_96105_row3_col4\" class=\"data row3 col4\" >['MT-CO1', 'G6PD', 'UQCRB', 'MT-CYB', 'MT-CO2', 'COX7C', 'PRDX1', 'set:PRDX1,2,5', 'set:Active AKT', 'set:AMPK gamma']</td>\n",
       "      <td id=\"T_96105_row3_col5\" class=\"data row3 col5\" >http://model.geneontology.org/R-HSA-5628897</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_96105_level0_row4\" class=\"row_heading level0 row4\" >4</th>\n",
       "      <td id=\"T_96105_row4_col0\" class=\"data row4 col0\" >The fatty acid cycling model - Reactome</td>\n",
       "      <td id=\"T_96105_row4_col1\" class=\"data row4 col1\" >0.000071</td>\n",
       "      <td id=\"T_96105_row4_col2\" class=\"data row4 col2\" >11</td>\n",
       "      <td id=\"T_96105_row4_col3\" class=\"data row4 col3\" >70</td>\n",
       "      <td id=\"T_96105_row4_col4\" class=\"data row4 col4\" >['MT-ND2', 'MT-ND1', 'MT-CO1', 'MT-ND4', 'NDUFA1', 'UQCRB', 'MT-ND6', 'MT-CYB', 'MT-CO2', 'NDUFA3', 'COX7C']</td>\n",
       "      <td id=\"T_96105_row4_col5\" class=\"data row4 col5\" >http://model.geneontology.org/R-HSA-167826</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_96105_level0_row5\" class=\"row_heading level0 row5\" >5</th>\n",
       "      <td id=\"T_96105_row5_col0\" class=\"data row5 col0\" >The proton buffering model - Reactome</td>\n",
       "      <td id=\"T_96105_row5_col1\" class=\"data row5 col1\" >0.000071</td>\n",
       "      <td id=\"T_96105_row5_col2\" class=\"data row5 col2\" >11</td>\n",
       "      <td id=\"T_96105_row5_col3\" class=\"data row5 col3\" >70</td>\n",
       "      <td id=\"T_96105_row5_col4\" class=\"data row5 col4\" >['MT-ND2', 'MT-ND1', 'MT-CO1', 'MT-ND4', 'NDUFA1', 'UQCRB', 'MT-ND6', 'MT-CYB', 'MT-CO2', 'NDUFA3', 'COX7C']</td>\n",
       "      <td id=\"T_96105_row5_col5\" class=\"data row5 col5\" >http://model.geneontology.org/R-HSA-167827</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_96105_level0_row6\" class=\"row_heading level0 row6\" >6</th>\n",
       "      <td id=\"T_96105_row6_col0\" class=\"data row6 col0\" >Regulation of Complement cascade - Reactome</td>\n",
       "      <td id=\"T_96105_row6_col1\" class=\"data row6 col1\" >0.000178</td>\n",
       "      <td id=\"T_96105_row6_col2\" class=\"data row6 col2\" >7</td>\n",
       "      <td id=\"T_96105_row6_col3\" class=\"data row6 col3\" >11</td>\n",
       "      <td id=\"T_96105_row6_col4\" class=\"data row6 col4\" >['C3', 'C1R', 'set:C3 convertases', 'set:Complement factor D', 'set:C4 activators', 'set:Activated thrombin, (ELANE)', 'set:CD46, CR1:C4b:C3b complexes']</td>\n",
       "      <td id=\"T_96105_row6_col5\" class=\"data row6 col5\" >http://model.geneontology.org/R-HSA-977606</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x12fd7b790>"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "LMNA_weighted = enrich.enrich_wrapper('LMNA_FB_comb.csv','Gene Symbol',method='ncHGT',FDR = 0.05,fpath = '../test_data/processed/',table_number = 21)\n",
    "LMNA_weighted"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d26bf791-4b41-4e9a-a23f-d471e7236879",
   "metadata": {
    "tags": []
   },
   "source": [
    "### RBM20"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "be2a17c8-7bd2-4f95-875b-1b2b17928afc",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 283/283 [00:00<00:00, 17465.72it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 193 entities from 163 out of 625 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_fa884 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_fa884\">\n",
       "  <caption>Table S22: RBM20 fibroblasts combined up/downregulated genes, unweighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_fa884_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_fa884_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_fa884_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_fa884_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_fa884_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_fa884_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_fa884_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_fa884_row0_col0\" class=\"data row0 col0\" >Formation of ATP by chemiosmotic coupling - Reactome</td>\n",
       "      <td id=\"T_fa884_row0_col1\" class=\"data row0 col1\" >0.000015</td>\n",
       "      <td id=\"T_fa884_row0_col2\" class=\"data row0 col2\" >15</td>\n",
       "      <td id=\"T_fa884_row0_col3\" class=\"data row0 col3\" >88</td>\n",
       "      <td id=\"T_fa884_row0_col4\" class=\"data row0 col4\" >['MT-ND4', 'MT-ND2', 'MT-ND1', 'MT-ATP6', 'MT-ND6', 'MT-CYB', 'MT-CO1', 'UQCRB', 'MT-CO2', 'MT-ND3', 'MT-CO3', 'MT-ND5', 'NDUFA1', 'COX7C', 'set:ANT homodimer']</td>\n",
       "      <td id=\"T_fa884_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-163210</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_fa884_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_fa884_row1_col0\" class=\"data row1 col0\" >Peptide chain elongation - Reactome</td>\n",
       "      <td id=\"T_fa884_row1_col1\" class=\"data row1 col1\" >0.000019</td>\n",
       "      <td id=\"T_fa884_row1_col2\" class=\"data row1 col2\" >14</td>\n",
       "      <td id=\"T_fa884_row1_col3\" class=\"data row1 col3\" >79</td>\n",
       "      <td id=\"T_fa884_row1_col4\" class=\"data row1 col4\" >['RPS17', 'RPL13A', 'RPS18', 'RPS15A', 'RPL27A', 'RPLP2', 'RPL34', 'RPS20', 'RPLP0', 'RPS5', 'RPL7', 'RPL38', 'RPL27', 'set:RPL3,RPL3L']</td>\n",
       "      <td id=\"T_fa884_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-156902</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_fa884_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "      <td id=\"T_fa884_row2_col0\" class=\"data row2 col0\" >The proton buffering model - Reactome</td>\n",
       "      <td id=\"T_fa884_row2_col1\" class=\"data row2 col1\" >0.000022</td>\n",
       "      <td id=\"T_fa884_row2_col2\" class=\"data row2 col2\" >13</td>\n",
       "      <td id=\"T_fa884_row2_col3\" class=\"data row2 col3\" >70</td>\n",
       "      <td id=\"T_fa884_row2_col4\" class=\"data row2 col4\" >['MT-ND4', 'MT-ND2', 'MT-ND1', 'MT-ND6', 'MT-CYB', 'MT-CO1', 'UQCRB', 'MT-CO2', 'MT-ND3', 'MT-CO3', 'MT-ND5', 'NDUFA1', 'COX7C']</td>\n",
       "      <td id=\"T_fa884_row2_col5\" class=\"data row2 col5\" >http://model.geneontology.org/R-HSA-167827</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_fa884_level0_row3\" class=\"row_heading level0 row3\" >3</th>\n",
       "      <td id=\"T_fa884_row3_col0\" class=\"data row3 col0\" >The fatty acid cycling model - Reactome</td>\n",
       "      <td id=\"T_fa884_row3_col1\" class=\"data row3 col1\" >0.000022</td>\n",
       "      <td id=\"T_fa884_row3_col2\" class=\"data row3 col2\" >13</td>\n",
       "      <td id=\"T_fa884_row3_col3\" class=\"data row3 col3\" >70</td>\n",
       "      <td id=\"T_fa884_row3_col4\" class=\"data row3 col4\" >['MT-ND4', 'MT-ND2', 'MT-ND1', 'MT-ND6', 'MT-CYB', 'MT-CO1', 'UQCRB', 'MT-CO2', 'MT-ND3', 'MT-CO3', 'MT-ND5', 'NDUFA1', 'COX7C']</td>\n",
       "      <td id=\"T_fa884_row3_col5\" class=\"data row3 col5\" >http://model.geneontology.org/R-HSA-167826</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_fa884_level0_row4\" class=\"row_heading level0 row4\" >4</th>\n",
       "      <td id=\"T_fa884_row4_col0\" class=\"data row4 col0\" >Eukaryotic Translation Elongation - Reactome</td>\n",
       "      <td id=\"T_fa884_row4_col1\" class=\"data row4 col1\" >0.000034</td>\n",
       "      <td id=\"T_fa884_row4_col2\" class=\"data row4 col2\" >14</td>\n",
       "      <td id=\"T_fa884_row4_col3\" class=\"data row4 col3\" >83</td>\n",
       "      <td id=\"T_fa884_row4_col4\" class=\"data row4 col4\" >['RPS17', 'RPL13A', 'RPS18', 'RPS15A', 'RPL27A', 'RPLP2', 'RPL34', 'RPS20', 'RPLP0', 'RPS5', 'RPL7', 'RPL38', 'RPL27', 'set:RPL3,RPL3L']</td>\n",
       "      <td id=\"T_fa884_row4_col5\" class=\"data row4 col5\" >http://model.geneontology.org/R-HSA-156842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_fa884_level0_row5\" class=\"row_heading level0 row5\" >5</th>\n",
       "      <td id=\"T_fa884_row5_col0\" class=\"data row5 col0\" >Regulation of Complement cascade - Reactome</td>\n",
       "      <td id=\"T_fa884_row5_col1\" class=\"data row5 col1\" >0.000092</td>\n",
       "      <td id=\"T_fa884_row5_col2\" class=\"data row5 col2\" >5</td>\n",
       "      <td id=\"T_fa884_row5_col3\" class=\"data row5 col3\" >11</td>\n",
       "      <td id=\"T_fa884_row5_col4\" class=\"data row5 col4\" >['C3', 'set:C3 convertases', 'set:C4 activators', 'set:Activated thrombin, (ELANE)', 'set:CD46, CR1:C4b:C3b complexes']</td>\n",
       "      <td id=\"T_fa884_row5_col5\" class=\"data row5 col5\" >http://model.geneontology.org/R-HSA-977606</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x12fd7b6d0>"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "RBM20_set = enrich.enrich_wrapper('RBM20_FB_comb.csv','Gene Symbol',method='set',FDR = 0.05,fpath = '../test_data/processed/',table_number = 22)\n",
    "RBM20_set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "0432287a-1671-4221-abde-a9ba6c95db09",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 283/283 [00:00<00:00, 15974.32it/s]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_36cb2 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_36cb2\">\n",
       "  <caption>Table S23: RBM20 fibroblasts combined up/downregulated genes, standard method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_36cb2_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_36cb2_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_36cb2_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_36cb2_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_36cb2_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_36cb2_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_36cb2_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_36cb2_row0_col0\" class=\"data row0 col0\" >Formation of ATP by chemiosmotic coupling - Reactome</td>\n",
       "      <td id=\"T_36cb2_row0_col1\" class=\"data row0 col1\" >0.000000</td>\n",
       "      <td id=\"T_36cb2_row0_col2\" class=\"data row0 col2\" >15</td>\n",
       "      <td id=\"T_36cb2_row0_col3\" class=\"data row0 col3\" >89</td>\n",
       "      <td id=\"T_36cb2_row0_col4\" class=\"data row0 col4\" >['MT-ND4', 'MT-ND2', 'MT-ND1', 'MT-ATP6', 'MT-ND6', 'MT-CYB', 'MT-CO1', 'UQCRB', 'MT-CO2', 'MT-ND3', 'MT-CO3', 'MT-ND5', 'NDUFA1', 'SLC25A6', 'COX7C']</td>\n",
       "      <td id=\"T_36cb2_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-163210</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_36cb2_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_36cb2_row1_col0\" class=\"data row1 col0\" >The proton buffering model - Reactome</td>\n",
       "      <td id=\"T_36cb2_row1_col1\" class=\"data row1 col1\" >0.000000</td>\n",
       "      <td id=\"T_36cb2_row1_col2\" class=\"data row1 col2\" >13</td>\n",
       "      <td id=\"T_36cb2_row1_col3\" class=\"data row1 col3\" >74</td>\n",
       "      <td id=\"T_36cb2_row1_col4\" class=\"data row1 col4\" >['MT-ND4', 'MT-ND2', 'MT-ND1', 'MT-ND6', 'MT-CYB', 'MT-CO1', 'UQCRB', 'MT-CO2', 'MT-ND3', 'MT-CO3', 'MT-ND5', 'NDUFA1', 'COX7C']</td>\n",
       "      <td id=\"T_36cb2_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-167827</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_36cb2_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "      <td id=\"T_36cb2_row2_col0\" class=\"data row2 col0\" >The fatty acid cycling model - Reactome</td>\n",
       "      <td id=\"T_36cb2_row2_col1\" class=\"data row2 col1\" >0.000000</td>\n",
       "      <td id=\"T_36cb2_row2_col2\" class=\"data row2 col2\" >13</td>\n",
       "      <td id=\"T_36cb2_row2_col3\" class=\"data row2 col3\" >74</td>\n",
       "      <td id=\"T_36cb2_row2_col4\" class=\"data row2 col4\" >['MT-ND4', 'MT-ND2', 'MT-ND1', 'MT-ND6', 'MT-CYB', 'MT-CO1', 'UQCRB', 'MT-CO2', 'MT-ND3', 'MT-CO3', 'MT-ND5', 'NDUFA1', 'COX7C']</td>\n",
       "      <td id=\"T_36cb2_row2_col5\" class=\"data row2 col5\" >http://model.geneontology.org/R-HSA-167826</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_36cb2_level0_row3\" class=\"row_heading level0 row3\" >3</th>\n",
       "      <td id=\"T_36cb2_row3_col0\" class=\"data row3 col0\" >Peptide chain elongation - Reactome</td>\n",
       "      <td id=\"T_36cb2_row3_col1\" class=\"data row3 col1\" >0.000000</td>\n",
       "      <td id=\"T_36cb2_row3_col2\" class=\"data row3 col2\" >14</td>\n",
       "      <td id=\"T_36cb2_row3_col3\" class=\"data row3 col3\" >88</td>\n",
       "      <td id=\"T_36cb2_row3_col4\" class=\"data row3 col4\" >['RPS17', 'RPL13A', 'RPS18', 'RPS15A', 'RPL27A', 'RPLP2', 'RPL34', 'RPL3L', 'RPS20', 'RPLP0', 'RPS5', 'RPL7', 'RPL38', 'RPL27']</td>\n",
       "      <td id=\"T_36cb2_row3_col5\" class=\"data row3 col5\" >http://model.geneontology.org/R-HSA-156902</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_36cb2_level0_row4\" class=\"row_heading level0 row4\" >4</th>\n",
       "      <td id=\"T_36cb2_row4_col0\" class=\"data row4 col0\" >Eukaryotic Translation Elongation - Reactome</td>\n",
       "      <td id=\"T_36cb2_row4_col1\" class=\"data row4 col1\" >0.000001</td>\n",
       "      <td id=\"T_36cb2_row4_col2\" class=\"data row4 col2\" >14</td>\n",
       "      <td id=\"T_36cb2_row4_col3\" class=\"data row4 col3\" >94</td>\n",
       "      <td id=\"T_36cb2_row4_col4\" class=\"data row4 col4\" >['RPS17', 'RPL13A', 'RPS18', 'RPS15A', 'RPL27A', 'RPLP2', 'RPL34', 'RPL3L', 'RPS20', 'RPLP0', 'RPS5', 'RPL7', 'RPL38', 'RPL27']</td>\n",
       "      <td id=\"T_36cb2_row4_col5\" class=\"data row4 col5\" >http://model.geneontology.org/R-HSA-156842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_36cb2_level0_row5\" class=\"row_heading level0 row5\" >5</th>\n",
       "      <td id=\"T_36cb2_row5_col0\" class=\"data row5 col0\" >Respiratory electron transport - Reactome</td>\n",
       "      <td id=\"T_36cb2_row5_col1\" class=\"data row5 col1\" >0.000004</td>\n",
       "      <td id=\"T_36cb2_row5_col2\" class=\"data row5 col2\" >13</td>\n",
       "      <td id=\"T_36cb2_row5_col3\" class=\"data row5 col3\" >91</td>\n",
       "      <td id=\"T_36cb2_row5_col4\" class=\"data row5 col4\" >['MT-ND4', 'MT-ND2', 'MT-ND1', 'MT-ND6', 'MT-CYB', 'MT-CO1', 'UQCRB', 'MT-CO2', 'MT-ND3', 'MT-CO3', 'MT-ND5', 'NDUFA1', 'COX7C']</td>\n",
       "      <td id=\"T_36cb2_row5_col5\" class=\"data row5 col5\" >http://model.geneontology.org/R-HSA-611105</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x12fd787f0>"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "RBM20_standard = enrich.enrich_wrapper('RBM20_FB_comb.csv','Gene Symbol',method='standard',FDR = 0.05,fpath = '../test_data/processed/',table_number = 23)\n",
    "RBM20_standard"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "bdf99a47-b0e8-4d03-846a-cd073883e014",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|█████████████████████████████████████████| 283/283 [02:12<00:00,  2.14it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 193 entities from 163 out of 625 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_f5f15 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_f5f15\">\n",
       "  <caption>Table S24: RBM20 fibroblasts combined up/downregulated genes, weighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_f5f15_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_f5f15_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_f5f15_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_f5f15_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_f5f15_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_f5f15_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_f5f15_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_f5f15_row0_col0\" class=\"data row0 col0\" >Formation of ATP by chemiosmotic coupling - Reactome</td>\n",
       "      <td id=\"T_f5f15_row0_col1\" class=\"data row0 col1\" >0.000000</td>\n",
       "      <td id=\"T_f5f15_row0_col2\" class=\"data row0 col2\" >15</td>\n",
       "      <td id=\"T_f5f15_row0_col3\" class=\"data row0 col3\" >88</td>\n",
       "      <td id=\"T_f5f15_row0_col4\" class=\"data row0 col4\" >['MT-ND4', 'MT-ND2', 'MT-ND1', 'MT-ATP6', 'MT-ND6', 'MT-CYB', 'MT-CO1', 'UQCRB', 'MT-CO2', 'MT-ND3', 'MT-CO3', 'MT-ND5', 'NDUFA1', 'COX7C', 'set:ANT homodimer']</td>\n",
       "      <td id=\"T_f5f15_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-163210</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f5f15_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_f5f15_row1_col0\" class=\"data row1 col0\" >Peptide chain elongation - Reactome</td>\n",
       "      <td id=\"T_f5f15_row1_col1\" class=\"data row1 col1\" >0.000000</td>\n",
       "      <td id=\"T_f5f15_row1_col2\" class=\"data row1 col2\" >14</td>\n",
       "      <td id=\"T_f5f15_row1_col3\" class=\"data row1 col3\" >79</td>\n",
       "      <td id=\"T_f5f15_row1_col4\" class=\"data row1 col4\" >['RPS17', 'RPL13A', 'RPS18', 'RPS15A', 'RPL27A', 'RPLP2', 'RPL34', 'RPS20', 'RPLP0', 'RPS5', 'RPL7', 'RPL38', 'RPL27', 'set:RPL3,RPL3L']</td>\n",
       "      <td id=\"T_f5f15_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-156902</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f5f15_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "      <td id=\"T_f5f15_row2_col0\" class=\"data row2 col0\" >The fatty acid cycling model - Reactome</td>\n",
       "      <td id=\"T_f5f15_row2_col1\" class=\"data row2 col1\" >0.000000</td>\n",
       "      <td id=\"T_f5f15_row2_col2\" class=\"data row2 col2\" >13</td>\n",
       "      <td id=\"T_f5f15_row2_col3\" class=\"data row2 col3\" >70</td>\n",
       "      <td id=\"T_f5f15_row2_col4\" class=\"data row2 col4\" >['MT-ND4', 'MT-ND2', 'MT-ND1', 'MT-ND6', 'MT-CYB', 'MT-CO1', 'UQCRB', 'MT-CO2', 'MT-ND3', 'MT-CO3', 'MT-ND5', 'NDUFA1', 'COX7C']</td>\n",
       "      <td id=\"T_f5f15_row2_col5\" class=\"data row2 col5\" >http://model.geneontology.org/R-HSA-167826</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f5f15_level0_row3\" class=\"row_heading level0 row3\" >3</th>\n",
       "      <td id=\"T_f5f15_row3_col0\" class=\"data row3 col0\" >The proton buffering model - Reactome</td>\n",
       "      <td id=\"T_f5f15_row3_col1\" class=\"data row3 col1\" >0.000000</td>\n",
       "      <td id=\"T_f5f15_row3_col2\" class=\"data row3 col2\" >13</td>\n",
       "      <td id=\"T_f5f15_row3_col3\" class=\"data row3 col3\" >70</td>\n",
       "      <td id=\"T_f5f15_row3_col4\" class=\"data row3 col4\" >['MT-ND4', 'MT-ND2', 'MT-ND1', 'MT-ND6', 'MT-CYB', 'MT-CO1', 'UQCRB', 'MT-CO2', 'MT-ND3', 'MT-CO3', 'MT-ND5', 'NDUFA1', 'COX7C']</td>\n",
       "      <td id=\"T_f5f15_row3_col5\" class=\"data row3 col5\" >http://model.geneontology.org/R-HSA-167827</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f5f15_level0_row4\" class=\"row_heading level0 row4\" >4</th>\n",
       "      <td id=\"T_f5f15_row4_col0\" class=\"data row4 col0\" >Eukaryotic Translation Elongation - Reactome</td>\n",
       "      <td id=\"T_f5f15_row4_col1\" class=\"data row4 col1\" >0.000000</td>\n",
       "      <td id=\"T_f5f15_row4_col2\" class=\"data row4 col2\" >14</td>\n",
       "      <td id=\"T_f5f15_row4_col3\" class=\"data row4 col3\" >83</td>\n",
       "      <td id=\"T_f5f15_row4_col4\" class=\"data row4 col4\" >['RPS17', 'RPL13A', 'RPS18', 'RPS15A', 'RPL27A', 'RPLP2', 'RPL34', 'RPS20', 'RPLP0', 'RPS5', 'RPL7', 'RPL38', 'RPL27', 'set:RPL3,RPL3L']</td>\n",
       "      <td id=\"T_f5f15_row4_col5\" class=\"data row4 col5\" >http://model.geneontology.org/R-HSA-156842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f5f15_level0_row5\" class=\"row_heading level0 row5\" >5</th>\n",
       "      <td id=\"T_f5f15_row5_col0\" class=\"data row5 col0\" >Respiratory electron transport - Reactome</td>\n",
       "      <td id=\"T_f5f15_row5_col1\" class=\"data row5 col1\" >0.000000</td>\n",
       "      <td id=\"T_f5f15_row5_col2\" class=\"data row5 col2\" >13</td>\n",
       "      <td id=\"T_f5f15_row5_col3\" class=\"data row5 col3\" >91</td>\n",
       "      <td id=\"T_f5f15_row5_col4\" class=\"data row5 col4\" >['MT-ND4', 'MT-ND2', 'MT-ND1', 'MT-ND6', 'MT-CYB', 'MT-CO1', 'UQCRB', 'MT-CO2', 'MT-ND3', 'MT-CO3', 'MT-ND5', 'NDUFA1', 'COX7C']</td>\n",
       "      <td id=\"T_f5f15_row5_col5\" class=\"data row5 col5\" >http://model.geneontology.org/R-HSA-611105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f5f15_level0_row6\" class=\"row_heading level0 row6\" >6</th>\n",
       "      <td id=\"T_f5f15_row6_col0\" class=\"data row6 col0\" >TP53 Regulates Metabolic Genes - Reactome</td>\n",
       "      <td id=\"T_f5f15_row6_col1\" class=\"data row6 col1\" >0.000200</td>\n",
       "      <td id=\"T_f5f15_row6_col2\" class=\"data row6 col2\" >8</td>\n",
       "      <td id=\"T_f5f15_row6_col3\" class=\"data row6 col3\" >46</td>\n",
       "      <td id=\"T_f5f15_row6_col4\" class=\"data row6 col4\" >['MT-CYB', 'MT-CO1', 'UQCRB', 'MT-CO2', 'MT-CO3', 'COX7C', 'G6PD', 'set:AMPK gamma']</td>\n",
       "      <td id=\"T_f5f15_row6_col5\" class=\"data row6 col5\" >http://model.geneontology.org/R-HSA-5628897</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x14e649900>"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "RBM20_weighted = enrich.enrich_wrapper('RBM20_FB_comb.csv','Gene Symbol',method='ncHGT',FDR = 0.05,fpath = '../test_data/processed/',table_number = 24)\n",
    "RBM20_weighted"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0a57c514-71ab-44a6-9511-a2ac4db9672e",
   "metadata": {
    "tags": []
   },
   "source": [
    "### PKP2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "af7599c9-3a1c-4ca5-8b85-48765aca9243",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 358/358 [00:00<00:00, 17384.61it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 239 entities from 191 out of 725 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_65721 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_65721\">\n",
       "  <caption>Table S25: PKP2 fibroblasts combined up/downregulated genes, unweighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_65721_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_65721_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_65721_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_65721_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_65721_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_65721_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_65721_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_65721_row0_col0\" class=\"data row0 col0\" >Synthesis of PIPs at the plasma membrane - Reactome</td>\n",
       "      <td id=\"T_65721_row0_col1\" class=\"data row0 col1\" >0.000000</td>\n",
       "      <td id=\"T_65721_row0_col2\" class=\"data row0 col2\" >12</td>\n",
       "      <td id=\"T_65721_row0_col3\" class=\"data row0 col3\" >21</td>\n",
       "      <td id=\"T_65721_row0_col4\" class=\"data row0 col4\" >['PTEN', 'set:PI3K-catalytic subunit', 'set:PIP4K2/5K1', 'set:PIK3C(1)', 'set:PIK3(2)', 'set:Activator:PI3K', 'set:PI3K-regulatory subunit', 'set:INPP4A/B', 'set:SYNJs,OCRL', 'set:SYNJ/INPP5(1)', 'set:SYNJ/MTM(1)', 'set:PIP4K2 dimers']</td>\n",
       "      <td id=\"T_65721_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-1660499</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_65721_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_65721_row1_col0\" class=\"data row1 col0\" >PI3K/AKT Signaling - Reactome</td>\n",
       "      <td id=\"T_65721_row1_col1\" class=\"data row1 col1\" >0.000011</td>\n",
       "      <td id=\"T_65721_row1_col2\" class=\"data row1 col2\" >7</td>\n",
       "      <td id=\"T_65721_row1_col3\" class=\"data row1 col3\" >15</td>\n",
       "      <td id=\"T_65721_row1_col4\" class=\"data row1 col4\" >['PIK3R1', 'PIK3CA', 'set:PI3K-catalytic subunit', 'set:Activator:PI3K', 'set:PI3K-regulatory subunit', 'set:PI3K alpha, beta, gamma', 'set:PIP4K2 dimers']</td>\n",
       "      <td id=\"T_65721_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-1257604</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_65721_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "      <td id=\"T_65721_row2_col0\" class=\"data row2 col0\" >Negative regulation of the PI3K/AKT network - Reactome</td>\n",
       "      <td id=\"T_65721_row2_col1\" class=\"data row2 col1\" >0.000014</td>\n",
       "      <td id=\"T_65721_row2_col2\" class=\"data row2 col2\" >5</td>\n",
       "      <td id=\"T_65721_row2_col3\" class=\"data row2 col3\" >7</td>\n",
       "      <td id=\"T_65721_row2_col4\" class=\"data row2 col4\" >['PTEN', 'set:PI3K-catalytic subunit', 'set:PIK3C(1)', 'set:Activator:PI3K', 'set:PI3K-regulatory subunit']</td>\n",
       "      <td id=\"T_65721_row2_col5\" class=\"data row2 col5\" >http://model.geneontology.org/R-HSA-199418</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_65721_level0_row3\" class=\"row_heading level0 row3\" >3</th>\n",
       "      <td id=\"T_65721_row3_col0\" class=\"data row3 col0\" >CD28 dependent PI3K/Akt signaling - Reactome</td>\n",
       "      <td id=\"T_65721_row3_col1\" class=\"data row3 col1\" >0.000018</td>\n",
       "      <td id=\"T_65721_row3_col2\" class=\"data row3 col2\" >7</td>\n",
       "      <td id=\"T_65721_row3_col3\" class=\"data row3 col3\" >16</td>\n",
       "      <td id=\"T_65721_row3_col4\" class=\"data row3 col4\" >['PIK3R1', 'PIK3CA', 'set:PI3K-catalytic subunit', 'set:Active AKT', 'set:Activator:PI3K', 'set:PI3K-regulatory subunit', 'set:PI3K alpha, beta, gamma']</td>\n",
       "      <td id=\"T_65721_row3_col5\" class=\"data row3 col5\" >http://model.geneontology.org/R-HSA-389357</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_65721_level0_row4\" class=\"row_heading level0 row4\" >4</th>\n",
       "      <td id=\"T_65721_row4_col0\" class=\"data row4 col0\" >G alpha (i) signalling events - Reactome</td>\n",
       "      <td id=\"T_65721_row4_col1\" class=\"data row4 col1\" >0.000036</td>\n",
       "      <td id=\"T_65721_row4_col2\" class=\"data row4 col2\" >5</td>\n",
       "      <td id=\"T_65721_row4_col3\" class=\"data row4 col3\" >8</td>\n",
       "      <td id=\"T_65721_row4_col4\" class=\"data row4 col4\" >['set:Adenylate cyclase', 'set:RGS1,3,4,5,6,7,8,9,10,11,12,13,14,16,17,18,19,20,21', 'set:Ligand:GPCR complexes that activate Gi', 'set:G alpha (i)', 'set:Light-sensing opsins']</td>\n",
       "      <td id=\"T_65721_row4_col5\" class=\"data row4 col5\" >http://model.geneontology.org/R-HSA-418594</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_65721_level0_row5\" class=\"row_heading level0 row5\" >5</th>\n",
       "      <td id=\"T_65721_row5_col0\" class=\"data row5 col0\" >PI5P, PP2A and IER3 Regulate PI3K/AKT Signaling - Reactome</td>\n",
       "      <td id=\"T_65721_row5_col1\" class=\"data row5 col1\" >0.000053</td>\n",
       "      <td id=\"T_65721_row5_col2\" class=\"data row5 col2\" >6</td>\n",
       "      <td id=\"T_65721_row5_col3\" class=\"data row5 col3\" >13</td>\n",
       "      <td id=\"T_65721_row5_col4\" class=\"data row5 col4\" >['PTEN', 'set:Activator:PI3K', 'set:SYNJ/INPP5(1)', 'set:SYNJ/MTM(1)', 'set:Activated SRC,LCK,EGFR,INSR', 'set:PIP4K2 dimers']</td>\n",
       "      <td id=\"T_65721_row5_col5\" class=\"data row5 col5\" >http://model.geneontology.org/R-HSA-6811558</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_65721_level0_row6\" class=\"row_heading level0 row6\" >6</th>\n",
       "      <td id=\"T_65721_row6_col0\" class=\"data row6 col0\" >VEGFR2 mediated cell proliferation - Reactome</td>\n",
       "      <td id=\"T_65721_row6_col1\" class=\"data row6 col1\" >0.000122</td>\n",
       "      <td id=\"T_65721_row6_col2\" class=\"data row6 col2\" >8</td>\n",
       "      <td id=\"T_65721_row6_col3\" class=\"data row6 col3\" >27</td>\n",
       "      <td id=\"T_65721_row6_col4\" class=\"data row6 col4\" >['PTEN', 'SPHK1', 'set:G(q) alpha 11,14,15,Q', 'set:PI3K-catalytic subunit', 'set:PI3K-regulatory subunit', 'set:PLCdegh', 'set:G-protein alpha (q/11/14/15)', 'set:ORAI dimer']</td>\n",
       "      <td id=\"T_65721_row6_col5\" class=\"data row6 col5\" >http://model.geneontology.org/R-HSA-5218921</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_65721_level0_row7\" class=\"row_heading level0 row7\" >7</th>\n",
       "      <td id=\"T_65721_row7_col0\" class=\"data row7 col0\" >G beta:gamma signalling through PI3Kgamma - Reactome</td>\n",
       "      <td id=\"T_65721_row7_col1\" class=\"data row7 col1\" >0.000147</td>\n",
       "      <td id=\"T_65721_row7_col2\" class=\"data row7 col2\" >5</td>\n",
       "      <td id=\"T_65721_row7_col3\" class=\"data row7 col3\" >10</td>\n",
       "      <td id=\"T_65721_row7_col4\" class=\"data row7 col4\" >['PIK3R1', 'PIK3CA', 'set:PI3K-catalytic subunit', 'set:PI3K-regulatory subunit', 'set:PI3K alpha, beta, gamma']</td>\n",
       "      <td id=\"T_65721_row7_col5\" class=\"data row7 col5\" >http://model.geneontology.org/R-HSA-392451</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x14e64aa70>"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PKP2_set = enrich.enrich_wrapper('PKP2_FB_comb.csv','Gene Symbol',method='set',FDR = 0.05,fpath = '../test_data/processed/',table_number = 25)\n",
    "PKP2_set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "b471b1a5-adfc-48e9-a3b8-6be9e02d1ab5",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 358/358 [00:00<00:00, 16778.15it/s]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_06e00 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_06e00\">\n",
       "  <caption>Table S26: PKP2 fibroblasts combined up/downregulated genes, standard method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_06e00_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_06e00_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_06e00_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_06e00_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_06e00_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_06e00_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x14e648e20>"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PKP2_standard = enrich.enrich_wrapper('PKP2_FB_comb.csv','Gene Symbol',method='standard',FDR = 0.05,fpath = '../test_data/processed/',table_number = 26)\n",
    "PKP2_standard"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "053966f1-5249-4f4a-84ee-a06d2a6d081a",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|█████████████████████████████████████████| 358/358 [02:09<00:00,  2.76it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 239 entities from 191 out of 725 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_5eeec caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_5eeec\">\n",
       "  <caption>Table S27: PKP2 fibroblasts combined up/downregulated genes, weighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_5eeec_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_5eeec_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_5eeec_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_5eeec_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_5eeec_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_5eeec_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_5eeec_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_5eeec_row0_col0\" class=\"data row0 col0\" >Synthesis of PIPs at the plasma membrane - Reactome</td>\n",
       "      <td id=\"T_5eeec_row0_col1\" class=\"data row0 col1\" >0.000000</td>\n",
       "      <td id=\"T_5eeec_row0_col2\" class=\"data row0 col2\" >12</td>\n",
       "      <td id=\"T_5eeec_row0_col3\" class=\"data row0 col3\" >21</td>\n",
       "      <td id=\"T_5eeec_row0_col4\" class=\"data row0 col4\" >['PTEN', 'set:PI3K-catalytic subunit', 'set:PIP4K2/5K1', 'set:PIK3C(1)', 'set:PIK3(2)', 'set:Activator:PI3K', 'set:PI3K-regulatory subunit', 'set:INPP4A/B', 'set:SYNJs,OCRL', 'set:SYNJ/INPP5(1)', 'set:SYNJ/MTM(1)', 'set:PIP4K2 dimers']</td>\n",
       "      <td id=\"T_5eeec_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-1660499</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x13bb6a830>"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PKP2_weighted = enrich.enrich_wrapper('PKP2_FB_comb.csv','Gene Symbol',method='ncHGT',FDR = 0.05,fpath = '../test_data/processed/',table_number = 27)\n",
    "PKP2_weighted"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b527f6d3-8ee8-407f-b2d5-76ed1a3830c4",
   "metadata": {
    "tags": []
   },
   "source": [
    "### TTN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "fb356ab9-da5f-4209-a95b-9ad29ec3fa12",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 461/461 [00:00<00:00, 17204.15it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 359 entities from 311 out of 1092 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_c69bf caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_c69bf\">\n",
       "  <caption>Table S28: TTN fibroblasts combined up/downregulated genes, unweighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_c69bf_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_c69bf_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_c69bf_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_c69bf_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_c69bf_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_c69bf_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_c69bf_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_c69bf_row0_col0\" class=\"data row0 col0\" >Peptide chain elongation - Reactome</td>\n",
       "      <td id=\"T_c69bf_row0_col1\" class=\"data row0 col1\" >0.000000</td>\n",
       "      <td id=\"T_c69bf_row0_col2\" class=\"data row0 col2\" >54</td>\n",
       "      <td id=\"T_c69bf_row0_col3\" class=\"data row0 col3\" >79</td>\n",
       "      <td id=\"T_c69bf_row0_col4\" class=\"data row0 col4\" >['RPS17', 'RPL13A', 'RPS18', 'RPLP0', 'RPL34', 'RPLP2', 'RPS15A', 'RPS5', 'RPL7', 'RPL18', 'RPS20', 'RPL27A', 'RPL35', 'RPL27', 'RPS25', 'RPS14', 'RPS19', 'RPL36', 'RPS15', 'RPL38', 'RPS28', 'RPL28', 'UBA52', 'RPLP1', 'RPL21', 'RPS11', 'RPL23A', 'RPS2', 'RPS13', 'RPS6', 'RPL13', 'RPL23', 'RPL7A', 'RPS23', 'RPS27A', 'RPL41', 'RPL15', 'RPL35A', 'RPS8', 'RPL10A', 'RPL12', 'RPL37A', 'FAU', 'RPL29', 'RPL31', 'RPS9', 'RPS12', 'RPL11', 'RPL8', 'set:RPL36A,RPL36AL', 'set:RPS4X,RPS4Y1,RPS4Y2', 'set:RPL3,RPL3L', 'set:RPS27,RPS27L', 'set:RPL22,RPL22L1']</td>\n",
       "      <td id=\"T_c69bf_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-156902</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_c69bf_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_c69bf_row1_col0\" class=\"data row1 col0\" >Eukaryotic Translation Elongation - Reactome</td>\n",
       "      <td id=\"T_c69bf_row1_col1\" class=\"data row1 col1\" >0.000000</td>\n",
       "      <td id=\"T_c69bf_row1_col2\" class=\"data row1 col2\" >55</td>\n",
       "      <td id=\"T_c69bf_row1_col3\" class=\"data row1 col3\" >83</td>\n",
       "      <td id=\"T_c69bf_row1_col4\" class=\"data row1 col4\" >['RPS17', 'RPL13A', 'RPS18', 'RPLP0', 'RPL34', 'RPLP2', 'RPS15A', 'RPS5', 'RPL7', 'RPL18', 'RPS20', 'RPL27A', 'RPL35', 'RPL27', 'RPS25', 'RPS14', 'RPS19', 'RPL36', 'RPS15', 'RPL38', 'RPS28', 'RPL28', 'UBA52', 'RPLP1', 'RPL21', 'RPS11', 'RPL23A', 'RPS2', 'RPS13', 'RPS6', 'RPL13', 'RPL23', 'RPL7A', 'RPS23', 'RPS27A', 'RPL41', 'RPL15', 'RPL35A', 'RPS8', 'RPL10A', 'RPL12', 'RPL37A', 'FAU', 'RPL29', 'RPL31', 'RPS9', 'RPS12', 'RPL11', 'RPL8', 'set:RPL36A,RPL36AL', 'set:RPS4X,RPS4Y1,RPS4Y2', 'set:RPL3,RPL3L', 'set:EEF1A1-like proteins', 'set:RPS27,RPS27L', 'set:RPL22,RPL22L1']</td>\n",
       "      <td id=\"T_c69bf_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-156842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_c69bf_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "      <td id=\"T_c69bf_row2_col0\" class=\"data row2 col0\" >Regulation of Complement cascade - Reactome</td>\n",
       "      <td id=\"T_c69bf_row2_col1\" class=\"data row2 col1\" >0.000011</td>\n",
       "      <td id=\"T_c69bf_row2_col2\" class=\"data row2 col2\" >7</td>\n",
       "      <td id=\"T_c69bf_row2_col3\" class=\"data row2 col3\" >11</td>\n",
       "      <td id=\"T_c69bf_row2_col4\" class=\"data row2 col4\" >['C3', 'C1R', 'set:C3 convertases', 'set:Complement factor D', 'set:C4 activators', 'set:Activated thrombin, (ELANE)', 'set:CD46, CR1:C4b:C3b complexes']</td>\n",
       "      <td id=\"T_c69bf_row2_col5\" class=\"data row2 col5\" >http://model.geneontology.org/R-HSA-977606</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_c69bf_level0_row3\" class=\"row_heading level0 row3\" >3</th>\n",
       "      <td id=\"T_c69bf_row3_col0\" class=\"data row3 col0\" >G alpha (i) signalling events - Reactome</td>\n",
       "      <td id=\"T_c69bf_row3_col1\" class=\"data row3 col1\" >0.000012</td>\n",
       "      <td id=\"T_c69bf_row3_col2\" class=\"data row3 col2\" >6</td>\n",
       "      <td id=\"T_c69bf_row3_col3\" class=\"data row3 col3\" >8</td>\n",
       "      <td id=\"T_c69bf_row3_col4\" class=\"data row3 col4\" >['set:Adenylate cyclase', 'set:RGS1,3,4,5,6,7,8,9,10,11,12,13,14,16,17,18,19,20,21', 'set:GNB', 'set:Ligand:GPCR complexes that activate Gi', 'set:G alpha (i)', 'set:Light-sensing opsins']</td>\n",
       "      <td id=\"T_c69bf_row3_col5\" class=\"data row3 col5\" >http://model.geneontology.org/R-HSA-418594</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x14f5e0fd0>"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TTN_set = enrich.enrich_wrapper('TTN_FB_comb.csv','Gene Symbol',method='set',FDR = 0.05,fpath = '../test_data/processed/',table_number = 28)\n",
    "TTN_set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "50af92ca-809b-4321-9e3c-50def7fb0d86",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 461/461 [00:00<00:00, 16936.07it/s]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_dd2f7 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_dd2f7\">\n",
       "  <caption>Table S29: TTN fibroblasts combined up/downregulated genes, standard method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_dd2f7_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_dd2f7_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_dd2f7_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_dd2f7_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_dd2f7_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_dd2f7_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_dd2f7_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_dd2f7_row0_col0\" class=\"data row0 col0\" >Eukaryotic Translation Elongation - Reactome</td>\n",
       "      <td id=\"T_dd2f7_row0_col1\" class=\"data row0 col1\" >0.000000</td>\n",
       "      <td id=\"T_dd2f7_row0_col2\" class=\"data row0 col2\" >56</td>\n",
       "      <td id=\"T_dd2f7_row0_col3\" class=\"data row0 col3\" >94</td>\n",
       "      <td id=\"T_dd2f7_row0_col4\" class=\"data row0 col4\" >['EEF1A2', 'RPS17', 'RPL13A', 'RPS18', 'RPLP0', 'RPL34', 'RPLP2', 'RPL3', 'RPS15A', 'RPS5', 'RPL7', 'RPL18', 'RPS20', 'RPS27', 'RPL27A', 'RPL35', 'RPL27', 'RPS25', 'RPS14', 'RPS19', 'RPL36', 'RPS15', 'RPL38', 'RPS28', 'RPL28', 'UBA52', 'RPLP1', 'RPL21', 'RPS11', 'RPL23A', 'RPS2', 'RPS13', 'RPS6', 'RPL13', 'RPL23', 'RPL7A', 'RPS23', 'RPS4X', 'RPS27A', 'RPL41', 'RPL15', 'RPL35A', 'RPS8', 'RPL10A', 'RPL36AL', 'RPL22L1', 'RPL12', 'RPL37A', 'FAU', 'RPL29', 'EEF1A1', 'RPL31', 'RPS9', 'RPS12', 'RPL11', 'RPL8']</td>\n",
       "      <td id=\"T_dd2f7_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-156842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_dd2f7_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_dd2f7_row1_col0\" class=\"data row1 col0\" >Peptide chain elongation - Reactome</td>\n",
       "      <td id=\"T_dd2f7_row1_col1\" class=\"data row1 col1\" >0.000000</td>\n",
       "      <td id=\"T_dd2f7_row1_col2\" class=\"data row1 col2\" >54</td>\n",
       "      <td id=\"T_dd2f7_row1_col3\" class=\"data row1 col3\" >88</td>\n",
       "      <td id=\"T_dd2f7_row1_col4\" class=\"data row1 col4\" >['RPS17', 'RPL13A', 'RPS18', 'RPLP0', 'RPL34', 'RPLP2', 'RPL3', 'RPS15A', 'RPS5', 'RPL7', 'RPL18', 'RPS20', 'RPS27', 'RPL27A', 'RPL35', 'RPL27', 'RPS25', 'RPS14', 'RPS19', 'RPL36', 'RPS15', 'RPL38', 'RPS28', 'RPL28', 'UBA52', 'RPLP1', 'RPL21', 'RPS11', 'RPL23A', 'RPS2', 'RPS13', 'RPS6', 'RPL13', 'RPL23', 'RPL7A', 'RPS23', 'RPS4X', 'RPS27A', 'RPL41', 'RPL15', 'RPL35A', 'RPS8', 'RPL10A', 'RPL36AL', 'RPL22L1', 'RPL12', 'RPL37A', 'FAU', 'RPL29', 'RPL31', 'RPS9', 'RPS12', 'RPL11', 'RPL8']</td>\n",
       "      <td id=\"T_dd2f7_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-156902</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x14e64a050>"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TTN_standard = enrich.enrich_wrapper('TTN_FB_comb.csv','Gene Symbol',method='standard',FDR = 0.05,fpath = '../test_data/processed/',table_number = 29)\n",
    "TTN_standard"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "79061df4-5d20-40f7-9d4c-ee9c0ebfd5fa",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|█████████████████████████████████████████| 461/461 [05:03<00:00,  1.52it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 359 entities from 311 out of 1092 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_6b193 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_6b193\">\n",
       "  <caption>Table S30: TTN fibroblasts combined up/downregulated genes, weighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_6b193_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_6b193_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_6b193_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_6b193_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_6b193_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_6b193_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_6b193_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_6b193_row0_col0\" class=\"data row0 col0\" >Peptide chain elongation - Reactome</td>\n",
       "      <td id=\"T_6b193_row0_col1\" class=\"data row0 col1\" >0.000000</td>\n",
       "      <td id=\"T_6b193_row0_col2\" class=\"data row0 col2\" >54</td>\n",
       "      <td id=\"T_6b193_row0_col3\" class=\"data row0 col3\" >79</td>\n",
       "      <td id=\"T_6b193_row0_col4\" class=\"data row0 col4\" >['RPS17', 'RPL13A', 'RPS18', 'RPLP0', 'RPL34', 'RPLP2', 'RPS15A', 'RPS5', 'RPL7', 'RPL18', 'RPS20', 'RPL27A', 'RPL35', 'RPL27', 'RPS25', 'RPS14', 'RPS19', 'RPL36', 'RPS15', 'RPL38', 'RPS28', 'RPL28', 'UBA52', 'RPLP1', 'RPL21', 'RPS11', 'RPL23A', 'RPS2', 'RPS13', 'RPS6', 'RPL13', 'RPL23', 'RPL7A', 'RPS23', 'RPS27A', 'RPL41', 'RPL15', 'RPL35A', 'RPS8', 'RPL10A', 'RPL12', 'RPL37A', 'FAU', 'RPL29', 'RPL31', 'RPS9', 'RPS12', 'RPL11', 'RPL8', 'set:RPL36A,RPL36AL', 'set:RPS4X,RPS4Y1,RPS4Y2', 'set:RPL3,RPL3L', 'set:RPS27,RPS27L', 'set:RPL22,RPL22L1']</td>\n",
       "      <td id=\"T_6b193_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-156902</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_6b193_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_6b193_row1_col0\" class=\"data row1 col0\" >Eukaryotic Translation Elongation - Reactome</td>\n",
       "      <td id=\"T_6b193_row1_col1\" class=\"data row1 col1\" >0.000000</td>\n",
       "      <td id=\"T_6b193_row1_col2\" class=\"data row1 col2\" >55</td>\n",
       "      <td id=\"T_6b193_row1_col3\" class=\"data row1 col3\" >83</td>\n",
       "      <td id=\"T_6b193_row1_col4\" class=\"data row1 col4\" >['RPS17', 'RPL13A', 'RPS18', 'RPLP0', 'RPL34', 'RPLP2', 'RPS15A', 'RPS5', 'RPL7', 'RPL18', 'RPS20', 'RPL27A', 'RPL35', 'RPL27', 'RPS25', 'RPS14', 'RPS19', 'RPL36', 'RPS15', 'RPL38', 'RPS28', 'RPL28', 'UBA52', 'RPLP1', 'RPL21', 'RPS11', 'RPL23A', 'RPS2', 'RPS13', 'RPS6', 'RPL13', 'RPL23', 'RPL7A', 'RPS23', 'RPS27A', 'RPL41', 'RPL15', 'RPL35A', 'RPS8', 'RPL10A', 'RPL12', 'RPL37A', 'FAU', 'RPL29', 'RPL31', 'RPS9', 'RPS12', 'RPL11', 'RPL8', 'set:RPL36A,RPL36AL', 'set:RPS4X,RPS4Y1,RPS4Y2', 'set:RPL3,RPL3L', 'set:EEF1A1-like proteins', 'set:RPS27,RPS27L', 'set:RPL22,RPL22L1']</td>\n",
       "      <td id=\"T_6b193_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-156842</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x13bc041c0>"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TTN_weighted = enrich.enrich_wrapper('TTN_FB_comb.csv','Gene Symbol',method='ncHGT',FDR = 0.05,fpath = '../test_data/processed/',table_number = 30)\n",
    "TTN_weighted"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a90b030e-2667-4793-8ab4-cfdc11c6ceb4",
   "metadata": {
    "tags": []
   },
   "source": [
    "### PVneg"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "399f5b18-bdc5-4c3a-81a5-052590296057",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 452/452 [00:00<00:00, 16281.29it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 396 entities from 338 out of 1135 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_ae3e6 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_ae3e6\">\n",
       "  <caption>Table S31: PVneg fibroblasts combined up/downregulated genes, unweighted set method, FDR = 0.1.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_ae3e6_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_ae3e6_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_ae3e6_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_ae3e6_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_ae3e6_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_ae3e6_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_ae3e6_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_ae3e6_row0_col0\" class=\"data row0 col0\" >Peptide chain elongation - Reactome</td>\n",
       "      <td id=\"T_ae3e6_row0_col1\" class=\"data row0 col1\" >0.000000</td>\n",
       "      <td id=\"T_ae3e6_row0_col2\" class=\"data row0 col2\" >50</td>\n",
       "      <td id=\"T_ae3e6_row0_col3\" class=\"data row0 col3\" >79</td>\n",
       "      <td id=\"T_ae3e6_row0_col4\" class=\"data row0 col4\" >['RPS17', 'RPL13A', 'RPLP0', 'RPS18', 'RPLP2', 'RPL27A', 'RPS15A', 'RPS2', 'RPL7', 'RPL35', 'RPLP1', 'RPS20', 'RPS28', 'RPS5', 'RPS14', 'RPL34', 'RPL15', 'RPL23A', 'FAU', 'RPL21', 'RPL27', 'RPS25', 'RPL12', 'RPL10A', 'RPS15', 'RPS19', 'RPS16', 'RPL18', 'RPS21', 'RPL13', 'UBA52', 'RPL41', 'RPL7A', 'RPS11', 'RPS8', 'RPL29', 'RPS6', 'RPL8', 'RPS23', 'RPS9', 'RPSA', 'RPL4', 'RPS13', 'RPS27A', 'RPL19', 'set:RPL36A,RPL36AL', 'set:RPL3,RPL3L', 'set:RPL10,RPL10L', 'set:RPS27,RPS27L', 'set:RPL22,RPL22L1']</td>\n",
       "      <td id=\"T_ae3e6_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-156902</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_ae3e6_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_ae3e6_row1_col0\" class=\"data row1 col0\" >Eukaryotic Translation Elongation - Reactome</td>\n",
       "      <td id=\"T_ae3e6_row1_col1\" class=\"data row1 col1\" >0.000000</td>\n",
       "      <td id=\"T_ae3e6_row1_col2\" class=\"data row1 col2\" >51</td>\n",
       "      <td id=\"T_ae3e6_row1_col3\" class=\"data row1 col3\" >83</td>\n",
       "      <td id=\"T_ae3e6_row1_col4\" class=\"data row1 col4\" >['RPS17', 'RPL13A', 'RPLP0', 'RPS18', 'RPLP2', 'RPL27A', 'RPS15A', 'RPS2', 'RPL7', 'RPL35', 'RPLP1', 'RPS20', 'RPS28', 'RPS5', 'RPS14', 'RPL34', 'RPL15', 'RPL23A', 'FAU', 'RPL21', 'RPL27', 'RPS25', 'RPL12', 'RPL10A', 'RPS15', 'RPS19', 'RPS16', 'RPL18', 'RPS21', 'RPL13', 'UBA52', 'RPL41', 'RPL7A', 'RPS11', 'RPS8', 'RPL29', 'RPS6', 'EEF1B2', 'RPL8', 'RPS23', 'RPS9', 'RPSA', 'RPL4', 'RPS13', 'RPS27A', 'RPL19', 'set:RPL36A,RPL36AL', 'set:RPL3,RPL3L', 'set:RPL10,RPL10L', 'set:RPS27,RPS27L', 'set:RPL22,RPL22L1']</td>\n",
       "      <td id=\"T_ae3e6_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-156842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_ae3e6_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "      <td id=\"T_ae3e6_row2_col0\" class=\"data row2 col0\" >Formation of ATP by chemiosmotic coupling - Reactome</td>\n",
       "      <td id=\"T_ae3e6_row2_col1\" class=\"data row2 col1\" >0.000000</td>\n",
       "      <td id=\"T_ae3e6_row2_col2\" class=\"data row2 col2\" >29</td>\n",
       "      <td id=\"T_ae3e6_row2_col3\" class=\"data row2 col3\" >88</td>\n",
       "      <td id=\"T_ae3e6_row2_col4\" class=\"data row2 col4\" >['MT-CO1', 'MT-ND4', 'MT-ND2', 'ATP5MC1', 'CYC1', 'COX5B', 'MT-CYB', 'UQCRB', 'MT-ATP6', 'ATP5MC3', 'COX7C', 'NDUFA1', 'MT-CO2', 'MT-CO3', 'MT-ND1', 'UQCRFS1', 'NDUFA3', 'NDUFA4', 'NDUFA6', 'ATP5ME', 'UQCRQ', 'ATP5F1D', 'UQCR10', 'MT-ND6', 'NDUFB10', 'NDUFA2', 'COX4I1', 'COX6B1', 'set:ANT homodimer']</td>\n",
       "      <td id=\"T_ae3e6_row2_col5\" class=\"data row2 col5\" >http://model.geneontology.org/R-HSA-163210</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_ae3e6_level0_row3\" class=\"row_heading level0 row3\" >3</th>\n",
       "      <td id=\"T_ae3e6_row3_col0\" class=\"data row3 col0\" >Cytoprotection by HMOX1 - Reactome</td>\n",
       "      <td id=\"T_ae3e6_row3_col1\" class=\"data row3 col1\" >0.000000</td>\n",
       "      <td id=\"T_ae3e6_row3_col2\" class=\"data row3 col2\" >15</td>\n",
       "      <td id=\"T_ae3e6_row3_col3\" class=\"data row3 col3\" >29</td>\n",
       "      <td id=\"T_ae3e6_row3_col4\" class=\"data row3 col4\" >['MT-CO1', 'CYC1', 'COX5B', 'MT-CYB', 'UQCRB', 'COX7C', 'MT-CO2', 'MT-CO3', 'UQCRFS1', 'NDUFA4', 'UQCRQ', 'UQCR10', 'COX4I1', 'COX6B1', 'set:BLVRA:Zn2+,BLVRB']</td>\n",
       "      <td id=\"T_ae3e6_row3_col5\" class=\"data row3 col5\" >http://model.geneontology.org/R-HSA-9707564</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_ae3e6_level0_row4\" class=\"row_heading level0 row4\" >4</th>\n",
       "      <td id=\"T_ae3e6_row4_col0\" class=\"data row4 col0\" >The fatty acid cycling model - Reactome</td>\n",
       "      <td id=\"T_ae3e6_row4_col1\" class=\"data row4 col1\" >0.000000</td>\n",
       "      <td id=\"T_ae3e6_row4_col2\" class=\"data row4 col2\" >23</td>\n",
       "      <td id=\"T_ae3e6_row4_col3\" class=\"data row4 col3\" >70</td>\n",
       "      <td id=\"T_ae3e6_row4_col4\" class=\"data row4 col4\" >['MT-CO1', 'MT-ND4', 'MT-ND2', 'CYC1', 'COX5B', 'MT-CYB', 'UQCRB', 'COX7C', 'NDUFA1', 'MT-CO2', 'MT-CO3', 'MT-ND1', 'UQCRFS1', 'NDUFA3', 'NDUFA4', 'NDUFA6', 'UQCRQ', 'UQCR10', 'MT-ND6', 'NDUFB10', 'NDUFA2', 'COX4I1', 'COX6B1']</td>\n",
       "      <td id=\"T_ae3e6_row4_col5\" class=\"data row4 col5\" >http://model.geneontology.org/R-HSA-167826</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_ae3e6_level0_row5\" class=\"row_heading level0 row5\" >5</th>\n",
       "      <td id=\"T_ae3e6_row5_col0\" class=\"data row5 col0\" >The proton buffering model - Reactome</td>\n",
       "      <td id=\"T_ae3e6_row5_col1\" class=\"data row5 col1\" >0.000000</td>\n",
       "      <td id=\"T_ae3e6_row5_col2\" class=\"data row5 col2\" >23</td>\n",
       "      <td id=\"T_ae3e6_row5_col3\" class=\"data row5 col3\" >70</td>\n",
       "      <td id=\"T_ae3e6_row5_col4\" class=\"data row5 col4\" >['MT-CO1', 'MT-ND4', 'MT-ND2', 'CYC1', 'COX5B', 'MT-CYB', 'UQCRB', 'COX7C', 'NDUFA1', 'MT-CO2', 'MT-CO3', 'MT-ND1', 'UQCRFS1', 'NDUFA3', 'NDUFA4', 'NDUFA6', 'UQCRQ', 'UQCR10', 'MT-ND6', 'NDUFB10', 'NDUFA2', 'COX4I1', 'COX6B1']</td>\n",
       "      <td id=\"T_ae3e6_row5_col5\" class=\"data row5 col5\" >http://model.geneontology.org/R-HSA-167827</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_ae3e6_level0_row6\" class=\"row_heading level0 row6\" >6</th>\n",
       "      <td id=\"T_ae3e6_row6_col0\" class=\"data row6 col0\" >TP53 Regulates Metabolic Genes - Reactome</td>\n",
       "      <td id=\"T_ae3e6_row6_col1\" class=\"data row6 col1\" >0.000000</td>\n",
       "      <td id=\"T_ae3e6_row6_col2\" class=\"data row6 col2\" >18</td>\n",
       "      <td id=\"T_ae3e6_row6_col3\" class=\"data row6 col3\" >46</td>\n",
       "      <td id=\"T_ae3e6_row6_col4\" class=\"data row6 col4\" >['G6PD', 'MT-CO1', 'CYC1', 'COX5B', 'MT-CYB', 'UQCRB', 'COX7C', 'PRDX1', 'MT-CO2', 'MT-CO3', 'UQCRFS1', 'NDUFA4', 'UQCRQ', 'UQCR10', 'COX4I1', 'COX6B1', 'set:PRDX1,2,5', 'set:AMPK gamma']</td>\n",
       "      <td id=\"T_ae3e6_row6_col5\" class=\"data row6 col5\" >http://model.geneontology.org/R-HSA-5628897</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_ae3e6_level0_row7\" class=\"row_heading level0 row7\" >7</th>\n",
       "      <td id=\"T_ae3e6_row7_col0\" class=\"data row7 col0\" >Respiratory electron transport - Reactome</td>\n",
       "      <td id=\"T_ae3e6_row7_col1\" class=\"data row7 col1\" >0.000001</td>\n",
       "      <td id=\"T_ae3e6_row7_col2\" class=\"data row7 col2\" >25</td>\n",
       "      <td id=\"T_ae3e6_row7_col3\" class=\"data row7 col3\" >91</td>\n",
       "      <td id=\"T_ae3e6_row7_col4\" class=\"data row7 col4\" >['MT-CO1', 'MT-ND4', 'MT-ND2', 'CYC1', 'COX5B', 'MT-CYB', 'UQCRB', 'COX7C', 'NDUFA1', 'MT-CO2', 'MT-CO3', 'MT-ND1', 'UQCRFS1', 'NDUFA3', 'NDUFA4', 'NDUFA6', 'UQCRQ', 'UQCR10', 'ACADVL', 'MT-ND6', 'NDUFB10', 'NDUFA2', 'IDH3B', 'COX4I1', 'COX6B1']</td>\n",
       "      <td id=\"T_ae3e6_row7_col5\" class=\"data row7 col5\" >http://model.geneontology.org/R-HSA-611105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_ae3e6_level0_row8\" class=\"row_heading level0 row8\" >8</th>\n",
       "      <td id=\"T_ae3e6_row8_col0\" class=\"data row8 col0\" >Regulation of Complement cascade - Reactome</td>\n",
       "      <td id=\"T_ae3e6_row8_col1\" class=\"data row8 col1\" >0.000021</td>\n",
       "      <td id=\"T_ae3e6_row8_col2\" class=\"data row8 col2\" >7</td>\n",
       "      <td id=\"T_ae3e6_row8_col3\" class=\"data row8 col3\" >11</td>\n",
       "      <td id=\"T_ae3e6_row8_col4\" class=\"data row8 col4\" >['C3', 'C1R', 'set:C3 convertases', 'set:Complement factor D', 'set:C4 activators', 'set:Activated thrombin, (ELANE)', 'set:CD46, CR1:C4b:C3b complexes']</td>\n",
       "      <td id=\"T_ae3e6_row8_col5\" class=\"data row8 col5\" >http://model.geneontology.org/R-HSA-977606</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_ae3e6_level0_row9\" class=\"row_heading level0 row9\" >9</th>\n",
       "      <td id=\"T_ae3e6_row9_col0\" class=\"data row9 col0\" >Collagen biosynthesis and modifying enzymes - Reactome</td>\n",
       "      <td id=\"T_ae3e6_row9_col1\" class=\"data row9 col1\" >0.000510</td>\n",
       "      <td id=\"T_ae3e6_row9_col2\" class=\"data row9 col2\" >6</td>\n",
       "      <td id=\"T_ae3e6_row9_col3\" class=\"data row9 col3\" >12</td>\n",
       "      <td id=\"T_ae3e6_row9_col4\" class=\"data row9 col4\" >['PPIB', 'set:Procollagen C-proteinases', 'set:Procollagen N-proteinases', 'set:Prolyl 3-hydroxylases', 'set:COLGALT1,COLGALT2', 'set:4-Hyp collagen propeptides']</td>\n",
       "      <td id=\"T_ae3e6_row9_col5\" class=\"data row9 col5\" >http://model.geneontology.org/R-HSA-1650814</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x13bcf6950>"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PVneg_set = enrich.enrich_wrapper('PVneg_FB_comb.csv','Gene Symbol',method='set',FDR = 0.1 ,fpath = '../test_data/processed/',table_number = 31)\n",
    "PVneg_set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "e6dd4d86-e273-4a83-a5a3-69401c29dd6e",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 452/452 [00:00<00:00, 15846.88it/s]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_0c12c caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_0c12c\">\n",
       "  <caption>Table S32: PVneg fibroblasts combined up/downregulated genes, standard method, FDR = 0.1.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_0c12c_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_0c12c_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_0c12c_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_0c12c_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_0c12c_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_0c12c_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_0c12c_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_0c12c_row0_col0\" class=\"data row0 col0\" >Peptide chain elongation - Reactome</td>\n",
       "      <td id=\"T_0c12c_row0_col1\" class=\"data row0 col1\" >0.000000</td>\n",
       "      <td id=\"T_0c12c_row0_col2\" class=\"data row0 col2\" >51</td>\n",
       "      <td id=\"T_0c12c_row0_col3\" class=\"data row0 col3\" >88</td>\n",
       "      <td id=\"T_0c12c_row0_col4\" class=\"data row0 col4\" >['RPS17', 'RPL13A', 'RPLP0', 'RPS18', 'RPLP2', 'RPL27A', 'RPS15A', 'RPS2', 'RPL7', 'RPL36AL', 'RPL35', 'RPLP1', 'RPS20', 'RPL3L', 'RPS28', 'RPS5', 'RPL3', 'RPS14', 'RPL34', 'RPL15', 'RPL23A', 'RPS27', 'FAU', 'RPL21', 'RPL27', 'RPS25', 'RPL12', 'RPL10A', 'RPS15', 'RPS19', 'RPS16', 'RPL10', 'RPL18', 'RPS21', 'RPL13', 'UBA52', 'RPL41', 'RPL7A', 'RPS11', 'RPS8', 'RPL29', 'RPS6', 'RPL8', 'RPS23', 'RPS9', 'RPSA', 'RPL4', 'RPS13', 'RPS27A', 'RPL22L1', 'RPL19']</td>\n",
       "      <td id=\"T_0c12c_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-156902</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0c12c_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_0c12c_row1_col0\" class=\"data row1 col0\" >Eukaryotic Translation Elongation - Reactome</td>\n",
       "      <td id=\"T_0c12c_row1_col1\" class=\"data row1 col1\" >0.000000</td>\n",
       "      <td id=\"T_0c12c_row1_col2\" class=\"data row1 col2\" >52</td>\n",
       "      <td id=\"T_0c12c_row1_col3\" class=\"data row1 col3\" >94</td>\n",
       "      <td id=\"T_0c12c_row1_col4\" class=\"data row1 col4\" >['RPS17', 'RPL13A', 'RPLP0', 'RPS18', 'RPLP2', 'RPL27A', 'RPS15A', 'RPS2', 'RPL7', 'RPL36AL', 'RPL35', 'RPLP1', 'RPS20', 'RPL3L', 'RPS28', 'RPS5', 'RPL3', 'RPS14', 'RPL34', 'RPL15', 'RPL23A', 'RPS27', 'FAU', 'RPL21', 'RPL27', 'RPS25', 'RPL12', 'RPL10A', 'RPS15', 'RPS19', 'RPS16', 'RPL10', 'RPL18', 'RPS21', 'RPL13', 'UBA52', 'RPL41', 'RPL7A', 'RPS11', 'RPS8', 'RPL29', 'RPS6', 'EEF1B2', 'RPL8', 'RPS23', 'RPS9', 'RPSA', 'RPL4', 'RPS13', 'RPS27A', 'RPL22L1', 'RPL19']</td>\n",
       "      <td id=\"T_0c12c_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-156842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0c12c_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "      <td id=\"T_0c12c_row2_col0\" class=\"data row2 col0\" >Formation of ATP by chemiosmotic coupling - Reactome</td>\n",
       "      <td id=\"T_0c12c_row2_col1\" class=\"data row2 col1\" >0.000000</td>\n",
       "      <td id=\"T_0c12c_row2_col2\" class=\"data row2 col2\" >30</td>\n",
       "      <td id=\"T_0c12c_row2_col3\" class=\"data row2 col3\" >89</td>\n",
       "      <td id=\"T_0c12c_row2_col4\" class=\"data row2 col4\" >['MT-CO1', 'MT-ND4', 'SLC25A6', 'MT-ND2', 'ATP5MC1', 'CYC1', 'COX5B', 'MT-CYB', 'UQCRB', 'MT-ATP6', 'ATP5MC3', 'COX7C', 'SLC25A5', 'NDUFA1', 'MT-CO2', 'MT-CO3', 'MT-ND1', 'UQCRFS1', 'NDUFA3', 'NDUFA4', 'NDUFA6', 'ATP5ME', 'UQCRQ', 'ATP5F1D', 'UQCR10', 'MT-ND6', 'NDUFB10', 'NDUFA2', 'COX4I1', 'COX6B1']</td>\n",
       "      <td id=\"T_0c12c_row2_col5\" class=\"data row2 col5\" >http://model.geneontology.org/R-HSA-163210</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0c12c_level0_row3\" class=\"row_heading level0 row3\" >3</th>\n",
       "      <td id=\"T_0c12c_row3_col0\" class=\"data row3 col0\" >The proton buffering model - Reactome</td>\n",
       "      <td id=\"T_0c12c_row3_col1\" class=\"data row3 col1\" >0.000000</td>\n",
       "      <td id=\"T_0c12c_row3_col2\" class=\"data row3 col2\" >23</td>\n",
       "      <td id=\"T_0c12c_row3_col3\" class=\"data row3 col3\" >74</td>\n",
       "      <td id=\"T_0c12c_row3_col4\" class=\"data row3 col4\" >['MT-CO1', 'MT-ND4', 'MT-ND2', 'CYC1', 'COX5B', 'MT-CYB', 'UQCRB', 'COX7C', 'NDUFA1', 'MT-CO2', 'MT-CO3', 'MT-ND1', 'UQCRFS1', 'NDUFA3', 'NDUFA4', 'NDUFA6', 'UQCRQ', 'UQCR10', 'MT-ND6', 'NDUFB10', 'NDUFA2', 'COX4I1', 'COX6B1']</td>\n",
       "      <td id=\"T_0c12c_row3_col5\" class=\"data row3 col5\" >http://model.geneontology.org/R-HSA-167827</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0c12c_level0_row4\" class=\"row_heading level0 row4\" >4</th>\n",
       "      <td id=\"T_0c12c_row4_col0\" class=\"data row4 col0\" >The fatty acid cycling model - Reactome</td>\n",
       "      <td id=\"T_0c12c_row4_col1\" class=\"data row4 col1\" >0.000000</td>\n",
       "      <td id=\"T_0c12c_row4_col2\" class=\"data row4 col2\" >23</td>\n",
       "      <td id=\"T_0c12c_row4_col3\" class=\"data row4 col3\" >74</td>\n",
       "      <td id=\"T_0c12c_row4_col4\" class=\"data row4 col4\" >['MT-CO1', 'MT-ND4', 'MT-ND2', 'CYC1', 'COX5B', 'MT-CYB', 'UQCRB', 'COX7C', 'NDUFA1', 'MT-CO2', 'MT-CO3', 'MT-ND1', 'UQCRFS1', 'NDUFA3', 'NDUFA4', 'NDUFA6', 'UQCRQ', 'UQCR10', 'MT-ND6', 'NDUFB10', 'NDUFA2', 'COX4I1', 'COX6B1']</td>\n",
       "      <td id=\"T_0c12c_row4_col5\" class=\"data row4 col5\" >http://model.geneontology.org/R-HSA-167826</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0c12c_level0_row5\" class=\"row_heading level0 row5\" >5</th>\n",
       "      <td id=\"T_0c12c_row5_col0\" class=\"data row5 col0\" >Cytoprotection by HMOX1 - Reactome</td>\n",
       "      <td id=\"T_0c12c_row5_col1\" class=\"data row5 col1\" >0.000000</td>\n",
       "      <td id=\"T_0c12c_row5_col2\" class=\"data row5 col2\" >15</td>\n",
       "      <td id=\"T_0c12c_row5_col3\" class=\"data row5 col3\" >31</td>\n",
       "      <td id=\"T_0c12c_row5_col4\" class=\"data row5 col4\" >['MT-CO1', 'CYC1', 'COX5B', 'MT-CYB', 'UQCRB', 'COX7C', 'BLVRB', 'MT-CO2', 'MT-CO3', 'UQCRFS1', 'NDUFA4', 'UQCRQ', 'UQCR10', 'COX4I1', 'COX6B1']</td>\n",
       "      <td id=\"T_0c12c_row5_col5\" class=\"data row5 col5\" >http://model.geneontology.org/R-HSA-9707564</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0c12c_level0_row6\" class=\"row_heading level0 row6\" >6</th>\n",
       "      <td id=\"T_0c12c_row6_col0\" class=\"data row6 col0\" >Respiratory electron transport - Reactome</td>\n",
       "      <td id=\"T_0c12c_row6_col1\" class=\"data row6 col1\" >0.000000</td>\n",
       "      <td id=\"T_0c12c_row6_col2\" class=\"data row6 col2\" >25</td>\n",
       "      <td id=\"T_0c12c_row6_col3\" class=\"data row6 col3\" >91</td>\n",
       "      <td id=\"T_0c12c_row6_col4\" class=\"data row6 col4\" >['MT-CO1', 'MT-ND4', 'MT-ND2', 'CYC1', 'COX5B', 'MT-CYB', 'UQCRB', 'COX7C', 'NDUFA1', 'MT-CO2', 'MT-CO3', 'MT-ND1', 'UQCRFS1', 'NDUFA3', 'NDUFA4', 'NDUFA6', 'UQCRQ', 'UQCR10', 'ACADVL', 'MT-ND6', 'NDUFB10', 'NDUFA2', 'IDH3B', 'COX4I1', 'COX6B1']</td>\n",
       "      <td id=\"T_0c12c_row6_col5\" class=\"data row6 col5\" >http://model.geneontology.org/R-HSA-611105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0c12c_level0_row7\" class=\"row_heading level0 row7\" >7</th>\n",
       "      <td id=\"T_0c12c_row7_col0\" class=\"data row7 col0\" >TP53 Regulates Metabolic Genes - Reactome</td>\n",
       "      <td id=\"T_0c12c_row7_col1\" class=\"data row7 col1\" >0.000000</td>\n",
       "      <td id=\"T_0c12c_row7_col2\" class=\"data row7 col2\" >17</td>\n",
       "      <td id=\"T_0c12c_row7_col3\" class=\"data row7 col3\" >59</td>\n",
       "      <td id=\"T_0c12c_row7_col4\" class=\"data row7 col4\" >['PRKAG3', 'G6PD', 'MT-CO1', 'CYC1', 'COX5B', 'MT-CYB', 'UQCRB', 'COX7C', 'PRDX1', 'MT-CO2', 'MT-CO3', 'UQCRFS1', 'NDUFA4', 'UQCRQ', 'UQCR10', 'COX4I1', 'COX6B1']</td>\n",
       "      <td id=\"T_0c12c_row7_col5\" class=\"data row7 col5\" >http://model.geneontology.org/R-HSA-5628897</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x14f5df970>"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PVneg_standard = enrich.enrich_wrapper('PVneg_FB_comb.csv','Gene Symbol',method='standard',FDR = 0.1 ,fpath = '../test_data/processed/',table_number = 32)\n",
    "PVneg_standard"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "994be1ef-3dc8-40db-85fb-d8232574ec7e",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|█████████████████████████████████████████| 452/452 [03:01<00:00,  2.49it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 396 entities from 338 out of 1135 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_d9cec caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_d9cec\">\n",
       "  <caption>Table S33: PVneg fibroblasts combined up/downregulated genes, weighted set method, FDR = 0.1.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_d9cec_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_d9cec_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_d9cec_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_d9cec_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_d9cec_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_d9cec_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_d9cec_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_d9cec_row0_col0\" class=\"data row0 col0\" >Peptide chain elongation - Reactome</td>\n",
       "      <td id=\"T_d9cec_row0_col1\" class=\"data row0 col1\" >0.000000</td>\n",
       "      <td id=\"T_d9cec_row0_col2\" class=\"data row0 col2\" >50</td>\n",
       "      <td id=\"T_d9cec_row0_col3\" class=\"data row0 col3\" >79</td>\n",
       "      <td id=\"T_d9cec_row0_col4\" class=\"data row0 col4\" >['RPS17', 'RPL13A', 'RPLP0', 'RPS18', 'RPLP2', 'RPL27A', 'RPS15A', 'RPS2', 'RPL7', 'RPL35', 'RPLP1', 'RPS20', 'RPS28', 'RPS5', 'RPS14', 'RPL34', 'RPL15', 'RPL23A', 'FAU', 'RPL21', 'RPL27', 'RPS25', 'RPL12', 'RPL10A', 'RPS15', 'RPS19', 'RPS16', 'RPL18', 'RPS21', 'RPL13', 'UBA52', 'RPL41', 'RPL7A', 'RPS11', 'RPS8', 'RPL29', 'RPS6', 'RPL8', 'RPS23', 'RPS9', 'RPSA', 'RPL4', 'RPS13', 'RPS27A', 'RPL19', 'set:RPL36A,RPL36AL', 'set:RPL3,RPL3L', 'set:RPL10,RPL10L', 'set:RPS27,RPS27L', 'set:RPL22,RPL22L1']</td>\n",
       "      <td id=\"T_d9cec_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-156902</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_d9cec_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_d9cec_row1_col0\" class=\"data row1 col0\" >Eukaryotic Translation Elongation - Reactome</td>\n",
       "      <td id=\"T_d9cec_row1_col1\" class=\"data row1 col1\" >0.000000</td>\n",
       "      <td id=\"T_d9cec_row1_col2\" class=\"data row1 col2\" >51</td>\n",
       "      <td id=\"T_d9cec_row1_col3\" class=\"data row1 col3\" >83</td>\n",
       "      <td id=\"T_d9cec_row1_col4\" class=\"data row1 col4\" >['RPS17', 'RPL13A', 'RPLP0', 'RPS18', 'RPLP2', 'RPL27A', 'RPS15A', 'RPS2', 'RPL7', 'RPL35', 'RPLP1', 'RPS20', 'RPS28', 'RPS5', 'RPS14', 'RPL34', 'RPL15', 'RPL23A', 'FAU', 'RPL21', 'RPL27', 'RPS25', 'RPL12', 'RPL10A', 'RPS15', 'RPS19', 'RPS16', 'RPL18', 'RPS21', 'RPL13', 'UBA52', 'RPL41', 'RPL7A', 'RPS11', 'RPS8', 'RPL29', 'RPS6', 'EEF1B2', 'RPL8', 'RPS23', 'RPS9', 'RPSA', 'RPL4', 'RPS13', 'RPS27A', 'RPL19', 'set:RPL36A,RPL36AL', 'set:RPL3,RPL3L', 'set:RPL10,RPL10L', 'set:RPS27,RPS27L', 'set:RPL22,RPL22L1']</td>\n",
       "      <td id=\"T_d9cec_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-156842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_d9cec_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "      <td id=\"T_d9cec_row2_col0\" class=\"data row2 col0\" >Formation of ATP by chemiosmotic coupling - Reactome</td>\n",
       "      <td id=\"T_d9cec_row2_col1\" class=\"data row2 col1\" >0.000000</td>\n",
       "      <td id=\"T_d9cec_row2_col2\" class=\"data row2 col2\" >29</td>\n",
       "      <td id=\"T_d9cec_row2_col3\" class=\"data row2 col3\" >88</td>\n",
       "      <td id=\"T_d9cec_row2_col4\" class=\"data row2 col4\" >['MT-CO1', 'MT-ND4', 'MT-ND2', 'ATP5MC1', 'CYC1', 'COX5B', 'MT-CYB', 'UQCRB', 'MT-ATP6', 'ATP5MC3', 'COX7C', 'NDUFA1', 'MT-CO2', 'MT-CO3', 'MT-ND1', 'UQCRFS1', 'NDUFA3', 'NDUFA4', 'NDUFA6', 'ATP5ME', 'UQCRQ', 'ATP5F1D', 'UQCR10', 'MT-ND6', 'NDUFB10', 'NDUFA2', 'COX4I1', 'COX6B1', 'set:ANT homodimer']</td>\n",
       "      <td id=\"T_d9cec_row2_col5\" class=\"data row2 col5\" >http://model.geneontology.org/R-HSA-163210</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_d9cec_level0_row3\" class=\"row_heading level0 row3\" >3</th>\n",
       "      <td id=\"T_d9cec_row3_col0\" class=\"data row3 col0\" >The proton buffering model - Reactome</td>\n",
       "      <td id=\"T_d9cec_row3_col1\" class=\"data row3 col1\" >0.000000</td>\n",
       "      <td id=\"T_d9cec_row3_col2\" class=\"data row3 col2\" >23</td>\n",
       "      <td id=\"T_d9cec_row3_col3\" class=\"data row3 col3\" >70</td>\n",
       "      <td id=\"T_d9cec_row3_col4\" class=\"data row3 col4\" >['MT-CO1', 'MT-ND4', 'MT-ND2', 'CYC1', 'COX5B', 'MT-CYB', 'UQCRB', 'COX7C', 'NDUFA1', 'MT-CO2', 'MT-CO3', 'MT-ND1', 'UQCRFS1', 'NDUFA3', 'NDUFA4', 'NDUFA6', 'UQCRQ', 'UQCR10', 'MT-ND6', 'NDUFB10', 'NDUFA2', 'COX4I1', 'COX6B1']</td>\n",
       "      <td id=\"T_d9cec_row3_col5\" class=\"data row3 col5\" >http://model.geneontology.org/R-HSA-167827</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_d9cec_level0_row4\" class=\"row_heading level0 row4\" >4</th>\n",
       "      <td id=\"T_d9cec_row4_col0\" class=\"data row4 col0\" >The fatty acid cycling model - Reactome</td>\n",
       "      <td id=\"T_d9cec_row4_col1\" class=\"data row4 col1\" >0.000000</td>\n",
       "      <td id=\"T_d9cec_row4_col2\" class=\"data row4 col2\" >23</td>\n",
       "      <td id=\"T_d9cec_row4_col3\" class=\"data row4 col3\" >70</td>\n",
       "      <td id=\"T_d9cec_row4_col4\" class=\"data row4 col4\" >['MT-CO1', 'MT-ND4', 'MT-ND2', 'CYC1', 'COX5B', 'MT-CYB', 'UQCRB', 'COX7C', 'NDUFA1', 'MT-CO2', 'MT-CO3', 'MT-ND1', 'UQCRFS1', 'NDUFA3', 'NDUFA4', 'NDUFA6', 'UQCRQ', 'UQCR10', 'MT-ND6', 'NDUFB10', 'NDUFA2', 'COX4I1', 'COX6B1']</td>\n",
       "      <td id=\"T_d9cec_row4_col5\" class=\"data row4 col5\" >http://model.geneontology.org/R-HSA-167826</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_d9cec_level0_row5\" class=\"row_heading level0 row5\" >5</th>\n",
       "      <td id=\"T_d9cec_row5_col0\" class=\"data row5 col0\" >Respiratory electron transport - Reactome</td>\n",
       "      <td id=\"T_d9cec_row5_col1\" class=\"data row5 col1\" >0.000000</td>\n",
       "      <td id=\"T_d9cec_row5_col2\" class=\"data row5 col2\" >25</td>\n",
       "      <td id=\"T_d9cec_row5_col3\" class=\"data row5 col3\" >91</td>\n",
       "      <td id=\"T_d9cec_row5_col4\" class=\"data row5 col4\" >['MT-CO1', 'MT-ND4', 'MT-ND2', 'CYC1', 'COX5B', 'MT-CYB', 'UQCRB', 'COX7C', 'NDUFA1', 'MT-CO2', 'MT-CO3', 'MT-ND1', 'UQCRFS1', 'NDUFA3', 'NDUFA4', 'NDUFA6', 'UQCRQ', 'UQCR10', 'ACADVL', 'MT-ND6', 'NDUFB10', 'NDUFA2', 'IDH3B', 'COX4I1', 'COX6B1']</td>\n",
       "      <td id=\"T_d9cec_row5_col5\" class=\"data row5 col5\" >http://model.geneontology.org/R-HSA-611105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_d9cec_level0_row6\" class=\"row_heading level0 row6\" >6</th>\n",
       "      <td id=\"T_d9cec_row6_col0\" class=\"data row6 col0\" >Cytoprotection by HMOX1 - Reactome</td>\n",
       "      <td id=\"T_d9cec_row6_col1\" class=\"data row6 col1\" >0.000000</td>\n",
       "      <td id=\"T_d9cec_row6_col2\" class=\"data row6 col2\" >15</td>\n",
       "      <td id=\"T_d9cec_row6_col3\" class=\"data row6 col3\" >29</td>\n",
       "      <td id=\"T_d9cec_row6_col4\" class=\"data row6 col4\" >['MT-CO1', 'CYC1', 'COX5B', 'MT-CYB', 'UQCRB', 'COX7C', 'MT-CO2', 'MT-CO3', 'UQCRFS1', 'NDUFA4', 'UQCRQ', 'UQCR10', 'COX4I1', 'COX6B1', 'set:BLVRA:Zn2+,BLVRB']</td>\n",
       "      <td id=\"T_d9cec_row6_col5\" class=\"data row6 col5\" >http://model.geneontology.org/R-HSA-9707564</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_d9cec_level0_row7\" class=\"row_heading level0 row7\" >7</th>\n",
       "      <td id=\"T_d9cec_row7_col0\" class=\"data row7 col0\" >TP53 Regulates Metabolic Genes - Reactome</td>\n",
       "      <td id=\"T_d9cec_row7_col1\" class=\"data row7 col1\" >0.000000</td>\n",
       "      <td id=\"T_d9cec_row7_col2\" class=\"data row7 col2\" >18</td>\n",
       "      <td id=\"T_d9cec_row7_col3\" class=\"data row7 col3\" >46</td>\n",
       "      <td id=\"T_d9cec_row7_col4\" class=\"data row7 col4\" >['G6PD', 'MT-CO1', 'CYC1', 'COX5B', 'MT-CYB', 'UQCRB', 'COX7C', 'PRDX1', 'MT-CO2', 'MT-CO3', 'UQCRFS1', 'NDUFA4', 'UQCRQ', 'UQCR10', 'COX4I1', 'COX6B1', 'set:PRDX1,2,5', 'set:AMPK gamma']</td>\n",
       "      <td id=\"T_d9cec_row7_col5\" class=\"data row7 col5\" >http://model.geneontology.org/R-HSA-5628897</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x14ecafa00>"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PVneg_weighted = enrich.enrich_wrapper('PVneg_FB_comb.csv','Gene Symbol',method='ncHGT',FDR = 0.1 ,fpath = '../test_data/processed/',table_number = 33)\n",
    "PVneg_weighted"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e69586b5-b669-46d7-a335-23639aeb6b70",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Left Ventricle Mural Smooth Muscle Cells"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "30233f54-1b6e-4318-b387-a80de72415f8",
   "metadata": {
    "tags": []
   },
   "source": [
    "### LMNA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "0ef1092e-e5f6-4a09-a4f0-1104af1570f3",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 326/326 [00:00<00:00, 16314.02it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 210 entities from 161 out of 633 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_88f77 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_88f77\">\n",
       "  <caption>Table S34: LMNA smooth muscle cells combined up/downregulated genes, unweighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_88f77_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_88f77_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_88f77_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_88f77_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_88f77_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_88f77_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_88f77_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_88f77_row0_col0\" class=\"data row0 col0\" >Eukaryotic Translation Elongation - Reactome</td>\n",
       "      <td id=\"T_88f77_row0_col1\" class=\"data row0 col1\" >0.000004</td>\n",
       "      <td id=\"T_88f77_row0_col2\" class=\"data row0 col2\" >16</td>\n",
       "      <td id=\"T_88f77_row0_col3\" class=\"data row0 col3\" >83</td>\n",
       "      <td id=\"T_88f77_row0_col4\" class=\"data row0 col4\" >['RPL13A', 'RPLP2', 'RPS18', 'RPL34', 'RPLP0', 'RPL7', 'RPS15A', 'RPS20', 'RPL23A', 'RPS15', 'RPL27A', 'UBA52', 'RPL35', 'RPS5', 'set:RPL3,RPL3L', 'set:EEF1A1-like proteins']</td>\n",
       "      <td id=\"T_88f77_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-156842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_88f77_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_88f77_row1_col0\" class=\"data row1 col0\" >Regulation of Complement cascade - Reactome</td>\n",
       "      <td id=\"T_88f77_row1_col1\" class=\"data row1 col1\" >0.000007</td>\n",
       "      <td id=\"T_88f77_row1_col2\" class=\"data row1 col2\" >6</td>\n",
       "      <td id=\"T_88f77_row1_col3\" class=\"data row1 col3\" >11</td>\n",
       "      <td id=\"T_88f77_row1_col4\" class=\"data row1 col4\" >['C3', 'C1R', 'set:C3 convertases', 'set:Complement factor D', 'set:C4 activators', 'set:CD46, CR1:C4b:C3b complexes']</td>\n",
       "      <td id=\"T_88f77_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-977606</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_88f77_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "      <td id=\"T_88f77_row2_col0\" class=\"data row2 col0\" >Peptide chain elongation - Reactome</td>\n",
       "      <td id=\"T_88f77_row2_col1\" class=\"data row2 col1\" >0.000011</td>\n",
       "      <td id=\"T_88f77_row2_col2\" class=\"data row2 col2\" >15</td>\n",
       "      <td id=\"T_88f77_row2_col3\" class=\"data row2 col3\" >79</td>\n",
       "      <td id=\"T_88f77_row2_col4\" class=\"data row2 col4\" >['RPL13A', 'RPLP2', 'RPS18', 'RPL34', 'RPLP0', 'RPL7', 'RPS15A', 'RPS20', 'RPL23A', 'RPS15', 'RPL27A', 'UBA52', 'RPL35', 'RPS5', 'set:RPL3,RPL3L']</td>\n",
       "      <td id=\"T_88f77_row2_col5\" class=\"data row2 col5\" >http://model.geneontology.org/R-HSA-156902</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_88f77_level0_row3\" class=\"row_heading level0 row3\" >3</th>\n",
       "      <td id=\"T_88f77_row3_col0\" class=\"data row3 col0\" >Acyl chain remodelling of PC - Reactome</td>\n",
       "      <td id=\"T_88f77_row3_col1\" class=\"data row3 col1\" >0.000041</td>\n",
       "      <td id=\"T_88f77_row3_col2\" class=\"data row3 col2\" >5</td>\n",
       "      <td id=\"T_88f77_row3_col3\" class=\"data row3 col3\" >9</td>\n",
       "      <td id=\"T_88f77_row3_col4\" class=\"data row3 col4\" >['set:PLA2(7)', 'set:LPCAT', 'set:PLA2(5)', 'set:PLA2(16)', 'set:PLA2(6)']</td>\n",
       "      <td id=\"T_88f77_row3_col5\" class=\"data row3 col5\" >http://model.geneontology.org/R-HSA-1482788</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_88f77_level0_row4\" class=\"row_heading level0 row4\" >4</th>\n",
       "      <td id=\"T_88f77_row4_col0\" class=\"data row4 col0\" >Acyl chain remodelling of PE - Reactome</td>\n",
       "      <td id=\"T_88f77_row4_col1\" class=\"data row4 col1\" >0.000079</td>\n",
       "      <td id=\"T_88f77_row4_col2\" class=\"data row4 col2\" >5</td>\n",
       "      <td id=\"T_88f77_row4_col3\" class=\"data row4 col3\" >10</td>\n",
       "      <td id=\"T_88f77_row4_col4\" class=\"data row4 col4\" >['set:PLA2(4)', 'set:LPEAT', 'set:PLA2(3)', 'set:PLA2(2)', 'set:PLA2(16)']</td>\n",
       "      <td id=\"T_88f77_row4_col5\" class=\"data row4 col5\" >http://model.geneontology.org/R-HSA-1482839</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_88f77_level0_row5\" class=\"row_heading level0 row5\" >5</th>\n",
       "      <td id=\"T_88f77_row5_col0\" class=\"data row5 col0\" >Hydrolysis of LPC - Reactome</td>\n",
       "      <td id=\"T_88f77_row5_col1\" class=\"data row5 col1\" >0.000232</td>\n",
       "      <td id=\"T_88f77_row5_col2\" class=\"data row5 col2\" >4</td>\n",
       "      <td id=\"T_88f77_row5_col3\" class=\"data row5 col3\" >7</td>\n",
       "      <td id=\"T_88f77_row5_col4\" class=\"data row5 col4\" >['set:PLA2(7)', 'set:PLA2(8)', 'set:PLA2(5)', 'set:PLA2(6)']</td>\n",
       "      <td id=\"T_88f77_row5_col5\" class=\"data row5 col5\" >http://model.geneontology.org/R-HSA-1483115</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x14f56cc70>"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "LMNA_set = enrich.enrich_wrapper('LMNA_SMC_comb.csv','Gene Symbol',method='set',FDR = 0.05,fpath = '../test_data/processed/',table_number = 34)\n",
    "LMNA_set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "515619b7-7173-4054-87d4-7af13bbf504b",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 326/326 [00:00<00:00, 15945.32it/s]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_a9b09 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_a9b09\">\n",
       "  <caption>Table S35: LMNA smooth muscle cells combined up/downregulated genes, standard method, FDR = 0.1.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_a9b09_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_a9b09_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_a9b09_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_a9b09_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_a9b09_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_a9b09_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_a9b09_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_a9b09_row0_col0\" class=\"data row0 col0\" >Eukaryotic Translation Elongation - Reactome</td>\n",
       "      <td id=\"T_a9b09_row0_col1\" class=\"data row0 col1\" >0.000000</td>\n",
       "      <td id=\"T_a9b09_row0_col2\" class=\"data row0 col2\" >16</td>\n",
       "      <td id=\"T_a9b09_row0_col3\" class=\"data row0 col3\" >94</td>\n",
       "      <td id=\"T_a9b09_row0_col4\" class=\"data row0 col4\" >['EEF1A2', 'RPL13A', 'RPLP2', 'RPS18', 'RPL34', 'RPLP0', 'RPL7', 'RPL3', 'RPS15A', 'RPS20', 'RPL23A', 'RPS15', 'RPL27A', 'UBA52', 'RPL35', 'RPS5']</td>\n",
       "      <td id=\"T_a9b09_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-156842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a9b09_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_a9b09_row1_col0\" class=\"data row1 col0\" >Peptide chain elongation - Reactome</td>\n",
       "      <td id=\"T_a9b09_row1_col1\" class=\"data row1 col1\" >0.000000</td>\n",
       "      <td id=\"T_a9b09_row1_col2\" class=\"data row1 col2\" >15</td>\n",
       "      <td id=\"T_a9b09_row1_col3\" class=\"data row1 col3\" >88</td>\n",
       "      <td id=\"T_a9b09_row1_col4\" class=\"data row1 col4\" >['RPL13A', 'RPLP2', 'RPS18', 'RPL34', 'RPLP0', 'RPL7', 'RPL3', 'RPS15A', 'RPS20', 'RPL23A', 'RPS15', 'RPL27A', 'UBA52', 'RPL35', 'RPS5']</td>\n",
       "      <td id=\"T_a9b09_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-156902</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x14f56f850>"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "LMNA_standard = enrich.enrich_wrapper('LMNA_SMC_comb.csv','Gene Symbol',method='standard',FDR = 0.1 ,fpath = '../test_data/processed/',table_number = 35)\n",
    "LMNA_standard"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "bfafd0d1-8b7b-4d9d-b703-f91109e1593b",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|█████████████████████████████████████████| 326/326 [02:09<00:00,  2.52it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 210 entities from 161 out of 633 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_de975 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_de975\">\n",
       "  <caption>Table S36: LMNA smooth muscle cells combined up/downregulated genes, weighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_de975_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_de975_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_de975_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_de975_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_de975_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_de975_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_de975_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_de975_row0_col0\" class=\"data row0 col0\" >Eukaryotic Translation Elongation - Reactome</td>\n",
       "      <td id=\"T_de975_row0_col1\" class=\"data row0 col1\" >0.000000</td>\n",
       "      <td id=\"T_de975_row0_col2\" class=\"data row0 col2\" >16</td>\n",
       "      <td id=\"T_de975_row0_col3\" class=\"data row0 col3\" >83</td>\n",
       "      <td id=\"T_de975_row0_col4\" class=\"data row0 col4\" >['RPL13A', 'RPLP2', 'RPS18', 'RPL34', 'RPLP0', 'RPL7', 'RPS15A', 'RPS20', 'RPL23A', 'RPS15', 'RPL27A', 'UBA52', 'RPL35', 'RPS5', 'set:RPL3,RPL3L', 'set:EEF1A1-like proteins']</td>\n",
       "      <td id=\"T_de975_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-156842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_de975_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_de975_row1_col0\" class=\"data row1 col0\" >Peptide chain elongation - Reactome</td>\n",
       "      <td id=\"T_de975_row1_col1\" class=\"data row1 col1\" >0.000000</td>\n",
       "      <td id=\"T_de975_row1_col2\" class=\"data row1 col2\" >15</td>\n",
       "      <td id=\"T_de975_row1_col3\" class=\"data row1 col3\" >79</td>\n",
       "      <td id=\"T_de975_row1_col4\" class=\"data row1 col4\" >['RPL13A', 'RPLP2', 'RPS18', 'RPL34', 'RPLP0', 'RPL7', 'RPS15A', 'RPS20', 'RPL23A', 'RPS15', 'RPL27A', 'UBA52', 'RPL35', 'RPS5', 'set:RPL3,RPL3L']</td>\n",
       "      <td id=\"T_de975_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-156902</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x14f56d0f0>"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "LMNA_weighted = enrich.enrich_wrapper('LMNA_SMC_comb.csv','Gene Symbol',method='ncHGT',FDR = 0.05,fpath = '../test_data/processed/',table_number = 36)\n",
    "LMNA_weighted"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cb8a3779-9227-4e28-93f9-ded918265700",
   "metadata": {
    "tags": []
   },
   "source": [
    "### RBM20"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "0ec120b0-552a-4ac4-a67e-4e5f155e08db",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 195/195 [00:00<00:00, 16203.85it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 135 entities from 103 out of 435 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_d8089 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_d8089\">\n",
       "  <caption>Table S37: RBM20 smooth muscle cells combined up/downregulated genes, unweighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_d8089_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_d8089_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_d8089_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_d8089_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_d8089_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_d8089_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_d8089_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_d8089_row0_col0\" class=\"data row0 col0\" >Regulation of Complement cascade - Reactome</td>\n",
       "      <td id=\"T_d8089_row0_col1\" class=\"data row0 col1\" >0.000016</td>\n",
       "      <td id=\"T_d8089_row0_col2\" class=\"data row0 col2\" >5</td>\n",
       "      <td id=\"T_d8089_row0_col3\" class=\"data row0 col3\" >11</td>\n",
       "      <td id=\"T_d8089_row0_col4\" class=\"data row0 col4\" >['C3', 'set:C3 convertases', 'set:C4 activators', 'set:Complement factor D', 'set:CD46, CR1:C4b:C3b complexes']</td>\n",
       "      <td id=\"T_d8089_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-977606</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x14e3e1300>"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "RBM20_set = enrich.enrich_wrapper('RBM20_SMC_comb.csv','Gene Symbol',method='set',FDR = 0.05,fpath = '../test_data/processed/',table_number = 37)\n",
    "RBM20_set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "0e4f430d-37d1-4585-9ec4-bc71f09371a0",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 195/195 [00:00<00:00, 17262.70it/s]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_109a9 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_109a9\">\n",
       "  <caption>Table S38: RBM20 smooth muscle cells combined up/downregulated genes, standard method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_109a9_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_109a9_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_109a9_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_109a9_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_109a9_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_109a9_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_109a9_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_109a9_row0_col0\" class=\"data row0 col0\" >Eukaryotic Translation Elongation - Reactome</td>\n",
       "      <td id=\"T_109a9_row0_col1\" class=\"data row0 col1\" >0.000017</td>\n",
       "      <td id=\"T_109a9_row0_col2\" class=\"data row0 col2\" >10</td>\n",
       "      <td id=\"T_109a9_row0_col3\" class=\"data row0 col3\" >94</td>\n",
       "      <td id=\"T_109a9_row0_col4\" class=\"data row0 col4\" >['EEF1A2', 'RPL13A', 'RPS18', 'RPLP0', 'RPS15A', 'RPLP2', 'RPL34', 'RPL27A', 'RPS20', 'RPL38']</td>\n",
       "      <td id=\"T_109a9_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-156842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_109a9_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_109a9_row1_col0\" class=\"data row1 col0\" >Peptide chain elongation - Reactome</td>\n",
       "      <td id=\"T_109a9_row1_col1\" class=\"data row1 col1\" >0.000062</td>\n",
       "      <td id=\"T_109a9_row1_col2\" class=\"data row1 col2\" >9</td>\n",
       "      <td id=\"T_109a9_row1_col3\" class=\"data row1 col3\" >88</td>\n",
       "      <td id=\"T_109a9_row1_col4\" class=\"data row1 col4\" >['RPL13A', 'RPS18', 'RPLP0', 'RPS15A', 'RPLP2', 'RPL34', 'RPL27A', 'RPS20', 'RPL38']</td>\n",
       "      <td id=\"T_109a9_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-156902</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x13bc3e890>"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "RBM20_standard = enrich.enrich_wrapper('RBM20_SMC_comb.csv','Gene Symbol',method='standard',FDR = 0.05,fpath = '../test_data/processed/',table_number = 38)\n",
    "RBM20_standard"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "3742a5c8-977b-41f3-a1ab-9483e1add8f2",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|█████████████████████████████████████████| 195/195 [01:15<00:00,  2.58it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 135 entities from 103 out of 435 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_2e4f8 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_2e4f8\">\n",
       "  <caption>Table S39: RBM20 smooth muscle cells combined up/downregulated genes, weighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_2e4f8_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_2e4f8_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_2e4f8_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_2e4f8_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_2e4f8_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_2e4f8_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_2e4f8_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_2e4f8_row0_col0\" class=\"data row0 col0\" >Eukaryotic Translation Elongation - Reactome</td>\n",
       "      <td id=\"T_2e4f8_row0_col1\" class=\"data row0 col1\" >0.000004</td>\n",
       "      <td id=\"T_2e4f8_row0_col2\" class=\"data row0 col2\" >10</td>\n",
       "      <td id=\"T_2e4f8_row0_col3\" class=\"data row0 col3\" >83</td>\n",
       "      <td id=\"T_2e4f8_row0_col4\" class=\"data row0 col4\" >['RPL13A', 'RPS18', 'RPLP0', 'RPS15A', 'RPLP2', 'RPL34', 'RPL27A', 'RPS20', 'RPL38', 'set:EEF1A1-like proteins']</td>\n",
       "      <td id=\"T_2e4f8_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-156842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2e4f8_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_2e4f8_row1_col0\" class=\"data row1 col0\" >Peptide chain elongation - Reactome</td>\n",
       "      <td id=\"T_2e4f8_row1_col1\" class=\"data row1 col1\" >0.000019</td>\n",
       "      <td id=\"T_2e4f8_row1_col2\" class=\"data row1 col2\" >9</td>\n",
       "      <td id=\"T_2e4f8_row1_col3\" class=\"data row1 col3\" >79</td>\n",
       "      <td id=\"T_2e4f8_row1_col4\" class=\"data row1 col4\" >['RPL13A', 'RPS18', 'RPLP0', 'RPS15A', 'RPLP2', 'RPL34', 'RPL27A', 'RPS20', 'RPL38']</td>\n",
       "      <td id=\"T_2e4f8_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-156902</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x12fd7ae90>"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "RBM20_weighted = enrich.enrich_wrapper('RBM20_SMC_comb.csv','Gene Symbol',method='ncHGT',FDR = 0.05,fpath = '../test_data/processed/',table_number = 39)\n",
    "RBM20_weighted"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "19c5f98e-ff5e-458a-a2f7-89dbdae22db2",
   "metadata": {
    "tags": []
   },
   "source": [
    "### PKP2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "b201bb3c-d3f3-4fb0-a062-7b9bf461b1cd",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 185/185 [00:00<00:00, 16042.22it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 92 entities from 65 out of 309 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_354a6 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_354a6\">\n",
       "  <caption>Table S40: PKP2 smooth muscle cells combined up/downregulated genes, unweighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_354a6_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_354a6_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_354a6_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_354a6_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_354a6_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_354a6_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x12fd7a5c0>"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PKP2_set = enrich.enrich_wrapper('PKP2_SMC_comb.csv','Gene Symbol',method='set',FDR = 0.05,fpath = '../test_data/processed/',table_number = 40)\n",
    "PKP2_set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "995f3fe0-3d5d-4e7a-9c7d-a3137d873718",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 185/185 [00:00<00:00, 15716.64it/s]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_5449a caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_5449a\">\n",
       "  <caption>Table S41: PKP2 smooth muscle cells combined up/downregulated genes, standard method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_5449a_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_5449a_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_5449a_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_5449a_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_5449a_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_5449a_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x12fd79f30>"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PKP2_standard = enrich.enrich_wrapper('PKP2_SMC_comb.csv','Gene Symbol',method='standard',FDR = 0.05,fpath = '../test_data/processed/',table_number = 41)\n",
    "PKP2_standard"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "59398e15-6769-432d-8c63-ccefeaa82340",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|█████████████████████████████████████████| 185/185 [01:21<00:00,  2.27it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 92 entities from 65 out of 309 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_49489 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_49489\">\n",
       "  <caption>Table S42: PKP2 smooth muscle cells combined up/downregulated genes, weighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_49489_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_49489_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_49489_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_49489_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_49489_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_49489_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x14e64b340>"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PKP2_weighted = enrich.enrich_wrapper('PKP2_SMC_comb.csv','Gene Symbol',method='ncHGT',FDR = 0.05,fpath = '../test_data/processed/',table_number = 42)\n",
    "PKP2_weighted"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9fe55e77-d3af-4544-8e39-dcd614d4909d",
   "metadata": {
    "tags": []
   },
   "source": [
    "### TTN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "5d5a4bf2-af43-448e-a110-5844ac10e295",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 268/268 [00:00<00:00, 16117.37it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 139 entities from 103 out of 396 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_1f09d caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_1f09d\">\n",
       "  <caption>Table S43: TTN smooth muscle cells combined up/downregulated genes, unweighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_1f09d_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_1f09d_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_1f09d_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_1f09d_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_1f09d_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_1f09d_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_1f09d_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_1f09d_row0_col0\" class=\"data row0 col0\" >Eukaryotic Translation Elongation - Reactome</td>\n",
       "      <td id=\"T_1f09d_row0_col1\" class=\"data row0 col1\" >0.000000</td>\n",
       "      <td id=\"T_1f09d_row0_col2\" class=\"data row0 col2\" >16</td>\n",
       "      <td id=\"T_1f09d_row0_col3\" class=\"data row0 col3\" >83</td>\n",
       "      <td id=\"T_1f09d_row0_col4\" class=\"data row0 col4\" >['RPL13A', 'RPS18', 'RPLP0', 'RPLP2', 'RPS5', 'RPL18', 'RPL34', 'RPL7', 'RPS14', 'RPL27', 'RPS15', 'RPS15A', 'RPS25', 'RPSA', 'set:RPL3,RPL3L', 'set:EEF1A1-like proteins']</td>\n",
       "      <td id=\"T_1f09d_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-156842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_1f09d_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_1f09d_row1_col0\" class=\"data row1 col0\" >Peptide chain elongation - Reactome</td>\n",
       "      <td id=\"T_1f09d_row1_col1\" class=\"data row1 col1\" >0.000000</td>\n",
       "      <td id=\"T_1f09d_row1_col2\" class=\"data row1 col2\" >15</td>\n",
       "      <td id=\"T_1f09d_row1_col3\" class=\"data row1 col3\" >79</td>\n",
       "      <td id=\"T_1f09d_row1_col4\" class=\"data row1 col4\" >['RPL13A', 'RPS18', 'RPLP0', 'RPLP2', 'RPS5', 'RPL18', 'RPL34', 'RPL7', 'RPS14', 'RPL27', 'RPS15', 'RPS15A', 'RPS25', 'RPSA', 'set:RPL3,RPL3L']</td>\n",
       "      <td id=\"T_1f09d_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-156902</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_1f09d_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "      <td id=\"T_1f09d_row2_col0\" class=\"data row2 col0\" >Regulation of Complement cascade - Reactome</td>\n",
       "      <td id=\"T_1f09d_row2_col1\" class=\"data row2 col1\" >0.000001</td>\n",
       "      <td id=\"T_1f09d_row2_col2\" class=\"data row2 col2\" >6</td>\n",
       "      <td id=\"T_1f09d_row2_col3\" class=\"data row2 col3\" >11</td>\n",
       "      <td id=\"T_1f09d_row2_col4\" class=\"data row2 col4\" >['C3', 'C1R', 'set:C3 convertases', 'set:Complement factor D', 'set:C4 activators', 'set:CD46, CR1:C4b:C3b complexes']</td>\n",
       "      <td id=\"T_1f09d_row2_col5\" class=\"data row2 col5\" >http://model.geneontology.org/R-HSA-977606</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x14f56e170>"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TTN_set = enrich.enrich_wrapper('TTN_SMC_comb.csv','Gene Symbol',method='set',FDR = 0.05,fpath = '../test_data/processed/',table_number = 43)\n",
    "TTN_set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "5f50d9d4-ea4a-4184-8ce5-02e245c06f9a",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 268/268 [00:00<00:00, 15864.64it/s]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_3a5df caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_3a5df\">\n",
       "  <caption>Table S44: TTN smooth muscle cells combined up/downregulated genes, standard method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_3a5df_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_3a5df_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_3a5df_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_3a5df_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_3a5df_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_3a5df_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_3a5df_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_3a5df_row0_col0\" class=\"data row0 col0\" >Eukaryotic Translation Elongation - Reactome</td>\n",
       "      <td id=\"T_3a5df_row0_col1\" class=\"data row0 col1\" >0.000000</td>\n",
       "      <td id=\"T_3a5df_row0_col2\" class=\"data row0 col2\" >16</td>\n",
       "      <td id=\"T_3a5df_row0_col3\" class=\"data row0 col3\" >94</td>\n",
       "      <td id=\"T_3a5df_row0_col4\" class=\"data row0 col4\" >['EEF1A2', 'RPL13A', 'RPS18', 'RPL3', 'RPLP0', 'RPLP2', 'RPS5', 'RPL18', 'RPL34', 'RPL7', 'RPS14', 'RPL27', 'RPS15', 'RPS15A', 'RPS25', 'RPSA']</td>\n",
       "      <td id=\"T_3a5df_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-156842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_3a5df_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_3a5df_row1_col0\" class=\"data row1 col0\" >Peptide chain elongation - Reactome</td>\n",
       "      <td id=\"T_3a5df_row1_col1\" class=\"data row1 col1\" >0.000000</td>\n",
       "      <td id=\"T_3a5df_row1_col2\" class=\"data row1 col2\" >15</td>\n",
       "      <td id=\"T_3a5df_row1_col3\" class=\"data row1 col3\" >88</td>\n",
       "      <td id=\"T_3a5df_row1_col4\" class=\"data row1 col4\" >['RPL13A', 'RPS18', 'RPL3', 'RPLP0', 'RPLP2', 'RPS5', 'RPL18', 'RPL34', 'RPL7', 'RPS14', 'RPL27', 'RPS15', 'RPS15A', 'RPS25', 'RPSA']</td>\n",
       "      <td id=\"T_3a5df_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-156902</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x14e3e2200>"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TTN_standard = enrich.enrich_wrapper('TTN_SMC_comb.csv','Gene Symbol',method='standard',FDR = 0.05,fpath = '../test_data/processed/',table_number = 44)\n",
    "TTN_standard"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "c8648810-a79e-47a5-b448-457e71901dfb",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|█████████████████████████████████████████| 268/268 [01:43<00:00,  2.59it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 139 entities from 103 out of 396 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_0779d caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_0779d\">\n",
       "  <caption>Table S45: TTN smooth muscle cells combined up/downregulated genes, weighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_0779d_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_0779d_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_0779d_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_0779d_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_0779d_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_0779d_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_0779d_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_0779d_row0_col0\" class=\"data row0 col0\" >Eukaryotic Translation Elongation - Reactome</td>\n",
       "      <td id=\"T_0779d_row0_col1\" class=\"data row0 col1\" >0.000000</td>\n",
       "      <td id=\"T_0779d_row0_col2\" class=\"data row0 col2\" >16</td>\n",
       "      <td id=\"T_0779d_row0_col3\" class=\"data row0 col3\" >83</td>\n",
       "      <td id=\"T_0779d_row0_col4\" class=\"data row0 col4\" >['RPL13A', 'RPS18', 'RPLP0', 'RPLP2', 'RPS5', 'RPL18', 'RPL34', 'RPL7', 'RPS14', 'RPL27', 'RPS15', 'RPS15A', 'RPS25', 'RPSA', 'set:RPL3,RPL3L', 'set:EEF1A1-like proteins']</td>\n",
       "      <td id=\"T_0779d_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-156842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0779d_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_0779d_row1_col0\" class=\"data row1 col0\" >Peptide chain elongation - Reactome</td>\n",
       "      <td id=\"T_0779d_row1_col1\" class=\"data row1 col1\" >0.000000</td>\n",
       "      <td id=\"T_0779d_row1_col2\" class=\"data row1 col2\" >15</td>\n",
       "      <td id=\"T_0779d_row1_col3\" class=\"data row1 col3\" >79</td>\n",
       "      <td id=\"T_0779d_row1_col4\" class=\"data row1 col4\" >['RPL13A', 'RPS18', 'RPLP0', 'RPLP2', 'RPS5', 'RPL18', 'RPL34', 'RPL7', 'RPS14', 'RPL27', 'RPS15', 'RPS15A', 'RPS25', 'RPSA', 'set:RPL3,RPL3L']</td>\n",
       "      <td id=\"T_0779d_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-156902</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0779d_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "      <td id=\"T_0779d_row2_col0\" class=\"data row2 col0\" >Regulation of Complement cascade - Reactome</td>\n",
       "      <td id=\"T_0779d_row2_col1\" class=\"data row2 col1\" >0.000082</td>\n",
       "      <td id=\"T_0779d_row2_col2\" class=\"data row2 col2\" >6</td>\n",
       "      <td id=\"T_0779d_row2_col3\" class=\"data row2 col3\" >11</td>\n",
       "      <td id=\"T_0779d_row2_col4\" class=\"data row2 col4\" >['C3', 'C1R', 'set:C3 convertases', 'set:Complement factor D', 'set:C4 activators', 'set:CD46, CR1:C4b:C3b complexes']</td>\n",
       "      <td id=\"T_0779d_row2_col5\" class=\"data row2 col5\" >http://model.geneontology.org/R-HSA-977606</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x12fd7bac0>"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TTN_weighted = enrich.enrich_wrapper('TTN_SMC_comb.csv','Gene Symbol',method='ncHGT',FDR = 0.05,fpath = '../test_data/processed/',table_number = 45)\n",
    "TTN_weighted"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7d784530-b1e1-4b87-b33a-4874d21410d0",
   "metadata": {
    "tags": []
   },
   "source": [
    "### PVneg"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "32010cd5-fa8c-4990-ab67-d6be597d2566",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 324/324 [00:00<00:00, 16269.25it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 256 entities from 221 out of 686 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_4dc5f caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_4dc5f\">\n",
       "  <caption>Table S46: PVneg smooth muscle cells combined up/downregulated genes, unweighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_4dc5f_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_4dc5f_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_4dc5f_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_4dc5f_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_4dc5f_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_4dc5f_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_4dc5f_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_4dc5f_row0_col0\" class=\"data row0 col0\" >Peptide chain elongation - Reactome</td>\n",
       "      <td id=\"T_4dc5f_row0_col1\" class=\"data row0 col1\" >0.000000</td>\n",
       "      <td id=\"T_4dc5f_row0_col2\" class=\"data row0 col2\" >41</td>\n",
       "      <td id=\"T_4dc5f_row0_col3\" class=\"data row0 col3\" >79</td>\n",
       "      <td id=\"T_4dc5f_row0_col4\" class=\"data row0 col4\" >['RPL13A', 'RPLP2', 'RPLP0', 'RPS18', 'RPL35', 'RPL27A', 'RPS15A', 'RPL7', 'RPS5', 'RPS2', 'RPS20', 'RPLP1', 'RPSA', 'RPL10A', 'RPS25', 'RPL15', 'RPL27', 'RPL23A', 'RPL34', 'RPS16', 'RPS9', 'RPS14', 'RPL8', 'RPS15', 'RPL38', 'RPS29', 'RPL13', 'RPL4', 'RPS19', 'RPL12', 'RPL29', 'RPS28', 'RPS6', 'RPS23', 'RPL21', 'UBA52', 'RPL41', 'set:RPL36A,RPL36AL', 'set:RPL3,RPL3L', 'set:RPL10,RPL10L', 'set:RPS27,RPS27L']</td>\n",
       "      <td id=\"T_4dc5f_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-156902</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_4dc5f_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_4dc5f_row1_col0\" class=\"data row1 col0\" >Eukaryotic Translation Elongation - Reactome</td>\n",
       "      <td id=\"T_4dc5f_row1_col1\" class=\"data row1 col1\" >0.000000</td>\n",
       "      <td id=\"T_4dc5f_row1_col2\" class=\"data row1 col2\" >41</td>\n",
       "      <td id=\"T_4dc5f_row1_col3\" class=\"data row1 col3\" >83</td>\n",
       "      <td id=\"T_4dc5f_row1_col4\" class=\"data row1 col4\" >['RPL13A', 'RPLP2', 'RPLP0', 'RPS18', 'RPL35', 'RPL27A', 'RPS15A', 'RPL7', 'RPS5', 'RPS2', 'RPS20', 'RPLP1', 'RPSA', 'RPL10A', 'RPS25', 'RPL15', 'RPL27', 'RPL23A', 'RPL34', 'RPS16', 'RPS9', 'RPS14', 'RPL8', 'RPS15', 'RPL38', 'RPS29', 'RPL13', 'RPL4', 'RPS19', 'RPL12', 'RPL29', 'RPS28', 'RPS6', 'RPS23', 'RPL21', 'UBA52', 'RPL41', 'set:RPL36A,RPL36AL', 'set:RPL3,RPL3L', 'set:RPL10,RPL10L', 'set:RPS27,RPS27L']</td>\n",
       "      <td id=\"T_4dc5f_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-156842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_4dc5f_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "      <td id=\"T_4dc5f_row2_col0\" class=\"data row2 col0\" >Formation of ATP by chemiosmotic coupling - Reactome</td>\n",
       "      <td id=\"T_4dc5f_row2_col1\" class=\"data row2 col1\" >0.000000</td>\n",
       "      <td id=\"T_4dc5f_row2_col2\" class=\"data row2 col2\" >23</td>\n",
       "      <td id=\"T_4dc5f_row2_col3\" class=\"data row2 col3\" >88</td>\n",
       "      <td id=\"T_4dc5f_row2_col4\" class=\"data row2 col4\" >['MT-ND4', 'MT-ND2', 'CYC1', 'UQCRQ', 'COX5B', 'ATP5MC1', 'COX7C', 'NDUFA1', 'MT-CO2', 'MT-CO1', 'ATP5F1D', 'ATP5MC3', 'UQCRB', 'COX6B1', 'NDUFA4', 'ATP5ME', 'NDUFA6', 'NDUFB3', 'ATP5PD', 'ATP5F1E', 'COX4I1', 'NDUFB11', 'set:ANT homodimer']</td>\n",
       "      <td id=\"T_4dc5f_row2_col5\" class=\"data row2 col5\" >http://model.geneontology.org/R-HSA-163210</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_4dc5f_level0_row3\" class=\"row_heading level0 row3\" >3</th>\n",
       "      <td id=\"T_4dc5f_row3_col0\" class=\"data row3 col0\" >Respiratory electron transport - Reactome</td>\n",
       "      <td id=\"T_4dc5f_row3_col1\" class=\"data row3 col1\" >0.000003</td>\n",
       "      <td id=\"T_4dc5f_row3_col2\" class=\"data row3 col2\" >19</td>\n",
       "      <td id=\"T_4dc5f_row3_col3\" class=\"data row3 col3\" >91</td>\n",
       "      <td id=\"T_4dc5f_row3_col4\" class=\"data row3 col4\" >['MT-ND4', 'MT-ND2', 'CYC1', 'UQCRQ', 'COX5B', 'IDH3B', 'ACADVL', 'COX7C', 'NDUFA1', 'MT-CO2', 'MT-CO1', 'UQCRB', 'COX6B1', 'NDUFA4', 'NDUFA6', 'IDH3G', 'NDUFB3', 'COX4I1', 'NDUFB11']</td>\n",
       "      <td id=\"T_4dc5f_row3_col5\" class=\"data row3 col5\" >http://model.geneontology.org/R-HSA-611105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_4dc5f_level0_row4\" class=\"row_heading level0 row4\" >4</th>\n",
       "      <td id=\"T_4dc5f_row4_col0\" class=\"data row4 col0\" >The proton buffering model - Reactome</td>\n",
       "      <td id=\"T_4dc5f_row4_col1\" class=\"data row4 col1\" >0.000006</td>\n",
       "      <td id=\"T_4dc5f_row4_col2\" class=\"data row4 col2\" >16</td>\n",
       "      <td id=\"T_4dc5f_row4_col3\" class=\"data row4 col3\" >70</td>\n",
       "      <td id=\"T_4dc5f_row4_col4\" class=\"data row4 col4\" >['MT-ND4', 'MT-ND2', 'CYC1', 'UQCRQ', 'COX5B', 'COX7C', 'NDUFA1', 'MT-CO2', 'MT-CO1', 'UQCRB', 'COX6B1', 'NDUFA4', 'NDUFA6', 'NDUFB3', 'COX4I1', 'NDUFB11']</td>\n",
       "      <td id=\"T_4dc5f_row4_col5\" class=\"data row4 col5\" >http://model.geneontology.org/R-HSA-167827</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_4dc5f_level0_row5\" class=\"row_heading level0 row5\" >5</th>\n",
       "      <td id=\"T_4dc5f_row5_col0\" class=\"data row5 col0\" >The fatty acid cycling model - Reactome</td>\n",
       "      <td id=\"T_4dc5f_row5_col1\" class=\"data row5 col1\" >0.000006</td>\n",
       "      <td id=\"T_4dc5f_row5_col2\" class=\"data row5 col2\" >16</td>\n",
       "      <td id=\"T_4dc5f_row5_col3\" class=\"data row5 col3\" >70</td>\n",
       "      <td id=\"T_4dc5f_row5_col4\" class=\"data row5 col4\" >['MT-ND4', 'MT-ND2', 'CYC1', 'UQCRQ', 'COX5B', 'COX7C', 'NDUFA1', 'MT-CO2', 'MT-CO1', 'UQCRB', 'COX6B1', 'NDUFA4', 'NDUFA6', 'NDUFB3', 'COX4I1', 'NDUFB11']</td>\n",
       "      <td id=\"T_4dc5f_row5_col5\" class=\"data row5 col5\" >http://model.geneontology.org/R-HSA-167826</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_4dc5f_level0_row6\" class=\"row_heading level0 row6\" >6</th>\n",
       "      <td id=\"T_4dc5f_row6_col0\" class=\"data row6 col0\" >Cytoprotection by HMOX1 - Reactome</td>\n",
       "      <td id=\"T_4dc5f_row6_col1\" class=\"data row6 col1\" >0.000007</td>\n",
       "      <td id=\"T_4dc5f_row6_col2\" class=\"data row6 col2\" >10</td>\n",
       "      <td id=\"T_4dc5f_row6_col3\" class=\"data row6 col3\" >29</td>\n",
       "      <td id=\"T_4dc5f_row6_col4\" class=\"data row6 col4\" >['CYC1', 'UQCRQ', 'COX5B', 'COX7C', 'MT-CO2', 'MT-CO1', 'UQCRB', 'COX6B1', 'NDUFA4', 'COX4I1']</td>\n",
       "      <td id=\"T_4dc5f_row6_col5\" class=\"data row6 col5\" >http://model.geneontology.org/R-HSA-9707564</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_4dc5f_level0_row7\" class=\"row_heading level0 row7\" >7</th>\n",
       "      <td id=\"T_4dc5f_row7_col0\" class=\"data row7 col0\" >Transport of nucleosides and free purine and pyrimidine bases across the plasma membrane - Reactome</td>\n",
       "      <td id=\"T_4dc5f_row7_col1\" class=\"data row7 col1\" >0.000010</td>\n",
       "      <td id=\"T_4dc5f_row7_col2\" class=\"data row7 col2\" >10</td>\n",
       "      <td id=\"T_4dc5f_row7_col3\" class=\"data row7 col3\" >30</td>\n",
       "      <td id=\"T_4dc5f_row7_col4\" class=\"data row7 col4\" >['ATP5MC1', 'ATP5F1D', 'ATP5MC3', 'SLC25A4', 'ATP5ME', 'ATP5PD', 'ATP5F1E', 'set:ANT homodimer', 'set:SLC29A1-like proteins', 'set:SLC29A2-like proteins']</td>\n",
       "      <td id=\"T_4dc5f_row7_col5\" class=\"data row7 col5\" >http://model.geneontology.org/R-HSA-83936</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_4dc5f_level0_row8\" class=\"row_heading level0 row8\" >8</th>\n",
       "      <td id=\"T_4dc5f_row8_col0\" class=\"data row8 col0\" >TP53 Regulates Metabolic Genes - Reactome</td>\n",
       "      <td id=\"T_4dc5f_row8_col1\" class=\"data row8 col1\" >0.000021</td>\n",
       "      <td id=\"T_4dc5f_row8_col2\" class=\"data row8 col2\" >12</td>\n",
       "      <td id=\"T_4dc5f_row8_col3\" class=\"data row8 col3\" >46</td>\n",
       "      <td id=\"T_4dc5f_row8_col4\" class=\"data row8 col4\" >['CYC1', 'UQCRQ', 'COX5B', 'COX7C', 'MT-CO2', 'MT-CO1', 'UQCRB', 'COX6B1', 'NDUFA4', 'COX4I1', 'PRDX1', 'set:PRDX1,2,5']</td>\n",
       "      <td id=\"T_4dc5f_row8_col5\" class=\"data row8 col5\" >http://model.geneontology.org/R-HSA-5628897</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_4dc5f_level0_row9\" class=\"row_heading level0 row9\" >9</th>\n",
       "      <td id=\"T_4dc5f_row9_col0\" class=\"data row9 col0\" >Chondroitin sulfate biosynthesis - Reactome</td>\n",
       "      <td id=\"T_4dc5f_row9_col1\" class=\"data row9 col1\" >0.000107</td>\n",
       "      <td id=\"T_4dc5f_row9_col2\" class=\"data row9 col2\" >5</td>\n",
       "      <td id=\"T_4dc5f_row9_col3\" class=\"data row9 col3\" >9</td>\n",
       "      <td id=\"T_4dc5f_row9_col4\" class=\"data row9 col4\" >['set:PAPST1,2', 'set:B3GAT dimers', 'set:CHPF,CHSY3', 'set:CHST9,11,12,13', 'set:CHPF,CHPF2,CHSY3']</td>\n",
       "      <td id=\"T_4dc5f_row9_col5\" class=\"data row9 col5\" >http://model.geneontology.org/R-HSA-2022870</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x14e64a530>"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PVneg_set = enrich.enrich_wrapper('PVneg_SMC_comb.csv','Gene Symbol',method='set',FDR = 0.05,fpath = '../test_data/processed/',table_number = 46)\n",
    "PVneg_set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "c93e4a6a-ca63-4aa9-893b-4a843c7d3daf",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 324/324 [00:00<00:00, 15890.12it/s]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_74bd9 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_74bd9\">\n",
       "  <caption>Table S47: PVneg smooth muscle cells combined up/downregulated genes, standard method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_74bd9_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_74bd9_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_74bd9_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_74bd9_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_74bd9_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_74bd9_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_74bd9_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_74bd9_row0_col0\" class=\"data row0 col0\" >Peptide chain elongation - Reactome</td>\n",
       "      <td id=\"T_74bd9_row0_col1\" class=\"data row0 col1\" >0.000000</td>\n",
       "      <td id=\"T_74bd9_row0_col2\" class=\"data row0 col2\" >42</td>\n",
       "      <td id=\"T_74bd9_row0_col3\" class=\"data row0 col3\" >88</td>\n",
       "      <td id=\"T_74bd9_row0_col4\" class=\"data row0 col4\" >['RPL13A', 'RPLP2', 'RPLP0', 'RPS18', 'RPL35', 'RPL3', 'RPL27A', 'RPS15A', 'RPL7', 'RPS5', 'RPS2', 'RPS20', 'RPS27', 'RPLP1', 'RPL3L', 'RPSA', 'RPL10A', 'RPS25', 'RPL15', 'RPL27', 'RPL23A', 'RPL34', 'RPS16', 'RPS9', 'RPS14', 'RPL8', 'RPS15', 'RPL38', 'RPL10', 'RPL36AL', 'RPS29', 'RPL13', 'RPL4', 'RPS19', 'RPL12', 'RPL29', 'RPS28', 'RPS6', 'RPS23', 'RPL21', 'UBA52', 'RPL41']</td>\n",
       "      <td id=\"T_74bd9_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-156902</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_74bd9_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_74bd9_row1_col0\" class=\"data row1 col0\" >Eukaryotic Translation Elongation - Reactome</td>\n",
       "      <td id=\"T_74bd9_row1_col1\" class=\"data row1 col1\" >0.000000</td>\n",
       "      <td id=\"T_74bd9_row1_col2\" class=\"data row1 col2\" >42</td>\n",
       "      <td id=\"T_74bd9_row1_col3\" class=\"data row1 col3\" >94</td>\n",
       "      <td id=\"T_74bd9_row1_col4\" class=\"data row1 col4\" >['RPL13A', 'RPLP2', 'RPLP0', 'RPS18', 'RPL35', 'RPL3', 'RPL27A', 'RPS15A', 'RPL7', 'RPS5', 'RPS2', 'RPS20', 'RPS27', 'RPLP1', 'RPL3L', 'RPSA', 'RPL10A', 'RPS25', 'RPL15', 'RPL27', 'RPL23A', 'RPL34', 'RPS16', 'RPS9', 'RPS14', 'RPL8', 'RPS15', 'RPL38', 'RPL10', 'RPL36AL', 'RPS29', 'RPL13', 'RPL4', 'RPS19', 'RPL12', 'RPL29', 'RPS28', 'RPS6', 'RPS23', 'RPL21', 'UBA52', 'RPL41']</td>\n",
       "      <td id=\"T_74bd9_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-156842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_74bd9_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "      <td id=\"T_74bd9_row2_col0\" class=\"data row2 col0\" >Formation of ATP by chemiosmotic coupling - Reactome</td>\n",
       "      <td id=\"T_74bd9_row2_col1\" class=\"data row2 col1\" >0.000000</td>\n",
       "      <td id=\"T_74bd9_row2_col2\" class=\"data row2 col2\" >24</td>\n",
       "      <td id=\"T_74bd9_row2_col3\" class=\"data row2 col3\" >89</td>\n",
       "      <td id=\"T_74bd9_row2_col4\" class=\"data row2 col4\" >['MT-ND4', 'MT-ND2', 'SLC25A6', 'CYC1', 'UQCRQ', 'COX5B', 'ATP5MC1', 'SLC25A5', 'COX7C', 'NDUFA1', 'MT-CO2', 'MT-CO1', 'ATP5F1D', 'ATP5MC3', 'UQCRB', 'COX6B1', 'NDUFA4', 'ATP5ME', 'NDUFA6', 'NDUFB3', 'ATP5PD', 'ATP5F1E', 'COX4I1', 'NDUFB11']</td>\n",
       "      <td id=\"T_74bd9_row2_col5\" class=\"data row2 col5\" >http://model.geneontology.org/R-HSA-163210</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_74bd9_level0_row3\" class=\"row_heading level0 row3\" >3</th>\n",
       "      <td id=\"T_74bd9_row3_col0\" class=\"data row3 col0\" >Respiratory electron transport - Reactome</td>\n",
       "      <td id=\"T_74bd9_row3_col1\" class=\"data row3 col1\" >0.000000</td>\n",
       "      <td id=\"T_74bd9_row3_col2\" class=\"data row3 col2\" >19</td>\n",
       "      <td id=\"T_74bd9_row3_col3\" class=\"data row3 col3\" >91</td>\n",
       "      <td id=\"T_74bd9_row3_col4\" class=\"data row3 col4\" >['MT-ND4', 'MT-ND2', 'CYC1', 'UQCRQ', 'COX5B', 'IDH3B', 'ACADVL', 'COX7C', 'NDUFA1', 'MT-CO2', 'MT-CO1', 'UQCRB', 'COX6B1', 'NDUFA4', 'NDUFA6', 'IDH3G', 'NDUFB3', 'COX4I1', 'NDUFB11']</td>\n",
       "      <td id=\"T_74bd9_row3_col5\" class=\"data row3 col5\" >http://model.geneontology.org/R-HSA-611105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_74bd9_level0_row4\" class=\"row_heading level0 row4\" >4</th>\n",
       "      <td id=\"T_74bd9_row4_col0\" class=\"data row4 col0\" >The fatty acid cycling model - Reactome</td>\n",
       "      <td id=\"T_74bd9_row4_col1\" class=\"data row4 col1\" >0.000000</td>\n",
       "      <td id=\"T_74bd9_row4_col2\" class=\"data row4 col2\" >16</td>\n",
       "      <td id=\"T_74bd9_row4_col3\" class=\"data row4 col3\" >74</td>\n",
       "      <td id=\"T_74bd9_row4_col4\" class=\"data row4 col4\" >['MT-ND4', 'MT-ND2', 'CYC1', 'UQCRQ', 'COX5B', 'COX7C', 'NDUFA1', 'MT-CO2', 'MT-CO1', 'UQCRB', 'COX6B1', 'NDUFA4', 'NDUFA6', 'NDUFB3', 'COX4I1', 'NDUFB11']</td>\n",
       "      <td id=\"T_74bd9_row4_col5\" class=\"data row4 col5\" >http://model.geneontology.org/R-HSA-167826</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_74bd9_level0_row5\" class=\"row_heading level0 row5\" >5</th>\n",
       "      <td id=\"T_74bd9_row5_col0\" class=\"data row5 col0\" >The proton buffering model - Reactome</td>\n",
       "      <td id=\"T_74bd9_row5_col1\" class=\"data row5 col1\" >0.000000</td>\n",
       "      <td id=\"T_74bd9_row5_col2\" class=\"data row5 col2\" >16</td>\n",
       "      <td id=\"T_74bd9_row5_col3\" class=\"data row5 col3\" >74</td>\n",
       "      <td id=\"T_74bd9_row5_col4\" class=\"data row5 col4\" >['MT-ND4', 'MT-ND2', 'CYC1', 'UQCRQ', 'COX5B', 'COX7C', 'NDUFA1', 'MT-CO2', 'MT-CO1', 'UQCRB', 'COX6B1', 'NDUFA4', 'NDUFA6', 'NDUFB3', 'COX4I1', 'NDUFB11']</td>\n",
       "      <td id=\"T_74bd9_row5_col5\" class=\"data row5 col5\" >http://model.geneontology.org/R-HSA-167827</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_74bd9_level0_row6\" class=\"row_heading level0 row6\" >6</th>\n",
       "      <td id=\"T_74bd9_row6_col0\" class=\"data row6 col0\" >Cytoprotection by HMOX1 - Reactome</td>\n",
       "      <td id=\"T_74bd9_row6_col1\" class=\"data row6 col1\" >0.000000</td>\n",
       "      <td id=\"T_74bd9_row6_col2\" class=\"data row6 col2\" >10</td>\n",
       "      <td id=\"T_74bd9_row6_col3\" class=\"data row6 col3\" >31</td>\n",
       "      <td id=\"T_74bd9_row6_col4\" class=\"data row6 col4\" >['CYC1', 'UQCRQ', 'COX5B', 'COX7C', 'MT-CO2', 'MT-CO1', 'UQCRB', 'COX6B1', 'NDUFA4', 'COX4I1']</td>\n",
       "      <td id=\"T_74bd9_row6_col5\" class=\"data row6 col5\" >http://model.geneontology.org/R-HSA-9707564</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_74bd9_level0_row7\" class=\"row_heading level0 row7\" >7</th>\n",
       "      <td id=\"T_74bd9_row7_col0\" class=\"data row7 col0\" >Transport of nucleosides and free purine and pyrimidine bases across the plasma membrane - Reactome</td>\n",
       "      <td id=\"T_74bd9_row7_col1\" class=\"data row7 col1\" >0.000000</td>\n",
       "      <td id=\"T_74bd9_row7_col2\" class=\"data row7 col2\" >10</td>\n",
       "      <td id=\"T_74bd9_row7_col3\" class=\"data row7 col3\" >32</td>\n",
       "      <td id=\"T_74bd9_row7_col4\" class=\"data row7 col4\" >['SLC25A6', 'ATP5MC1', 'SLC25A5', 'ATP5F1D', 'ATP5MC3', 'SLC25A4', 'ATP5ME', 'SLC29A1', 'ATP5PD', 'ATP5F1E']</td>\n",
       "      <td id=\"T_74bd9_row7_col5\" class=\"data row7 col5\" >http://model.geneontology.org/R-HSA-83936</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_74bd9_level0_row8\" class=\"row_heading level0 row8\" >8</th>\n",
       "      <td id=\"T_74bd9_row8_col0\" class=\"data row8 col0\" >TP53 Regulates Metabolic Genes - Reactome</td>\n",
       "      <td id=\"T_74bd9_row8_col1\" class=\"data row8 col1\" >0.000028</td>\n",
       "      <td id=\"T_74bd9_row8_col2\" class=\"data row8 col2\" >11</td>\n",
       "      <td id=\"T_74bd9_row8_col3\" class=\"data row8 col3\" >59</td>\n",
       "      <td id=\"T_74bd9_row8_col4\" class=\"data row8 col4\" >['CYC1', 'UQCRQ', 'COX5B', 'COX7C', 'MT-CO2', 'MT-CO1', 'UQCRB', 'COX6B1', 'NDUFA4', 'COX4I1', 'PRDX1']</td>\n",
       "      <td id=\"T_74bd9_row8_col5\" class=\"data row8 col5\" >http://model.geneontology.org/R-HSA-5628897</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x12fd784f0>"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PVneg_standard = enrich.enrich_wrapper('PVneg_SMC_comb.csv','Gene Symbol',method='standard',FDR = 0.05,fpath = '../test_data/processed/',table_number = 47)\n",
    "PVneg_standard"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "d30ad193-e1f3-465b-be98-86be25c82dc6",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|█████████████████████████████████████████| 324/324 [00:39<00:00,  8.12it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 256 entities from 221 out of 686 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_875e1 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_875e1\">\n",
       "  <caption>Table S48: PVneg smooth muscle cells combined up/downregulated genes, weighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_875e1_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_875e1_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_875e1_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_875e1_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_875e1_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_875e1_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_875e1_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_875e1_row0_col0\" class=\"data row0 col0\" >Peptide chain elongation - Reactome</td>\n",
       "      <td id=\"T_875e1_row0_col1\" class=\"data row0 col1\" >0.000000</td>\n",
       "      <td id=\"T_875e1_row0_col2\" class=\"data row0 col2\" >41</td>\n",
       "      <td id=\"T_875e1_row0_col3\" class=\"data row0 col3\" >79</td>\n",
       "      <td id=\"T_875e1_row0_col4\" class=\"data row0 col4\" >['RPL13A', 'RPLP2', 'RPLP0', 'RPS18', 'RPL35', 'RPL27A', 'RPS15A', 'RPL7', 'RPS5', 'RPS2', 'RPS20', 'RPLP1', 'RPSA', 'RPL10A', 'RPS25', 'RPL15', 'RPL27', 'RPL23A', 'RPL34', 'RPS16', 'RPS9', 'RPS14', 'RPL8', 'RPS15', 'RPL38', 'RPS29', 'RPL13', 'RPL4', 'RPS19', 'RPL12', 'RPL29', 'RPS28', 'RPS6', 'RPS23', 'RPL21', 'UBA52', 'RPL41', 'set:RPL36A,RPL36AL', 'set:RPL3,RPL3L', 'set:RPL10,RPL10L', 'set:RPS27,RPS27L']</td>\n",
       "      <td id=\"T_875e1_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-156902</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_875e1_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_875e1_row1_col0\" class=\"data row1 col0\" >Eukaryotic Translation Elongation - Reactome</td>\n",
       "      <td id=\"T_875e1_row1_col1\" class=\"data row1 col1\" >0.000000</td>\n",
       "      <td id=\"T_875e1_row1_col2\" class=\"data row1 col2\" >41</td>\n",
       "      <td id=\"T_875e1_row1_col3\" class=\"data row1 col3\" >83</td>\n",
       "      <td id=\"T_875e1_row1_col4\" class=\"data row1 col4\" >['RPL13A', 'RPLP2', 'RPLP0', 'RPS18', 'RPL35', 'RPL27A', 'RPS15A', 'RPL7', 'RPS5', 'RPS2', 'RPS20', 'RPLP1', 'RPSA', 'RPL10A', 'RPS25', 'RPL15', 'RPL27', 'RPL23A', 'RPL34', 'RPS16', 'RPS9', 'RPS14', 'RPL8', 'RPS15', 'RPL38', 'RPS29', 'RPL13', 'RPL4', 'RPS19', 'RPL12', 'RPL29', 'RPS28', 'RPS6', 'RPS23', 'RPL21', 'UBA52', 'RPL41', 'set:RPL36A,RPL36AL', 'set:RPL3,RPL3L', 'set:RPL10,RPL10L', 'set:RPS27,RPS27L']</td>\n",
       "      <td id=\"T_875e1_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-156842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_875e1_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "      <td id=\"T_875e1_row2_col0\" class=\"data row2 col0\" >Formation of ATP by chemiosmotic coupling - Reactome</td>\n",
       "      <td id=\"T_875e1_row2_col1\" class=\"data row2 col1\" >0.000000</td>\n",
       "      <td id=\"T_875e1_row2_col2\" class=\"data row2 col2\" >23</td>\n",
       "      <td id=\"T_875e1_row2_col3\" class=\"data row2 col3\" >88</td>\n",
       "      <td id=\"T_875e1_row2_col4\" class=\"data row2 col4\" >['MT-ND4', 'MT-ND2', 'CYC1', 'UQCRQ', 'COX5B', 'ATP5MC1', 'COX7C', 'NDUFA1', 'MT-CO2', 'MT-CO1', 'ATP5F1D', 'ATP5MC3', 'UQCRB', 'COX6B1', 'NDUFA4', 'ATP5ME', 'NDUFA6', 'NDUFB3', 'ATP5PD', 'ATP5F1E', 'COX4I1', 'NDUFB11', 'set:ANT homodimer']</td>\n",
       "      <td id=\"T_875e1_row2_col5\" class=\"data row2 col5\" >http://model.geneontology.org/R-HSA-163210</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_875e1_level0_row3\" class=\"row_heading level0 row3\" >3</th>\n",
       "      <td id=\"T_875e1_row3_col0\" class=\"data row3 col0\" >Respiratory electron transport - Reactome</td>\n",
       "      <td id=\"T_875e1_row3_col1\" class=\"data row3 col1\" >0.000000</td>\n",
       "      <td id=\"T_875e1_row3_col2\" class=\"data row3 col2\" >19</td>\n",
       "      <td id=\"T_875e1_row3_col3\" class=\"data row3 col3\" >91</td>\n",
       "      <td id=\"T_875e1_row3_col4\" class=\"data row3 col4\" >['MT-ND4', 'MT-ND2', 'CYC1', 'UQCRQ', 'COX5B', 'IDH3B', 'ACADVL', 'COX7C', 'NDUFA1', 'MT-CO2', 'MT-CO1', 'UQCRB', 'COX6B1', 'NDUFA4', 'NDUFA6', 'IDH3G', 'NDUFB3', 'COX4I1', 'NDUFB11']</td>\n",
       "      <td id=\"T_875e1_row3_col5\" class=\"data row3 col5\" >http://model.geneontology.org/R-HSA-611105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_875e1_level0_row4\" class=\"row_heading level0 row4\" >4</th>\n",
       "      <td id=\"T_875e1_row4_col0\" class=\"data row4 col0\" >The fatty acid cycling model - Reactome</td>\n",
       "      <td id=\"T_875e1_row4_col1\" class=\"data row4 col1\" >0.000000</td>\n",
       "      <td id=\"T_875e1_row4_col2\" class=\"data row4 col2\" >16</td>\n",
       "      <td id=\"T_875e1_row4_col3\" class=\"data row4 col3\" >70</td>\n",
       "      <td id=\"T_875e1_row4_col4\" class=\"data row4 col4\" >['MT-ND4', 'MT-ND2', 'CYC1', 'UQCRQ', 'COX5B', 'COX7C', 'NDUFA1', 'MT-CO2', 'MT-CO1', 'UQCRB', 'COX6B1', 'NDUFA4', 'NDUFA6', 'NDUFB3', 'COX4I1', 'NDUFB11']</td>\n",
       "      <td id=\"T_875e1_row4_col5\" class=\"data row4 col5\" >http://model.geneontology.org/R-HSA-167826</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_875e1_level0_row5\" class=\"row_heading level0 row5\" >5</th>\n",
       "      <td id=\"T_875e1_row5_col0\" class=\"data row5 col0\" >The proton buffering model - Reactome</td>\n",
       "      <td id=\"T_875e1_row5_col1\" class=\"data row5 col1\" >0.000000</td>\n",
       "      <td id=\"T_875e1_row5_col2\" class=\"data row5 col2\" >16</td>\n",
       "      <td id=\"T_875e1_row5_col3\" class=\"data row5 col3\" >70</td>\n",
       "      <td id=\"T_875e1_row5_col4\" class=\"data row5 col4\" >['MT-ND4', 'MT-ND2', 'CYC1', 'UQCRQ', 'COX5B', 'COX7C', 'NDUFA1', 'MT-CO2', 'MT-CO1', 'UQCRB', 'COX6B1', 'NDUFA4', 'NDUFA6', 'NDUFB3', 'COX4I1', 'NDUFB11']</td>\n",
       "      <td id=\"T_875e1_row5_col5\" class=\"data row5 col5\" >http://model.geneontology.org/R-HSA-167827</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_875e1_level0_row6\" class=\"row_heading level0 row6\" >6</th>\n",
       "      <td id=\"T_875e1_row6_col0\" class=\"data row6 col0\" >Cytoprotection by HMOX1 - Reactome</td>\n",
       "      <td id=\"T_875e1_row6_col1\" class=\"data row6 col1\" >0.000000</td>\n",
       "      <td id=\"T_875e1_row6_col2\" class=\"data row6 col2\" >10</td>\n",
       "      <td id=\"T_875e1_row6_col3\" class=\"data row6 col3\" >29</td>\n",
       "      <td id=\"T_875e1_row6_col4\" class=\"data row6 col4\" >['CYC1', 'UQCRQ', 'COX5B', 'COX7C', 'MT-CO2', 'MT-CO1', 'UQCRB', 'COX6B1', 'NDUFA4', 'COX4I1']</td>\n",
       "      <td id=\"T_875e1_row6_col5\" class=\"data row6 col5\" >http://model.geneontology.org/R-HSA-9707564</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_875e1_level0_row7\" class=\"row_heading level0 row7\" >7</th>\n",
       "      <td id=\"T_875e1_row7_col0\" class=\"data row7 col0\" >Transport of nucleosides and free purine and pyrimidine bases across the plasma membrane - Reactome</td>\n",
       "      <td id=\"T_875e1_row7_col1\" class=\"data row7 col1\" >0.000000</td>\n",
       "      <td id=\"T_875e1_row7_col2\" class=\"data row7 col2\" >10</td>\n",
       "      <td id=\"T_875e1_row7_col3\" class=\"data row7 col3\" >30</td>\n",
       "      <td id=\"T_875e1_row7_col4\" class=\"data row7 col4\" >['ATP5MC1', 'ATP5F1D', 'ATP5MC3', 'SLC25A4', 'ATP5ME', 'ATP5PD', 'ATP5F1E', 'set:ANT homodimer', 'set:SLC29A1-like proteins', 'set:SLC29A2-like proteins']</td>\n",
       "      <td id=\"T_875e1_row7_col5\" class=\"data row7 col5\" >http://model.geneontology.org/R-HSA-83936</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_875e1_level0_row8\" class=\"row_heading level0 row8\" >8</th>\n",
       "      <td id=\"T_875e1_row8_col0\" class=\"data row8 col0\" >TP53 Regulates Metabolic Genes - Reactome</td>\n",
       "      <td id=\"T_875e1_row8_col1\" class=\"data row8 col1\" >0.000000</td>\n",
       "      <td id=\"T_875e1_row8_col2\" class=\"data row8 col2\" >12</td>\n",
       "      <td id=\"T_875e1_row8_col3\" class=\"data row8 col3\" >46</td>\n",
       "      <td id=\"T_875e1_row8_col4\" class=\"data row8 col4\" >['CYC1', 'UQCRQ', 'COX5B', 'COX7C', 'MT-CO2', 'MT-CO1', 'UQCRB', 'COX6B1', 'NDUFA4', 'COX4I1', 'PRDX1', 'set:PRDX1,2,5']</td>\n",
       "      <td id=\"T_875e1_row8_col5\" class=\"data row8 col5\" >http://model.geneontology.org/R-HSA-5628897</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_875e1_level0_row9\" class=\"row_heading level0 row9\" >9</th>\n",
       "      <td id=\"T_875e1_row9_col0\" class=\"data row9 col0\" >KEAP1-NFE2L2 pathway - Reactome</td>\n",
       "      <td id=\"T_875e1_row9_col1\" class=\"data row9 col1\" >0.000228</td>\n",
       "      <td id=\"T_875e1_row9_col2\" class=\"data row9 col2\" >9</td>\n",
       "      <td id=\"T_875e1_row9_col3\" class=\"data row9 col3\" >55</td>\n",
       "      <td id=\"T_875e1_row9_col4\" class=\"data row9 col4\" >['PSMC3', 'PSMC4', 'PSMB6', 'PSMC5', 'PSMB4', 'PSMA7', 'PRDX1', 'PSMB3', 'set:PRDX1,2,5']</td>\n",
       "      <td id=\"T_875e1_row9_col5\" class=\"data row9 col5\" >http://model.geneontology.org/R-HSA-9755511</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x14e649570>"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PVneg_weighted = enrich.enrich_wrapper('PVneg_SMC_comb.csv','Gene Symbol',method='ncHGT',FDR = 0.05,fpath = '../test_data/processed/',table_number = 48)\n",
    "PVneg_weighted"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5975cb0d-8fa1-4985-b44f-f237400e9085",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "# Platelets in SARS-CoV-2\n",
    "\n",
    "Kanth manne et al., 2020\n",
    "\n",
    "https://ashpublications.org/blood/article/136/11/1317/461106/Platelet-gene-expression-and-function-in-patients\n",
    "\n",
    "RNAseq in platelets from patients with covid-19 vs healthy donors. There were 6 ICU and 4 nonICU patients."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "3b18dbef-fb4b-4dcc-ac2a-aa14adda23ad",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 482/482 [00:00<00:00, 16302.48it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 423 entities from 365 out of 1172 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_2c84f caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_2c84f\">\n",
       "  <caption>Table S49: platelets up, unweighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_2c84f_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_2c84f_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_2c84f_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_2c84f_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_2c84f_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_2c84f_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_2c84f_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_2c84f_row0_col0\" class=\"data row0 col0\" >Collagen biosynthesis and modifying enzymes - Reactome</td>\n",
       "      <td id=\"T_2c84f_row0_col1\" class=\"data row0 col1\" >0.000000</td>\n",
       "      <td id=\"T_2c84f_row0_col2\" class=\"data row0 col2\" >10</td>\n",
       "      <td id=\"T_2c84f_row0_col3\" class=\"data row0 col3\" >12</td>\n",
       "      <td id=\"T_2c84f_row0_col4\" class=\"data row0 col4\" >['PPIB', 'P3H1', 'PLOD3', 'P4HB', 'set:Procollagen C-proteinases', 'set:Procollagen N-proteinases', 'set:Prolyl 3-hydroxylases', 'set:COLGALT1,COLGALT2', 'set:Lysyl hydroxylases', 'set:4-Hyp collagen propeptides']</td>\n",
       "      <td id=\"T_2c84f_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-1650814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c84f_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_2c84f_row1_col0\" class=\"data row1 col0\" >Synthesis of PE - Reactome</td>\n",
       "      <td id=\"T_2c84f_row1_col1\" class=\"data row1 col1\" >0.000050</td>\n",
       "      <td id=\"T_2c84f_row1_col2\" class=\"data row1 col2\" >8</td>\n",
       "      <td id=\"T_2c84f_row1_col3\" class=\"data row1 col3\" >15</td>\n",
       "      <td id=\"T_2c84f_row1_col4\" class=\"data row1 col4\" >['PTDSS2', 'PCYT2', 'PISD', 'PHOSPHO1', 'set:LPIN', 'set:CHK/ETNK', 'set:AGPAT', 'set:PNPLA2/3']</td>\n",
       "      <td id=\"T_2c84f_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-1483213</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x14e3e1120>"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "platelets_up_set = enrich.enrich_wrapper('platelets_up.csv','Gene Symbol',method='set',FDR = 0.05,fpath = '../test_data/processed/',table_number = 49)\n",
    "platelets_up_set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "5cc96f69-30be-4b82-b069-d07dacd25725",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 482/482 [00:00<00:00, 15927.57it/s]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_cc9fd caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_cc9fd\">\n",
       "  <caption>Table S50: platelets up, standard method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_cc9fd_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_cc9fd_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_cc9fd_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_cc9fd_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_cc9fd_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_cc9fd_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_cc9fd_row0_col0\" class=\"data row0 col0\" >ER-Phagosome pathway - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row0_col1\" class=\"data row0 col1\" >0.000045</td>\n",
       "      <td id=\"T_cc9fd_row0_col2\" class=\"data row0 col2\" >13</td>\n",
       "      <td id=\"T_cc9fd_row0_col3\" class=\"data row0 col3\" >52</td>\n",
       "      <td id=\"T_cc9fd_row0_col4\" class=\"data row0 col4\" >['SEC61A1', 'PSMD13', 'PSMA5', 'PSMD11', 'SEC61B', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-1236974</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_cc9fd_row1_col0\" class=\"data row1 col0\" >Hedgehog ligand biogenesis - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row1_col1\" class=\"data row1 col1\" >0.000045</td>\n",
       "      <td id=\"T_cc9fd_row1_col2\" class=\"data row1 col2\" >13</td>\n",
       "      <td id=\"T_cc9fd_row1_col3\" class=\"data row1 col3\" >52</td>\n",
       "      <td id=\"T_cc9fd_row1_col4\" class=\"data row1 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'P4HB', 'PSMB6', 'SYVN1']</td>\n",
       "      <td id=\"T_cc9fd_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-5358346</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "      <td id=\"T_cc9fd_row2_col0\" class=\"data row2 col0\" >Regulation of APC/C activators between G1/S and early anaphase - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row2_col1\" class=\"data row2 col1\" >0.000069</td>\n",
       "      <td id=\"T_cc9fd_row2_col2\" class=\"data row2 col2\" >13</td>\n",
       "      <td id=\"T_cc9fd_row2_col3\" class=\"data row2 col3\" >54</td>\n",
       "      <td id=\"T_cc9fd_row2_col4\" class=\"data row2 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'CDC25B', 'PSMB7', 'CDK1', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row2_col5\" class=\"data row2 col5\" >http://model.geneontology.org/R-HSA-176408</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row3\" class=\"row_heading level0 row3\" >3</th>\n",
       "      <td id=\"T_cc9fd_row3_col0\" class=\"data row3 col0\" >Conversion from APC/C:Cdc20 to APC/C:Cdh1 in late anaphase - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row3_col1\" class=\"data row3 col1\" >0.000111</td>\n",
       "      <td id=\"T_cc9fd_row3_col2\" class=\"data row3 col2\" >12</td>\n",
       "      <td id=\"T_cc9fd_row3_col3\" class=\"data row3 col3\" >49</td>\n",
       "      <td id=\"T_cc9fd_row3_col4\" class=\"data row3 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'CDK1', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row3_col5\" class=\"data row3 col5\" >http://model.geneontology.org/R-HSA-176407</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row4\" class=\"row_heading level0 row4\" >4</th>\n",
       "      <td id=\"T_cc9fd_row4_col0\" class=\"data row4 col0\" >Neddylation - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row4_col1\" class=\"data row4 col1\" >0.000113</td>\n",
       "      <td id=\"T_cc9fd_row4_col2\" class=\"data row4 col2\" >14</td>\n",
       "      <td id=\"T_cc9fd_row4_col3\" class=\"data row4 col3\" >64</td>\n",
       "      <td id=\"T_cc9fd_row4_col4\" class=\"data row4 col4\" >['UBE2M', 'PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'CUL9', 'PSMB7', 'UCHL3', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row4_col5\" class=\"data row4 col5\" >http://model.geneontology.org/R-HSA-8951664</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row5\" class=\"row_heading level0 row5\" >5</th>\n",
       "      <td id=\"T_cc9fd_row5_col0\" class=\"data row5 col0\" >KEAP1-NFE2L2 pathway - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row5_col1\" class=\"data row5 col1\" >0.000126</td>\n",
       "      <td id=\"T_cc9fd_row5_col2\" class=\"data row5 col2\" >13</td>\n",
       "      <td id=\"T_cc9fd_row5_col3\" class=\"data row5 col3\" >57</td>\n",
       "      <td id=\"T_cc9fd_row5_col4\" class=\"data row5 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'CSNK2B', 'PSMD8', 'PSMB5', 'PSMB6', 'PRDX2']</td>\n",
       "      <td id=\"T_cc9fd_row5_col5\" class=\"data row5 col5\" >http://model.geneontology.org/R-HSA-9755511</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row6\" class=\"row_heading level0 row6\" >6</th>\n",
       "      <td id=\"T_cc9fd_row6_col0\" class=\"data row6 col0\" >SCF(Skp2)-mediated degradation of p27/p21 - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row6_col1\" class=\"data row6 col1\" >0.000136</td>\n",
       "      <td id=\"T_cc9fd_row6_col2\" class=\"data row6 col2\" >12</td>\n",
       "      <td id=\"T_cc9fd_row6_col3\" class=\"data row6 col3\" >50</td>\n",
       "      <td id=\"T_cc9fd_row6_col4\" class=\"data row6 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6', 'CKS1B']</td>\n",
       "      <td id=\"T_cc9fd_row6_col5\" class=\"data row6 col5\" >http://model.geneontology.org/R-HSA-187577</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row7\" class=\"row_heading level0 row7\" >7</th>\n",
       "      <td id=\"T_cc9fd_row7_col0\" class=\"data row7 col0\" >The role of GTSE1 in G2/M progression after G2 checkpoint - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row7_col1\" class=\"data row7 col1\" >0.000136</td>\n",
       "      <td id=\"T_cc9fd_row7_col2\" class=\"data row7 col2\" >12</td>\n",
       "      <td id=\"T_cc9fd_row7_col3\" class=\"data row7 col3\" >50</td>\n",
       "      <td id=\"T_cc9fd_row7_col4\" class=\"data row7 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'CDC25B', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row7_col5\" class=\"data row7 col5\" >http://model.geneontology.org/R-HSA-8852276</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row8\" class=\"row_heading level0 row8\" >8</th>\n",
       "      <td id=\"T_cc9fd_row8_col0\" class=\"data row8 col0\" >SCF-beta-TrCP mediated degradation of Emi1 - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row8_col1\" class=\"data row8 col1\" >0.000167</td>\n",
       "      <td id=\"T_cc9fd_row8_col2\" class=\"data row8 col2\" >12</td>\n",
       "      <td id=\"T_cc9fd_row8_col3\" class=\"data row8 col3\" >51</td>\n",
       "      <td id=\"T_cc9fd_row8_col4\" class=\"data row8 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'CDK1', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row8_col5\" class=\"data row8 col5\" >http://model.geneontology.org/R-HSA-174113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row9\" class=\"row_heading level0 row9\" >9</th>\n",
       "      <td id=\"T_cc9fd_row9_col0\" class=\"data row9 col0\" >GSK3B and BTRC:CUL1-mediated-degradation of NFE2L2 - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row9_col1\" class=\"data row9 col1\" >0.000175</td>\n",
       "      <td id=\"T_cc9fd_row9_col2\" class=\"data row9 col2\" >11</td>\n",
       "      <td id=\"T_cc9fd_row9_col3\" class=\"data row9 col3\" >44</td>\n",
       "      <td id=\"T_cc9fd_row9_col4\" class=\"data row9 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row9_col5\" class=\"data row9 col5\" >http://model.geneontology.org/R-HSA-9762114</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row10\" class=\"row_heading level0 row10\" >10</th>\n",
       "      <td id=\"T_cc9fd_row10_col0\" class=\"data row10 col0\" >Degradation of beta-catenin by the destruction complex - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row10_col1\" class=\"data row10 col1\" >0.000183</td>\n",
       "      <td id=\"T_cc9fd_row10_col2\" class=\"data row10 col2\" >13</td>\n",
       "      <td id=\"T_cc9fd_row10_col3\" class=\"data row10 col3\" >59</td>\n",
       "      <td id=\"T_cc9fd_row10_col4\" class=\"data row10 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PPP2R5B', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PPP2R1B', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row10_col5\" class=\"data row10 col5\" >http://model.geneontology.org/R-HSA-195253</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row11\" class=\"row_heading level0 row11\" >11</th>\n",
       "      <td id=\"T_cc9fd_row11_col0\" class=\"data row11 col0\" >Cross-presentation of soluble exogenous antigens (endosomes) - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row11_col1\" class=\"data row11 col1\" >0.000217</td>\n",
       "      <td id=\"T_cc9fd_row11_col2\" class=\"data row11 col2\" >11</td>\n",
       "      <td id=\"T_cc9fd_row11_col3\" class=\"data row11 col3\" >45</td>\n",
       "      <td id=\"T_cc9fd_row11_col4\" class=\"data row11 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row11_col5\" class=\"data row11 col5\" >http://model.geneontology.org/R-HSA-1236978</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row12\" class=\"row_heading level0 row12\" >12</th>\n",
       "      <td id=\"T_cc9fd_row12_col0\" class=\"data row12 col0\" >AUF1 (hnRNP D0) binds and destabilizes mRNA - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row12_col1\" class=\"data row12 col1\" >0.000217</td>\n",
       "      <td id=\"T_cc9fd_row12_col2\" class=\"data row12 col2\" >11</td>\n",
       "      <td id=\"T_cc9fd_row12_col3\" class=\"data row12 col3\" >45</td>\n",
       "      <td id=\"T_cc9fd_row12_col4\" class=\"data row12 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row12_col5\" class=\"data row12 col5\" >http://model.geneontology.org/R-HSA-450408</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row13\" class=\"row_heading level0 row13\" >13</th>\n",
       "      <td id=\"T_cc9fd_row13_col0\" class=\"data row13 col0\" >APC/C:Cdc20 mediated degradation of mitotic proteins - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row13_col1\" class=\"data row13 col1\" >0.000217</td>\n",
       "      <td id=\"T_cc9fd_row13_col2\" class=\"data row13 col2\" >11</td>\n",
       "      <td id=\"T_cc9fd_row13_col3\" class=\"data row13 col3\" >45</td>\n",
       "      <td id=\"T_cc9fd_row13_col4\" class=\"data row13 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row13_col5\" class=\"data row13 col5\" >http://model.geneontology.org/R-HSA-176409</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row14\" class=\"row_heading level0 row14\" >14</th>\n",
       "      <td id=\"T_cc9fd_row14_col0\" class=\"data row14 col0\" >Regulation of activated PAK-2p34 by proteasome mediated degradation - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row14_col1\" class=\"data row14 col1\" >0.000268</td>\n",
       "      <td id=\"T_cc9fd_row14_col2\" class=\"data row14 col2\" >11</td>\n",
       "      <td id=\"T_cc9fd_row14_col3\" class=\"data row14 col3\" >46</td>\n",
       "      <td id=\"T_cc9fd_row14_col4\" class=\"data row14 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row14_col5\" class=\"data row14 col5\" >http://model.geneontology.org/R-HSA-211733</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row15\" class=\"row_heading level0 row15\" >15</th>\n",
       "      <td id=\"T_cc9fd_row15_col0\" class=\"data row15 col0\" >Ubiquitin-dependent degradation of Cyclin D - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row15_col1\" class=\"data row15 col1\" >0.000268</td>\n",
       "      <td id=\"T_cc9fd_row15_col2\" class=\"data row15 col2\" >11</td>\n",
       "      <td id=\"T_cc9fd_row15_col3\" class=\"data row15 col3\" >46</td>\n",
       "      <td id=\"T_cc9fd_row15_col4\" class=\"data row15 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row15_col5\" class=\"data row15 col5\" >http://model.geneontology.org/R-HSA-75815</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row16\" class=\"row_heading level0 row16\" >16</th>\n",
       "      <td id=\"T_cc9fd_row16_col0\" class=\"data row16 col0\" >Regulation of RUNX3 expression and activity - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row16_col1\" class=\"data row16 col1\" >0.000268</td>\n",
       "      <td id=\"T_cc9fd_row16_col2\" class=\"data row16 col2\" >11</td>\n",
       "      <td id=\"T_cc9fd_row16_col3\" class=\"data row16 col3\" >46</td>\n",
       "      <td id=\"T_cc9fd_row16_col4\" class=\"data row16 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row16_col5\" class=\"data row16 col5\" >http://model.geneontology.org/R-HSA-8941858</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row17\" class=\"row_heading level0 row17\" >17</th>\n",
       "      <td id=\"T_cc9fd_row17_col0\" class=\"data row17 col0\" >Antigen Presentation: Folding, assembly and peptide loading of class I MHC - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row17_col1\" class=\"data row17 col1\" >0.000297</td>\n",
       "      <td id=\"T_cc9fd_row17_col2\" class=\"data row17 col2\" >12</td>\n",
       "      <td id=\"T_cc9fd_row17_col3\" class=\"data row17 col3\" >54</td>\n",
       "      <td id=\"T_cc9fd_row17_col4\" class=\"data row17 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6', 'ERAP2']</td>\n",
       "      <td id=\"T_cc9fd_row17_col5\" class=\"data row17 col5\" >http://model.geneontology.org/R-HSA-983170</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row18\" class=\"row_heading level0 row18\" >18</th>\n",
       "      <td id=\"T_cc9fd_row18_col0\" class=\"data row18 col0\" >Degradation of DVL - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row18_col1\" class=\"data row18 col1\" >0.000328</td>\n",
       "      <td id=\"T_cc9fd_row18_col2\" class=\"data row18 col2\" >11</td>\n",
       "      <td id=\"T_cc9fd_row18_col3\" class=\"data row18 col3\" >47</td>\n",
       "      <td id=\"T_cc9fd_row18_col4\" class=\"data row18 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row18_col5\" class=\"data row18 col5\" >http://model.geneontology.org/R-HSA-4641258</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row19\" class=\"row_heading level0 row19\" >19</th>\n",
       "      <td id=\"T_cc9fd_row19_col0\" class=\"data row19 col0\" >Autodegradation of the E3 ubiquitin ligase COP1 - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row19_col1\" class=\"data row19 col1\" >0.000328</td>\n",
       "      <td id=\"T_cc9fd_row19_col2\" class=\"data row19 col2\" >11</td>\n",
       "      <td id=\"T_cc9fd_row19_col3\" class=\"data row19 col3\" >47</td>\n",
       "      <td id=\"T_cc9fd_row19_col4\" class=\"data row19 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row19_col5\" class=\"data row19 col5\" >http://model.geneontology.org/R-HSA-349425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row20\" class=\"row_heading level0 row20\" >20</th>\n",
       "      <td id=\"T_cc9fd_row20_col0\" class=\"data row20 col0\" >Asymmetric localization of PCP proteins - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row20_col1\" class=\"data row20 col1\" >0.000328</td>\n",
       "      <td id=\"T_cc9fd_row20_col2\" class=\"data row20 col2\" >11</td>\n",
       "      <td id=\"T_cc9fd_row20_col3\" class=\"data row20 col3\" >47</td>\n",
       "      <td id=\"T_cc9fd_row20_col4\" class=\"data row20 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row20_col5\" class=\"data row20 col5\" >http://model.geneontology.org/R-HSA-4608870</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row21\" class=\"row_heading level0 row21\" >21</th>\n",
       "      <td id=\"T_cc9fd_row21_col0\" class=\"data row21 col0\" >TNFR2 non-canonical NF-kB pathway - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row21_col1\" class=\"data row21 col1\" >0.000328</td>\n",
       "      <td id=\"T_cc9fd_row21_col2\" class=\"data row21 col2\" >11</td>\n",
       "      <td id=\"T_cc9fd_row21_col3\" class=\"data row21 col3\" >47</td>\n",
       "      <td id=\"T_cc9fd_row21_col4\" class=\"data row21 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row21_col5\" class=\"data row21 col5\" >http://model.geneontology.org/R-HSA-5668541</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row22\" class=\"row_heading level0 row22\" >22</th>\n",
       "      <td id=\"T_cc9fd_row22_col0\" class=\"data row22 col0\" >Synthesis of PE - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row22_col1\" class=\"data row22 col1\" >0.000338</td>\n",
       "      <td id=\"T_cc9fd_row22_col2\" class=\"data row22 col2\" >9</td>\n",
       "      <td id=\"T_cc9fd_row22_col3\" class=\"data row22 col3\" >33</td>\n",
       "      <td id=\"T_cc9fd_row22_col4\" class=\"data row22 col4\" >['PTDSS2', 'LPIN1', 'PCYT2', 'PNPLA2', 'PISD', 'AGPAT2', 'CHKB', 'AGPAT4', 'PHOSPHO1']</td>\n",
       "      <td id=\"T_cc9fd_row22_col5\" class=\"data row22 col5\" >http://model.geneontology.org/R-HSA-1483213</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row23\" class=\"row_heading level0 row23\" >23</th>\n",
       "      <td id=\"T_cc9fd_row23_col0\" class=\"data row23 col0\" >Heme biosynthesis - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row23_col1\" class=\"data row23 col1\" >0.000348</td>\n",
       "      <td id=\"T_cc9fd_row23_col2\" class=\"data row23 col2\" >6</td>\n",
       "      <td id=\"T_cc9fd_row23_col3\" class=\"data row23 col3\" >15</td>\n",
       "      <td id=\"T_cc9fd_row23_col4\" class=\"data row23 col4\" >['ALAS2', 'UROS', 'COX10', 'HMBS', 'CPOX', 'PPOX']</td>\n",
       "      <td id=\"T_cc9fd_row23_col5\" class=\"data row23 col5\" >http://model.geneontology.org/R-HSA-189451</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row24\" class=\"row_heading level0 row24\" >24</th>\n",
       "      <td id=\"T_cc9fd_row24_col0\" class=\"data row24 col0\" >G2/M Checkpoints - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row24_col1\" class=\"data row24 col1\" >0.000399</td>\n",
       "      <td id=\"T_cc9fd_row24_col2\" class=\"data row24 col2\" >11</td>\n",
       "      <td id=\"T_cc9fd_row24_col3\" class=\"data row24 col3\" >48</td>\n",
       "      <td id=\"T_cc9fd_row24_col4\" class=\"data row24 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row24_col5\" class=\"data row24 col5\" >http://model.geneontology.org/R-HSA-69481</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row25\" class=\"row_heading level0 row25\" >25</th>\n",
       "      <td id=\"T_cc9fd_row25_col0\" class=\"data row25 col0\" >Degradation of AXIN - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row25_col1\" class=\"data row25 col1\" >0.000483</td>\n",
       "      <td id=\"T_cc9fd_row25_col2\" class=\"data row25 col2\" >11</td>\n",
       "      <td id=\"T_cc9fd_row25_col3\" class=\"data row25 col3\" >49</td>\n",
       "      <td id=\"T_cc9fd_row25_col4\" class=\"data row25 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row25_col5\" class=\"data row25 col5\" >http://model.geneontology.org/R-HSA-4641257</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row26\" class=\"row_heading level0 row26\" >26</th>\n",
       "      <td id=\"T_cc9fd_row26_col0\" class=\"data row26 col0\" >Oxygen-dependent proline hydroxylation of Hypoxia-inducible Factor Alpha - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row26_col1\" class=\"data row26 col1\" >0.000483</td>\n",
       "      <td id=\"T_cc9fd_row26_col2\" class=\"data row26 col2\" >11</td>\n",
       "      <td id=\"T_cc9fd_row26_col3\" class=\"data row26 col3\" >49</td>\n",
       "      <td id=\"T_cc9fd_row26_col4\" class=\"data row26 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row26_col5\" class=\"data row26 col5\" >http://model.geneontology.org/R-HSA-1234176</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row27\" class=\"row_heading level0 row27\" >27</th>\n",
       "      <td id=\"T_cc9fd_row27_col0\" class=\"data row27 col0\" >RUNX1 regulates transcription of genes involved in differentiation of HSCs - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row27_col1\" class=\"data row27 col1\" >0.000483</td>\n",
       "      <td id=\"T_cc9fd_row27_col2\" class=\"data row27 col2\" >11</td>\n",
       "      <td id=\"T_cc9fd_row27_col3\" class=\"data row27 col3\" >49</td>\n",
       "      <td id=\"T_cc9fd_row27_col4\" class=\"data row27 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row27_col5\" class=\"data row27 col5\" >http://model.geneontology.org/R-HSA-8939236</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row28\" class=\"row_heading level0 row28\" >28</th>\n",
       "      <td id=\"T_cc9fd_row28_col0\" class=\"data row28 col0\" >Ubiquitin Mediated Degradation of Phosphorylated Cdc25A - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row28_col1\" class=\"data row28 col1\" >0.000483</td>\n",
       "      <td id=\"T_cc9fd_row28_col2\" class=\"data row28 col2\" >11</td>\n",
       "      <td id=\"T_cc9fd_row28_col3\" class=\"data row28 col3\" >49</td>\n",
       "      <td id=\"T_cc9fd_row28_col4\" class=\"data row28 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row28_col5\" class=\"data row28 col5\" >http://model.geneontology.org/R-HSA-69601</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row29\" class=\"row_heading level0 row29\" >29</th>\n",
       "      <td id=\"T_cc9fd_row29_col0\" class=\"data row29 col0\" >Cyclin A:Cdk2-associated events at S phase entry - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row29_col1\" class=\"data row29 col1\" >0.000505</td>\n",
       "      <td id=\"T_cc9fd_row29_col2\" class=\"data row29 col2\" >12</td>\n",
       "      <td id=\"T_cc9fd_row29_col3\" class=\"data row29 col3\" >57</td>\n",
       "      <td id=\"T_cc9fd_row29_col4\" class=\"data row29 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'CDC25B', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row29_col5\" class=\"data row29 col5\" >http://model.geneontology.org/R-HSA-69656</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row30\" class=\"row_heading level0 row30\" >30</th>\n",
       "      <td id=\"T_cc9fd_row30_col0\" class=\"data row30 col0\" >Respiratory electron transport - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row30_col1\" class=\"data row30 col1\" >0.000546</td>\n",
       "      <td id=\"T_cc9fd_row30_col2\" class=\"data row30 col2\" >16</td>\n",
       "      <td id=\"T_cc9fd_row30_col3\" class=\"data row30 col3\" >91</td>\n",
       "      <td id=\"T_cc9fd_row30_col4\" class=\"data row30 col4\" >['NDUFS8', 'ACADVL', 'COX5B', 'NDUFS5', 'NDUFA8', 'NDUFB7', 'ETFB', 'NDUFB8', 'NDUFV2', 'COX5A', 'COX7C', 'NDUFV1', 'NDUFA11', 'NDUFA13', 'ACADS', 'UQCRC1']</td>\n",
       "      <td id=\"T_cc9fd_row30_col5\" class=\"data row30 col5\" >http://model.geneontology.org/R-HSA-611105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row31\" class=\"row_heading level0 row31\" >31</th>\n",
       "      <td id=\"T_cc9fd_row31_col0\" class=\"data row31 col0\" >FBXL7 down-regulates AURKA during mitotic entry and in early mitosis - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row31_col1\" class=\"data row31 col1\" >0.000580</td>\n",
       "      <td id=\"T_cc9fd_row31_col2\" class=\"data row31 col2\" >11</td>\n",
       "      <td id=\"T_cc9fd_row31_col3\" class=\"data row31 col3\" >50</td>\n",
       "      <td id=\"T_cc9fd_row31_col4\" class=\"data row31 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row31_col5\" class=\"data row31 col5\" >http://model.geneontology.org/R-HSA-8854050</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row32\" class=\"row_heading level0 row32\" >32</th>\n",
       "      <td id=\"T_cc9fd_row32_col0\" class=\"data row32 col0\" >Hedgehog 'on' state - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row32_col1\" class=\"data row32 col1\" >0.000580</td>\n",
       "      <td id=\"T_cc9fd_row32_col2\" class=\"data row32 col2\" >11</td>\n",
       "      <td id=\"T_cc9fd_row32_col3\" class=\"data row32 col3\" >50</td>\n",
       "      <td id=\"T_cc9fd_row32_col4\" class=\"data row32 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row32_col5\" class=\"data row32 col5\" >http://model.geneontology.org/R-HSA-5632684</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row33\" class=\"row_heading level0 row33\" >33</th>\n",
       "      <td id=\"T_cc9fd_row33_col0\" class=\"data row33 col0\" >NIK-->noncanonical NF-kB signaling - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row33_col1\" class=\"data row33 col1\" >0.000694</td>\n",
       "      <td id=\"T_cc9fd_row33_col2\" class=\"data row33 col2\" >11</td>\n",
       "      <td id=\"T_cc9fd_row33_col3\" class=\"data row33 col3\" >51</td>\n",
       "      <td id=\"T_cc9fd_row33_col4\" class=\"data row33 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row33_col5\" class=\"data row33 col5\" >http://model.geneontology.org/R-HSA-5676590</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row34\" class=\"row_heading level0 row34\" >34</th>\n",
       "      <td id=\"T_cc9fd_row34_col0\" class=\"data row34 col0\" >Negative regulation of NOTCH4 signaling - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row34_col1\" class=\"data row34 col1\" >0.000694</td>\n",
       "      <td id=\"T_cc9fd_row34_col2\" class=\"data row34 col2\" >11</td>\n",
       "      <td id=\"T_cc9fd_row34_col3\" class=\"data row34 col3\" >51</td>\n",
       "      <td id=\"T_cc9fd_row34_col4\" class=\"data row34 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row34_col5\" class=\"data row34 col5\" >http://model.geneontology.org/R-HSA-9604323</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row35\" class=\"row_heading level0 row35\" >35</th>\n",
       "      <td id=\"T_cc9fd_row35_col0\" class=\"data row35 col0\" >Regulation of expression of SLITs and ROBOs - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row35_col1\" class=\"data row35 col1\" >0.000694</td>\n",
       "      <td id=\"T_cc9fd_row35_col2\" class=\"data row35 col2\" >11</td>\n",
       "      <td id=\"T_cc9fd_row35_col3\" class=\"data row35 col3\" >51</td>\n",
       "      <td id=\"T_cc9fd_row35_col4\" class=\"data row35 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row35_col5\" class=\"data row35 col5\" >http://model.geneontology.org/R-HSA-9010553</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row36\" class=\"row_heading level0 row36\" >36</th>\n",
       "      <td id=\"T_cc9fd_row36_col0\" class=\"data row36 col0\" >Dectin-1 mediated noncanonical NF-kB signaling - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row36_col1\" class=\"data row36 col1\" >0.000694</td>\n",
       "      <td id=\"T_cc9fd_row36_col2\" class=\"data row36 col2\" >11</td>\n",
       "      <td id=\"T_cc9fd_row36_col3\" class=\"data row36 col3\" >51</td>\n",
       "      <td id=\"T_cc9fd_row36_col4\" class=\"data row36 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row36_col5\" class=\"data row36 col5\" >http://model.geneontology.org/R-HSA-5607761</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row37\" class=\"row_heading level0 row37\" >37</th>\n",
       "      <td id=\"T_cc9fd_row37_col0\" class=\"data row37 col0\" >Cyclin E associated events during G1/S transition - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row37_col1\" class=\"data row37 col1\" >0.000703</td>\n",
       "      <td id=\"T_cc9fd_row37_col2\" class=\"data row37 col2\" >12</td>\n",
       "      <td id=\"T_cc9fd_row37_col3\" class=\"data row37 col3\" >59</td>\n",
       "      <td id=\"T_cc9fd_row37_col4\" class=\"data row37 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'CDK4', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row37_col5\" class=\"data row37 col5\" >http://model.geneontology.org/R-HSA-69202</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row38\" class=\"row_heading level0 row38\" >38</th>\n",
       "      <td id=\"T_cc9fd_row38_col0\" class=\"data row38 col0\" >Orc1 removal from chromatin - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row38_col1\" class=\"data row38 col1\" >0.000825</td>\n",
       "      <td id=\"T_cc9fd_row38_col2\" class=\"data row38 col2\" >11</td>\n",
       "      <td id=\"T_cc9fd_row38_col3\" class=\"data row38 col3\" >52</td>\n",
       "      <td id=\"T_cc9fd_row38_col4\" class=\"data row38 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row38_col5\" class=\"data row38 col5\" >http://model.geneontology.org/R-HSA-68949</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row39\" class=\"row_heading level0 row39\" >39</th>\n",
       "      <td id=\"T_cc9fd_row39_col0\" class=\"data row39 col0\" >Degradation of GLI1 by the proteasome - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row39_col1\" class=\"data row39 col1\" >0.000977</td>\n",
       "      <td id=\"T_cc9fd_row39_col2\" class=\"data row39 col2\" >11</td>\n",
       "      <td id=\"T_cc9fd_row39_col3\" class=\"data row39 col3\" >53</td>\n",
       "      <td id=\"T_cc9fd_row39_col4\" class=\"data row39 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row39_col5\" class=\"data row39 col5\" >http://model.geneontology.org/R-HSA-5610780</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row40\" class=\"row_heading level0 row40\" >40</th>\n",
       "      <td id=\"T_cc9fd_row40_col0\" class=\"data row40 col0\" >GLI3 is processed to GLI3R by the proteasome - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row40_col1\" class=\"data row40 col1\" >0.001151</td>\n",
       "      <td id=\"T_cc9fd_row40_col2\" class=\"data row40 col2\" >11</td>\n",
       "      <td id=\"T_cc9fd_row40_col3\" class=\"data row40 col3\" >54</td>\n",
       "      <td id=\"T_cc9fd_row40_col4\" class=\"data row40 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row40_col5\" class=\"data row40 col5\" >http://model.geneontology.org/R-HSA-5610785</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row41\" class=\"row_heading level0 row41\" >41</th>\n",
       "      <td id=\"T_cc9fd_row41_col0\" class=\"data row41 col0\" >Regulation of RUNX2 expression and activity - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row41_col1\" class=\"data row41 col1\" >0.001151</td>\n",
       "      <td id=\"T_cc9fd_row41_col2\" class=\"data row41 col2\" >11</td>\n",
       "      <td id=\"T_cc9fd_row41_col3\" class=\"data row41 col3\" >54</td>\n",
       "      <td id=\"T_cc9fd_row41_col4\" class=\"data row41 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row41_col5\" class=\"data row41 col5\" >http://model.geneontology.org/R-HSA-8939902</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row42\" class=\"row_heading level0 row42\" >42</th>\n",
       "      <td id=\"T_cc9fd_row42_col0\" class=\"data row42 col0\" >Degradation of GLI2 by the proteasome - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row42_col1\" class=\"data row42 col1\" >0.001151</td>\n",
       "      <td id=\"T_cc9fd_row42_col2\" class=\"data row42 col2\" >11</td>\n",
       "      <td id=\"T_cc9fd_row42_col3\" class=\"data row42 col3\" >54</td>\n",
       "      <td id=\"T_cc9fd_row42_col4\" class=\"data row42 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row42_col5\" class=\"data row42 col5\" >http://model.geneontology.org/R-HSA-5610783</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row43\" class=\"row_heading level0 row43\" >43</th>\n",
       "      <td id=\"T_cc9fd_row43_col0\" class=\"data row43 col0\" >Hedgehog 'off' state - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row43_col1\" class=\"data row43 col1\" >0.001349</td>\n",
       "      <td id=\"T_cc9fd_row43_col2\" class=\"data row43 col2\" >11</td>\n",
       "      <td id=\"T_cc9fd_row43_col3\" class=\"data row43 col3\" >55</td>\n",
       "      <td id=\"T_cc9fd_row43_col4\" class=\"data row43 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row43_col5\" class=\"data row43 col5\" >http://model.geneontology.org/R-HSA-5610787</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row44\" class=\"row_heading level0 row44\" >44</th>\n",
       "      <td id=\"T_cc9fd_row44_col0\" class=\"data row44 col0\" >Regulation of PTEN stability and activity - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row44_col1\" class=\"data row44 col1\" >0.001495</td>\n",
       "      <td id=\"T_cc9fd_row44_col2\" class=\"data row44 col2\" >12</td>\n",
       "      <td id=\"T_cc9fd_row44_col3\" class=\"data row44 col3\" >64</td>\n",
       "      <td id=\"T_cc9fd_row44_col4\" class=\"data row44 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'CSNK2B', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row44_col5\" class=\"data row44 col5\" >http://model.geneontology.org/R-HSA-8948751</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row45\" class=\"row_heading level0 row45\" >45</th>\n",
       "      <td id=\"T_cc9fd_row45_col0\" class=\"data row45 col0\" >CLEC7A (Dectin-1) signaling - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row45_col1\" class=\"data row45 col1\" >0.001575</td>\n",
       "      <td id=\"T_cc9fd_row45_col2\" class=\"data row45 col2\" >11</td>\n",
       "      <td id=\"T_cc9fd_row45_col3\" class=\"data row45 col3\" >56</td>\n",
       "      <td id=\"T_cc9fd_row45_col4\" class=\"data row45 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row45_col5\" class=\"data row45 col5\" >http://model.geneontology.org/R-HSA-5607764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row46\" class=\"row_heading level0 row46\" >46</th>\n",
       "      <td id=\"T_cc9fd_row46_col0\" class=\"data row46 col0\" >FCERI mediated NF-kB activation - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row46_col1\" class=\"data row46 col1\" >0.001575</td>\n",
       "      <td id=\"T_cc9fd_row46_col2\" class=\"data row46 col2\" >11</td>\n",
       "      <td id=\"T_cc9fd_row46_col3\" class=\"data row46 col3\" >56</td>\n",
       "      <td id=\"T_cc9fd_row46_col4\" class=\"data row46 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_cc9fd_row46_col5\" class=\"data row46 col5\" >http://model.geneontology.org/R-HSA-2871837</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row47\" class=\"row_heading level0 row47\" >47</th>\n",
       "      <td id=\"T_cc9fd_row47_col0\" class=\"data row47 col0\" >The proton buffering model - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row47_col1\" class=\"data row47 col1\" >0.001795</td>\n",
       "      <td id=\"T_cc9fd_row47_col2\" class=\"data row47 col2\" >13</td>\n",
       "      <td id=\"T_cc9fd_row47_col3\" class=\"data row47 col3\" >74</td>\n",
       "      <td id=\"T_cc9fd_row47_col4\" class=\"data row47 col4\" >['NDUFS8', 'COX5B', 'NDUFS5', 'NDUFA8', 'NDUFB7', 'NDUFB8', 'NDUFV2', 'COX5A', 'COX7C', 'NDUFV1', 'NDUFA11', 'NDUFA13', 'UQCRC1']</td>\n",
       "      <td id=\"T_cc9fd_row47_col5\" class=\"data row47 col5\" >http://model.geneontology.org/R-HSA-167827</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_cc9fd_level0_row48\" class=\"row_heading level0 row48\" >48</th>\n",
       "      <td id=\"T_cc9fd_row48_col0\" class=\"data row48 col0\" >The fatty acid cycling model - Reactome</td>\n",
       "      <td id=\"T_cc9fd_row48_col1\" class=\"data row48 col1\" >0.001795</td>\n",
       "      <td id=\"T_cc9fd_row48_col2\" class=\"data row48 col2\" >13</td>\n",
       "      <td id=\"T_cc9fd_row48_col3\" class=\"data row48 col3\" >74</td>\n",
       "      <td id=\"T_cc9fd_row48_col4\" class=\"data row48 col4\" >['NDUFS8', 'COX5B', 'NDUFS5', 'NDUFA8', 'NDUFB7', 'NDUFB8', 'NDUFV2', 'COX5A', 'COX7C', 'NDUFV1', 'NDUFA11', 'NDUFA13', 'UQCRC1']</td>\n",
       "      <td id=\"T_cc9fd_row48_col5\" class=\"data row48 col5\" >http://model.geneontology.org/R-HSA-167826</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x12fd79d20>"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "platelets_up_standard = enrich.enrich_wrapper('platelets_up.csv','Gene Symbol',method='standard',FDR = 0.05,fpath = '../test_data/processed/',table_number = 50)\n",
    "platelets_up_standard"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "026b50c2-5dae-47df-a472-0b7d0efefe8f",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|█████████████████████████████████████████| 482/482 [01:23<00:00,  5.75it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 423 entities from 365 out of 1172 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_a56fb caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_a56fb\">\n",
       "  <caption>Table S51: platelets up, weighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_a56fb_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_a56fb_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_a56fb_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_a56fb_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_a56fb_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_a56fb_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_a56fb_row0_col0\" class=\"data row0 col0\" >Collagen biosynthesis and modifying enzymes - Reactome</td>\n",
       "      <td id=\"T_a56fb_row0_col1\" class=\"data row0 col1\" >0.000000</td>\n",
       "      <td id=\"T_a56fb_row0_col2\" class=\"data row0 col2\" >10</td>\n",
       "      <td id=\"T_a56fb_row0_col3\" class=\"data row0 col3\" >12</td>\n",
       "      <td id=\"T_a56fb_row0_col4\" class=\"data row0 col4\" >['PPIB', 'P3H1', 'PLOD3', 'P4HB', 'set:Procollagen C-proteinases', 'set:Procollagen N-proteinases', 'set:Prolyl 3-hydroxylases', 'set:COLGALT1,COLGALT2', 'set:Lysyl hydroxylases', 'set:4-Hyp collagen propeptides']</td>\n",
       "      <td id=\"T_a56fb_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-1650814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_a56fb_row1_col0\" class=\"data row1 col0\" >Hedgehog ligand biogenesis - Reactome</td>\n",
       "      <td id=\"T_a56fb_row1_col1\" class=\"data row1 col1\" >0.000005</td>\n",
       "      <td id=\"T_a56fb_row1_col2\" class=\"data row1 col2\" >13</td>\n",
       "      <td id=\"T_a56fb_row1_col3\" class=\"data row1 col3\" >50</td>\n",
       "      <td id=\"T_a56fb_row1_col4\" class=\"data row1 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'P4HB', 'PSMB6', 'SYVN1']</td>\n",
       "      <td id=\"T_a56fb_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-5358346</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "      <td id=\"T_a56fb_row2_col0\" class=\"data row2 col0\" >ER-Phagosome pathway - Reactome</td>\n",
       "      <td id=\"T_a56fb_row2_col1\" class=\"data row2 col1\" >0.000005</td>\n",
       "      <td id=\"T_a56fb_row2_col2\" class=\"data row2 col2\" >13</td>\n",
       "      <td id=\"T_a56fb_row2_col3\" class=\"data row2 col3\" >51</td>\n",
       "      <td id=\"T_a56fb_row2_col4\" class=\"data row2 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'SEC61B', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6', 'set:SEC61 alpha']</td>\n",
       "      <td id=\"T_a56fb_row2_col5\" class=\"data row2 col5\" >http://model.geneontology.org/R-HSA-1236974</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row3\" class=\"row_heading level0 row3\" >3</th>\n",
       "      <td id=\"T_a56fb_row3_col0\" class=\"data row3 col0\" >Regulation of APC/C activators between G1/S and early anaphase - Reactome</td>\n",
       "      <td id=\"T_a56fb_row3_col1\" class=\"data row3 col1\" >0.000007</td>\n",
       "      <td id=\"T_a56fb_row3_col2\" class=\"data row3 col2\" >13</td>\n",
       "      <td id=\"T_a56fb_row3_col3\" class=\"data row3 col3\" >52</td>\n",
       "      <td id=\"T_a56fb_row3_col4\" class=\"data row3 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'CDK1', 'PSMD8', 'PSMB5', 'PSMB6', 'set:CDC25']</td>\n",
       "      <td id=\"T_a56fb_row3_col5\" class=\"data row3 col5\" >http://model.geneontology.org/R-HSA-176408</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row4\" class=\"row_heading level0 row4\" >4</th>\n",
       "      <td id=\"T_a56fb_row4_col0\" class=\"data row4 col0\" >Neddylation - Reactome</td>\n",
       "      <td id=\"T_a56fb_row4_col1\" class=\"data row4 col1\" >0.000009</td>\n",
       "      <td id=\"T_a56fb_row4_col2\" class=\"data row4 col2\" >14</td>\n",
       "      <td id=\"T_a56fb_row4_col3\" class=\"data row4 col3\" >62</td>\n",
       "      <td id=\"T_a56fb_row4_col4\" class=\"data row4 col4\" >['UBE2M', 'PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'CUL9', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6', 'set:UCHL3,SENP8']</td>\n",
       "      <td id=\"T_a56fb_row4_col5\" class=\"data row4 col5\" >http://model.geneontology.org/R-HSA-8951664</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row5\" class=\"row_heading level0 row5\" >5</th>\n",
       "      <td id=\"T_a56fb_row5_col0\" class=\"data row5 col0\" >KEAP1-NFE2L2 pathway - Reactome</td>\n",
       "      <td id=\"T_a56fb_row5_col1\" class=\"data row5 col1\" >0.000015</td>\n",
       "      <td id=\"T_a56fb_row5_col2\" class=\"data row5 col2\" >13</td>\n",
       "      <td id=\"T_a56fb_row5_col3\" class=\"data row5 col3\" >55</td>\n",
       "      <td id=\"T_a56fb_row5_col4\" class=\"data row5 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'CSNK2B', 'PSMD8', 'PSMB5', 'PSMB6', 'set:PRDX1,2,5']</td>\n",
       "      <td id=\"T_a56fb_row5_col5\" class=\"data row5 col5\" >http://model.geneontology.org/R-HSA-9755511</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row6\" class=\"row_heading level0 row6\" >6</th>\n",
       "      <td id=\"T_a56fb_row6_col0\" class=\"data row6 col0\" >Conversion from APC/C:Cdc20 to APC/C:Cdh1 in late anaphase - Reactome</td>\n",
       "      <td id=\"T_a56fb_row6_col1\" class=\"data row6 col1\" >0.000018</td>\n",
       "      <td id=\"T_a56fb_row6_col2\" class=\"data row6 col2\" >12</td>\n",
       "      <td id=\"T_a56fb_row6_col3\" class=\"data row6 col3\" >49</td>\n",
       "      <td id=\"T_a56fb_row6_col4\" class=\"data row6 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'CDK1', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row6_col5\" class=\"data row6 col5\" >http://model.geneontology.org/R-HSA-176407</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row7\" class=\"row_heading level0 row7\" >7</th>\n",
       "      <td id=\"T_a56fb_row7_col0\" class=\"data row7 col0\" >The role of GTSE1 in G2/M progression after G2 checkpoint - Reactome</td>\n",
       "      <td id=\"T_a56fb_row7_col1\" class=\"data row7 col1\" >0.000018</td>\n",
       "      <td id=\"T_a56fb_row7_col2\" class=\"data row7 col2\" >12</td>\n",
       "      <td id=\"T_a56fb_row7_col3\" class=\"data row7 col3\" >50</td>\n",
       "      <td id=\"T_a56fb_row7_col4\" class=\"data row7 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'CDC25B', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row7_col5\" class=\"data row7 col5\" >http://model.geneontology.org/R-HSA-8852276</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row8\" class=\"row_heading level0 row8\" >8</th>\n",
       "      <td id=\"T_a56fb_row8_col0\" class=\"data row8 col0\" >Degradation of beta-catenin by the destruction complex - Reactome</td>\n",
       "      <td id=\"T_a56fb_row8_col1\" class=\"data row8 col1\" >0.000020</td>\n",
       "      <td id=\"T_a56fb_row8_col2\" class=\"data row8 col2\" >13</td>\n",
       "      <td id=\"T_a56fb_row8_col3\" class=\"data row8 col3\" >54</td>\n",
       "      <td id=\"T_a56fb_row8_col4\" class=\"data row8 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6', 'set:PP2A-subunit A', 'set:PP2A regulatory subunit B56']</td>\n",
       "      <td id=\"T_a56fb_row8_col5\" class=\"data row8 col5\" >http://model.geneontology.org/R-HSA-195253</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row9\" class=\"row_heading level0 row9\" >9</th>\n",
       "      <td id=\"T_a56fb_row9_col0\" class=\"data row9 col0\" >SCF(Skp2)-mediated degradation of p27/p21 - Reactome</td>\n",
       "      <td id=\"T_a56fb_row9_col1\" class=\"data row9 col1\" >0.000021</td>\n",
       "      <td id=\"T_a56fb_row9_col2\" class=\"data row9 col2\" >12</td>\n",
       "      <td id=\"T_a56fb_row9_col3\" class=\"data row9 col3\" >49</td>\n",
       "      <td id=\"T_a56fb_row9_col4\" class=\"data row9 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6', 'CKS1B']</td>\n",
       "      <td id=\"T_a56fb_row9_col5\" class=\"data row9 col5\" >http://model.geneontology.org/R-HSA-187577</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row10\" class=\"row_heading level0 row10\" >10</th>\n",
       "      <td id=\"T_a56fb_row10_col0\" class=\"data row10 col0\" >Antigen processing: Ubiquitination & Proteasome degradation - Reactome</td>\n",
       "      <td id=\"T_a56fb_row10_col1\" class=\"data row10 col1\" >0.000022</td>\n",
       "      <td id=\"T_a56fb_row10_col2\" class=\"data row10 col2\" >14</td>\n",
       "      <td id=\"T_a56fb_row10_col3\" class=\"data row10 col3\" >50</td>\n",
       "      <td id=\"T_a56fb_row10_col4\" class=\"data row10 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6', 'set:E3 ligases in proteasomal degradation', 'set:Activating enzymes E1', 'set:E2 Ubiquitin-conjugating enzyme']</td>\n",
       "      <td id=\"T_a56fb_row10_col5\" class=\"data row10 col5\" >http://model.geneontology.org/R-HSA-983168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row11\" class=\"row_heading level0 row11\" >11</th>\n",
       "      <td id=\"T_a56fb_row11_col0\" class=\"data row11 col0\" >SCF-beta-TrCP mediated degradation of Emi1 - Reactome</td>\n",
       "      <td id=\"T_a56fb_row11_col1\" class=\"data row11 col1\" >0.000027</td>\n",
       "      <td id=\"T_a56fb_row11_col2\" class=\"data row11 col2\" >12</td>\n",
       "      <td id=\"T_a56fb_row11_col3\" class=\"data row11 col3\" >51</td>\n",
       "      <td id=\"T_a56fb_row11_col4\" class=\"data row11 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'CDK1', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row11_col5\" class=\"data row11 col5\" >http://model.geneontology.org/R-HSA-174113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row12\" class=\"row_heading level0 row12\" >12</th>\n",
       "      <td id=\"T_a56fb_row12_col0\" class=\"data row12 col0\" >Respiratory electron transport - Reactome</td>\n",
       "      <td id=\"T_a56fb_row12_col1\" class=\"data row12 col1\" >0.000031</td>\n",
       "      <td id=\"T_a56fb_row12_col2\" class=\"data row12 col2\" >16</td>\n",
       "      <td id=\"T_a56fb_row12_col3\" class=\"data row12 col3\" >91</td>\n",
       "      <td id=\"T_a56fb_row12_col4\" class=\"data row12 col4\" >['NDUFS8', 'ACADVL', 'COX5B', 'NDUFS5', 'NDUFA8', 'NDUFB7', 'ETFB', 'NDUFB8', 'NDUFV2', 'COX5A', 'COX7C', 'NDUFV1', 'NDUFA11', 'NDUFA13', 'ACADS', 'UQCRC1']</td>\n",
       "      <td id=\"T_a56fb_row12_col5\" class=\"data row12 col5\" >http://model.geneontology.org/R-HSA-611105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row13\" class=\"row_heading level0 row13\" >13</th>\n",
       "      <td id=\"T_a56fb_row13_col0\" class=\"data row13 col0\" >Antigen Presentation: Folding, assembly and peptide loading of class I MHC - Reactome</td>\n",
       "      <td id=\"T_a56fb_row13_col1\" class=\"data row13 col1\" >0.000037</td>\n",
       "      <td id=\"T_a56fb_row13_col2\" class=\"data row13 col2\" >12</td>\n",
       "      <td id=\"T_a56fb_row13_col3\" class=\"data row13 col3\" >50</td>\n",
       "      <td id=\"T_a56fb_row13_col4\" class=\"data row13 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6', 'set:ERAP1/2']</td>\n",
       "      <td id=\"T_a56fb_row13_col5\" class=\"data row13 col5\" >http://model.geneontology.org/R-HSA-983170</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row14\" class=\"row_heading level0 row14\" >14</th>\n",
       "      <td id=\"T_a56fb_row14_col0\" class=\"data row14 col0\" >GSK3B and BTRC:CUL1-mediated-degradation of NFE2L2 - Reactome</td>\n",
       "      <td id=\"T_a56fb_row14_col1\" class=\"data row14 col1\" >0.000038</td>\n",
       "      <td id=\"T_a56fb_row14_col2\" class=\"data row14 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row14_col3\" class=\"data row14 col3\" >44</td>\n",
       "      <td id=\"T_a56fb_row14_col4\" class=\"data row14 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row14_col5\" class=\"data row14 col5\" >http://model.geneontology.org/R-HSA-9762114</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row15\" class=\"row_heading level0 row15\" >15</th>\n",
       "      <td id=\"T_a56fb_row15_col0\" class=\"data row15 col0\" >Cross-presentation of soluble exogenous antigens (endosomes) - Reactome</td>\n",
       "      <td id=\"T_a56fb_row15_col1\" class=\"data row15 col1\" >0.000047</td>\n",
       "      <td id=\"T_a56fb_row15_col2\" class=\"data row15 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row15_col3\" class=\"data row15 col3\" >45</td>\n",
       "      <td id=\"T_a56fb_row15_col4\" class=\"data row15 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row15_col5\" class=\"data row15 col5\" >http://model.geneontology.org/R-HSA-1236978</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row16\" class=\"row_heading level0 row16\" >16</th>\n",
       "      <td id=\"T_a56fb_row16_col0\" class=\"data row16 col0\" >APC/C:Cdc20 mediated degradation of mitotic proteins - Reactome</td>\n",
       "      <td id=\"T_a56fb_row16_col1\" class=\"data row16 col1\" >0.000047</td>\n",
       "      <td id=\"T_a56fb_row16_col2\" class=\"data row16 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row16_col3\" class=\"data row16 col3\" >45</td>\n",
       "      <td id=\"T_a56fb_row16_col4\" class=\"data row16 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row16_col5\" class=\"data row16 col5\" >http://model.geneontology.org/R-HSA-176409</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row17\" class=\"row_heading level0 row17\" >17</th>\n",
       "      <td id=\"T_a56fb_row17_col0\" class=\"data row17 col0\" >AUF1 (hnRNP D0) binds and destabilizes mRNA - Reactome</td>\n",
       "      <td id=\"T_a56fb_row17_col1\" class=\"data row17 col1\" >0.000047</td>\n",
       "      <td id=\"T_a56fb_row17_col2\" class=\"data row17 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row17_col3\" class=\"data row17 col3\" >45</td>\n",
       "      <td id=\"T_a56fb_row17_col4\" class=\"data row17 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row17_col5\" class=\"data row17 col5\" >http://model.geneontology.org/R-HSA-450408</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row18\" class=\"row_heading level0 row18\" >18</th>\n",
       "      <td id=\"T_a56fb_row18_col0\" class=\"data row18 col0\" >Regulation of RUNX3 expression and activity - Reactome</td>\n",
       "      <td id=\"T_a56fb_row18_col1\" class=\"data row18 col1\" >0.000055</td>\n",
       "      <td id=\"T_a56fb_row18_col2\" class=\"data row18 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row18_col3\" class=\"data row18 col3\" >45</td>\n",
       "      <td id=\"T_a56fb_row18_col4\" class=\"data row18 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row18_col5\" class=\"data row18 col5\" >http://model.geneontology.org/R-HSA-8941858</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row19\" class=\"row_heading level0 row19\" >19</th>\n",
       "      <td id=\"T_a56fb_row19_col0\" class=\"data row19 col0\" >Ubiquitin-dependent degradation of Cyclin D - Reactome</td>\n",
       "      <td id=\"T_a56fb_row19_col1\" class=\"data row19 col1\" >0.000057</td>\n",
       "      <td id=\"T_a56fb_row19_col2\" class=\"data row19 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row19_col3\" class=\"data row19 col3\" >46</td>\n",
       "      <td id=\"T_a56fb_row19_col4\" class=\"data row19 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row19_col5\" class=\"data row19 col5\" >http://model.geneontology.org/R-HSA-75815</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row20\" class=\"row_heading level0 row20\" >20</th>\n",
       "      <td id=\"T_a56fb_row20_col0\" class=\"data row20 col0\" >Regulation of activated PAK-2p34 by proteasome mediated degradation - Reactome</td>\n",
       "      <td id=\"T_a56fb_row20_col1\" class=\"data row20 col1\" >0.000057</td>\n",
       "      <td id=\"T_a56fb_row20_col2\" class=\"data row20 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row20_col3\" class=\"data row20 col3\" >46</td>\n",
       "      <td id=\"T_a56fb_row20_col4\" class=\"data row20 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row20_col5\" class=\"data row20 col5\" >http://model.geneontology.org/R-HSA-211733</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row21\" class=\"row_heading level0 row21\" >21</th>\n",
       "      <td id=\"T_a56fb_row21_col0\" class=\"data row21 col0\" >Cyclin A:Cdk2-associated events at S phase entry - Reactome</td>\n",
       "      <td id=\"T_a56fb_row21_col1\" class=\"data row21 col1\" >0.000064</td>\n",
       "      <td id=\"T_a56fb_row21_col2\" class=\"data row21 col2\" >12</td>\n",
       "      <td id=\"T_a56fb_row21_col3\" class=\"data row21 col3\" >52</td>\n",
       "      <td id=\"T_a56fb_row21_col4\" class=\"data row21 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6', 'set:Cdc25 A/B']</td>\n",
       "      <td id=\"T_a56fb_row21_col5\" class=\"data row21 col5\" >http://model.geneontology.org/R-HSA-69656</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row22\" class=\"row_heading level0 row22\" >22</th>\n",
       "      <td id=\"T_a56fb_row22_col0\" class=\"data row22 col0\" >Asymmetric localization of PCP proteins - Reactome</td>\n",
       "      <td id=\"T_a56fb_row22_col1\" class=\"data row22 col1\" >0.000067</td>\n",
       "      <td id=\"T_a56fb_row22_col2\" class=\"data row22 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row22_col3\" class=\"data row22 col3\" >46</td>\n",
       "      <td id=\"T_a56fb_row22_col4\" class=\"data row22 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row22_col5\" class=\"data row22 col5\" >http://model.geneontology.org/R-HSA-4608870</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row23\" class=\"row_heading level0 row23\" >23</th>\n",
       "      <td id=\"T_a56fb_row23_col0\" class=\"data row23 col0\" >G2/M Checkpoints - Reactome</td>\n",
       "      <td id=\"T_a56fb_row23_col1\" class=\"data row23 col1\" >0.000070</td>\n",
       "      <td id=\"T_a56fb_row23_col2\" class=\"data row23 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row23_col3\" class=\"data row23 col3\" >48</td>\n",
       "      <td id=\"T_a56fb_row23_col4\" class=\"data row23 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row23_col5\" class=\"data row23 col5\" >http://model.geneontology.org/R-HSA-69481</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row24\" class=\"row_heading level0 row24\" >24</th>\n",
       "      <td id=\"T_a56fb_row24_col0\" class=\"data row24 col0\" >Autodegradation of the E3 ubiquitin ligase COP1 - Reactome</td>\n",
       "      <td id=\"T_a56fb_row24_col1\" class=\"data row24 col1\" >0.000070</td>\n",
       "      <td id=\"T_a56fb_row24_col2\" class=\"data row24 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row24_col3\" class=\"data row24 col3\" >47</td>\n",
       "      <td id=\"T_a56fb_row24_col4\" class=\"data row24 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row24_col5\" class=\"data row24 col5\" >http://model.geneontology.org/R-HSA-349425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row25\" class=\"row_heading level0 row25\" >25</th>\n",
       "      <td id=\"T_a56fb_row25_col0\" class=\"data row25 col0\" >Degradation of DVL - Reactome</td>\n",
       "      <td id=\"T_a56fb_row25_col1\" class=\"data row25 col1\" >0.000070</td>\n",
       "      <td id=\"T_a56fb_row25_col2\" class=\"data row25 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row25_col3\" class=\"data row25 col3\" >47</td>\n",
       "      <td id=\"T_a56fb_row25_col4\" class=\"data row25 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row25_col5\" class=\"data row25 col5\" >http://model.geneontology.org/R-HSA-4641258</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row26\" class=\"row_heading level0 row26\" >26</th>\n",
       "      <td id=\"T_a56fb_row26_col0\" class=\"data row26 col0\" >Cyclin E associated events during G1/S transition - Reactome</td>\n",
       "      <td id=\"T_a56fb_row26_col1\" class=\"data row26 col1\" >0.000075</td>\n",
       "      <td id=\"T_a56fb_row26_col2\" class=\"data row26 col2\" >12</td>\n",
       "      <td id=\"T_a56fb_row26_col3\" class=\"data row26 col3\" >51</td>\n",
       "      <td id=\"T_a56fb_row26_col4\" class=\"data row26 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6', 'set:p-T172-CDK4,p-T177-CDK6:CCND:CDKN1A,CDKN1B,(CDKN1C); p-T172-CDK4,p-T177-CDK6:CCND; (p-T172-CDK4,p-T177-CDK6:CCND:p-Y88-CDKN1B)']</td>\n",
       "      <td id=\"T_a56fb_row26_col5\" class=\"data row26 col5\" >http://model.geneontology.org/R-HSA-69202</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row27\" class=\"row_heading level0 row27\" >27</th>\n",
       "      <td id=\"T_a56fb_row27_col0\" class=\"data row27 col0\" >TNFR2 non-canonical NF-kB pathway - Reactome</td>\n",
       "      <td id=\"T_a56fb_row27_col1\" class=\"data row27 col1\" >0.000094</td>\n",
       "      <td id=\"T_a56fb_row27_col2\" class=\"data row27 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row27_col3\" class=\"data row27 col3\" >47</td>\n",
       "      <td id=\"T_a56fb_row27_col4\" class=\"data row27 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row27_col5\" class=\"data row27 col5\" >http://model.geneontology.org/R-HSA-5668541</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row28\" class=\"row_heading level0 row28\" >28</th>\n",
       "      <td id=\"T_a56fb_row28_col0\" class=\"data row28 col0\" >Degradation of AXIN - Reactome</td>\n",
       "      <td id=\"T_a56fb_row28_col1\" class=\"data row28 col1\" >0.000098</td>\n",
       "      <td id=\"T_a56fb_row28_col2\" class=\"data row28 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row28_col3\" class=\"data row28 col3\" >48</td>\n",
       "      <td id=\"T_a56fb_row28_col4\" class=\"data row28 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row28_col5\" class=\"data row28 col5\" >http://model.geneontology.org/R-HSA-4641257</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row29\" class=\"row_heading level0 row29\" >29</th>\n",
       "      <td id=\"T_a56fb_row29_col0\" class=\"data row29 col0\" >RUNX1 regulates transcription of genes involved in differentiation of HSCs - Reactome</td>\n",
       "      <td id=\"T_a56fb_row29_col1\" class=\"data row29 col1\" >0.000102</td>\n",
       "      <td id=\"T_a56fb_row29_col2\" class=\"data row29 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row29_col3\" class=\"data row29 col3\" >49</td>\n",
       "      <td id=\"T_a56fb_row29_col4\" class=\"data row29 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row29_col5\" class=\"data row29 col5\" >http://model.geneontology.org/R-HSA-8939236</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row30\" class=\"row_heading level0 row30\" >30</th>\n",
       "      <td id=\"T_a56fb_row30_col0\" class=\"data row30 col0\" >Ubiquitin Mediated Degradation of Phosphorylated Cdc25A - Reactome</td>\n",
       "      <td id=\"T_a56fb_row30_col1\" class=\"data row30 col1\" >0.000102</td>\n",
       "      <td id=\"T_a56fb_row30_col2\" class=\"data row30 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row30_col3\" class=\"data row30 col3\" >49</td>\n",
       "      <td id=\"T_a56fb_row30_col4\" class=\"data row30 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row30_col5\" class=\"data row30 col5\" >http://model.geneontology.org/R-HSA-69601</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row31\" class=\"row_heading level0 row31\" >31</th>\n",
       "      <td id=\"T_a56fb_row31_col0\" class=\"data row31 col0\" >Oxygen-dependent proline hydroxylation of Hypoxia-inducible Factor Alpha - Reactome</td>\n",
       "      <td id=\"T_a56fb_row31_col1\" class=\"data row31 col1\" >0.000112</td>\n",
       "      <td id=\"T_a56fb_row31_col2\" class=\"data row31 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row31_col3\" class=\"data row31 col3\" >48</td>\n",
       "      <td id=\"T_a56fb_row31_col4\" class=\"data row31 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row31_col5\" class=\"data row31 col5\" >http://model.geneontology.org/R-HSA-1234176</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row32\" class=\"row_heading level0 row32\" >32</th>\n",
       "      <td id=\"T_a56fb_row32_col0\" class=\"data row32 col0\" >Hedgehog 'on' state - Reactome</td>\n",
       "      <td id=\"T_a56fb_row32_col1\" class=\"data row32 col1\" >0.000117</td>\n",
       "      <td id=\"T_a56fb_row32_col2\" class=\"data row32 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row32_col3\" class=\"data row32 col3\" >49</td>\n",
       "      <td id=\"T_a56fb_row32_col4\" class=\"data row32 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row32_col5\" class=\"data row32 col5\" >http://model.geneontology.org/R-HSA-5632684</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row33\" class=\"row_heading level0 row33\" >33</th>\n",
       "      <td id=\"T_a56fb_row33_col0\" class=\"data row33 col0\" >Hedgehog 'off' state - Reactome</td>\n",
       "      <td id=\"T_a56fb_row33_col1\" class=\"data row33 col1\" >0.000119</td>\n",
       "      <td id=\"T_a56fb_row33_col2\" class=\"data row33 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row33_col3\" class=\"data row33 col3\" >46</td>\n",
       "      <td id=\"T_a56fb_row33_col4\" class=\"data row33 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row33_col5\" class=\"data row33 col5\" >http://model.geneontology.org/R-HSA-5610787</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row34\" class=\"row_heading level0 row34\" >34</th>\n",
       "      <td id=\"T_a56fb_row34_col0\" class=\"data row34 col0\" >FBXL7 down-regulates AURKA during mitotic entry and in early mitosis - Reactome</td>\n",
       "      <td id=\"T_a56fb_row34_col1\" class=\"data row34 col1\" >0.000122</td>\n",
       "      <td id=\"T_a56fb_row34_col2\" class=\"data row34 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row34_col3\" class=\"data row34 col3\" >50</td>\n",
       "      <td id=\"T_a56fb_row34_col4\" class=\"data row34 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row34_col5\" class=\"data row34 col5\" >http://model.geneontology.org/R-HSA-8854050</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row35\" class=\"row_heading level0 row35\" >35</th>\n",
       "      <td id=\"T_a56fb_row35_col0\" class=\"data row35 col0\" >Synthesis of PE - Reactome</td>\n",
       "      <td id=\"T_a56fb_row35_col1\" class=\"data row35 col1\" >0.000124</td>\n",
       "      <td id=\"T_a56fb_row35_col2\" class=\"data row35 col2\" >8</td>\n",
       "      <td id=\"T_a56fb_row35_col3\" class=\"data row35 col3\" >15</td>\n",
       "      <td id=\"T_a56fb_row35_col4\" class=\"data row35 col4\" >['PTDSS2', 'PCYT2', 'PISD', 'PHOSPHO1', 'set:LPIN', 'set:CHK/ETNK', 'set:AGPAT', 'set:PNPLA2/3']</td>\n",
       "      <td id=\"T_a56fb_row35_col5\" class=\"data row35 col5\" >http://model.geneontology.org/R-HSA-1483213</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row36\" class=\"row_heading level0 row36\" >36</th>\n",
       "      <td id=\"T_a56fb_row36_col0\" class=\"data row36 col0\" >Negative regulation of NOTCH4 signaling - Reactome</td>\n",
       "      <td id=\"T_a56fb_row36_col1\" class=\"data row36 col1\" >0.000134</td>\n",
       "      <td id=\"T_a56fb_row36_col2\" class=\"data row36 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row36_col3\" class=\"data row36 col3\" >49</td>\n",
       "      <td id=\"T_a56fb_row36_col4\" class=\"data row36 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row36_col5\" class=\"data row36 col5\" >http://model.geneontology.org/R-HSA-9604323</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row37\" class=\"row_heading level0 row37\" >37</th>\n",
       "      <td id=\"T_a56fb_row37_col0\" class=\"data row37 col0\" >NIK-->noncanonical NF-kB signaling - Reactome</td>\n",
       "      <td id=\"T_a56fb_row37_col1\" class=\"data row37 col1\" >0.000140</td>\n",
       "      <td id=\"T_a56fb_row37_col2\" class=\"data row37 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row37_col3\" class=\"data row37 col3\" >50</td>\n",
       "      <td id=\"T_a56fb_row37_col4\" class=\"data row37 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row37_col5\" class=\"data row37 col5\" >http://model.geneontology.org/R-HSA-5676590</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row38\" class=\"row_heading level0 row38\" >38</th>\n",
       "      <td id=\"T_a56fb_row38_col0\" class=\"data row38 col0\" >Dectin-1 mediated noncanonical NF-kB signaling - Reactome</td>\n",
       "      <td id=\"T_a56fb_row38_col1\" class=\"data row38 col1\" >0.000140</td>\n",
       "      <td id=\"T_a56fb_row38_col2\" class=\"data row38 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row38_col3\" class=\"data row38 col3\" >50</td>\n",
       "      <td id=\"T_a56fb_row38_col4\" class=\"data row38 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row38_col5\" class=\"data row38 col5\" >http://model.geneontology.org/R-HSA-5607761</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row39\" class=\"row_heading level0 row39\" >39</th>\n",
       "      <td id=\"T_a56fb_row39_col0\" class=\"data row39 col0\" >Regulation of expression of SLITs and ROBOs - Reactome</td>\n",
       "      <td id=\"T_a56fb_row39_col1\" class=\"data row39 col1\" >0.000145</td>\n",
       "      <td id=\"T_a56fb_row39_col2\" class=\"data row39 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row39_col3\" class=\"data row39 col3\" >51</td>\n",
       "      <td id=\"T_a56fb_row39_col4\" class=\"data row39 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row39_col5\" class=\"data row39 col5\" >http://model.geneontology.org/R-HSA-9010553</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row40\" class=\"row_heading level0 row40\" >40</th>\n",
       "      <td id=\"T_a56fb_row40_col0\" class=\"data row40 col0\" >Orc1 removal from chromatin - Reactome</td>\n",
       "      <td id=\"T_a56fb_row40_col1\" class=\"data row40 col1\" >0.000166</td>\n",
       "      <td id=\"T_a56fb_row40_col2\" class=\"data row40 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row40_col3\" class=\"data row40 col3\" >51</td>\n",
       "      <td id=\"T_a56fb_row40_col4\" class=\"data row40 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row40_col5\" class=\"data row40 col5\" >http://model.geneontology.org/R-HSA-68949</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row41\" class=\"row_heading level0 row41\" >41</th>\n",
       "      <td id=\"T_a56fb_row41_col0\" class=\"data row41 col0\" >Regulation of PTEN stability and activity - Reactome</td>\n",
       "      <td id=\"T_a56fb_row41_col1\" class=\"data row41 col1\" >0.000172</td>\n",
       "      <td id=\"T_a56fb_row41_col2\" class=\"data row41 col2\" >12</td>\n",
       "      <td id=\"T_a56fb_row41_col3\" class=\"data row41 col3\" >56</td>\n",
       "      <td id=\"T_a56fb_row41_col4\" class=\"data row41 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'CSNK2B', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row41_col5\" class=\"data row41 col5\" >http://model.geneontology.org/R-HSA-8948751</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row42\" class=\"row_heading level0 row42\" >42</th>\n",
       "      <td id=\"T_a56fb_row42_col0\" class=\"data row42 col0\" >The fatty acid cycling model - Reactome</td>\n",
       "      <td id=\"T_a56fb_row42_col1\" class=\"data row42 col1\" >0.000183</td>\n",
       "      <td id=\"T_a56fb_row42_col2\" class=\"data row42 col2\" >13</td>\n",
       "      <td id=\"T_a56fb_row42_col3\" class=\"data row42 col3\" >70</td>\n",
       "      <td id=\"T_a56fb_row42_col4\" class=\"data row42 col4\" >['NDUFS8', 'COX5B', 'NDUFS5', 'NDUFA8', 'NDUFB7', 'NDUFB8', 'NDUFV2', 'COX5A', 'COX7C', 'NDUFV1', 'NDUFA11', 'NDUFA13', 'UQCRC1']</td>\n",
       "      <td id=\"T_a56fb_row42_col5\" class=\"data row42 col5\" >http://model.geneontology.org/R-HSA-167826</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row43\" class=\"row_heading level0 row43\" >43</th>\n",
       "      <td id=\"T_a56fb_row43_col0\" class=\"data row43 col0\" >The proton buffering model - Reactome</td>\n",
       "      <td id=\"T_a56fb_row43_col1\" class=\"data row43 col1\" >0.000183</td>\n",
       "      <td id=\"T_a56fb_row43_col2\" class=\"data row43 col2\" >13</td>\n",
       "      <td id=\"T_a56fb_row43_col3\" class=\"data row43 col3\" >70</td>\n",
       "      <td id=\"T_a56fb_row43_col4\" class=\"data row43 col4\" >['NDUFS8', 'COX5B', 'NDUFS5', 'NDUFA8', 'NDUFB7', 'NDUFB8', 'NDUFV2', 'COX5A', 'COX7C', 'NDUFV1', 'NDUFA11', 'NDUFA13', 'UQCRC1']</td>\n",
       "      <td id=\"T_a56fb_row43_col5\" class=\"data row43 col5\" >http://model.geneontology.org/R-HSA-167827</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row44\" class=\"row_heading level0 row44\" >44</th>\n",
       "      <td id=\"T_a56fb_row44_col0\" class=\"data row44 col0\" >Degradation of GLI1 by the proteasome - Reactome</td>\n",
       "      <td id=\"T_a56fb_row44_col1\" class=\"data row44 col1\" >0.000185</td>\n",
       "      <td id=\"T_a56fb_row44_col2\" class=\"data row44 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row44_col3\" class=\"data row44 col3\" >51</td>\n",
       "      <td id=\"T_a56fb_row44_col4\" class=\"data row44 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row44_col5\" class=\"data row44 col5\" >http://model.geneontology.org/R-HSA-5610780</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row45\" class=\"row_heading level0 row45\" >45</th>\n",
       "      <td id=\"T_a56fb_row45_col0\" class=\"data row45 col0\" >GLI3 is processed to GLI3R by the proteasome - Reactome</td>\n",
       "      <td id=\"T_a56fb_row45_col1\" class=\"data row45 col1\" >0.000218</td>\n",
       "      <td id=\"T_a56fb_row45_col2\" class=\"data row45 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row45_col3\" class=\"data row45 col3\" >52</td>\n",
       "      <td id=\"T_a56fb_row45_col4\" class=\"data row45 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row45_col5\" class=\"data row45 col5\" >http://model.geneontology.org/R-HSA-5610785</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row46\" class=\"row_heading level0 row46\" >46</th>\n",
       "      <td id=\"T_a56fb_row46_col0\" class=\"data row46 col0\" >Degradation of GLI2 by the proteasome - Reactome</td>\n",
       "      <td id=\"T_a56fb_row46_col1\" class=\"data row46 col1\" >0.000218</td>\n",
       "      <td id=\"T_a56fb_row46_col2\" class=\"data row46 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row46_col3\" class=\"data row46 col3\" >52</td>\n",
       "      <td id=\"T_a56fb_row46_col4\" class=\"data row46 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row46_col5\" class=\"data row46 col5\" >http://model.geneontology.org/R-HSA-5610783</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row47\" class=\"row_heading level0 row47\" >47</th>\n",
       "      <td id=\"T_a56fb_row47_col0\" class=\"data row47 col0\" >Regulation of RUNX2 expression and activity - Reactome</td>\n",
       "      <td id=\"T_a56fb_row47_col1\" class=\"data row47 col1\" >0.000239</td>\n",
       "      <td id=\"T_a56fb_row47_col2\" class=\"data row47 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row47_col3\" class=\"data row47 col3\" >54</td>\n",
       "      <td id=\"T_a56fb_row47_col4\" class=\"data row47 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row47_col5\" class=\"data row47 col5\" >http://model.geneontology.org/R-HSA-8939902</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row48\" class=\"row_heading level0 row48\" >48</th>\n",
       "      <td id=\"T_a56fb_row48_col0\" class=\"data row48 col0\" >Regulation of RAS by GAPs - Reactome</td>\n",
       "      <td id=\"T_a56fb_row48_col1\" class=\"data row48 col1\" >0.000264</td>\n",
       "      <td id=\"T_a56fb_row48_col2\" class=\"data row48 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row48_col3\" class=\"data row48 col3\" >48</td>\n",
       "      <td id=\"T_a56fb_row48_col4\" class=\"data row48 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row48_col5\" class=\"data row48 col5\" >http://model.geneontology.org/R-HSA-5658442</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row49\" class=\"row_heading level0 row49\" >49</th>\n",
       "      <td id=\"T_a56fb_row49_col0\" class=\"data row49 col0\" >FCERI mediated NF-kB activation - Reactome</td>\n",
       "      <td id=\"T_a56fb_row49_col1\" class=\"data row49 col1\" >0.000307</td>\n",
       "      <td id=\"T_a56fb_row49_col2\" class=\"data row49 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row49_col3\" class=\"data row49 col3\" >54</td>\n",
       "      <td id=\"T_a56fb_row49_col4\" class=\"data row49 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row49_col5\" class=\"data row49 col5\" >http://model.geneontology.org/R-HSA-2871837</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row50\" class=\"row_heading level0 row50\" >50</th>\n",
       "      <td id=\"T_a56fb_row50_col0\" class=\"data row50 col0\" >CLEC7A (Dectin-1) signaling - Reactome</td>\n",
       "      <td id=\"T_a56fb_row50_col1\" class=\"data row50 col1\" >0.000317</td>\n",
       "      <td id=\"T_a56fb_row50_col2\" class=\"data row50 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row50_col3\" class=\"data row50 col3\" >55</td>\n",
       "      <td id=\"T_a56fb_row50_col4\" class=\"data row50 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row50_col5\" class=\"data row50 col5\" >http://model.geneontology.org/R-HSA-5607764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row51\" class=\"row_heading level0 row51\" >51</th>\n",
       "      <td id=\"T_a56fb_row51_col0\" class=\"data row51 col0\" >Formation of ATP by chemiosmotic coupling - Reactome</td>\n",
       "      <td id=\"T_a56fb_row51_col1\" class=\"data row51 col1\" >0.000358</td>\n",
       "      <td id=\"T_a56fb_row51_col2\" class=\"data row51 col2\" >14</td>\n",
       "      <td id=\"T_a56fb_row51_col3\" class=\"data row51 col3\" >88</td>\n",
       "      <td id=\"T_a56fb_row51_col4\" class=\"data row51 col4\" >['ATP5F1D', 'NDUFS8', 'COX5B', 'NDUFS5', 'NDUFA8', 'NDUFB7', 'NDUFB8', 'NDUFV2', 'COX5A', 'COX7C', 'NDUFV1', 'NDUFA11', 'NDUFA13', 'UQCRC1']</td>\n",
       "      <td id=\"T_a56fb_row51_col5\" class=\"data row51 col5\" >http://model.geneontology.org/R-HSA-163210</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row52\" class=\"row_heading level0 row52\" >52</th>\n",
       "      <td id=\"T_a56fb_row52_col0\" class=\"data row52 col0\" >Cyclin A/B1/B2 associated events during G2/M transition - Reactome</td>\n",
       "      <td id=\"T_a56fb_row52_col1\" class=\"data row52 col1\" >0.000379</td>\n",
       "      <td id=\"T_a56fb_row52_col2\" class=\"data row52 col2\" >7</td>\n",
       "      <td id=\"T_a56fb_row52_col3\" class=\"data row52 col3\" >17</td>\n",
       "      <td id=\"T_a56fb_row52_col4\" class=\"data row52 col4\" >['LCMT1', 'CDC25B', 'CDK1', 'PPP2R2A', 'set:PP2A-subunit A', 'set:CDC25', 'set:CRS kinase']</td>\n",
       "      <td id=\"T_a56fb_row52_col5\" class=\"data row52 col5\" >http://model.geneontology.org/R-HSA-69273</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row53\" class=\"row_heading level0 row53\" >53</th>\n",
       "      <td id=\"T_a56fb_row53_col0\" class=\"data row53 col0\" >Activation of NF-kappaB in B cells - Reactome</td>\n",
       "      <td id=\"T_a56fb_row53_col1\" class=\"data row53 col1\" >0.000383</td>\n",
       "      <td id=\"T_a56fb_row53_col2\" class=\"data row53 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row53_col3\" class=\"data row53 col3\" >54</td>\n",
       "      <td id=\"T_a56fb_row53_col4\" class=\"data row53 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row53_col5\" class=\"data row53 col5\" >http://model.geneontology.org/R-HSA-1169091</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row54\" class=\"row_heading level0 row54\" >54</th>\n",
       "      <td id=\"T_a56fb_row54_col0\" class=\"data row54 col0\" >Heme biosynthesis - Reactome</td>\n",
       "      <td id=\"T_a56fb_row54_col1\" class=\"data row54 col1\" >0.000479</td>\n",
       "      <td id=\"T_a56fb_row54_col2\" class=\"data row54 col2\" >6</td>\n",
       "      <td id=\"T_a56fb_row54_col3\" class=\"data row54 col3\" >14</td>\n",
       "      <td id=\"T_a56fb_row54_col4\" class=\"data row54 col4\" >['UROS', 'COX10', 'HMBS', 'CPOX', 'PPOX', 'set:ALAS1,2']</td>\n",
       "      <td id=\"T_a56fb_row54_col5\" class=\"data row54 col5\" >http://model.geneontology.org/R-HSA-189451</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row55\" class=\"row_heading level0 row55\" >55</th>\n",
       "      <td id=\"T_a56fb_row55_col0\" class=\"data row55 col0\" >Autodegradation of Cdh1 by Cdh1:APC/C - Reactome</td>\n",
       "      <td id=\"T_a56fb_row55_col1\" class=\"data row55 col1\" >0.000507</td>\n",
       "      <td id=\"T_a56fb_row55_col2\" class=\"data row55 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row55_col3\" class=\"data row55 col3\" >59</td>\n",
       "      <td id=\"T_a56fb_row55_col4\" class=\"data row55 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row55_col5\" class=\"data row55 col5\" >http://model.geneontology.org/R-HSA-174084</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row56\" class=\"row_heading level0 row56\" >56</th>\n",
       "      <td id=\"T_a56fb_row56_col0\" class=\"data row56 col0\" >MAPK6/MAPK4 signaling - Reactome</td>\n",
       "      <td id=\"T_a56fb_row56_col1\" class=\"data row56 col1\" >0.000570</td>\n",
       "      <td id=\"T_a56fb_row56_col2\" class=\"data row56 col2\" >12</td>\n",
       "      <td id=\"T_a56fb_row56_col3\" class=\"data row56 col3\" >62</td>\n",
       "      <td id=\"T_a56fb_row56_col4\" class=\"data row56 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'CDK1', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row56_col5\" class=\"data row56 col5\" >http://model.geneontology.org/R-HSA-5687128</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row57\" class=\"row_heading level0 row57\" >57</th>\n",
       "      <td id=\"T_a56fb_row57_col0\" class=\"data row57 col0\" >Assembly of the pre-replicative complex - Reactome</td>\n",
       "      <td id=\"T_a56fb_row57_col1\" class=\"data row57 col1\" >0.000583</td>\n",
       "      <td id=\"T_a56fb_row57_col2\" class=\"data row57 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row57_col3\" class=\"data row57 col3\" >60</td>\n",
       "      <td id=\"T_a56fb_row57_col4\" class=\"data row57 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row57_col5\" class=\"data row57 col5\" >http://model.geneontology.org/R-HSA-68867</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row58\" class=\"row_heading level0 row58\" >58</th>\n",
       "      <td id=\"T_a56fb_row58_col0\" class=\"data row58 col0\" >ABC-family proteins mediated transport - Reactome</td>\n",
       "      <td id=\"T_a56fb_row58_col1\" class=\"data row58 col1\" >0.000624</td>\n",
       "      <td id=\"T_a56fb_row58_col2\" class=\"data row58 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row58_col3\" class=\"data row58 col3\" >54</td>\n",
       "      <td id=\"T_a56fb_row58_col4\" class=\"data row58 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row58_col5\" class=\"data row58 col5\" >http://model.geneontology.org/R-HSA-382556</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row59\" class=\"row_heading level0 row59\" >59</th>\n",
       "      <td id=\"T_a56fb_row59_col0\" class=\"data row59 col0\" >APC/C:Cdc20 mediated degradation of Securin - Reactome</td>\n",
       "      <td id=\"T_a56fb_row59_col1\" class=\"data row59 col1\" >0.000764</td>\n",
       "      <td id=\"T_a56fb_row59_col2\" class=\"data row59 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row59_col3\" class=\"data row59 col3\" >62</td>\n",
       "      <td id=\"T_a56fb_row59_col4\" class=\"data row59 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row59_col5\" class=\"data row59 col5\" >http://model.geneontology.org/R-HSA-174154</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row60\" class=\"row_heading level0 row60\" >60</th>\n",
       "      <td id=\"T_a56fb_row60_col0\" class=\"data row60 col0\" >Cdc20:Phospho-APC/C mediated degradation of Cyclin A - Reactome</td>\n",
       "      <td id=\"T_a56fb_row60_col1\" class=\"data row60 col1\" >0.000764</td>\n",
       "      <td id=\"T_a56fb_row60_col2\" class=\"data row60 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row60_col3\" class=\"data row60 col3\" >62</td>\n",
       "      <td id=\"T_a56fb_row60_col4\" class=\"data row60 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row60_col5\" class=\"data row60 col5\" >http://model.geneontology.org/R-HSA-174184</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row61\" class=\"row_heading level0 row61\" >61</th>\n",
       "      <td id=\"T_a56fb_row61_col0\" class=\"data row61 col0\" >APC/C:Cdh1 mediated degradation of Cdc20 and other APC/C:Cdh1 targeted proteins in late mitosis/early G1 - Reactome</td>\n",
       "      <td id=\"T_a56fb_row61_col1\" class=\"data row61 col1\" >0.000764</td>\n",
       "      <td id=\"T_a56fb_row61_col2\" class=\"data row61 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row61_col3\" class=\"data row61 col3\" >62</td>\n",
       "      <td id=\"T_a56fb_row61_col4\" class=\"data row61 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row61_col5\" class=\"data row61 col5\" >http://model.geneontology.org/R-HSA-174178</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row62\" class=\"row_heading level0 row62\" >62</th>\n",
       "      <td id=\"T_a56fb_row62_col0\" class=\"data row62 col0\" >Separation of Sister Chromatids - Reactome</td>\n",
       "      <td id=\"T_a56fb_row62_col1\" class=\"data row62 col1\" >0.000870</td>\n",
       "      <td id=\"T_a56fb_row62_col2\" class=\"data row62 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row62_col3\" class=\"data row62 col3\" >63</td>\n",
       "      <td id=\"T_a56fb_row62_col4\" class=\"data row62 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row62_col5\" class=\"data row62 col5\" >http://model.geneontology.org/R-HSA-2467813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row63\" class=\"row_heading level0 row63\" >63</th>\n",
       "      <td id=\"T_a56fb_row63_col0\" class=\"data row63 col0\" >CDK-mediated phosphorylation and removal of Cdc6 - Reactome</td>\n",
       "      <td id=\"T_a56fb_row63_col1\" class=\"data row63 col1\" >0.001441</td>\n",
       "      <td id=\"T_a56fb_row63_col2\" class=\"data row63 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row63_col3\" class=\"data row63 col3\" >65</td>\n",
       "      <td id=\"T_a56fb_row63_col4\" class=\"data row63 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row63_col5\" class=\"data row63 col5\" >http://model.geneontology.org/R-HSA-69017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row64\" class=\"row_heading level0 row64\" >64</th>\n",
       "      <td id=\"T_a56fb_row64_col0\" class=\"data row64 col0\" >Interleukin-1 signaling - Reactome</td>\n",
       "      <td id=\"T_a56fb_row64_col1\" class=\"data row64 col1\" >0.001727</td>\n",
       "      <td id=\"T_a56fb_row64_col2\" class=\"data row64 col2\" >11</td>\n",
       "      <td id=\"T_a56fb_row64_col3\" class=\"data row64 col3\" >64</td>\n",
       "      <td id=\"T_a56fb_row64_col4\" class=\"data row64 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6']</td>\n",
       "      <td id=\"T_a56fb_row64_col5\" class=\"data row64 col5\" >http://model.geneontology.org/R-HSA-9020702</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a56fb_level0_row65\" class=\"row_heading level0 row65\" >65</th>\n",
       "      <td id=\"T_a56fb_row65_col0\" class=\"data row65 col0\" >Downstream TCR signaling - Reactome</td>\n",
       "      <td id=\"T_a56fb_row65_col1\" class=\"data row65 col1\" >0.001814</td>\n",
       "      <td id=\"T_a56fb_row65_col2\" class=\"data row65 col2\" >14</td>\n",
       "      <td id=\"T_a56fb_row65_col3\" class=\"data row65 col3\" >74</td>\n",
       "      <td id=\"T_a56fb_row65_col4\" class=\"data row65 col4\" >['PSMD13', 'PSMA5', 'PSMD11', 'PSMA7', 'PSME2', 'PSMD4', 'PSMB1', 'PSMB7', 'PSMD8', 'PSMB5', 'PSMB6', 'CD3D', 'set:PIK3C(1)', 'set:Activator:PI3K']</td>\n",
       "      <td id=\"T_a56fb_row65_col5\" class=\"data row65 col5\" >http://model.geneontology.org/R-HSA-202424</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x14e64a560>"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "platelets_up_weighted = enrich.enrich_wrapper('platelets_up.csv','Gene Symbol',method='ncHGT',FDR = 0.05,fpath = '../test_data/processed/',table_number = 51)\n",
    "platelets_up_weighted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "9ac9f1ce-9932-494a-b22b-653e9f7fb07c",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 436/436 [00:00<00:00, 16523.30it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 309 entities from 267 out of 1088 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_0dfe4 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_0dfe4\">\n",
       "  <caption>Table S52: platelets down, unweighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_0dfe4_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_0dfe4_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_0dfe4_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_0dfe4_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_0dfe4_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_0dfe4_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_0dfe4_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_0dfe4_row0_col0\" class=\"data row0 col0\" >Role of phospholipids in phagocytosis - Reactome</td>\n",
       "      <td id=\"T_0dfe4_row0_col1\" class=\"data row0 col1\" >0.000000</td>\n",
       "      <td id=\"T_0dfe4_row0_col2\" class=\"data row0 col2\" >11</td>\n",
       "      <td id=\"T_0dfe4_row0_col3\" class=\"data row0 col3\" >22</td>\n",
       "      <td id=\"T_0dfe4_row0_col4\" class=\"data row0 col4\" >['set:G(q) alpha 11,14,15,Q', 'set:PL(C)D4:3xCa2+', 'set:PLC-beta 1/2/3', 'set:G-protein gamma subunit', 'set:PI3K-regulatory subunit', 'set:PLCdegh', 'set:PLD', 'set:PLC beta1,2,3', 'set:PLC-beta', 'set:PLCbz', 'set:G-protein alpha (q/11/14/15)']</td>\n",
       "      <td id=\"T_0dfe4_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-2029485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0dfe4_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_0dfe4_row1_col0\" class=\"data row1 col0\" >DAG and IP3 signaling - Reactome</td>\n",
       "      <td id=\"T_0dfe4_row1_col1\" class=\"data row1 col1\" >0.000000</td>\n",
       "      <td id=\"T_0dfe4_row1_col2\" class=\"data row1 col2\" >10</td>\n",
       "      <td id=\"T_0dfe4_row1_col3\" class=\"data row1 col3\" >18</td>\n",
       "      <td id=\"T_0dfe4_row1_col4\" class=\"data row1 col4\" >['set:G(q) alpha 11,14,15,Q', 'set:PL(C)D4:3xCa2+', 'set:PLC-beta 1/2/3', 'set:G-protein gamma subunit', 'set:PI3K-regulatory subunit', 'set:PLCdegh', 'set:PLC beta1,2,3', 'set:PLC-beta', 'set:PLCbz', 'set:G-protein alpha (q/11/14/15)']</td>\n",
       "      <td id=\"T_0dfe4_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-1489509</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0dfe4_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "      <td id=\"T_0dfe4_row2_col0\" class=\"data row2 col0\" >PLC beta mediated events - Reactome</td>\n",
       "      <td id=\"T_0dfe4_row2_col1\" class=\"data row2 col1\" >0.000000</td>\n",
       "      <td id=\"T_0dfe4_row2_col2\" class=\"data row2 col2\" >9</td>\n",
       "      <td id=\"T_0dfe4_row2_col3\" class=\"data row2 col3\" >15</td>\n",
       "      <td id=\"T_0dfe4_row2_col4\" class=\"data row2 col4\" >['set:G(q) alpha 11,14,15,Q', 'set:PL(C)D4:3xCa2+', 'set:PLC-beta 1/2/3', 'set:G-protein gamma subunit', 'set:PLCdegh', 'set:PLC beta1,2,3', 'set:PLC-beta', 'set:PLCbz', 'set:G-protein alpha (q/11/14/15)']</td>\n",
       "      <td id=\"T_0dfe4_row2_col5\" class=\"data row2 col5\" >http://model.geneontology.org/R-HSA-112043</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0dfe4_level0_row3\" class=\"row_heading level0 row3\" >3</th>\n",
       "      <td id=\"T_0dfe4_row3_col0\" class=\"data row3 col0\" >Antigen activates B Cell Receptor (BCR) leading to generation of second messengers - Reactome</td>\n",
       "      <td id=\"T_0dfe4_row3_col1\" class=\"data row3 col1\" >0.000001</td>\n",
       "      <td id=\"T_0dfe4_row3_col2\" class=\"data row3 col2\" >11</td>\n",
       "      <td id=\"T_0dfe4_row3_col3\" class=\"data row3 col3\" >25</td>\n",
       "      <td id=\"T_0dfe4_row3_col4\" class=\"data row3 col4\" >['PIK3R1', 'set:G(q) alpha 11,14,15,Q', 'set:PL(C)D4:3xCa2+', 'set:PLC-beta 1/2/3', 'set:G-protein gamma subunit', 'set:PLCdegh', 'set:PLC beta1,2,3', 'set:PLC-beta', 'set:PLCbz', 'set:G-protein alpha (q/11/14/15)', 'set:LYN, p-SYK']</td>\n",
       "      <td id=\"T_0dfe4_row3_col5\" class=\"data row3 col5\" >http://model.geneontology.org/R-HSA-983695</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0dfe4_level0_row4\" class=\"row_heading level0 row4\" >4</th>\n",
       "      <td id=\"T_0dfe4_row4_col0\" class=\"data row4 col0\" >G alpha (12/13) signalling events - Reactome</td>\n",
       "      <td id=\"T_0dfe4_row4_col1\" class=\"data row4 col1\" >0.000001</td>\n",
       "      <td id=\"T_0dfe4_row4_col2\" class=\"data row4 col2\" >8</td>\n",
       "      <td id=\"T_0dfe4_row4_col3\" class=\"data row4 col3\" >13</td>\n",
       "      <td id=\"T_0dfe4_row4_col4\" class=\"data row4 col4\" >['GNA13', 'set:GEFs', 'set:G-protein G12/G13 (inactive)', 'set:G-protein gamma subunit', 'set:Ligand:GPCR complexes that activate Gi', 'set:G alpha (i)', 'set:ARHGEF11,ARHGEF12', 'set:G-protein alpha (12/13):GTP']</td>\n",
       "      <td id=\"T_0dfe4_row4_col5\" class=\"data row4 col5\" >http://model.geneontology.org/R-HSA-416482</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0dfe4_level0_row5\" class=\"row_heading level0 row5\" >5</th>\n",
       "      <td id=\"T_0dfe4_row5_col0\" class=\"data row5 col0\" >VEGFR2 mediated cell proliferation - Reactome</td>\n",
       "      <td id=\"T_0dfe4_row5_col1\" class=\"data row5 col1\" >0.000002</td>\n",
       "      <td id=\"T_0dfe4_row5_col2\" class=\"data row5 col2\" >11</td>\n",
       "      <td id=\"T_0dfe4_row5_col3\" class=\"data row5 col3\" >27</td>\n",
       "      <td id=\"T_0dfe4_row5_col4\" class=\"data row5 col4\" >['KDR', 'set:G(q) alpha 11,14,15,Q', 'set:PL(C)D4:3xCa2+', 'set:PLC-beta 1/2/3', 'set:G-protein gamma subunit', 'set:PI3K-regulatory subunit', 'set:PLCdegh', 'set:PLC beta1,2,3', 'set:PLC-beta', 'set:PLCbz', 'set:G-protein alpha (q/11/14/15)']</td>\n",
       "      <td id=\"T_0dfe4_row5_col5\" class=\"data row5 col5\" >http://model.geneontology.org/R-HSA-5218921</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0dfe4_level0_row6\" class=\"row_heading level0 row6\" >6</th>\n",
       "      <td id=\"T_0dfe4_row6_col0\" class=\"data row6 col0\" >CLEC7A (Dectin-1) induces NFAT activation - Reactome</td>\n",
       "      <td id=\"T_0dfe4_row6_col1\" class=\"data row6 col1\" >0.000018</td>\n",
       "      <td id=\"T_0dfe4_row6_col2\" class=\"data row6 col2\" >9</td>\n",
       "      <td id=\"T_0dfe4_row6_col3\" class=\"data row6 col3\" >22</td>\n",
       "      <td id=\"T_0dfe4_row6_col4\" class=\"data row6 col4\" >['set:G(q) alpha 11,14,15,Q', 'set:PL(C)D4:3xCa2+', 'set:PLC-beta 1/2/3', 'set:G-protein gamma subunit', 'set:PLCdegh', 'set:PLC beta1,2,3', 'set:PLC-beta', 'set:PLCbz', 'set:G-protein alpha (q/11/14/15)']</td>\n",
       "      <td id=\"T_0dfe4_row6_col5\" class=\"data row6 col5\" >http://model.geneontology.org/R-HSA-5607763</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0dfe4_level0_row7\" class=\"row_heading level0 row7\" >7</th>\n",
       "      <td id=\"T_0dfe4_row7_col0\" class=\"data row7 col0\" >Regulation of insulin secretion - Reactome</td>\n",
       "      <td id=\"T_0dfe4_row7_col1\" class=\"data row7 col1\" >0.000018</td>\n",
       "      <td id=\"T_0dfe4_row7_col2\" class=\"data row7 col2\" >9</td>\n",
       "      <td id=\"T_0dfe4_row7_col3\" class=\"data row7 col3\" >22</td>\n",
       "      <td id=\"T_0dfe4_row7_col4\" class=\"data row7 col4\" >['set:G(q) alpha 11,14,15,Q', 'set:PL(C)D4:3xCa2+', 'set:PLC-beta 1/2/3', 'set:G-protein gamma subunit', 'set:PLCdegh', 'set:PLC beta1,2,3', 'set:PLC-beta', 'set:PLCbz', 'set:G-protein alpha (q/11/14/15)']</td>\n",
       "      <td id=\"T_0dfe4_row7_col5\" class=\"data row7 col5\" >http://model.geneontology.org/R-HSA-422356</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0dfe4_level0_row8\" class=\"row_heading level0 row8\" >8</th>\n",
       "      <td id=\"T_0dfe4_row8_col0\" class=\"data row8 col0\" >VEGFR2 mediated vascular permeability - Reactome</td>\n",
       "      <td id=\"T_0dfe4_row8_col1\" class=\"data row8 col1\" >0.000066</td>\n",
       "      <td id=\"T_0dfe4_row8_col2\" class=\"data row8 col2\" >8</td>\n",
       "      <td id=\"T_0dfe4_row8_col3\" class=\"data row8 col3\" >20</td>\n",
       "      <td id=\"T_0dfe4_row8_col4\" class=\"data row8 col4\" >['PIK3R1', 'PIK3CG', 'set:p-S,T-PAK1,2,3', 'set:G-protein gamma subunit', 'set:Activator:PI3K', 'set:PI3K-regulatory subunit', 'set:PI3K alpha, beta, gamma', 'set:PAK1,2,3 dimer']</td>\n",
       "      <td id=\"T_0dfe4_row8_col5\" class=\"data row8 col5\" >http://model.geneontology.org/R-HSA-5218920</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0dfe4_level0_row9\" class=\"row_heading level0 row9\" >9</th>\n",
       "      <td id=\"T_0dfe4_row9_col0\" class=\"data row9 col0\" >FCERI mediated Ca+2 mobilization - Reactome</td>\n",
       "      <td id=\"T_0dfe4_row9_col1\" class=\"data row9 col1\" >0.000087</td>\n",
       "      <td id=\"T_0dfe4_row9_col2\" class=\"data row9 col2\" >9</td>\n",
       "      <td id=\"T_0dfe4_row9_col3\" class=\"data row9 col3\" >26</td>\n",
       "      <td id=\"T_0dfe4_row9_col4\" class=\"data row9 col4\" >['set:G(q) alpha 11,14,15,Q', 'set:PL(C)D4:3xCa2+', 'set:PLC-beta 1/2/3', 'set:G-protein gamma subunit', 'set:PLCdegh', 'set:PLC beta1,2,3', 'set:PLC-beta', 'set:PLCbz', 'set:G-protein alpha (q/11/14/15)']</td>\n",
       "      <td id=\"T_0dfe4_row9_col5\" class=\"data row9 col5\" >http://model.geneontology.org/R-HSA-2871809</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0dfe4_level0_row10\" class=\"row_heading level0 row10\" >10</th>\n",
       "      <td id=\"T_0dfe4_row10_col0\" class=\"data row10 col0\" >Ca2+ pathway - Reactome</td>\n",
       "      <td id=\"T_0dfe4_row10_col1\" class=\"data row10 col1\" >0.000121</td>\n",
       "      <td id=\"T_0dfe4_row10_col2\" class=\"data row10 col2\" >9</td>\n",
       "      <td id=\"T_0dfe4_row10_col3\" class=\"data row10 col3\" >27</td>\n",
       "      <td id=\"T_0dfe4_row10_col4\" class=\"data row10 col4\" >['set:G(q) alpha 11,14,15,Q', 'set:PL(C)D4:3xCa2+', 'set:PLC-beta 1/2/3', 'set:G-protein gamma subunit', 'set:PLCdegh', 'set:PLC beta1,2,3', 'set:PLC-beta', 'set:PLCbz', 'set:G-protein alpha (q/11/14/15)']</td>\n",
       "      <td id=\"T_0dfe4_row10_col5\" class=\"data row10 col5\" >http://model.geneontology.org/R-HSA-4086398</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0dfe4_level0_row11\" class=\"row_heading level0 row11\" >11</th>\n",
       "      <td id=\"T_0dfe4_row11_col0\" class=\"data row11 col0\" >Synthesis of very long-chain fatty acyl-CoAs - Reactome</td>\n",
       "      <td id=\"T_0dfe4_row11_col1\" class=\"data row11 col1\" >0.000129</td>\n",
       "      <td id=\"T_0dfe4_row11_col2\" class=\"data row11 col2\" >6</td>\n",
       "      <td id=\"T_0dfe4_row11_col3\" class=\"data row11 col3\" >12</td>\n",
       "      <td id=\"T_0dfe4_row11_col4\" class=\"data row11 col4\" >['ELOVL7', 'set:ACSBG1,2', 'set:PTPLs', 'set:ACSL3,4', 'set:HSD17B3,12', 'set:ELOVL3,6,7']</td>\n",
       "      <td id=\"T_0dfe4_row11_col5\" class=\"data row11 col5\" >http://model.geneontology.org/R-HSA-75876</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0dfe4_level0_row12\" class=\"row_heading level0 row12\" >12</th>\n",
       "      <td id=\"T_0dfe4_row12_col0\" class=\"data row12 col0\" >Synthesis of IP2, IP, and Ins in the cytosol - Reactome</td>\n",
       "      <td id=\"T_0dfe4_row12_col1\" class=\"data row12 col1\" >0.000156</td>\n",
       "      <td id=\"T_0dfe4_row12_col2\" class=\"data row12 col2\" >7</td>\n",
       "      <td id=\"T_0dfe4_row12_col3\" class=\"data row12 col3\" >17</td>\n",
       "      <td id=\"T_0dfe4_row12_col4\" class=\"data row12 col4\" >['set:PL(C)D4:3xCa2+', 'set:Inositol polyphosphate 5-phosphatase', 'set:PLCdegh', 'set:INPP5(3)/ITPK1', 'set:IMPA1/2', 'set:INPP5(4)', 'set:PLCbz']</td>\n",
       "      <td id=\"T_0dfe4_row12_col5\" class=\"data row12 col5\" >http://model.geneontology.org/R-HSA-1855183</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0dfe4_level0_row13\" class=\"row_heading level0 row13\" >13</th>\n",
       "      <td id=\"T_0dfe4_row13_col0\" class=\"data row13 col0\" >G beta:gamma signalling through PI3Kgamma - Reactome</td>\n",
       "      <td id=\"T_0dfe4_row13_col1\" class=\"data row13 col1\" >0.000495</td>\n",
       "      <td id=\"T_0dfe4_row13_col2\" class=\"data row13 col2\" >5</td>\n",
       "      <td id=\"T_0dfe4_row13_col3\" class=\"data row13 col3\" >10</td>\n",
       "      <td id=\"T_0dfe4_row13_col4\" class=\"data row13 col4\" >['PIK3R1', 'PIK3CG', 'set:G-protein gamma subunit', 'set:PI3K-regulatory subunit', 'set:PI3K alpha, beta, gamma']</td>\n",
       "      <td id=\"T_0dfe4_row13_col5\" class=\"data row13 col5\" >http://model.geneontology.org/R-HSA-392451</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0dfe4_level0_row14\" class=\"row_heading level0 row14\" >14</th>\n",
       "      <td id=\"T_0dfe4_row14_col0\" class=\"data row14 col0\" >PI3K/AKT Signaling - Reactome</td>\n",
       "      <td id=\"T_0dfe4_row14_col1\" class=\"data row14 col1\" >0.000570</td>\n",
       "      <td id=\"T_0dfe4_row14_col2\" class=\"data row14 col2\" >6</td>\n",
       "      <td id=\"T_0dfe4_row14_col3\" class=\"data row14 col3\" >15</td>\n",
       "      <td id=\"T_0dfe4_row14_col4\" class=\"data row14 col4\" >['PIK3R1', 'PIK3CG', 'set:G-protein gamma subunit', 'set:Activator:PI3K', 'set:PI3K-regulatory subunit', 'set:PI3K alpha, beta, gamma']</td>\n",
       "      <td id=\"T_0dfe4_row14_col5\" class=\"data row14 col5\" >http://model.geneontology.org/R-HSA-1257604</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x14e649990>"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "platelets_down_set = enrich.enrich_wrapper('platelets_down.csv','Gene Symbol',method='set',FDR = 0.05,fpath = '../test_data/processed/',table_number = 52)\n",
    "platelets_down_set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "ad16f6e1-2718-4d43-93d9-2173c4e5f057",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 436/436 [00:00<00:00, 16395.90it/s]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_95f2d caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_95f2d\">\n",
       "  <caption>Table S53: platelets down, standard method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_95f2d_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_95f2d_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_95f2d_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_95f2d_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_95f2d_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_95f2d_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_95f2d_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_95f2d_row0_col0\" class=\"data row0 col0\" >RHOA GTPase cycle - Reactome</td>\n",
       "      <td id=\"T_95f2d_row0_col1\" class=\"data row0 col1\" >0.000007</td>\n",
       "      <td id=\"T_95f2d_row0_col2\" class=\"data row0 col2\" >16</td>\n",
       "      <td id=\"T_95f2d_row0_col3\" class=\"data row0 col3\" >87</td>\n",
       "      <td id=\"T_95f2d_row0_col4\" class=\"data row0 col4\" >['PIK3R1', 'ARHGAP5', 'ARHGAP21', 'OPHN1', 'ARHGEF12', 'ARHGEF3', 'ARHGAP29', 'ARHGAP31', 'STARD8', 'ABR', 'TIAM1', 'PREX2', 'FAM13A', 'ARHGEF15', 'ARHGEF10L', 'ARHGAP24']</td>\n",
       "      <td id=\"T_95f2d_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-8980692</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_95f2d_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_95f2d_row1_col0\" class=\"data row1 col0\" >CDC42 GTPase cycle - Reactome</td>\n",
       "      <td id=\"T_95f2d_row1_col1\" class=\"data row1 col1\" >0.000011</td>\n",
       "      <td id=\"T_95f2d_row1_col2\" class=\"data row1 col2\" >16</td>\n",
       "      <td id=\"T_95f2d_row1_col3\" class=\"data row1 col3\" >90</td>\n",
       "      <td id=\"T_95f2d_row1_col4\" class=\"data row1 col4\" >['PIK3R1', 'ARHGAP5', 'ARHGAP21', 'GNA13', 'OPHN1', 'ARHGEF12', 'ARHGAP29', 'DOCK9', 'ARHGAP31', 'STARD8', 'ABR', 'TIAM1', 'PREX2', 'ARHGEF15', 'PLEKHG1', 'ARHGAP24']</td>\n",
       "      <td id=\"T_95f2d_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-9013148</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_95f2d_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "      <td id=\"T_95f2d_row2_col0\" class=\"data row2 col0\" >RAC1 GTPase cycle - Reactome</td>\n",
       "      <td id=\"T_95f2d_row2_col1\" class=\"data row2 col1\" >0.000044</td>\n",
       "      <td id=\"T_95f2d_row2_col2\" class=\"data row2 col2\" >16</td>\n",
       "      <td id=\"T_95f2d_row2_col3\" class=\"data row2 col3\" >100</td>\n",
       "      <td id=\"T_95f2d_row2_col4\" class=\"data row2 col4\" >['PIK3R1', 'ARHGAP5', 'ARHGAP21', 'GNA13', 'OPHN1', 'ARHGAP29', 'FGD5', 'DOCK9', 'ARHGAP31', 'ABR', 'TIAM1', 'PREX2', 'FAM13A', 'ARHGEF15', 'PLEKHG1', 'ARHGAP24']</td>\n",
       "      <td id=\"T_95f2d_row2_col5\" class=\"data row2 col5\" >http://model.geneontology.org/R-HSA-9013149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_95f2d_level0_row3\" class=\"row_heading level0 row3\" >3</th>\n",
       "      <td id=\"T_95f2d_row3_col0\" class=\"data row3 col0\" >RHOB GTPase cycle - Reactome</td>\n",
       "      <td id=\"T_95f2d_row3_col1\" class=\"data row3 col1\" >0.000079</td>\n",
       "      <td id=\"T_95f2d_row3_col2\" class=\"data row3 col2\" >9</td>\n",
       "      <td id=\"T_95f2d_row3_col3\" class=\"data row3 col3\" >37</td>\n",
       "      <td id=\"T_95f2d_row3_col4\" class=\"data row3 col4\" >['PIK3R1', 'ARHGAP5', 'ARHGAP21', 'OPHN1', 'ARHGEF12', 'ARHGEF3', 'STARD8', 'ABR', 'ARHGEF10L']</td>\n",
       "      <td id=\"T_95f2d_row3_col5\" class=\"data row3 col5\" >http://model.geneontology.org/R-HSA-9013026</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x12fd7a2c0>"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "platelets_down_standard = enrich.enrich_wrapper('platelets_down.csv','Gene Symbol',method='standard',FDR = 0.05,fpath = '../test_data/processed/',table_number = 53)\n",
    "platelets_down_standard"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "e4c4fad5-e804-42f2-8ec7-74b5a1ba99e5",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|█████████████████████████████████████████| 436/436 [02:21<00:00,  3.08it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 309 entities from 267 out of 1088 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_8a8ad caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_8a8ad\">\n",
       "  <caption>Table S54: platelets down, weighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_8a8ad_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_8a8ad_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_8a8ad_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_8a8ad_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_8a8ad_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_8a8ad_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_8a8ad_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_8a8ad_row0_col0\" class=\"data row0 col0\" >Role of phospholipids in phagocytosis - Reactome</td>\n",
       "      <td id=\"T_8a8ad_row0_col1\" class=\"data row0 col1\" >0.000010</td>\n",
       "      <td id=\"T_8a8ad_row0_col2\" class=\"data row0 col2\" >11</td>\n",
       "      <td id=\"T_8a8ad_row0_col3\" class=\"data row0 col3\" >22</td>\n",
       "      <td id=\"T_8a8ad_row0_col4\" class=\"data row0 col4\" >['set:G(q) alpha 11,14,15,Q', 'set:PL(C)D4:3xCa2+', 'set:PLC-beta 1/2/3', 'set:G-protein gamma subunit', 'set:PI3K-regulatory subunit', 'set:PLCdegh', 'set:PLD', 'set:PLC beta1,2,3', 'set:PLC-beta', 'set:PLCbz', 'set:G-protein alpha (q/11/14/15)']</td>\n",
       "      <td id=\"T_8a8ad_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-2029485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_8a8ad_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_8a8ad_row1_col0\" class=\"data row1 col0\" >DAG and IP3 signaling - Reactome</td>\n",
       "      <td id=\"T_8a8ad_row1_col1\" class=\"data row1 col1\" >0.000019</td>\n",
       "      <td id=\"T_8a8ad_row1_col2\" class=\"data row1 col2\" >10</td>\n",
       "      <td id=\"T_8a8ad_row1_col3\" class=\"data row1 col3\" >18</td>\n",
       "      <td id=\"T_8a8ad_row1_col4\" class=\"data row1 col4\" >['set:G(q) alpha 11,14,15,Q', 'set:PL(C)D4:3xCa2+', 'set:PLC-beta 1/2/3', 'set:G-protein gamma subunit', 'set:PI3K-regulatory subunit', 'set:PLCdegh', 'set:PLC beta1,2,3', 'set:PLC-beta', 'set:PLCbz', 'set:G-protein alpha (q/11/14/15)']</td>\n",
       "      <td id=\"T_8a8ad_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-1489509</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_8a8ad_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "      <td id=\"T_8a8ad_row2_col0\" class=\"data row2 col0\" >VEGFR2 mediated cell proliferation - Reactome</td>\n",
       "      <td id=\"T_8a8ad_row2_col1\" class=\"data row2 col1\" >0.000046</td>\n",
       "      <td id=\"T_8a8ad_row2_col2\" class=\"data row2 col2\" >11</td>\n",
       "      <td id=\"T_8a8ad_row2_col3\" class=\"data row2 col3\" >27</td>\n",
       "      <td id=\"T_8a8ad_row2_col4\" class=\"data row2 col4\" >['KDR', 'set:G(q) alpha 11,14,15,Q', 'set:PL(C)D4:3xCa2+', 'set:PLC-beta 1/2/3', 'set:G-protein gamma subunit', 'set:PI3K-regulatory subunit', 'set:PLCdegh', 'set:PLC beta1,2,3', 'set:PLC-beta', 'set:PLCbz', 'set:G-protein alpha (q/11/14/15)']</td>\n",
       "      <td id=\"T_8a8ad_row2_col5\" class=\"data row2 col5\" >http://model.geneontology.org/R-HSA-5218921</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_8a8ad_level0_row3\" class=\"row_heading level0 row3\" >3</th>\n",
       "      <td id=\"T_8a8ad_row3_col0\" class=\"data row3 col0\" >PLC beta mediated events - Reactome</td>\n",
       "      <td id=\"T_8a8ad_row3_col1\" class=\"data row3 col1\" >0.000050</td>\n",
       "      <td id=\"T_8a8ad_row3_col2\" class=\"data row3 col2\" >9</td>\n",
       "      <td id=\"T_8a8ad_row3_col3\" class=\"data row3 col3\" >15</td>\n",
       "      <td id=\"T_8a8ad_row3_col4\" class=\"data row3 col4\" >['set:G(q) alpha 11,14,15,Q', 'set:PL(C)D4:3xCa2+', 'set:PLC-beta 1/2/3', 'set:G-protein gamma subunit', 'set:PLCdegh', 'set:PLC beta1,2,3', 'set:PLC-beta', 'set:PLCbz', 'set:G-protein alpha (q/11/14/15)']</td>\n",
       "      <td id=\"T_8a8ad_row3_col5\" class=\"data row3 col5\" >http://model.geneontology.org/R-HSA-112043</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_8a8ad_level0_row4\" class=\"row_heading level0 row4\" >4</th>\n",
       "      <td id=\"T_8a8ad_row4_col0\" class=\"data row4 col0\" >Antigen activates B Cell Receptor (BCR) leading to generation of second messengers - Reactome</td>\n",
       "      <td id=\"T_8a8ad_row4_col1\" class=\"data row4 col1\" >0.000071</td>\n",
       "      <td id=\"T_8a8ad_row4_col2\" class=\"data row4 col2\" >11</td>\n",
       "      <td id=\"T_8a8ad_row4_col3\" class=\"data row4 col3\" >25</td>\n",
       "      <td id=\"T_8a8ad_row4_col4\" class=\"data row4 col4\" >['PIK3R1', 'set:G(q) alpha 11,14,15,Q', 'set:PL(C)D4:3xCa2+', 'set:PLC-beta 1/2/3', 'set:G-protein gamma subunit', 'set:PLCdegh', 'set:PLC beta1,2,3', 'set:PLC-beta', 'set:PLCbz', 'set:G-protein alpha (q/11/14/15)', 'set:LYN, p-SYK']</td>\n",
       "      <td id=\"T_8a8ad_row4_col5\" class=\"data row4 col5\" >http://model.geneontology.org/R-HSA-983695</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x12fd7a5f0>"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "platelets_down_weighted = enrich.enrich_wrapper('platelets_down.csv','Gene Symbol',method='ncHGT',FDR = 0.05,fpath = '../test_data/processed/',table_number = 54)\n",
    "platelets_down_weighted"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d0707d19-67fc-4e00-9759-d5c281b2c89a",
   "metadata": {
    "tags": []
   },
   "source": [
    "# Non Alcoholic Fatty Liver Disease\n",
    "\n",
    "Govaere et al., 2023\n",
    "\n",
    "https://www.nature.com/articles/s42255-023-00775-1\n",
    "\n",
    "NAFLD. Paired transcriptomics and proteomics in 306 patients to find signatures of active steatohepatitis and severe fibrosis.\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "804e2df1-93c6-426a-aecc-30c7c160bf97",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|████████████████████████████████████████| 98/98 [00:00<00:00, 15363.18it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 68 entities from 48 out of 118 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_7fc0f caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_7fc0f\">\n",
       "  <caption>Table S55: NAFLD combined up/downregulated genes, unweighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_7fc0f_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_7fc0f_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_7fc0f_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_7fc0f_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_7fc0f_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_7fc0f_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_7fc0f_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_7fc0f_row0_col0\" class=\"data row0 col0\" >Activation of Matrix Metalloproteinases - Reactome</td>\n",
       "      <td id=\"T_7fc0f_row0_col1\" class=\"data row0 col1\" >0.000000</td>\n",
       "      <td id=\"T_7fc0f_row0_col2\" class=\"data row0 col2\" >7</td>\n",
       "      <td id=\"T_7fc0f_row0_col3\" class=\"data row0 col3\" >18</td>\n",
       "      <td id=\"T_7fc0f_row0_col4\" class=\"data row0 col4\" >['set:proMMP3 initial activators', 'set:proMMP10 activators', 'set:proMMP9 activating proteases', 'set:Trypsin, plasmin', 'set:proMMP8 initial activators', 'set:proMMP1 initial activators', 'set:MMP7 initial activators']</td>\n",
       "      <td id=\"T_7fc0f_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-1592389</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_7fc0f_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_7fc0f_row1_col0\" class=\"data row1 col0\" >Cytosolic sulfonation of small molecules - Reactome</td>\n",
       "      <td id=\"T_7fc0f_row1_col1\" class=\"data row1 col1\" >0.000012</td>\n",
       "      <td id=\"T_7fc0f_row1_col2\" class=\"data row1 col2\" >5</td>\n",
       "      <td id=\"T_7fc0f_row1_col3\" class=\"data row1 col3\" >19</td>\n",
       "      <td id=\"T_7fc0f_row1_col4\" class=\"data row1 col4\" >['SULT2A1', 'set:SULTs active on DHEA', 'set:SULT dimers (T2)', 'set:SULT dimers (T3)', 'set:SULT1E1,2A1']</td>\n",
       "      <td id=\"T_7fc0f_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-156584</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x12fd79630>"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "GS2_step= enrich.enrich_wrapper('Goavere_S2.csv','Gene Symbol',method='set',FDR = 0.05,fpath = '../test_data/processed/', table_number = 55)\n",
    "GS2_step"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "f206c197-c8ad-4aed-b254-4fa87769aaae",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|████████████████████████████████████████| 98/98 [00:00<00:00, 16556.91it/s]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_6281d caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_6281d\">\n",
       "  <caption>Table S56: NAFLD combined up/downregulated genes, standard method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_6281d_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_6281d_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_6281d_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_6281d_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_6281d_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_6281d_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x14f5f3ac0>"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "GS2_standard = enrich.enrich_wrapper('Goavere_S2.csv','Gene Symbol',method='standard',FDR = 0.05,fpath = '../test_data/processed/', table_number = 56)\n",
    "GS2_standard"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "a8959fee-e78a-41db-98be-ec5c22afc40d",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|███████████████████████████████████████████| 98/98 [00:01<00:00, 77.22it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 68 entities from 48 out of 118 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_3c3fa caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_3c3fa\">\n",
       "  <caption>Table S57: NAFLD combined up/downregulated genes, weighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_3c3fa_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_3c3fa_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_3c3fa_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_3c3fa_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_3c3fa_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_3c3fa_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_3c3fa_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_3c3fa_row0_col0\" class=\"data row0 col0\" >Activation of Matrix Metalloproteinases - Reactome</td>\n",
       "      <td id=\"T_3c3fa_row0_col1\" class=\"data row0 col1\" >0.000000</td>\n",
       "      <td id=\"T_3c3fa_row0_col2\" class=\"data row0 col2\" >7</td>\n",
       "      <td id=\"T_3c3fa_row0_col3\" class=\"data row0 col3\" >18</td>\n",
       "      <td id=\"T_3c3fa_row0_col4\" class=\"data row0 col4\" >['set:proMMP3 initial activators', 'set:proMMP10 activators', 'set:proMMP9 activating proteases', 'set:Trypsin, plasmin', 'set:proMMP8 initial activators', 'set:proMMP1 initial activators', 'set:MMP7 initial activators']</td>\n",
       "      <td id=\"T_3c3fa_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-1592389</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x14f5f1fc0>"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "GS2_weighted = enrich.enrich_wrapper('Goavere_S2.csv','Gene Symbol',method='ncHGT',FDR = 0.05,fpath = '../test_data/processed/', table_number = 57)\n",
    "GS2_weighted"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "500cb76b-fbe8-438f-93c1-40aea3334658",
   "metadata": {
    "tags": []
   },
   "source": [
    "# Astrocytes in Aging Mouse Brains across 4 brain regions\n",
    "\n",
    "Boisvert et al., 2018\n",
    "\n",
    "https://www.sciencedirect.com/science/article/pii/S221112471731848X?via%3Dihub#app3\n",
    "\n",
    "RNA seq on astrocytes isolated from different brain regions in 4 month and 2 year old mice. Most brain regions had ~200 up and ~200 downregulated genes, but most of those were also differentially expressed between brain regions.\n",
    "\n",
    "In fig 2B, it appears that cerebellum and hypothalamus are the best candidates, as the 3 biological replicates (3 mice) cluster together by region and age. The hypothalamus looks a little cleaner than cerebellum."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e91b0ea8-184b-42c7-be3e-bfb91e3ff8ac",
   "metadata": {},
   "source": [
    "The paper says they filtered based on \"FPKM > 1, Adj P value < 0.05, and pulldown/input > 0.75\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cd618b67-bda7-401f-85e9-53d11ba04f4f",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Hypothalamus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "9c72398e-d82d-4a9a-b36a-3ed3c3b087d8",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 246/246 [00:00<00:00, 16416.32it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 139 entities from 115 out of 416 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_52c2b caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_52c2b\">\n",
       "  <caption>Table S58: astroctyes hypothalamus combined up/downregulated genes, unweighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_52c2b_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_52c2b_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_52c2b_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_52c2b_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_52c2b_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_52c2b_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_52c2b_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_52c2b_row0_col0\" class=\"data row0 col0\" >Cholesterol biosynthesis - Reactome</td>\n",
       "      <td id=\"T_52c2b_row0_col1\" class=\"data row0 col1\" >0.000000</td>\n",
       "      <td id=\"T_52c2b_row0_col2\" class=\"data row0 col2\" >10</td>\n",
       "      <td id=\"T_52c2b_row0_col3\" class=\"data row0 col3\" >22</td>\n",
       "      <td id=\"T_52c2b_row0_col4\" class=\"data row0 col4\" >['HMGCS1', 'HSD17B7', 'SQLE', 'MSMO1', 'DHCR24', 'MVD', 'ACLY', 'FDFT1', 'LSS', 'CYP51A1']</td>\n",
       "      <td id=\"T_52c2b_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-191273</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x14f5f0e80>"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "astro_HTH_set = enrich.enrich_wrapper('astro_HTH_comb.csv','Gene Symbol',method='set',FDR = 0.05,fpath = '../test_data/processed/',table_number = 58)\n",
    "astro_HTH_set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "8f9ec97e-d4a4-4a3c-8240-04ee9f6284ca",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 246/246 [00:00<00:00, 16873.52it/s]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_a32f6 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_a32f6\">\n",
       "  <caption>Table S59: astroctyes hypothalamus combined up/downregulated genes, standard method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_a32f6_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_a32f6_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_a32f6_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_a32f6_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_a32f6_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_a32f6_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_a32f6_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_a32f6_row0_col0\" class=\"data row0 col0\" >Cholesterol biosynthesis - Reactome</td>\n",
       "      <td id=\"T_a32f6_row0_col1\" class=\"data row0 col1\" >0.000000</td>\n",
       "      <td id=\"T_a32f6_row0_col2\" class=\"data row0 col2\" >10</td>\n",
       "      <td id=\"T_a32f6_row0_col3\" class=\"data row0 col3\" >24</td>\n",
       "      <td id=\"T_a32f6_row0_col4\" class=\"data row0 col4\" >['HMGCS1', 'HSD17B7', 'SQLE', 'MSMO1', 'DHCR24', 'MVD', 'ACLY', 'FDFT1', 'LSS', 'CYP51A1']</td>\n",
       "      <td id=\"T_a32f6_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-191273</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a32f6_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_a32f6_row1_col0\" class=\"data row1 col0\" >Terminal pathway of complement - Reactome</td>\n",
       "      <td id=\"T_a32f6_row1_col1\" class=\"data row1 col1\" >0.000001</td>\n",
       "      <td id=\"T_a32f6_row1_col2\" class=\"data row1 col2\" >4</td>\n",
       "      <td id=\"T_a32f6_row1_col3\" class=\"data row1 col3\" >5</td>\n",
       "      <td id=\"T_a32f6_row1_col4\" class=\"data row1 col4\" >['C3', 'C4B', 'C4B', 'C2']</td>\n",
       "      <td id=\"T_a32f6_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-166665</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a32f6_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "      <td id=\"T_a32f6_row2_col0\" class=\"data row2 col0\" >Peptide ligand-binding receptors - Reactome</td>\n",
       "      <td id=\"T_a32f6_row2_col1\" class=\"data row2 col1\" >0.000071</td>\n",
       "      <td id=\"T_a32f6_row2_col2\" class=\"data row2 col2\" >4</td>\n",
       "      <td id=\"T_a32f6_row2_col3\" class=\"data row2 col3\" >11</td>\n",
       "      <td id=\"T_a32f6_row2_col4\" class=\"data row2 col4\" >['C3', 'C4B', 'C4B', 'C2']</td>\n",
       "      <td id=\"T_a32f6_row2_col5\" class=\"data row2 col5\" >http://model.geneontology.org/R-HSA-375276</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a32f6_level0_row3\" class=\"row_heading level0 row3\" >3</th>\n",
       "      <td id=\"T_a32f6_row3_col0\" class=\"data row3 col0\" >Kandutsch-Russell pathway - Reactome</td>\n",
       "      <td id=\"T_a32f6_row3_col1\" class=\"data row3 col1\" >0.000107</td>\n",
       "      <td id=\"T_a32f6_row3_col2\" class=\"data row3 col2\" >3</td>\n",
       "      <td id=\"T_a32f6_row3_col3\" class=\"data row3 col3\" >5</td>\n",
       "      <td id=\"T_a32f6_row3_col4\" class=\"data row3 col4\" >['HSD17B7', 'DHCR7', 'DHCR24']</td>\n",
       "      <td id=\"T_a32f6_row3_col5\" class=\"data row3 col5\" >http://model.geneontology.org/R-HSA-6807062</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a32f6_level0_row4\" class=\"row_heading level0 row4\" >4</th>\n",
       "      <td id=\"T_a32f6_row4_col0\" class=\"data row4 col0\" >Bloch pathway - Reactome</td>\n",
       "      <td id=\"T_a32f6_row4_col1\" class=\"data row4 col1\" >0.000107</td>\n",
       "      <td id=\"T_a32f6_row4_col2\" class=\"data row4 col2\" >3</td>\n",
       "      <td id=\"T_a32f6_row4_col3\" class=\"data row4 col3\" >5</td>\n",
       "      <td id=\"T_a32f6_row4_col4\" class=\"data row4 col4\" >['HSD17B7', 'DHCR7', 'DHCR24']</td>\n",
       "      <td id=\"T_a32f6_row4_col5\" class=\"data row4 col5\" >http://model.geneontology.org/R-HSA-6807047</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x14f5f0250>"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "astro_HTH_standard = enrich.enrich_wrapper('astro_HTH_comb.csv','Gene Symbol',method='standard',FDR = 0.05,fpath = '../test_data/processed/',table_number = 59)\n",
    "astro_HTH_standard"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b7a22ab9-91c7-42ea-a162-d2d6b988cdc6",
   "metadata": {},
   "source": [
    "**Note on C4B:** \n",
    "\n",
    "C4B is listed twice, but that is due to  P0C0L4 and P0C0L5 both being officially listed as uniprot accession IDs for C4B, which affected our results when we used SimpleMine. \n",
    "\n",
    "P0C0L4 seems to actually correspond to C4A (C4A and C4B are paralogs). However, C4b can be derived from either the C4A gene product or the C4B gene product, and so both genes are listed as members of a set, and both are annotated to the pathway. GO-CAMs use uniprot accession IDs, so P0C0L4 and P0C0L5 are listed in the model for enrichment. When the input list of genes was mapped from gene symbols to uniprot accession IDs, both P0C0L4 and P0C0L5 were hits for the GOCAM. \n",
    "\n",
    "Why both accession IDs are officially associated with C4B is unclear, since they correspond to C4A and C4B on UniProt's website. We believe the issue is that the genes are cross listed as aliases for one another, which led to both uniprot accession ID's being associated with C4B."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "175a1e52-472f-4365-b472-e751aded6f27",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|█████████████████████████████████████████| 246/246 [00:32<00:00,  7.55it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 139 entities from 115 out of 416 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_85f53 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_85f53\">\n",
       "  <caption>Table S60: astroctyes hypothalamus combined up/downregulated genes, weighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_85f53_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_85f53_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_85f53_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_85f53_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_85f53_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_85f53_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_85f53_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_85f53_row0_col0\" class=\"data row0 col0\" >Cholesterol biosynthesis - Reactome</td>\n",
       "      <td id=\"T_85f53_row0_col1\" class=\"data row0 col1\" >0.000000</td>\n",
       "      <td id=\"T_85f53_row0_col2\" class=\"data row0 col2\" >10</td>\n",
       "      <td id=\"T_85f53_row0_col3\" class=\"data row0 col3\" >22</td>\n",
       "      <td id=\"T_85f53_row0_col4\" class=\"data row0 col4\" >['HMGCS1', 'HSD17B7', 'SQLE', 'MSMO1', 'DHCR24', 'MVD', 'ACLY', 'FDFT1', 'LSS', 'CYP51A1']</td>\n",
       "      <td id=\"T_85f53_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-191273</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x12fd7b8b0>"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "astro_HTH_weighted = enrich.enrich_wrapper('astro_HTH_comb.csv','Gene Symbol',method='ncHGT',FDR = 0.05,fpath = '../test_data/processed/',table_number = 60)\n",
    "astro_HTH_weighted"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a8e6cd4e-06c4-4598-a5f8-fad3bfba49d7",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Cerebellum"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "7b9816d9-8f5e-49bc-8540-01defec9848c",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 310/310 [00:00<00:00, 17650.88it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 200 entities from 172 out of 605 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_0461c caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_0461c\">\n",
       "  <caption>Table S61: astroctyes cerebellum combined up/downregulated genes, unweighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_0461c_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_0461c_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_0461c_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_0461c_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_0461c_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_0461c_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x14e3e38e0>"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "astro_CB_set = enrich.enrich_wrapper('astro_CB_comb.csv','Gene Symbol',method='set',FDR = 0.05,fpath = '../test_data/processed/',table_number = 61)\n",
    "astro_CB_set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "65913ea3-8c25-43d2-825b-16be7c4cbb0a",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 310/310 [00:00<00:00, 16538.63it/s]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_e34d7 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_e34d7\">\n",
       "  <caption>Table S62: astroctyes cerebellum combined up/downregulated genes, standard method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_e34d7_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_e34d7_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_e34d7_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_e34d7_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_e34d7_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_e34d7_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x14f5f2350>"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "astro_CB_standard = enrich.enrich_wrapper('astro_CB_comb.csv','Gene Symbol',method='standard',FDR = 0.05,fpath = '../test_data/processed/',table_number = 62)\n",
    "astro_CB_standard"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "750e2ee7-084e-41d4-ba2c-993914468415",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|█████████████████████████████████████████| 310/310 [00:30<00:00, 10.19it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 200 entities from 172 out of 605 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_2b1dc caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_2b1dc\">\n",
       "  <caption>Table S63: astroctyes cerebellum combined up/downregulated genes, weighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_2b1dc_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_2b1dc_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_2b1dc_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_2b1dc_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_2b1dc_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_2b1dc_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x14f5f0dc0>"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "astro_CB_weighted = enrich.enrich_wrapper('astro_CB_comb.csv','Gene Symbol',method='ncHGT',FDR = 0.05,fpath = '../test_data/processed/',table_number = 63)\n",
    "astro_CB_weighted"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f0bf5420-b073-49b2-95eb-0a817f980b41",
   "metadata": {
    "tags": []
   },
   "source": [
    "# Macrophage activation\n",
    "\n",
    "Orrechioni et al., 2019\n",
    "\n",
    "https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6543837/pdf/fimmu-10-01084.pdf\n",
    "\n",
    "In vitro classically activated vs alternatively activated macrophpages"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "3c42237d-e4fd-4117-bc70-de05cf894176",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 667/667 [00:00<00:00, 16768.07it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 619 entities from 511 out of 1519 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_39934 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_39934\">\n",
       "  <caption>Table S64: macrophages combined up/downregulated genes, unweighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_39934_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_39934_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_39934_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_39934_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_39934_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_39934_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_39934_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_39934_row0_col0\" class=\"data row0 col0\" >Resolution of Sister Chromatid Cohesion - Reactome</td>\n",
       "      <td id=\"T_39934_row0_col1\" class=\"data row0 col1\" >0.000057</td>\n",
       "      <td id=\"T_39934_row0_col2\" class=\"data row0 col2\" >30</td>\n",
       "      <td id=\"T_39934_row0_col3\" class=\"data row0 col3\" >95</td>\n",
       "      <td id=\"T_39934_row0_col4\" class=\"data row0 col4\" >['BIRC5', 'CDC20', 'KIF2C', 'CENPF', 'CENPA', 'NUF2', 'ZWILCH', 'AURKB', 'CENPN', 'SPC24', 'KNTC1', 'CENPK', 'PLK1', 'SKA1', 'CDCA5', 'CENPI', 'BUB1', 'BUB1B', 'CENPP', 'CENPH', 'CDCA8', 'CENPT', 'CDK1', 'NDC80', 'ERCC6L', 'SPC25', 'CENPE', 'CENPM', 'MAD2L1', 'set:CCNB1,CCNB2:p-T161-CDK1']</td>\n",
       "      <td id=\"T_39934_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-2500257</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_39934_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_39934_row1_col0\" class=\"data row1 col0\" >Interleukin-35 Signalling - Reactome</td>\n",
       "      <td id=\"T_39934_row1_col1\" class=\"data row1 col1\" >0.000057</td>\n",
       "      <td id=\"T_39934_row1_col2\" class=\"data row1 col2\" >7</td>\n",
       "      <td id=\"T_39934_row1_col3\" class=\"data row1 col3\" >9</td>\n",
       "      <td id=\"T_39934_row1_col4\" class=\"data row1 col4\" >['IL6ST', 'JAK2', 'STAT4', 'EBI3', 'STAT1', 'IL12A', 'set:JAK1, JAK2, (TYK2)']</td>\n",
       "      <td id=\"T_39934_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-8984722</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_39934_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "      <td id=\"T_39934_row2_col0\" class=\"data row2 col0\" >Unwinding of DNA - Reactome</td>\n",
       "      <td id=\"T_39934_row2_col1\" class=\"data row2 col1\" >0.000084</td>\n",
       "      <td id=\"T_39934_row2_col2\" class=\"data row2 col2\" >6</td>\n",
       "      <td id=\"T_39934_row2_col3\" class=\"data row2 col3\" >7</td>\n",
       "      <td id=\"T_39934_row2_col4\" class=\"data row2 col4\" >['MCM5', 'MCM7', 'MCM2', 'MCM3', 'MCM4', 'MCM6']</td>\n",
       "      <td id=\"T_39934_row2_col5\" class=\"data row2 col5\" >http://model.geneontology.org/R-HSA-176974</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x14f5f33a0>"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mac_set = enrich.enrich_wrapper('mac_comb.csv','Gene Symbol',method='set',FDR = 0.05,fpath = '../test_data/processed/',table_number = 64)\n",
    "mac_set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "f3766949-03c2-46a0-a7a9-73363d56d7be",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████| 667/667 [00:00<00:00, 16580.14it/s]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_b8191 caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_b8191\">\n",
       "  <caption>Table S65: macrophages combined up/downregulated genes, standard method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_b8191_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_b8191_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_b8191_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_b8191_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_b8191_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_b8191_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_b8191_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_b8191_row0_col0\" class=\"data row0 col0\" >Resolution of Sister Chromatid Cohesion - Reactome</td>\n",
       "      <td id=\"T_b8191_row0_col1\" class=\"data row0 col1\" >0.000000</td>\n",
       "      <td id=\"T_b8191_row0_col2\" class=\"data row0 col2\" >30</td>\n",
       "      <td id=\"T_b8191_row0_col3\" class=\"data row0 col3\" >106</td>\n",
       "      <td id=\"T_b8191_row0_col4\" class=\"data row0 col4\" >['CCNB2', 'BIRC5', 'CDC20', 'KIF2C', 'CENPF', 'CENPA', 'NUF2', 'ZWILCH', 'AURKB', 'CENPN', 'SPC24', 'KNTC1', 'CENPK', 'PLK1', 'SKA1', 'CDCA5', 'CENPI', 'BUB1', 'BUB1B', 'CENPP', 'CENPH', 'CDCA8', 'CENPT', 'CDK1', 'NDC80', 'ERCC6L', 'SPC25', 'CENPE', 'CENPM', 'MAD2L1']</td>\n",
       "      <td id=\"T_b8191_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-2500257</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_b8191_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_b8191_row1_col0\" class=\"data row1 col0\" >Unwinding of DNA - Reactome</td>\n",
       "      <td id=\"T_b8191_row1_col1\" class=\"data row1 col1\" >0.000005</td>\n",
       "      <td id=\"T_b8191_row1_col2\" class=\"data row1 col2\" >6</td>\n",
       "      <td id=\"T_b8191_row1_col3\" class=\"data row1 col3\" >7</td>\n",
       "      <td id=\"T_b8191_row1_col4\" class=\"data row1 col4\" >['MCM5', 'MCM7', 'MCM2', 'MCM3', 'MCM4', 'MCM6']</td>\n",
       "      <td id=\"T_b8191_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-176974</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_b8191_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "      <td id=\"T_b8191_row2_col0\" class=\"data row2 col0\" >Interleukin-35 Signalling - Reactome</td>\n",
       "      <td id=\"T_b8191_row2_col1\" class=\"data row2 col1\" >0.000055</td>\n",
       "      <td id=\"T_b8191_row2_col2\" class=\"data row2 col2\" >6</td>\n",
       "      <td id=\"T_b8191_row2_col3\" class=\"data row2 col3\" >9</td>\n",
       "      <td id=\"T_b8191_row2_col4\" class=\"data row2 col4\" >['IL6ST', 'JAK2', 'STAT4', 'EBI3', 'STAT1', 'IL12A']</td>\n",
       "      <td id=\"T_b8191_row2_col5\" class=\"data row2 col5\" >http://model.geneontology.org/R-HSA-8984722</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x14f5f1300>"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mac_standard = enrich.enrich_wrapper('mac_comb.csv','Gene Symbol',method='standard',FDR = 0.05,fpath = '../test_data/processed/',table_number = 65)\n",
    "mac_standard"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "4da775aa-aea4-4a23-85de-ad09f360afcf",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|█████████████████████████████████████████| 667/667 [03:22<00:00,  3.30it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis run on 619 entities from 511 out of 1519 input genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_d1bef caption {\n",
       "  caption-side: bottom;\n",
       "  font-size: 1.25em;\n",
       "  text-align: left;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_d1bef\">\n",
       "  <caption>Table S66: macrophages combined up/downregulated genes, weighted set method, FDR = 0.05.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_d1bef_level0_col0\" class=\"col_heading level0 col0\" >title</th>\n",
       "      <th id=\"T_d1bef_level0_col1\" class=\"col_heading level0 col1\" >pval (uncorrected)</th>\n",
       "      <th id=\"T_d1bef_level0_col2\" class=\"col_heading level0 col2\" ># entities in list</th>\n",
       "      <th id=\"T_d1bef_level0_col3\" class=\"col_heading level0 col3\" >#entities in model</th>\n",
       "      <th id=\"T_d1bef_level0_col4\" class=\"col_heading level0 col4\" >shared entities in gocam</th>\n",
       "      <th id=\"T_d1bef_level0_col5\" class=\"col_heading level0 col5\" >url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_d1bef_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_d1bef_row0_col0\" class=\"data row0 col0\" >Resolution of Sister Chromatid Cohesion - Reactome</td>\n",
       "      <td id=\"T_d1bef_row0_col1\" class=\"data row0 col1\" >0.000000</td>\n",
       "      <td id=\"T_d1bef_row0_col2\" class=\"data row0 col2\" >30</td>\n",
       "      <td id=\"T_d1bef_row0_col3\" class=\"data row0 col3\" >95</td>\n",
       "      <td id=\"T_d1bef_row0_col4\" class=\"data row0 col4\" >['BIRC5', 'CDC20', 'KIF2C', 'CENPF', 'CENPA', 'NUF2', 'ZWILCH', 'AURKB', 'CENPN', 'SPC24', 'KNTC1', 'CENPK', 'PLK1', 'SKA1', 'CDCA5', 'CENPI', 'BUB1', 'BUB1B', 'CENPP', 'CENPH', 'CDCA8', 'CENPT', 'CDK1', 'NDC80', 'ERCC6L', 'SPC25', 'CENPE', 'CENPM', 'MAD2L1', 'set:CCNB1,CCNB2:p-T161-CDK1']</td>\n",
       "      <td id=\"T_d1bef_row0_col5\" class=\"data row0 col5\" >http://model.geneontology.org/R-HSA-2500257</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_d1bef_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_d1bef_row1_col0\" class=\"data row1 col0\" >Interleukin-35 Signalling - Reactome</td>\n",
       "      <td id=\"T_d1bef_row1_col1\" class=\"data row1 col1\" >0.000016</td>\n",
       "      <td id=\"T_d1bef_row1_col2\" class=\"data row1 col2\" >7</td>\n",
       "      <td id=\"T_d1bef_row1_col3\" class=\"data row1 col3\" >9</td>\n",
       "      <td id=\"T_d1bef_row1_col4\" class=\"data row1 col4\" >['IL6ST', 'JAK2', 'STAT4', 'EBI3', 'STAT1', 'IL12A', 'set:JAK1, JAK2, (TYK2)']</td>\n",
       "      <td id=\"T_d1bef_row1_col5\" class=\"data row1 col5\" >http://model.geneontology.org/R-HSA-8984722</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_d1bef_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "      <td id=\"T_d1bef_row2_col0\" class=\"data row2 col0\" >Unwinding of DNA - Reactome</td>\n",
       "      <td id=\"T_d1bef_row2_col1\" class=\"data row2 col1\" >0.000029</td>\n",
       "      <td id=\"T_d1bef_row2_col2\" class=\"data row2 col2\" >6</td>\n",
       "      <td id=\"T_d1bef_row2_col3\" class=\"data row2 col3\" >7</td>\n",
       "      <td id=\"T_d1bef_row2_col4\" class=\"data row2 col4\" >['MCM5', 'MCM7', 'MCM2', 'MCM3', 'MCM4', 'MCM6']</td>\n",
       "      <td id=\"T_d1bef_row2_col5\" class=\"data row2 col5\" >http://model.geneontology.org/R-HSA-176974</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x13bb1e3b0>"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mac_weighted = enrich.enrich_wrapper('mac_comb.csv','Gene Symbol',method='ncHGT',FDR = 0.05,fpath = '../test_data/processed/',table_number = 66)\n",
    "mac_weighted"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:gocam2]",
   "language": "python",
   "name": "conda-env-gocam2-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
